---
title: "Microglia VCP mutant LPS treated bulk RNAseq"
author: "Oliver Ziff"
date: "`r format(Sys.time(), '%d %h %Y %H:%M')`"
output:
  html_document:
    fig_width: 7
    fig_height: 6
    toc_depth: 3
    theme: simplex
    dev: jpeg
editor_options:
  chunk_output_type: console
---

```{r}
library(here)
proj_path = here("/Volumes/lab-patanir/home/users/ziffo/projects/microglia-bulk-rnaseq")
load(here(proj_path,"scripts/microglia-bulk-rnaseq.RData")) # load metadata before functions to avoid overwriting with old functions
load(here(proj_path, "scripts/microglia_mass_spec.RData"))

# Mac
nemo_path = here("/Volumes/lab-patanir/home/users/ziffo")
shared_path = here("/Volumes/lab-patanir/home/users/ziffo/proj-luscombn-patani/working")
collab_path = here("/Volumes/lab-patanir/home/shared/patani-collab")
proj_path = here("/Volumes/lab-patanir/home/users/ziffo/projects/microglia-bulk-rnaseq")

## OnDemand
# proj_path = here("/nemo/lab/patanir/home/users/ziffo/projects/ipsc-mn-als-meta")
# script_path = here(here(proj_path,"scripts/ipsc_mn_als_meta")
# shared_path = here("/nemo/project/proj-luscombn-patani/working")
# collab_path = here("/nemo/lab/patanir/home/shared/patani-collab")
# setwd(proj_path)
# set_here(proj_path)
here(proj_path)
# here::i_am("ipsc_mn_als_meta.Rmd")
here()
dr_here()

source(here(nemo_path,"scripts/functions/R_workspace.R"))

# Ben figures located at https://www.dropbox.com/work/PataniR/Ben/Microglia%20figures

```

Public Data:
Lorenzini et al C9orf72 iPSC microglia https://www.biorxiv.org/content/10.1101/2020.09.03.277459v1.full # request RNAseq - The iPSC-MG and iPSC-CN datasets used and/or analyzed during the current study are available from the corresponding author on reasonable request. 
Haenseler 2017 - microarray only
Bennett 2016 https://www.pnas.org/content/113/12/E1738 (mouse) https://www.ncbi.nlm.nih.gov/bioproject?term=PRJNA307271 - Table S2 list of microglia-enriched genes. S3 microglia gene cassette. S4 LPS regulated genees

# Fig S1: Public data comparison

A - VCP mutation allele depth ratio
B - Dendrogram hiPSC microglia vs other cells
C - Heatmap Patir 2019 microglia markers

## VCP VCF

Allele Depth is the number of reads that support each of the reported alleles. All reads at the position (including reads that did not pass the variant caller's filters) are included in this number, except reads that were considered uninformative. Reads are considered uninformative when they do not provide enough statistical evidence to support one allele over another.

Box plot for R155C & R191Q alleles together\
Box plot for R155C alone\
Box plot for R191Q alone\
Scatter plot for R155C alone (hover for sample names)\
Scatter plot for R191Q alone (hover for sample names)

```{r setup, include=FALSE}
# import R155C AD flag text file & modify
R155C <- read.csv(here(collab_path, "microglia-bulk-rnaseq/vcp-vcf/vcf-analysis/R155C_ADflag.txt"), sep="") %>% mutate(vcp = "R155C", ratio = ALT / (REF + ALT)) # create ratio for mutation to reference (mut / ref + alt)
R191Q <- read.csv(here(collab_path, "microglia-bulk-rnaseq/vcp-vcf/vcf-analysis/R191Q_ADflag.txt"), sep="") %>% mutate(vcp = "R191Q", ratio = ALT / (REF + ALT)) # create ratio for mutation to reference (mut / ref + alt)
VCP <- bind_rows(R155C, R191Q) %>% mutate(cellline = gsub("_lps|_untx","", sample_name), treatment = str_extract(sample_name, "lps|untx"), als = str_extract(sample_name, "ctrl|vcp"))

## Box plot R155C & R191Q together
vcp_vcf_r155c_r191q = ggplot(VCP, aes(x=cellline, y=ratio, colour = vcp, shape = treatment)) + ylim(0,1)  + geom_point(stat="identity", position=position_dodge(width = 0.2), size = 2) + 
  labs(caption = "Allele Depth ratio = Mutation / Reference + Mutation. R155C: chr9:35065364 (ref = C, mut = T). R191Q: chr9:35065255 (mut = A: ref = G)", x="Cell line", y = "Allele Depth ratio") +
  theme_bw() + theme(plot.title = element_text(color = "red", size = 12, face = "bold", hjust = 0.5), legend.title=element_blank())
vcp_vcf_r155c_r191q
```

## Dendrogram

```{r}
clarke_zhang_reich_abud.sampleDists <- dist(t(assay(clarke_zhang_reich_abud$vsd)))
clarke_zhang_reich_abud.hc <- hclust(clarke_zhang_reich_abud.sampleDists, method = "ward.D2")
clarke_zhang_reich_abud.pcaData <- plotPCA(clarke_zhang_reich_abud$vsd, intgroup=c("group", "sample", "age", "celltype", "author", "celltype_age", "celltype_author", "age_author", "celltype_age_author", "replicate"), returnData=TRUE)
clarke_zhang_reich_abud.pcaData$replicate[5] = 5
clarke_zhang_reich_abud.pcaData$replicate[6] = 6
clarke_zhang_reich_abud.pcaData$replicate[7] = 7
clarke_zhang_reich_abud.hc$labels = paste0(clarke_zhang_reich_abud.pcaData$celltype_age_author, clarke_zhang_reich_abud.pcaData$replicate)
brewer.pal(10, "Paired")
clarke_zhang_reich_abud.a <- c(rep("#FB9A99", 9), rep("#E31A1C", 7), rep("#FDBF6F", 5), rep("#33A02C", 3), rep("#B2DF8A", 3), rep("#A6CEE3", 12), rep("#FF7F00", 3), rep("#1F78B4", 2), rep("#CAB2D6", 1), rep("#6A3D9A", 5))
clarke_zhang_reich_abud.dendrogram <- ggdendrogram(clarke_zhang_reich_abud.hc, rotate = FALSE, theme_dendro = FALSE) +  theme_classic()  + theme(axis.line=element_blank(), axis.title.y=element_blank(), axis.title.x=element_blank(), axis.text.y=element_blank(), axis.ticks.y=element_blank(), axis.text.x= element_text(colour = clarke_zhang_reich_abud.a, size = 10, angle = 90), strip.background = element_rect(colour="black", fill="white"), panel.grid = element_blank())
clarke_zhang_reich_abud.dendrogram
# ggsave(clarke_zhang_reich_abud.dendrogram, filename = here(proj_path, "expression/deseq2/clarke_zhang_reich_abud.dendrogram.png", width = 13, height = 5)
# 
```

## Microglia database

```{r}
microglia_signature = read_excel(here(proj_path, "expression/glia23572-sup-0001-tables3.xlsx"), skip = 5) %>% clean_names() %>% filter(core == "Core", gene_symbol %in% gtf$gene_name) %>% pull(gene_symbol)
# microglia_signature = read_excel(here(proj_path, "expression/glia23572-sup-0001-tables3.xlsx", skip = 5) %>% clean_names() %>% filter(gene_symbol %in% gtf$gene_name) %>% pull(gene_symbol)
# microglia_signature = read_excel(here(proj_path, "expression/gossellin_human_microglia_science_tableS2.xlsx", sheet = "Human microglia gene signature") %>% clean_names() %>% filter(gene_name %in% gtf$gene_name) %>% pull(gene_name)
# 
# microglia_signature.df = tibble("gene_name" = microglia_signature) %>% mutate(group = "microglia") %>% arrange(gene_name)
microglia_signature.df = read_excel(here(proj_path, "expression/glia23572-sup-0001-tables3.xlsx"), skip = 5) %>% clean_names() %>% 
  mutate(gene_name = gene_symbol, group = biological_process, group = case_when(
    group %in% c("Antigen signaling",  "Cytokine signaling", "Complement system", "Fc fragment of IgG receptor singaling", "GPCR signaling", "Immune signaling", "Signal transduction", "TLR signaling", "Transporter", "Vesicle transport/formation", "Apoptosis, Inflammasome", "Reactive oxidation production") ~ "Immune signaling", group %in% c("Cell differentiation /Proliferation", "Development") ~ "Differentiation", group %in% c("Cell interaction", "Integrin singaling", "Cytoskeletal organisation") ~ "Cell organisation", 
    group %in% c("Lysosomal", "Migration", "Phagocytosis") ~ "Phagocytosis",
    group %in% c("Metabolism/homeostasis") ~ "Metabolism", TRUE ~ "Misc"), group = factor(group, levels = c("Immune signaling", "Cell organisation", "Differentiation", "Metabolism", "Phagocytosis", "Misc"))) %>% 
  filter(core == "Core", gene_name %in% gtf$gene_name) %>% select(gene_name, group) %>% arrange(group)

# "Antigen signaling",  "Cytokine signaling", "Complement system", "Fc fragment of IgG receptor singaling", "GPCR signaling", "Immune signaling", "Signal transduction", "TLR signaling", "Transporter", "Vesicle transport/formation", "Apoptosis, Inflammasome", "Reactive oxidation production", 
# "Cell differentiation /Proliferation", "Development"
# "Cell interaction", "Integrin singaling", "Cytoskeletal organisation", 
# "DAP12 signaling", "Uncharacterized"
# "Lysosomal", "Migration", "Phagocytosis", 
# "Metabolism/homeostasis"
#                     Antigen signaling               Apoptosis, Inflammasome   Cell differentiation /Proliferation                      Cell interaction                     Complement system 
#                                    21                                     2                                    16                                     9                                     6 
#                    Cytokine signaling             Cytoskeletal organisation                       DAP12 signaling                           Development Fc fragment of IgG receptor singaling 
#                                    16                                    17                                     2                                     2                                     8 
#                        GPCR signaling                      Immune signaling                    Integrin singaling                             Lysosomal                Metabolism/homeostasis 
#                                    11                                    31                                     4                                     2                                    20 
#                             Migration                          Phagocytosis         Reactive oxidation production                   Signal transduction                         TLR signaling 
#                                     6                                     4                                     5                                    12                                     8 
#                           Transporter                       Uncharacterized           Vesicle transport/formation 
#                                     8                                    30                                     4 
clarke_zhang_reich_abud$vsd.counts = as_tibble(assay(clarke_zhang_reich_abud$vsd), rownames = "gene_id") %>% left_join(gene2ens)
microglia_signature.marker <- clarke_zhang_reich_abud$vsd.counts %>% distinct(gene_name, .keep_all = TRUE) %>% 
  filter(gene_name %in% microglia_signature) %>% #, rowSums(across(where(is.numeric))) > 162) %>% dplyr::mutate_if(is.numeric, funs((. - 3.258)))  %>% #dplyr::mutate_if(is.numeric, funs(log2(.))) %>% dplyr::mutate_if(is.numeric, funs(ifelse(. == -Inf,0, .))) %>%
  arrange(factor(gene_name, levels = microglia_signature.df$gene_name)) %>%
  select(gene_name, CTRL1 = ctrl_untx_R1, CTRL2 = ctrl_untx_R2, CTRL3 = ctrl_untx_R3, CTRL4 = ctrl_untx_R4, VCP1 = vcp_untx_R1, VCP2 = vcp_untx_R2, VCP3 = vcp_untx_R3, 
         contains("ipsc-microglia-reich-2020_R"), 
         contains("ipsc-microglia-abud-2019_R"),
         contains("adult-microglia_"), 
         contains("myeloid_adult_temporal_lobe_"), 
         contains("astrocyte_adult_temporal_lobe_"), 
         contains("oligodendrocyte_adult_temporal_lobe_"), 
         contains("endothelial_adult_temporal_lobe_"), neuron_adult_temporal_lobe_R1) 
microglia_signature.marker.mat <- make_matrix(select(microglia_signature.marker,-gene_name), pull(microglia_signature.marker,gene_name))
# microglia_signature.marker.mat[microglia_signature.marker.mat > 6] <- 6 # crop outliers to improve colour distribution
microglia_signature.marker.scaled_mat = t(scale(t(microglia_signature.marker.mat))) # row scale
# microglia_signature.marker.scaled_mat <- microglia_signature.marker.scaled_mat[!is.infinite(rowSums(microglia_signature.marker.scaled_mat)),]

microglia_signature_markers = microglia_signature.marker$gene_name[microglia_signature.marker$gene_name %in% microglia_signature]

microglia_signature.rnaseq.heatmap <- Heatmap(microglia_signature.marker.scaled_mat, 
          heatmap_legend_param = list(title = ""), border = TRUE, 
          cluster_rows = FALSE, cluster_row_slices = FALSE, cluster_columns = FALSE,
          row_split = factor(microglia_signature.df$group, levels = unique(microglia_signature.df$group)),
          row_order = microglia_signature.df$gene_name, show_row_names = FALSE, show_column_names = FALSE,#row_names_gp = gpar(fontsize = 8), column_names_gp = gpar(fontsize = 8), #, #, col = c(rep("red", 10), rep("blue", 8)))
          bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = c("firebrick1","firebrick1","firebrick1","firebrick1","firebrick1","dodgerblue1","dodgerblue1","dodgerblue1","dodgerblue1")), labels = c("Clarke ipsc\nMicroglia", "Reich ipsc\nMicroglia", "Abud ipsc\nMicroglia", "Abud adult\nMicroglia", "Zhang adult\nMyeloid", "Zhang adult\nAstrocyte", "Zhang adult\nOligodendrocyte", "Zhang adult\nEndothelial", "Neuron"), labels_gp = gpar(fontsize = 10), labels_rot = 90, height = unit(25,"mm"))), 
          right_annotation = rowAnnotation(markers = anno_block(gp = gpar(fill = c("white", "white", "white", "white", "white")), labels = c("Immune\nSignaling", "Cell\nOrganisation", "Differentiation", "Metabolism", "Phagocytosis", "Misc."), labels_gp = gpar(fontsize = 10), labels_rot = 0, 
                                                                width = unit(25,"mm"))), 
          row_title = NULL, row_title_gp = gpar(fontsize = 10),
          column_title = NULL, column_names_rot = 90, column_names_centered = TRUE, 
          column_split =  c(rep(1,7),rep(2,5),rep(3,9),rep(4,3),rep(5,3),rep(6,12),rep(7,5),rep(8,2),rep(9,1)))
microglia_signature.rnaseq.heatmap

```

## Morphology

```{r}
microglia_morphology_ben = read_excel(here(proj_path,"expression/morphology/Morphology for Oli Fig 1H.xlsx")) %>% mutate(condition = paste0(celltype,"\n",mutation), condition = factor(condition, levels = c("Microglia\nCTRL", "Microglia\nVCP", "Macrophage\nCTRL", "Macrophage\nVCP")), type = gsub("Secondary branches","Secondary\nbranches",type)) %>% drop_na(value)
microglia_morphology_ben %>% glimpse
microglia_morphology_ben.summarised = microglia_morphology_ben %>% my_summarise(value, type, condition) %>% mutate(type = factor(type, levels = c("Secondary\nbranches","Multipolar","Bipolar","Round"))) %>% group_by(condition) %>% mutate(mean_stacked = cumsum(mean_value))
microglia_morphology_ben.summarised
# type                  condition          mean_value median_value sd_value Q1_value Q3_value min_value max_value sum_value n_value se_value
#    <chr>                 <fct>                   <dbl>        <dbl>    <dbl>    <dbl>    <dbl>     <dbl>     <dbl>     <dbl>   <int>    <dbl>
#  1 "Bipolar"             "Microglia\nCTRL"       25.1         22.6     13.7     13.3     35.9      10.7       54.0     428.       17     3.31
#  2 "Bipolar"             "Microglia\nVCP"        26.3         25.2     12.3     14.6     36.2      12.7       52.6     394.       15     3.18
#  3 "Bipolar"             "Macrophage\nCTRL"      15.1          9.56    15.3      4.67    18.8       0         58.3     256.       17     3.71
#  4 "Bipolar"             "Macrophage\nVCP"       13.3          8.62    12.4      3.73    19.7       0         44.0     200.       15     3.21
#  5 "Multipolar"          "Microglia\nCTRL"       35.8         41.5     13.7     21.3     46.7       3.39      48.7     609.       17     3.32
#  6 "Multipolar"          "Microglia\nVCP"        35.8         36.1     15.1     25.3     43.1      12.8       68.8     537.       15     3.89
#  7 "Multipolar"          "Macrophage\nCTRL"      22.7         22.4     12.4     10       32.0       2.49      39.6     386.       17     3.01
#  8 "Multipolar"          "Macrophage\nVCP"       28.2         23.1     15.7     17.6     39.7       4.28      63.5     422.       15     4.06
#  9 "Round"               "Microglia\nCTRL"       25.7         24.2     12.4     17.2     32.3       7.76      55.1     437.       17     3.01
# 10 "Round"               "Microglia\nVCP"        26.5         27.6     11.8     18.0     37.9       1.06      40.0     398.       15     3.04
# 11 "Round"               "Macrophage\nCTRL"      56.9         49.9     19.4     40.6     70.2      33.3       93.0     967.       17     4.69
# 12 "Round"               "Macrophage\nVCP"       49.2         49.5     22.4     36.8     62.6      14.8       91.4     737.       15     5.80
# 13 "Secondary\nbranches" "Microglia\nCTRL"       13.3         11.9      8.19     7.77    19.2       1.43      30.2     226.       17     1.99
# 14 "Secondary\nbranches" "Microglia\nVCP"        11.4         13.4      7.13     4.20    17.1       1.18      22.5     171.       15     1.84
# 15 "Secondary\nbranches" "Macrophage\nCTRL"       5.33         3.83     4.82     2.19     6.91      0         15.8      90.6      17     1.17
# 16 "Secondary\nbranches" "Macrophage\nVCP"        9.35         5.72     7.91     4.50    16.5       0         23.5     140.       15     2.04

# proportion barchart
microglia_morphology_ben.barplot <- ggplot(microglia_morphology_ben.summarised, aes(x=type, y=mean_value, fill=condition)) +
  geom_bar(stat="identity", position = position_dodge()) +
  geom_errorbar(aes(ymin=mean_value-se_value, ymax=mean_value+se_value), width=.2, position=position_dodge(0.9)) +
  # scale_fill_npg() +
  scale_fill_brewer(palette = "Paired") +
  theme_classic() +  theme(panel.grid = element_blank(), legend.title = element_blank(), legend.position = "top", legend.spacing.y = unit(1, "cm"), legend.text=element_text(size=8), legend.key.size = unit(0.4, "cm"), axis.text=element_text(size=8), axis.title=element_text(size=8)) +#axis.text.x = element_text(angle = 45, hjust = 1)
  guides(fill = guide_legend(reverse = FALSE, nrow = 1)) + labs(y = "% of cells", x = "")
# microglia_morphology_ben.barplot

# violin
morphology.violin = violin_plot(data = microglia_morphology_ben, color_by = "condition", continuous = "value", cols = brewer.pal(6, "Paired"), ylabel = "% of cells", facet_by = "type", facet_ncol = 4, plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos =c(56,58,62,52,55,55,  66,68,70,63,67,65,  86,90,95,85,80,80,  26,30,30,27,30,25), tip_length = 0.01) + theme(axis.text.x = element_text(angle = 30, vjust = 0.7, hjust=0.7))
# morphology.violin

# c(56,58,60,55,57,55,  66,68,70,65,67,65,  86,88,90,85,87,85,  26,28,30,25,27,25)


# facet                 .y.   group1             group2                n1    n2 statistic          p y.position groups        xmin  xmax p.signif
#    <chr>                 <chr> <chr>              <chr>              <int> <int>     <dbl>      <dbl>      <dbl> <named list> <dbl> <dbl> <chr>   
#  1 "Bipolar"             lfc   "Microglia\nCTRL"  "Microglia\nVCP"      17    15     112   0.576              10 <chr [2]>        1     2 ns      
#  2 "Bipolar"             lfc   "Microglia\nCTRL"  "Macrophage\nCTRL"    17    17     225   0.005              10 <chr [2]>        1     3 **      
#  3 "Bipolar"             lfc   "Microglia\nCTRL"  "Macrophage\nVCP"     17    15     189   0.02               10 <chr [2]>        1     4 *       
#  4 "Bipolar"             lfc   "Microglia\nVCP"   "Macrophage\nCTRL"    15    17     204   0.003              10 <chr [2]>        2     3 **      
#  5 "Bipolar"             lfc   "Microglia\nVCP"   "Macrophage\nVCP"     15    15     172   0.013              10 <chr [2]>        2     4 *       
#  6 "Bipolar"             lfc   "Macrophage\nCTRL" "Macrophage\nVCP"     17    15     140.  0.664              10 <chr [2]>        3     4 ns      
#  7 "Multipolar"          lfc   "Microglia\nCTRL"  "Microglia\nVCP"      17    15     147   0.478              10 <chr [2]>        1     2 ns      
#  8 "Multipolar"          lfc   "Microglia\nCTRL"  "Macrophage\nCTRL"    17    17     232   0.002              10 <chr [2]>        1     3 **      
#  9 "Multipolar"          lfc   "Microglia\nCTRL"  "Macrophage\nVCP"     17    15     174   0.082              10 <chr [2]>        1     4 ns      
# 10 "Multipolar"          lfc   "Microglia\nVCP"   "Macrophage\nCTRL"    15    17     193   0.013              10 <chr [2]>        2     3 *       
# 11 "Multipolar"          lfc   "Microglia\nVCP"   "Macrophage\nVCP"     15    15     144   0.202              10 <chr [2]>        2     4 ns      
# 12 "Multipolar"          lfc   "Macrophage\nCTRL" "Macrophage\nVCP"     17    15      98   0.278              10 <chr [2]>        3     4 ns      
# 13 "Round"               lfc   "Microglia\nCTRL"  "Microglia\nVCP"      17    15     114   0.628              10 <chr [2]>        1     2 ns      
# 14 "Round"               lfc   "Microglia\nCTRL"  "Macrophage\nCTRL"    17    17      19   0.00000178         10 <chr [2]>        1     3 ****    
# 15 "Round"               lfc   "Microglia\nCTRL"  "Macrophage\nVCP"     17    15      53   0.004              10 <chr [2]>        1     4 **      
# 16 "Round"               lfc   "Microglia\nVCP"   "Macrophage\nCTRL"    15    17      21   0.0000122          10 <chr [2]>        2     3 ****    
# 17 "Round"               lfc   "Microglia\nVCP"   "Macrophage\nVCP"     15    15      41   0.002              10 <chr [2]>        2     4 **      
# 18 "Round"               lfc   "Macrophage\nCTRL" "Macrophage\nVCP"     17    15     148   0.455              10 <chr [2]>        3     4 ns      
# 19 "Secondary\nbranches" lfc   "Microglia\nCTRL"  "Microglia\nVCP"      17    15     143   0.576              10 <chr [2]>        1     2 ns      
# 20 "Secondary\nbranches" lfc   "Microglia\nCTRL"  "Macrophage\nCTRL"    17    17     233   0.002              10 <chr [2]>        1     3 **      
# 21 "Secondary\nbranches" lfc   "Microglia\nCTRL"  "Macrophage\nVCP"     17    15     160   0.23               10 <chr [2]>        1     4 ns      
# 22 "Secondary\nbranches" lfc   "Microglia\nVCP"   "Macrophage\nCTRL"    15    17     189   0.021              10 <chr [2]>        2     3 *       
# 23 "Secondary\nbranches" lfc   "Microglia\nVCP"   "Macrophage\nVCP"     15    15     129   0.512              10 <chr [2]>        2     4 ns      
# 24 "Secondary\nbranches" lfc   "Macrophage\nCTRL" "Macrophage\nVCP"     17    15      93.5 0.205              10 <chr [2]>        3     4 ns      

```

## Merge

```{r}
figS1.merged = ggarrange(vcp_vcf_r155c_r191q, clarke_zhang_reich_abud.dendrogram, NULL, as.ggplot(microglia_signature.rnaseq.heatmap), microglia_morphology_ben.barplot, ncol = 1, nrow = 5, heights = c(0.9,1.3,0.1,2,1), labels = c("A","B","C","", "D"))
# figS1.merged
ggsave(figS1.merged, filename = here(proj_path, "expression/deseq2/figS1.merged.png"), width = 11, height = 17)
ggsave(figS1.merged, filename = here(proj_path, "expression/deseq2/figS1.merged.pdf"), width = 11, height = 17)

```

# Fig. 1 Untx VCP & CTRL Gene Expression

A. Schematic for MG differentiation
B. PCA all cell types
C. Heatmap RNAseq markers
D. Heatmap proteomic markers
E. IBA1 IF
F. P2RY12 IF
G. TMEM119 IF
H. Morphology

```{r}
# https://www.dropbox.com/work/PataniR/Ben/Microglia%20figures

# import with magick 
Fig1A_MG_schematic.png = magick::image_read(here(proj_path,"ben-figures/Fig1A_MG_schematic.png"))
Fig1A_MG_schematic.gg = ggdraw() + draw_image(Fig1A_MG_schematic.png)
# Fig1A_MG_schematic.gg

Fig1Ei_IBA1image.png = magick::image_read(here(proj_path,"ben-figures/Fig1Ei_IBA1image.png"))
Fig1Ei_IBA1image.gg = ggdraw() + draw_image(Fig1Ei_IBA1image.png)
# Fig1Ei_IBA1image.gg

Fig1Fi_P2RY12image.png = magick::image_read(here(proj_path,"ben-figures/Fig1Fi_P2RY12image.png"))
Fig1Fi_P2RY12image.gg = ggdraw() + draw_image(Fig1Fi_P2RY12image.png)
# Fig1Fi_P2RY12image.gg

Fig1Gi_TMEM119image.png = magick::image_read(here(proj_path,"ben-figures/Fig1Gi_TMEM119image.png"))
Fig1Gi_TMEM119image.gg = ggdraw() + draw_image(Fig1Gi_TMEM119image.png)
# Fig1Gi_TMEM119image.gg

# Fig1Eii_IBA1quantification.png = magick::image_read(here(proj_path,"ben-figures/Fig1Eii_IBA1quantification.png"))
# Fig1Eii_IBA1quantification.gg = ggdraw() + draw_image(Fig1Eii_IBA1quantification.png)
# # Fig1Eii_IBA1quantification.gg


# Fig1Fii_P2RY12quantification.png = magick::image_read(here(proj_path,"ben-figures/Fig1Fii_P2RY12quantification.png"))
# Fig1Fii_P2RY12quantification.gg = ggdraw() + draw_image(Fig1Fii_P2RY12quantification.png)
# # Fig1Fii_P2RY12quantification.gg


# Fig1Gii_TMEM119quantification.png = magick::image_read(here(proj_path,"ben-figures/Fig1Gii_TMEM119quantification.png"))
# Fig1Gii_TMEM119quantification.gg = ggdraw() + draw_image(Fig1Gii_TMEM119quantification.png)
# # Fig1Gii_TMEM119quantification.gg
# 
# Fig1H_Morphology_quantification.png = magick::image_read(here(proj_path,"ben-figures/Fig1H_Morphology_quantification.png"))
# Fig1H_Morphology_quantification.gg = ggdraw() + draw_image(Fig1H_Morphology_quantification.png)
# # Fig1H_Morphology_quantification.gg

```

## PCA

cell types from Zhang 2017

microglia ipsc from Reich and Abud

```{r}
library(wesanderson)

# with IPSC AC & MN
clarke_zhang_reich_abud_ac_mn.pcaData <- plotPCA(clarke_zhang_reich_abud_ac_mn$vsd, intgroup=c("group", "sample", "age", "celltype", "author", "age_author", "celltype_age_author", "replicate"), returnData=TRUE)
clarke_zhang_reich_abud_ac_mn.percentVar <- round(100 * attr(clarke_zhang_reich_abud_ac_mn.pcaData, "percentVar"))
clarke_zhang_reich_abud_ac_mn.pca <- ggplot(clarke_zhang_reich_abud_ac_mn.pcaData, aes(PC1, PC2, color = celltype_age_author)) + #geom_text_repel(aes(label=name), size = 3) + 
  scale_colour_manual(values = wes_palette("Zissou1", 12, type = "continuous")) + #scale_colour_brewer(palette = "Set3") + #scale_colour_brewer(palette = "Paired") + 
  theme_bw() + theme(panel.grid = element_blank(), legend.text=element_text(size=8), legend.position = "none", legend.box = "horizontal", legend.spacing = unit(0.1, 'cm'), legend.title = element_blank()) +
  xlab(paste0("PC1: ",clarke_zhang_reich_abud_ac_mn.percentVar[1],"% variance")) +  ylab(paste0("PC2: ",clarke_zhang_reich_abud_ac_mn.percentVar[2],"% variance")) + #coord_fixed()  + 
  ggforce::geom_mark_ellipse(aes(label = celltype_age_author), label.fontsize = 8, label.fontface = "plain", con.type = "straight", con.cap = 0.1, label.buffer = unit(5,"mm"), show.legend = FALSE) +
  geom_point(size=2) # fill = celltype, 
# clarke_zhang_reich_abud_ac_mn.pca

# clarke_zhang_reich_abud.pcaData <- plotPCA(clarke_zhang_reich_abud$vsd, intgroup=c("group", "sample", "age", "celltype", "author", "celltype_age", "celltype_author", "age_author", "celltype_age_author", "replicate"), returnData=TRUE)
# clarke_zhang_reich_abud.percentVar <- round(100 * attr(clarke_zhang_reich_abud.pcaData, "percentVar"))
# clarke_zhang_reich_abud.pca <- ggplot(clarke_zhang_reich_abud.pcaData, aes(PC1, PC2, color = celltype, shape = age_author)) + #geom_text_repel(aes(label=name), size = 3) + 
#  scale_colour_brewer(palette = "Paired") + # scale_shape_manual(values=c(1, 16)) + scale_colour_manual( values = c("firebrick2", "dodgerblue2") ) +  scale_fill_manual( values = c("firebrick2", "dodgerblue2") ) +  
#   theme_bw() + theme(panel.grid = element_blank(), legend.text=element_text(size=8), legend.position = "top", legend.box = "horizontal", legend.spacing = unit(0.1, 'cm'), legend.title = element_blank()) +
#   xlab(paste0("PC1: ",clarke_zhang_reich_abud.percentVar[1],"% variance")) +  ylab(paste0("PC2: ",clarke_zhang_reich_abud.percentVar[2],"% variance")) + #coord_fixed()  + 
#   ggforce::geom_mark_ellipse(aes(label = celltype_age_author), label.fontsize = 6, label.fontface = "plain", con.type = "straight", con.cap = 0.1, label.buffer = unit(5,"mm"), show.legend = FALSE) +
#   geom_point(size=2) # fill = celltype, 
# clarke_zhang_reich_abud.pca
# # ggsave(clarke_zhang_reich_abud.pca, filename = here(proj_path, "expression/deseq2/clarke_zhang_reich_abud.pca.png", height = 8, width = 10)



# # just clarke samples
# untx_vcp_vs_ctrl.pcaData <- plotPCA(untx_vcp_vs_ctrl$vsd, intgroup=c("group", "sample", "age", "cellline", "mutation", "genotype", "parent_cellline", "age_author", "celltype_age_author", "replicate"), returnData=TRUE)
# untx_vcp_vs_ctrl.percentVar <- round(100 * attr(untx_vcp_vs_ctrl.pcaData, "percentVar"))
# untx_vcp_vs_ctrl.pca <- ggplot(untx_vcp_vs_ctrl.pcaData, aes(PC1, PC2, color = mutation)) + #geom_text_repel(aes(label=name), size = 3) + 
#  # scale_colour_brewer(palette = "Paired") + # scale_shape_manual(values=c(1, 16)) + scale_colour_manual( values = c("firebrick2", "dodgerblue2") ) +  scale_fill_manual( values = c("firebrick2", "dodgerblue2") ) +  
#   theme_bw() + theme(panel.grid = element_blank(), legend.text=element_text(size=8), legend.position = "top", legend.box = "horizontal", legend.spacing = unit(0.1, 'cm'), legend.title = element_blank()) +
#   geom_text_repel(data = untx_vcp_vs_ctrl.pcaData, aes(x = PC1, y = PC2, label = cellline), colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size=2.6, max.overlaps = Inf) +
#   xlab(paste0("PC1: ",untx_vcp_vs_ctrl.percentVar[1],"% variance")) +  ylab(paste0("PC2: ",untx_vcp_vs_ctrl.percentVar[2],"% variance")) + #coord_fixed()  + 
#   # ggforce::geom_mark_ellipse(aes(label = cellline), label.fontsize = 8, label.fontface = "plain", con.type = "straight", con.cap = 0.5, show.legend = FALSE) +
#   geom_point(size=2) # fill = celltype, 
# untx_vcp_vs_ctrl.pca
# 
# 
# untx_vcp_vs_ctrl.sampleDists <- dist(t(assay(untx_vcp_vs_ctrl$vsd)))
# untx_vcp_vs_ctrl.hc <- hclust(untx_vcp_vs_ctrl.sampleDists, method = "ward.D2")
# untx_vcp_vs_ctrl.pcaData <- plotPCA(untx_vcp_vs_ctrl$vsd, intgroup=c("group", "sample", "age", "celltype", "author", "celltype_age", "celltype_author", "age_author", "celltype_age_author", "replicate"), returnData=TRUE)
# # untx_vcp_vs_ctrl.hc$labels = paste0(untx_vcp_vs_ctrl.pcaData$celltype_age_author, untx_vcp_vs_ctrl.pcaData$replicate)
# brewer.pal(10, "Paired")
# untx_vcp_vs_ctrl.a <- c(rep("#FB9A99", 9), rep("#E31A1C", 7), rep("#FDBF6F", 5), rep("#33A02C", 3), rep("#B2DF8A", 3), rep("#A6CEE3", 12), rep("#FF7F00", 3), rep("#1F78B4", 2), rep("#CAB2D6", 1), rep("#6A3D9A", 5))
# untx_vcp_vs_ctrl.dendrogram <- ggdendrogram(untx_vcp_vs_ctrl.hc, rotate = FALSE, theme_dendro = FALSE) +  
#   theme_classic()  + theme(axis.line=element_blank(), axis.title.y=element_blank(), axis.title.x=element_blank(), axis.text.y=element_blank(), axis.ticks.y=element_blank(), axis.text.x= element_text(colour = untx_vcp_vs_ctrl.a, size = 8, angle = 90), strip.background = element_rect(colour="black", fill="white"), panel.grid = element_blank())
# untx_vcp_vs_ctrl.dendrogram


```


## RNA-seq heatmap celltype markers

Only untreated samples included in heatmap

Bennett et al. PNAS 2016 Table.S2 markers <https://www.pnas.org/content/pnas/suppl/2016/02/10/1525528113.DCSupplemental/pnas.1525528113.sapp.pdf> Zhang et al. Neuron 2017

```{r}
astrocyte.markers = c("ALDH1L1", "AQP4", "GFAP","GJA1", "SOX9", "EDNRB", "DIO2", "FGFR3", "MLC1", "SLC4A4") #  "SLC1A3", 
endo.markers = c("CLDN5",  "ESAM", "ICAM2", "APOLD1", "NOSTRIN", "ADGRL4") # "PECAM1",
myeloid.markers = c("AIF1", "C1QA", "CSF1R", "ITGAM", "PTPRC", "CEMIP2", "TREM2", "RUNX1", "CD83",  "CCL2", "TLR2",  "MERTK", "HMGB1", "CCL3") # "TMEM119", "TNF", "IL1A", "CX3CR1","P2RY12",
# myeloid.markers = c("AIF1", "C1QA", "CSF1R", "ITGAM", "PTPRC", "CEMIP2", "TREM2", "RUNX1", "CD83",  "CCL2", "TLR2",  "MERTK", "HMGB1", "TMEM119", "TNF", "IL1A", "CCL3", "CX3CR1","P2RY12")
neuron.markers = c("NPAS4", "NPY", "SST", "CCK", "GAD1", "SYT1", "STMN2", "SYN1") # "ENO2", 
oligo.markers = c("MBP", "MOG",  "GPR37", "SOX10", "OPALIN", "MOBP", "PLP1") # "CNP", "GALC",

celltype.markers = c(astrocyte.markers, endo.markers, myeloid.markers, neuron.markers, oligo.markers)
celltype.markers[!celltype.markers %in% gtf$gene_name]

celltype.markers.df = tibble("gene_name" = celltype.markers) %>% filter(gene_name %in% celltype.markers) %>% 
  mutate(group = case_when(gene_name %in% astrocyte.markers ~ "astrocyte", gene_name %in% endo.markers ~ "endo", gene_name %in% myeloid.markers ~ "myeloid", gene_name %in% neuron.markers ~ "neuronal", gene_name %in% oligo.markers ~ "oligo"), group = group) %>% arrange(group)

untx_vcp_vs_ctrl$vsd.counts = as_tibble(assay(untx_vcp_vs_ctrl$vsd), rownames = "gene_id") %>% left_join(gene2ens)
celltype.markers.marker <- untx_vcp_vs_ctrl$vsd.counts %>% distinct(gene_name, .keep_all = TRUE) %>% 
  filter(gene_name %in% celltype.markers) %>% #, rowSums(across(where(is.numeric))) > 162) %>% 
  dplyr::mutate_if(is.numeric, funs(ifelse(. <0, 0, .))) %>%   # dplyr::mutate_if(is.numeric, funs(log2(.))) %>%
  arrange(factor(gene_name, levels = celltype.markers.df$gene_name)) %>%
  rename_with(~ gsub("_untx_R", "", .x, fixed = TRUE)) %>%
  select(gene_name, CTRL1 = ctrl1, CTRL2 = ctrl2, CTRL3 = ctrl3, CTRL4 = ctrl4, VCP1 = vcp1, VCP2 = vcp2, VCP3 = vcp3) #lps_ctrl3,lps_ctrl4,lps_ncrm1,lps_vcpf10,lps_cb1d,lps_glib,lps_ncrme6)   
celltype.markers.marker.mat <- make_matrix(select(celltype.markers.marker,-gene_name), pull(celltype.markers.marker,gene_name))
# celltype.markers.marker.mat[celltype.markers.marker.mat > 6] <- 6 # crop outliers to improve colour distribution
# celltype.markers.marker.scaled_mat = t(scale(t(celltype.markers.marker.mat))) # row scale
# celltype.markers.marker.scaled_mat <- celltype.markers.marker.scaled_mat[!is.infinite(rowSums(celltype.markers.marker.scaled_mat)),]

astrocyte_markers = celltype.markers.marker$gene_name[celltype.markers.marker$gene_name %in% astrocyte.markers]
endo_markers = celltype.markers.marker$gene_name[celltype.markers.marker$gene_name %in% endo.markers]
myeloid_markers = celltype.markers.marker$gene_name[celltype.markers.marker$gene_name %in% myeloid.markers]
neuron_markers = celltype.markers.marker$gene_name[celltype.markers.marker$gene_name %in% neuron.markers]
oligo_markers = celltype.markers.marker$gene_name[celltype.markers.marker$gene_name %in% oligo.markers]

celltype.markers.rnaseq.heatmap <- Heatmap(t(celltype.markers.marker.mat), 
          heatmap_legend_param = list(title = ""), border = TRUE, 
          cluster_columns = FALSE, cluster_column_slices = FALSE, cluster_rows = FALSE,
          column_split = factor(celltype.markers.df$group, levels = unique(celltype.markers.df$group)), column_order = celltype.markers.df$gene_name,
          column_names_gp = gpar(fontsize = 7), row_names_gp = gpar(fontsize = 8), #, #, col = c(rep("red", 10), rep("blue", 8)))
          bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = c("white", "white", "white", "white", "white")), labels = c("Astrocyte", "Endothel", "Myeloid", "Neuronal", "Oligoden"), labels_gp = gpar(fontsize = 7), labels_rot = 0)), 
          column_title = "RNA sequencing", column_title_gp = gpar(fontsize = 8), row_title = NULL, row_names_rot = 0, row_names_centered = TRUE, row_split =  c(1,1,1,1,2,2,2))
# celltype.markers.rnaseq.heatmap
# ggsave(as.ggplot(celltype.markers.marker.heatmap), filename = here(proj_path, "expression/deseq2/celltype.markers.marker.heatmap.png")
```

## Mass spec heatmap microglia markers

```{r}
dep.vcp_vs_ctrl$assay =  assay(dep.vcp_vs_ctrl$dep) %>% as_tibble(rownames = "gene_name") 
### mass spec
myeloid.markers = c("AIF1", "C1QA", "CSF1R", "ITGAM", "PTPRC", "CEMIP2", "TREM2", "RUNX1", "CD83",  "CCL2", "TLR2",  "MERTK", "HMGB1", "TMEM119", "TNF", "CX3CR1","P2RY12", "CCR2")#"IL1A", "CCL3", 
myeloid.markers.filt = myeloid.markers[myeloid.markers %in% dep.vcp_vs_ctrl$assay$gene_name]
myeloid.markers.df = tibble("gene_name" = myeloid.markers) %>% filter(gene_name %in% myeloid.markers.filt) %>% mutate(group = "myeloid") %>% arrange(group)
myeloid.markers.marker <- dep.vcp_vs_ctrl$assay %>%
  filter(gene_name %in% myeloid.markers) %>% #, rowSums(across(where(is.numeric))) > 162) %>% 
  # mutate(across(is.numeric, ~ .x - min(.))) %>%
  arrange(factor(gene_name, levels = myeloid.markers.df$gene_name)) %>% 
  # mutate(`CTRL 1` = (ctrl_1+ctrl_2+ctrl_3)/3,`CTRL 2` = (ctrl_4+ctrl_5+ctrl_6)/3,`CTRL 3` = (ctrl_7+ctrl_8+ctrl_9)/3,`CTRL 4` = (ctrl_10+ctrl_11+ctrl_12)/3,
  #        `VCP 1` = (vcp_1+vcp_2+vcp_3)/3,`VCP 2` = (vcp_4+vcp_5+vcp_6)/3,`VCP 3` = (vcp_7+vcp_8+vcp_9)/3) %>%
  # select(gene_name, `CTRL 1`,`CTRL 2`,`CTRL 3`,`CTRL 4`,`VCP 1`,`VCP 2`,`VCP 3`)
  mutate(CTRL1 = (ctrl_1+ctrl_2+ctrl_3)/3,CTRL2 = (ctrl_4+ctrl_5+ctrl_6)/3, CTRL3 = (ctrl_7+ctrl_8+ctrl_9)/3,CTRL4 = (ctrl_10+ctrl_11+ctrl_12)/3,CTRL5 = (ctrl_13+ctrl_14+ctrl_15)/3,
         VCP1 = (vcp_1+vcp_2+vcp_3)/3, VCP2 = (vcp_4+vcp_5+vcp_6)/3, VCP3 = (vcp_7+vcp_8+vcp_9)/3, VCP4 = (vcp_10+vcp_11+vcp_12)/3, VCP5 = (vcp_13+vcp_14+vcp_15)/3) %>%
  select(gene_name, CTRL1,CTRL2,CTRL3,CTRL4,CTRL5,VCP1,VCP2,VCP3,VCP4,VCP5)
  
  
myeloid.markers.marker.mat <- make_matrix(select(myeloid.markers.marker,-gene_name), pull(myeloid.markers.marker,gene_name))

celltype.markers.marker$gene_name[celltype.markers.marker$gene_name %in% myeloid.markers]

col_fun = colorRamp2(c(0, 16, 32), c("blue", "white", "red"))
lgd = Legend(col_fun = colorRamp2(c(0, 16, 32), c("blue", "white", "red")))

myeloid.markers.massspec.heatmap <- Heatmap(t(myeloid.markers.marker.mat), 
          heatmap_legend_param = list(title = "", at = c(0,16,32)),
          col = colorRamp2(c(0, 16, 32), c("blue", "white", "red")),
          cluster_columns = FALSE, cluster_column_slices = FALSE, cluster_rows = FALSE, border = TRUE, 
          # row_order = myeloid.markers.df$gene_name,
          column_names_gp = gpar(fontsize = 8), row_names_gp = gpar(fontsize = 8), #, #, col = c(rep("red", 10), rep("blue", 8)))
          bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = "white"), labels = "Myeloid markers", labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          column_title = "Proteomics", column_title_gp = gpar(fontsize = 8), row_title = NULL, row_names_rot = 0, row_names_centered = TRUE, row_split =  c(1,1,1,1,1,2,2,2,2,2))
# myeloid.markers.massspec.heatmap
# ggsave(as.ggplot(myeloid.markers.massspec.heatmap), filename = here(proj_path, "expression/mass-spectrometry/myeloid.markers.massspec.heatmap.png", height = 5, width = 10)

# celltype.markers.rnaseq.heatmap %v% myeloid.markers.massspec.heatmap

# ### all cell type markers
# astrocyte.markers = c("ALDH1L1", "AQP4", "GFAP","GJA1", "SOX9", "EDNRB", "DIO2", "FGFR3", "MLC1", "SLC4A4") #  "SLC1A3", 
# endo.markers = c("CLDN5",  "ESAM", "ICAM2", "APOLD1", "NOSTRIN", "ADGRL4") # "PECAM1",
# myeloid.markers = c("AIF1", "C1QA", "CSF1R", "ITGAM", "PTPRC", "CEMIP2", "TREM2", "RUNX1", "CD83",  "CCL2", "TLR2",  "MERTK", "HMGB1", "IL1A", "CCL3")
# neuron.markers = c("NPAS4", "NPY", "SST", "CCK", "GAD1", "SYT1", "STMN2", "SYN1") # "ENO2", 
# oligo.markers = c("MBP", "MOG",  "GPR37", "SOX10", "OPALIN", "MOBP", "PLP1") # "CNP", "GALC",
# 
# celltype.markers = c(astrocyte.markers, endo.markers, myeloid.markers, neuron.markers, oligo.markers)
# celltype.markers[!celltype.markers %in% dep.vcp_vs_ctrl$assay$gene_name]
# 
# celltype.markers.df = tibble("gene_name" = celltype.markers) %>% filter(gene_name %in% celltype.markers) %>% 
#   mutate(group = case_when(gene_name %in% astrocyte.markers ~ "astrocyte", gene_name %in% endo.markers ~ "endo", gene_name %in% myeloid.markers ~ "myeloid", gene_name %in% neuron.markers ~ "neuronal", gene_name %in% oligo.markers ~ "oligo"), group = group) %>% arrange(group)
# 
# celltype.markers.marker <- celltype.markers.df %>% left_join(dep.vcp_vs_ctrl$assay) %>% mutate_all(~replace(., is.na(.), 0)) %>%
#   arrange(factor(gene_name, levels = celltype.markers.df$gene_name)) %>% select(-group)
# celltype.markers.marker.mat <- make_matrix(select(celltype.markers.marker,-gene_name), pull(celltype.markers.marker,gene_name))
# 
# celltype.markers.massspec.heatmap <- Heatmap(celltype.markers.marker.mat, 
#           heatmap_legend_param = list(title = ""), 
#           cluster_rows = FALSE, cluster_row_slices = FALSE, cluster_columns = FALSE,
#           row_split = factor(celltype.markers.df$group, levels = unique(celltype.markers.df$group)), border = TRUE, 
#           row_order = celltype.markers.df$gene_name,
#           row_names_gp = gpar(fontsize = 8), column_names_gp = gpar(fontsize = 8), #, #, col = c(rep("red", 10), rep("blue", 8)))
#           right_annotation = rowAnnotation(markers = anno_block(gp = gpar(fill = c("white", "white", "white", "white", "white")), labels = c("Astrocyte", "Endothel", "Myeloid", "Neuronal", "Oligoden"), 
#                                                                 labels_gp = gpar(fontsize = 7), labels_rot = 0)), 
#           row_title = NULL, column_title = NULL, column_names_rot = 0, column_names_centered = TRUE) #, column_split =  c(1,1,1,1,2,2,2))
# celltype.markers.massspec.heatmap
# ggsave(as.ggplot(celltype.markers.massspec.heatmap), filename = here(proj_path, "expression/mass-spectrometry/celltype.markers.massspec.heatmap.png", height = 8, width = 10)



```


## Immunoflurescence quantification

```{r}
microglia_immunoflurescence_ben = read_excel(here(proj_path,"expression/immunoflurescence/microglia_immunoflurescence_ben.xlsx")) %>% mutate(condition = paste0(celltype,"\n",mutation), condition = factor(condition, levels = c("Microglia\nCTRL", "Microglia\nVCP", "Macrophage\nCTRL", "Macrophage\nVCP")))
microglia_immunoflurescence_ben %>% glimpse
# microglia_immunoflurescence_ben %>% my_summarise(value, marker, condition)
# marker  condition          mean_intensity median_intensity sd_intensity Q1_intensity Q3_intensity min_intensity max_intensity sum_intensity n_intensity
#    <fct>   <fct>                       <dbl>            <dbl>        <dbl>        <dbl>        <dbl>         <dbl>         <dbl>         <dbl>       <int>
#  1 IBA1    "Macrophage\nCTRL"           91.7             95.1        10.7          92.5         99.0          68.2          99.2          825.           9
#  2 IBA1    "Macrophage\nVCP"            86.3             96.6        19.0          84.9         97.0          45.3          98.3          604.           7
#  3 IBA1    "Microglia\nCTRL"            92.4             98.0        13.5          90.9         99.4          57.7          99.5          832.           9
#  4 IBA1    "Microglia\nVCP"             95.4             98.1         6.18         96.3         98.3          81.6          98.6          668.           7
#  5 P2RY12  "Macrophage\nCTRL"           64.5             65.5        13.7          59.3         70.5          36.6          85.3          581.           9
#  6 P2RY12  "Macrophage\nVCP"            66.3             63.3        15.6          55.5         76.5          47.4          89.2          464.           7
#  7 P2RY12  "Microglia\nCTRL"            86.1             90.4         9.19         86.1         90.6          64.6          93.1          775.           9
#  8 P2RY12  "Microglia\nVCP"             80.0             81.4         9.58         75.1         85.0          65.5          92.9          560.           7
#  9 TMEM119 "Macrophage\nCTRL"           70.5             79.2        22.1          59.1         85.3          23.8          93.3          635.           9
# 10 TMEM119 "Macrophage\nVCP"            73.9             77.0        21.3          67.0         88.4          27.7          99.5          665.           9
# 11 TMEM119 "Microglia\nCTRL"            85.1             86.2         5.08         81.0         89.4          76.3          90.4          766.           9
# 12 TMEM119 "Microglia\nVCP"             89.5             94.0        10.5          84.3         98.0          67.0          98.9          806.           9

microglia_immunoflurescence.violin = violin_plot(data = microglia_immunoflurescence_ben, color_by = "condition", continuous = "intensity", ylims = c(20,110), cols = brewer.pal(6, "Paired"), ylabel = "% Positive", facet_by = "marker", plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = rep(c(0,103,110,95,0,0),3), tip_length = 0.003) + theme(axis.text.x = element_text(angle = 30, vjust = 0.7, hjust=0.7))
# microglia_immunoflurescence.violin
#  facet   .y.   group1             group2               n1    n2 statistic     p y.position groups        xmin  xmax p.signif
#    <fct>   <chr> <chr>              <chr>             <int> <int>     <dbl> <dbl>      <dbl> <named list> <dbl> <dbl> <chr>   
#  1 IBA1    lfc   "Macrophage\nCTRL" "Macrophage\nVCP"     9     7      38   0.536         99 <chr [2]>        1     2 ns      
#  2 IBA1    lfc   "Macrophage\nCTRL" "Microglia\nCTRL"     9     9      33   0.546         99 <chr [2]>        1     3 ns      
#  3 IBA1    lfc   "Macrophage\nCTRL" "Microglia\nVCP"      9     7      27   0.681         99 <chr [2]>        1     4 ns      
#  4 IBA1    lfc   "Macrophage\nVCP"  "Microglia\nCTRL"     7     9      16   0.114         99 <chr [2]>        2     3 ns      
#  5 IBA1    lfc   "Macrophage\nVCP"  "Microglia\nVCP"      7     7      13   0.165         99 <chr [2]>        2     4 ns      
#  6 IBA1    lfc   "Microglia\nCTRL"  "Microglia\nVCP"      9     7      34   0.837         99 <chr [2]>        3     4 ns      
#  7 P2RY12  lfc   "Macrophage\nCTRL" "Macrophage\nVCP"     9     7      32   1             99 <chr [2]>        1     2 ns      
#  8 P2RY12  lfc   "Macrophage\nCTRL" "Microglia\nCTRL"     9     9       6   0.001         99 <chr [2]>        1     3 **      
#  9 P2RY12  lfc   "Macrophage\nCTRL" "Microglia\nVCP"      9     7      11   0.031         99 <chr [2]>        1     4 *       
# 10 P2RY12  lfc   "Macrophage\nVCP"  "Microglia\nCTRL"     7     9       7   0.008         99 <chr [2]>        2     3 **      
# 11 P2RY12  lfc   "Macrophage\nVCP"  "Microglia\nVCP"      7     7      12   0.128         99 <chr [2]>        2     4 ns      
# 12 P2RY12  lfc   "Microglia\nCTRL"  "Microglia\nVCP"      9     7      44   0.21          99 <chr [2]>        3     4 ns      
# 13 TMEM119 lfc   "Macrophage\nCTRL" "Macrophage\nVCP"     9     9      35   0.666         99 <chr [2]>        1     2 ns      
# 14 TMEM119 lfc   "Macrophage\nCTRL" "Microglia\nCTRL"     9     9      22   0.113         99 <chr [2]>        1     3 ns      
# 15 TMEM119 lfc   "Macrophage\nCTRL" "Microglia\nVCP"      9     9      15   0.024         99 <chr [2]>        1     4 *       
# 16 TMEM119 lfc   "Macrophage\nVCP"  "Microglia\nCTRL"     9     9      26   0.222         99 <chr [2]>        2     3 ns      
# 17 TMEM119 lfc   "Macrophage\nVCP"  "Microglia\nVCP"      9     9      19.5 0.07          99 <chr [2]>        2     4 ns      
# 18 TMEM119 lfc   "Microglia\nCTRL"  "Microglia\nVCP"      9     9      24   0.161         99 <chr [2]>        3     4 ns   

```

## Merge

Figure 1:
A - Microglia hiPSC protocol schematic
B - PCA hiPSC vs other models/cell types
C - Heatmap RNAseq cell specific markers
D - Heatmap proteomics cell type specific markers
E - IBA1 IF image+quantification
F - P2RY12 IF image+quantification
G - TMEM119 IF image+quantification
H - Morphology quantification


```{r}
fig1.merged <- plot_grid(
  Fig1A_MG_schematic.gg,
  clarke_zhang_reich_abud_ac_mn.pca, 
  plot_grid(as.ggplot(celltype.markers.rnaseq.heatmap), as.ggplot(myeloid.markers.massspec.heatmap), ncol = 2, labels = c("C","D"), rel_widths = c(3,1)),
  plot_grid(
      plot_grid(NULL, Fig1Ei_IBA1image.gg, NULL, Fig1Fi_P2RY12image.gg, NULL, Fig1Gi_TMEM119image.gg, ncol = 6, rel_widths = c(0.1,1,0.1,1,0.1,1), labels = c("E","","F","","G","")),
      microglia_immunoflurescence.violin, nrow = 2, rel_heights = c(1,1)),
  nrow = 4, labels = c("A", "B", "", ""), rel_heights = c(0.7,1.9,0.9,1.7))
# fig1.merged
ggsave(fig1.merged, filename = here(proj_path, "expression/deseq2/fig1.merged.characterisation.png"), height = 14, width = 10.5)
ggsave(fig1.merged, filename = here(proj_path, "expression/deseq2/fig1.merged.characterisation.pdf"), height = 14, width = 10.5)

```

# Table S3-4: Untx VCP vs CTRL GE & MS

```{r}
table.S1_untx_vcp_vs_ctrl_ge = untx_vcp_vs_ctrl$res %>% select(gene_name, gene_id, baseMean, log2FoldChange, pvalue, padj, stat)%>% filter(padj < 0.05)
write_csv(table.S1_untx_vcp_vs_ctrl_ge, here(proj_path, "expression/deseq2/table.S1_untx_vcp_vs_ctrl_ge.csv"))
library(googlesheets4)
gs4_browse("https://docs.google.com/spreadsheets/d/1CVDodcgGdB2rDqnDtfGWMUMyb-BdNFmfzyAX0F-bQqY/edit#gid=0")
write_sheet(table.S1_untx_vcp_vs_ctrl_ge, ss = "https://docs.google.com/spreadsheets/d/1CVDodcgGdB2rDqnDtfGWMUMyb-BdNFmfzyAX0F-bQqY/edit#gid=0", sheet = "S3. VCP RNAseq")
sheet_properties("https://docs.google.com/spreadsheets/d/1CVDodcgGdB2rDqnDtfGWMUMyb-BdNFmfzyAX0F-bQqY/edit#gid=0")


table.S2_untx_vcp_vs_ctrl_mass_spec = dep.vcp_vs_ctrl$res %>% select(gene_name, baseMean, log2FoldChange, stat, pvalue, padj, B)%>% filter(padj < 0.05)
write_csv(table.S2_untx_vcp_vs_ctrl_mass_spec, here(proj_path, "expression/deseq2/table.S2_untx_vcp_vs_ctrl_mass_spec.csv"))
write_sheet(table.S2_untx_vcp_vs_ctrl_mass_spec, ss = "https://docs.google.com/spreadsheets/d/1CVDodcgGdB2rDqnDtfGWMUMyb-BdNFmfzyAX0F-bQqY/edit#gid=0", sheet = "S4. VCP mass spec")

```

# Fig 2: Untx VCP vs CTRL

A - Volcano plot DEGs VCP microglia
B - GO terms DEGs VCP microglia
C - GSEA lysosomal lumen VCP microglia
D - Volcano plot DEPs VCP microglia
E - GO terms DEPs VCP microglia

## Volcano

All samples with isogenic adjustment

```{r}
untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, gene_name == "GPNMB") #183
untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, stat > 0) # 110

untx_vcp_vs_ctrl.labels <- untx_vcp_vs_ctrl$res  %>% filter(-log10(padj) > 2, gene_name %in% all.reactivity.go ) %>% pull(gene_name)
untx_vcp_vs_ctrl.volcano <- volcano_plot(untx_vcp_vs_ctrl$res, gene_labels = untx_vcp_vs_ctrl.labels, ymax = 10, xmax = 5, title = "RNA sequencing", numerator = "VCP", denomenator = "CTRL")
# untx_vcp_vs_ctrl.volcano

untx_vcp_vs_ctrl$res  %>% filter(padj < 0.05, gene_name %in% ALS_genes ) %>% pull(gene_name)
untx_vcp_vs_ctrl$res%>% filter(padj < 0.05, gene_name %in% ALS_genes) # PFN1, MAP2K6

untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, gene_name %in% all.microglia.go)
untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, gene_name %in% microglia_signature)
untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, gene_name %in% RNA_BINDING_PROTEINS)

```

## GO terms

```{r}
untx_vcp_vs_ctrl.gost <- untx_vcp_vs_ctrl$res %>% filter(padj < 0.05) %>% select(gene_name, gene_id, log2FoldChange, padj, pvalue) %>% 
  mutate(deg = case_when(padj < 0.05 & log2FoldChange > 0 ~ "VCP Up", padj < 0.05 & log2FoldChange < 0 ~ "VCP Down", TRUE ~ "None")) %>% 
  group_split(deg) %>% map("gene_id") %>% gost()
untx_vcp_vs_ctrl.gost_filt <- untx_vcp_vs_ctrl.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM|WP")) %>% arrange( -log10(p_value) ) %>% 
  mutate(query = case_when(query == "query_1" ~ "VCP\n Down", query == "query_2" ~ "VCP\n Up"), query = factor(query, levels = c("VCP\n Up", "VCP\n Down")))
table(untx_vcp_vs_ctrl.gost_filt$query)
# VCP\n Up VCP\n Down 
#          2         11 
untx_vcp_vs_ctrl.dge_gprofiles_down <- untx_vcp_vs_ctrl.gost_filt %>% filter(query == "VCP\n Down")
untx_vcp_vs_ctrl.dge_gprofiles_up <- untx_vcp_vs_ctrl.gost_filt %>% filter(query == "VCP\n Up")
paste('"',unique(untx_vcp_vs_ctrl.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list = c(
# "Viral protein interaction with cytokine and cytokine receptor",
"Chemokine receptors bind chemokines",
# "regulation of macromolecule metabolic process",
# "ERK1 and ERK2 cascade",
"negative regulation by host of viral transcription",
# "response to organic substance",
# "cellular response to organic substance",
"Parkin-Ubiquitin Proteasomal System pathway",
"Interleukin-10 signaling",
# "Oncostatin M Signaling Pathway",
"chemokine activity",
# "regulation of metabolic process",
# "Lung fibrosis",
"CCR chemokine receptor binding",
"natural killer cell chemotaxis",
"cellular response to chemical stimulus",
"regulation of natural killer cell chemotaxis",
"positive regulation of metabolic process")
untx_vcp_vs_ctrl.dge_gprofiles_filt_down <- untx_vcp_vs_ctrl.gost_filt %>% filter(query == "VCP\n Down") %>% filter(term_name %in% down_list)
# untx_vcp_vs_ctrl.dge_gprofiles_filt_down$term_name = gsub("Viral protein interaction with cytokine and cytokine receptor", "Viral interaction with cytokine receptor", untx_vcp_vs_ctrl.dge_gprofiles_filt_down$term_name) %>% gsub("E. coli ", "", .) %>% gsub("Absence of bactericidal oxidative respiratory burst in phagocytes", "Absent respiratory burst in phagocytes", .) %>%
  # gsub("Cross-presentation of particulate exogenous antigens [(]phagosomes[)]", "Phagosome antigen cross-presentation", .)
paste('"',unique(untx_vcp_vs_ctrl.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list = c(
"FOXO-mediated transcription of cell cycle genes",
"Transcription factor regulation in adipogenesis",
"Insulin resistance",
"Viral Acute Myocarditis",
"phosphatidylinositol 3-kinase binding")
untx_vcp_vs_ctrl.dge_gprofiles_filt_up <- untx_vcp_vs_ctrl.gost_filt %>% filter(query == "VCP\n Up")  %>% filter(term_name %in% up_list)
untx_vcp_vs_ctrl.dge_gprofiles_filtered = bind_rows(untx_vcp_vs_ctrl.dge_gprofiles_filt_up, untx_vcp_vs_ctrl.dge_gprofiles_filt_down) %>% mutate(query = fct_relevel(query,"VCP\n Up"))
untx_vcp_vs_ctrl_dge.go_curated <- curated_profiles(untx_vcp_vs_ctrl.dge_gprofiles_filtered, cols = c("firebrick2", "dodgerblue2"), label_angle = 0)# + theme(strip.text.y.right = element_text(angle = 0))# + theme(legend.position = "none", axis.text = element_text(size = 10))
# untx_vcp_vs_ctrl_dge.go_curated

untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_PHOSPHATIDYLINOSITOL_3_KINASE_BINDING)
untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_NATURAL_KILLER_CELL_CHEMOTAXIS)
untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_INTERLEUKIN_10_SECRETION)

untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`NK T cell activation`)
# wikipathways.msigdb <- gmtPathways(here(nemo_path,"/genomes/gene-ontology/c2.cp.wikipathways.v7.5.1.symbols.gmt"))# %>% mutate(term = gsub("GO|_", " ", term))
# hpo.msigdb <- gmtPathways(here(nemo_path,"/genomes/gene-ontology/c5.hpo.v7.5.1.symbols.gmt"))# %>% mutate(term = gsub("GO|_", " ", term))
# hpo.msigdb <- gmtPathways(here(nemo_path,"/genomes/gene-ontology/c5.hpo.v7.5.1.symbols.gmt"))# %>% mutate(term = gsub("GO|_", " ", term))


```

## GSEA

```{r}
untx_vcp_vs_ctrl.geneList <- untx_vcp_vs_ctrl$res %>% drop_na(stat) %>% pull(stat)
names(untx_vcp_vs_ctrl.geneList) <- untx_vcp_vs_ctrl$res %>% drop_na(stat) %>% pull(gene_name) %>% as.character()
untx_vcp_vs_ctrl.geneList = sort(untx_vcp_vs_ctrl.geneList, decreasing = TRUE)

# GSEA GO_LYSOSOMAL_LUMEN
GO_LYSOSOMAL_LUMEN_genes.list <- list("GO_LYSOSOMAL_LUMEN" = untx_vcp_vs_ctrl$res$gene_name[untx_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_LYSOSOMAL_LUMEN])
untx_vcp_vs_ctrl.GO_LYSOSOMAL_LUMEN_fgseaRes <- fgsea(pathways = GO_LYSOSOMAL_LUMEN_genes.list, stats = untx_vcp_vs_ctrl.geneList)
untx_vcp_vs_ctrl.GO_LYSOSOMAL_LUMEN_fgseaRes
# pathway         pval         padj   log2err        ES      NES size                            leadingEdge
# 1: GO_LYSOSOMAL_LUMEN 9.370255e-07 9.370255e-07 0.6594444 0.5466541 2.108624   88 RNASET2,ASAH1,MANBA,SGSH,HPSE,HEXB,...
untx_vcp_vs_ctrl.GO_LYSOSOMAL_LUMEN.gseaplot = plotEnrichment(GO_LYSOSOMAL_LUMEN_genes.list[["GO_LYSOSOMAL_LUMEN"]], untx_vcp_vs_ctrl.geneList) + 
  theme(axis.text=element_text(size=8), axis.title=element_text(size=10)) + ggtitle("Lysosomal Lumen GSEA") + xlab(expression(VCP~vs~CTRL)) +
  annotate("text", x = 20000, y = 0.4, label = paste0("NES = ", round(untx_vcp_vs_ctrl.GO_LYSOSOMAL_LUMEN_fgseaRes$NES, digits = 3),"\nP = ", round(untx_vcp_vs_ctrl.GO_LYSOSOMAL_LUMEN_fgseaRes$pval, digits = 3)), size = 3, hjust = 0) + theme_oz()
# untx_vcp_vs_ctrl.GO_LYSOSOMAL_LUMEN.gseaplot


# multi_gsea_list <- list("GO_LYSOSOMAL_LUMEN_ACIDIFICATION" = untx_vcp_vs_ctrl$res$gene_name[untx_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_LYSOSOMAL_LUMEN_ACIDIFICATION],
#                         "GO_LYSOSOMAL_LUMEN" = untx_vcp_vs_ctrl$res$gene_name[untx_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_LYSOSOMAL_LUMEN],
#                         "GO_LYSOSOMAL_PROTEIN_CATABOLIC_PROCESS" = untx_vcp_vs_ctrl$res$gene_name[untx_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_LYSOSOMAL_PROTEIN_CATABOLIC_PROCESS],
#                         "GO_SECRETION_OF_LYSOSOMAL_ENZYMES" = untx_vcp_vs_ctrl$res$gene_name[untx_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_SECRETION_OF_LYSOSOMAL_ENZYMES],
#                         "GO_CELLULAR_SENESCENCE" = untx_vcp_vs_ctrl$res$gene_name[untx_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_CELLULAR_SENESCENCE]
#                         )
# # ,
# #                         "GO_MITOCHONDRIAL_GENE_EXPRESSION" = untx_vcp_vs_ctrl$res$gene_name[untx_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_MITOCHONDRIAL_GENE_EXPRESSION],
# #                         "GO_REGULATION_OF_MITOCHONDRIAL_ATP_SYNTHESIS_COUPLED_ELECTRON_TRANSPORT" = untx_vcp_vs_ctrl$res$gene_name[untx_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_REGULATION_OF_MITOCHONDRIAL_ATP_SYNTHESIS_COUPLED_ELECTRON_TRANSPORT],
# #                         "GO_CHAPERONE_MEDIATED_AUTOPHAGY" = untx_vcp_vs_ctrl$res$gene_name[untx_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_CHAPERONE_MEDIATED_AUTOPHAGY],
# #                         "GO_MACROAUTOPHAGY" = untx_vcp_vs_ctrl$res$gene_name[untx_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_MACROAUTOPHAGY],
# #                         "GO_REGULATION_OF_AUTOPHAGY" = untx_vcp_vs_ctrl$res$gene_name[untx_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_REGULATION_OF_AUTOPHAGY],
# #                         "GO_MICROGLIAL_CELL_MIGRATION" = untx_vcp_vs_ctrl$res$gene_name[untx_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_MICROGLIAL_CELL_MIGRATION],
# #                         "GO_MICROGLIAL_CELL_ACTIVATION" = untx_vcp_vs_ctrl$res$gene_name[untx_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_MICROGLIAL_CELL_ACTIVATION],
# #                         "GO_REGULATION_OF_MICROGLIAL_CELL_ACTIVATION" = untx_vcp_vs_ctrl$res$gene_name[untx_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_REGULATION_OF_MICROGLIAL_CELL_ACTIVATION])
# untx_vcp_vs_ctrl.multi_fgseaRes <- fgsea(pathways = multi_gsea_list, stats = untx_vcp_vs_ctrl.geneList)
# untx_vcp_vs_ctrl.multi_fgseaRes
#                                  pathway       pval      padj    log2err         ES       NES size                      leadingEdge
# 1:       GO_LYSOSOMAL_LUMEN_ACIDIFICATION 0.01802717 0.0540815 0.35248786  0.7114219  1.606407    8     CLN3,CLN5,CCDC115,SNAPIN,GRN
# 2: GO_LYSOSOMAL_PROTEIN_CATABOLIC_PROCESS 0.39355993 0.3935599 0.08603472 -0.4448648 -1.072370   11                LDLR,LAPTM4B,LRP2
# 3:      GO_SECRETION_OF_LYSOSOMAL_ENZYMES 0.07359307 0.1103896 0.24504179  0.7018507  1.471695    6 GNPTAB,NAGPA,BLOC1S3,NR1H2,NR1H3

#                                                                    pathway         pval         padj    log2err         ES        NES size
# 1:                                         GO_CHAPERONE_MEDIATED_AUTOPHAGY 0.8302277433 0.8302277433 0.05582647  0.2746280  0.7357766   16
# 2:                                                       GO_MACROAUTOPHAGY 0.0085628855 0.0154131939 0.38073040  0.3045399  1.3647876  291
# 3:                                            GO_MICROGLIAL_CELL_MIGRATION 0.4354838710 0.4899193548 0.08705159 -0.4859035 -1.0486338    8
# 4:                                        GO_MITOCHONDRIAL_GENE_EXPRESSION 0.0000000001 0.0000000009         NA -0.5169575 -2.2042999  165
# 5:                                                         GO_PHAGOCYTOSIS 0.0070638143 0.0154131939 0.40701792 -0.3016557 -1.3712922  265
# 6:              GO_POSITIVE_REGULATION_OF_MITOCHONDRIAL_MEMBRANE_POTENTIAL 0.0009349657 0.0028048972 0.47727082 -0.8164475 -1.8543792    9
# 7:                                              GO_REGULATION_OF_AUTOPHAGY 0.0001242117 0.0005589529 0.51884808  0.3356496  1.5247087  319
# 8:                             GO_REGULATION_OF_MICROGLIAL_CELL_ACTIVATION 0.3643892340 0.4685004437 0.09889030  0.3941282  1.0826758   18
# 9: GO_REGULATION_OF_MITOCHONDRIAL_ATP_SYNTHESIS_COUPLED_ELECTRON_TRANSPORT 0.2147117296 0.3220675944 0.13145761 -0.5931868 -1.2543411    7

# GOBP_LYSOSOMAL_LUMEN_ACIDIFICATION
# GOBP_LYSOSOMAL_MICROAUTOPHAGY
# GOBP_LYSOSOMAL_PROTEIN_CATABOLIC_PROCESS
# GOBP_SECRETION_OF_LYSOSOMAL_ENZYMES

```

## Mass spec

```{r}
dep.vcp_vs_ctrl$res %>% filter(padj < 0.05) #101
dep.vcp_vs_ctrl$res %>% filter(padj < 0.05, stat > 0) #51

# Volcano
dep.vcp_vs_ctrl.volcano <- volcano_plot(dep.vcp_vs_ctrl$res, title = "Proteomics", gene_labels = pull(filter(dep.vcp_vs_ctrl$res, padj < 0.01, gene_name %in% all.reactivity.go), gene_name), ymax = 8, xmax = 5, numerator = "VCP", denomenator = "CTRL") 
# dep.vcp_vs_ctrl.volcano

dep.vcp_vs_ctrl$res$sig.direction %>% table
# downregulated     unchanged   upregulated 
#             6          2817            20 

# GO terms mass spec
patani.mass_spec.dge_genes <- dep.vcp_vs_ctrl$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_name") %>% gost() # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
patani.mass_spec.dge_gprofiles_filt <- patani.mass_spec.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM|HPA")) %>% arrange( -log10(p_value) ) %>% 
  mutate(query = case_when(query == "query_1" ~ "VCP\n Down", query == "query_2" ~ "VCP\n Up"), term_name = as.factor(term_name)) 
patani.mass_spec.dge_gprofiles_down <- patani.mass_spec.dge_gprofiles_filt %>% filter(query == "VCP\n Down")
patani.mass_spec.dge_gprofiles_up <- patani.mass_spec.dge_gprofiles_filt %>% filter(query == "VCP\n Up")
paste('"',unique(patani.mass_spec.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list = c(
# "MHC protein binding",
# "actomyosin",
# "focal adhesion",
# "stress fiber",
# "contractile actin filament bundle",
# "laminin-10 complex",
# "cytoplasm",
"Focal adhesion",
"endoplasmic reticulum",
"Laminin interactions",
"laminin complex",
"actin cytoskeleton")
# "laminin-1 complex")
# "vesicle")
patani.mass_spec.dge_gprofiles_filt_down <- patani.mass_spec.dge_gprofiles_filt %>% filter(query == "VCP\n Down") %>% filter(term_name %in% down_list)
paste('"',unique(patani.mass_spec.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list = c(
"cell death",
"extracellular matrix",
# "primary lysosome",
"endoplasmic reticulum",
"secretory granule lumen",
"Innate Immune System",
# "secretory vesicle",
"secretory granule",
"lysosome",
"lysosomal lumen")
# "extracellular vesicle",
# "extracellular exosome")
patani.mass_spec.dge_gprofiles_filt_up <- patani.mass_spec.dge_gprofiles_filt %>% filter(query == "VCP\n Up")  %>% filter(term_name %in% up_list) %>% mutate(term_name = gsub("myeloid cell activation involved in immune response", "myeloid activation in immune response", term_name))
patani.mass_spec.dge_gprofiles_filtered = bind_rows(patani.mass_spec.dge_gprofiles_filt_up, patani.mass_spec.dge_gprofiles_filt_down) %>% mutate(query = fct_relevel(query,"VCP\n Up"))
vcp_vs_ctrl_mass_spec.go_curated <- curated_profiles(patani.mass_spec.dge_gprofiles_filtered, gprofiles =, cols = c("firebrick2", "dodgerblue2")) + theme(strip.text.y.right = element_text(angle = 0))# + theme(legend.position = "none", axis.text = element_text(size = 10))
# vcp_vs_ctrl_mass_spec.go_curated


# what genes are driving the terms
dep.vcp_vs_ctrl$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_LYSOSOMAL_LUMEN) 
dep.vcp_vs_ctrl$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_SECRETORY_GRANULE) 
dep.vcp_vs_ctrl$res %>% filter(padj < 0.05, gene_name %in% reactome.pathways.msigdb$REACTOME_INNATE_IMMUNE_SYSTEM) 
  


# GSEA phagocytosis mass spec
massspec_vcp_vs_ctrl.geneList <- dep.vcp_vs_ctrl$res  %>% drop_na(log2FoldChange) %>% pull(log2FoldChange)
names(massspec_vcp_vs_ctrl.geneList) <- dep.vcp_vs_ctrl$res  %>% drop_na(log2FoldChange) %>% pull(gene_name) %>% as.character()
massspec_vcp_vs_ctrl.geneList = sort(massspec_vcp_vs_ctrl.geneList, decreasing = TRUE)
phagocytosis_genes.list <- list("phagocytosis" = dep.vcp_vs_ctrl$res$gene_name[dep.vcp_vs_ctrl$res $gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS])
massspec_vcp_vs_ctrl.phagocytosis_fgseaRes <- fgsea(pathways = phagocytosis_genes.list, stats = massspec_vcp_vs_ctrl.geneList)
massspec_vcp_vs_ctrl.phagocytosis_fgseaRes
#         pathway      pval      padj    log2err        ES       NES size                        leadingEdge
# 1: phagocytosis 0.9562738 0.9562738 0.04540592 0.2262784 0.7553406  107 ATG5,TLR2,IGHG1,MSR1,IGKC,CD14,...
massspec_vcp_vs_ctrl.phagocytosis.gseaplot = plotEnrichment(phagocytosis_genes.list[["phagocytosis"]], massspec_vcp_vs_ctrl.geneList) +
  theme(axis.text=element_text(size=8), axis.title=element_text(size=10)) + ylab(expression(Phagocytosis)) + xlab(expression(Untreated~VCP~vs~CTRL)) +
  annotate("text", x = 100, y = -0.15, label = paste0("NES = ", round(massspec_vcp_vs_ctrl.phagocytosis_fgseaRes$NES, digits = 3),"\nP = ", round(massspec_vcp_vs_ctrl.phagocytosis_fgseaRes$pval, digits = 3)), size = 3, hjust = 0)
massspec_vcp_vs_ctrl.phagocytosis.gseaplot

# GSEA autophagy mass spec
autophagy.go = unique(c(go.pathways.msigdb$GO_REGULATION_OF_AUTOPHAGY, go.pathways.msigdb$GO_MACROAUTOPHAGY, go.pathways.msigdb$GO_CHAPERONE_MEDIATED_AUTOPHAGY))
massspec_autophagy_genes.list <- list("autophagy" = dep.vcp_vs_ctrl$res$gene_name[dep.vcp_vs_ctrl$res$gene_name %in% autophagy.go])
massspec_untx_vcp_vs_ctrl.autophagy_fgseaRes <- fgsea(pathways = massspec_autophagy_genes.list, stats = massspec_vcp_vs_ctrl.geneList)
massspec_untx_vcp_vs_ctrl.autophagy_fgseaRes
#      pathway      pval      padj    log2err         ES        NES size                                 leadingEdge
# 1: autophagy 0.5601751 0.5601751 0.07788401 -0.2725893 -0.9647065  137 ATG3,TOMM40,ZMPSTE24,VPS26B,VPS28,RAB5A,...
massspec_untx_vcp_vs_ctrl.autophagy.gseaplot = plotEnrichment(massspec_autophagy_genes.list[["autophagy"]], massspec_vcp_vs_ctrl.geneList) + 
  theme(axis.text=element_text(size=8), axis.title=element_text(size=10)) + ylab(expression(Autophagy~GSEA)) + xlab(expression(VCP~vs~CTRL)) +
  annotate("text", x = 23000, y = 0.26, label = paste0("NES = ", round(massspec_untx_vcp_vs_ctrl.autophagy_fgseaRes$NES, digits = 3),"\nP = ", round(massspec_untx_vcp_vs_ctrl.autophagy_fgseaRes$pval, digits = 3)), size = 3, hjust = 0)
massspec_untx_vcp_vs_ctrl.autophagy.gseaplot

# GSEA mitochondrial mass spec
mitochondrial.go = unique(c(go.pathways.msigdb$GO_POSITIVE_REGULATION_OF_MITOCHONDRIAL_MEMBRANE_POTENTIAL, go.pathways.msigdb$GO_MITOCHONDRIAL_GENE_EXPRESSION, go.pathways.msigdb$GO_REGULATION_OF_MITOCHONDRIAL_ATP_SYNTHESIS_COUPLED_ELECTRON_TRANSPORT))
massspec_mitochondrial_genes.list <- list("mitochondrial" = dep.vcp_vs_ctrl$res$gene_name[dep.vcp_vs_ctrl$res$gene_name %in% mitochondrial.go])
massspec_untx_vcp_vs_ctrl.mitochondrial_fgseaRes <- fgsea(pathways = massspec_mitochondrial_genes.list, stats = massspec_vcp_vs_ctrl.geneList)
massspec_untx_vcp_vs_ctrl.mitochondrial_fgseaRes
#         pathway      pval      padj    log2err        ES      NES size                        leadingEdge
# 1: mitochondrial 0.6583493 0.6583493 0.06321912 0.3127872 0.882755   34 MRPS14,MRPS21,MRPL12,MRPS34,MRPL49
massspec_untx_vcp_vs_ctrl.mitochondrial.gseaplot = plotEnrichment(massspec_mitochondrial_genes.list[["mitochondrial"]], massspec_vcp_vs_ctrl.geneList) + 
  theme(axis.text=element_text(size=8), axis.title=element_text(size=10)) + ylab(expression(Mitochondrial~GSEA)) + xlab(expression(VCP~vs~CTRL)) +
  annotate("text", x = 1000, y = -0.35, label = paste0("NES = ", round(massspec_untx_vcp_vs_ctrl.mitochondrial_fgseaRes$NES, digits = 3),"\nP = ", round(massspec_untx_vcp_vs_ctrl.mitochondrial_fgseaRes$pval, digits = 3)), size = 3, hjust = 0)
massspec_untx_vcp_vs_ctrl.mitochondrial.gseaplot

# GSEA phagocytosis RNAseq
untx_vcp_vs_ctrl.geneList <- untx_vcp_vs_ctrl$res %>% drop_na(stat) %>% pull(stat)


names(untx_vcp_vs_ctrl.geneList) <- untx_vcp_vs_ctrl$res %>% drop_na(stat) %>% pull(gene_name) %>% as.character()
untx_vcp_vs_ctrl.geneList = sort(untx_vcp_vs_ctrl.geneList, decreasing = TRUE)
phagocytosis_genes.list <- list("phagocytosis" = untx_vcp_vs_ctrl$res$gene_name[untx_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS])
untx_vcp_vs_ctrl.phagocytosis_fgseaRes <- fgsea(pathways = phagocytosis_genes.list, stats = untx_vcp_vs_ctrl.geneList)
untx_vcp_vs_ctrl.phagocytosis_fgseaRes
# pathway       pval       padj   log2err         ES       NES size                         leadingEdge
# 1: phagocytosis 0.00961924 0.00961924 0.3807304 -0.3016557 -1.366053  265 FCGR1A,CCL2,PLD4,IL1B,TNF,ARPC2,...
mass_spec_untx_vcp_vs_ctrl.phagocytosis.gseaplot = plotEnrichment(phagocytosis_genes.list[["phagocytosis"]], untx_vcp_vs_ctrl.geneList) + 
  theme(axis.text=element_text(size=8), axis.title=element_text(size=10)) + ylab(expression(Phagocytosis~GSEA)) + xlab(expression(VCP~vs~CTRL)) +
  annotate("text", x = 100, y = -0.26, label = paste0("NES = ", round(untx_vcp_vs_ctrl.phagocytosis_fgseaRes$NES, digits = 3),"\nP = ", round(untx_vcp_vs_ctrl.phagocytosis_fgseaRes$pval, digits = 3)), size = 3, hjust = 0)
mass_spec_untx_vcp_vs_ctrl.phagocytosis.gseaplot

dep.vcp_vs_ctrl$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS)
dep.vcp_vs_ctrl$res %>% filter(MS.pvalue < 0.05, gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS)


dep.vcp_vs_ctrl$res %>% filter(gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS, log2FoldChange < 0) # 58

mass_spec_multi_gsea_list <- list("GO_PHAGOCYTOSIS" = dep.vcp_vs_ctrl$res$gene_name[dep.vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS],
                        "GO_POSITIVE_REGULATION_OF_MITOCHONDRIAL_MEMBRANE_POTENTIAL" = dep.vcp_vs_ctrl$res$gene_name[dep.vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_POSITIVE_REGULATION_OF_MITOCHONDRIAL_MEMBRANE_POTENTIAL],
                        "GO_MITOCHONDRIAL_GENE_EXPRESSION" = dep.vcp_vs_ctrl$res$gene_name[dep.vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_MITOCHONDRIAL_GENE_EXPRESSION],
                        "GO_REGULATION_OF_MITOCHONDRIAL_ATP_SYNTHESIS_COUPLED_ELECTRON_TRANSPORT" = dep.vcp_vs_ctrl$res$gene_name[dep.vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_REGULATION_OF_MITOCHONDRIAL_ATP_SYNTHESIS_COUPLED_ELECTRON_TRANSPORT],
                        "GO_CHAPERONE_MEDIATED_AUTOPHAGY" = dep.vcp_vs_ctrl$res$gene_name[dep.vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_CHAPERONE_MEDIATED_AUTOPHAGY],
                        "GO_MACROAUTOPHAGY" = dep.vcp_vs_ctrl$res$gene_name[dep.vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_MACROAUTOPHAGY],
                        "GO_REGULATION_OF_AUTOPHAGY" = dep.vcp_vs_ctrl$res$gene_name[dep.vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_REGULATION_OF_AUTOPHAGY],
                        "GO_MICROGLIAL_CELL_MIGRATION" = dep.vcp_vs_ctrl$res$gene_name[dep.vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_MICROGLIAL_CELL_MIGRATION],
                        "GO_MICROGLIAL_CELL_ACTIVATION" = dep.vcp_vs_ctrl$res$gene_name[dep.vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_MICROGLIAL_CELL_ACTIVATION],
                        "GO_REGULATION_OF_MICROGLIAL_CELL_ACTIVATION" = dep.vcp_vs_ctrl$res$gene_name[dep.vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_REGULATION_OF_MICROGLIAL_CELL_ACTIVATION])
mass_spec_untx_vcp_vs_ctrl.multi_fgseaRes <- fgsea(pathways = mass_spec_multi_gsea_list, stats = massspec_vcp_vs_ctrl.geneList)
mass_spec_untx_vcp_vs_ctrl.multi_fgseaRes
#                                                                    pathway      pval      padj    log2err         ES        NES size                               leadingEdge
# 1:                                         GO_CHAPERONE_MEDIATED_AUTOPHAGY 0.9047619 0.9522777 0.05184576  0.3101215  0.6274516    9         LAMP2,BAG3,CTSA,SNCA,EEF1A1,HSPA8
# 2:                                                       GO_MACROAUTOPHAGY 0.7362637 0.9522777 0.06435834 -0.2631507 -0.8988173  110  ATG3,TOMM40,VPS26B,VPS28,RAB5A,MAPK3,...
# 3:                                            GO_MICROGLIAL_CELL_MIGRATION 0.5613027 0.9522777 0.07096095 -0.6705385 -0.9790435    2                                P2RX4,CCL3
# 4:                                        GO_MITOCHONDRIAL_GENE_EXPRESSION 0.5822050 0.9522777 0.06961334  0.3457400  0.9109598   27        MRPS14,MRPS21,MRPL12,MRPS34,MRPL49
# 5:                                                         GO_PHAGOCYTOSIS 0.9478585 0.9522777 0.04486451  0.2273575  0.7592528  107        ATG5,IGHG1,MSR1,TLR2,IGKC,CD14,...
# 6:              GO_POSITIVE_REGULATION_OF_MITOCHONDRIAL_MEMBRANE_POTENTIAL 0.5220729 0.9522777 0.07473852 -0.5935188 -0.9845808    4                        VCP,NNT,BID,STOML2
# 7:                                              GO_REGULATION_OF_AUTOPHAGY 0.9522777 0.9522777 0.05163560 -0.2257787 -0.7557235   94 ZMPSTE24,VPS26B,RAB5A,MAPK3,RHEB,OPTN,...
# 8:                             GO_REGULATION_OF_MICROGLIAL_CELL_ACTIVATION 0.9106029 0.9522777 0.05174055  0.3362465  0.5679958    4                           CTSC,PTPRC,CCL3
# 9: GO_REGULATION_OF_MITOCHONDRIAL_ATP_SYNTHESIS_COUPLED_ELECTRON_TRANSPORT 0.6065259 0.9522777 0.06720651 -0.6029072 -0.9383013    3                                      ISCU


dep.vcp_vs_ctrl$res %>% filter(gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS) %>% mutate(phagocytosis_regulator = case_when(gene_name %in% go.pathways.msigdb$GO_POSITIVE_REGULATION_OF_PHAGOCYTOSIS ~ "positive", 
                                                                                             gene_name %in% gene_name %in% go.pathways.msigdb$GO_NEGATIVE_REGULATION_OF_PHAGOCYTOSIS ~ "negative", TRUE ~ "")) %>%
  write_csv(here(proj_path, "mass-spectrometry/mass_spec.vcp_vs_ctrl.phagocytosis_proteins.csv")
dep.vcp_vs_ctrl$res %>% filter(gene_name %in% go.pathways.msigdb$GO_POSITIVE_REGULATION_OF_PHAGOCYTOSIS) # 24
dep.vcp_vs_ctrl$res %>% filter(gene_name %in% go.pathways.msigdb$GO_NEGATIVE_REGULATION_OF_PHAGOCYTOSIS) # 11
```

## Correlate RNAseq with mass spec

```{r}
untx_vcp_vs_ctrl.rna.protein = select(untx_vcp_vs_ctrl$res, gene_name, rna.stat = stat, rna.padj = padj) %>% 
  left_join(select(dep.vcp_vs_ctrl$res, gene_name, protein.stat = stat, protein.padj = padj))
untx_vcp_vs_ctrl.rna.protein.overlapping = untx_vcp_vs_ctrl.rna.protein %>% filter(rna.padj < 0.05, protein.padj < 0.05, (rna.stat > 0 & protein.stat > 0) | (rna.stat < 0 & protein.stat < 0)) %>% top_n(n = 20, wt = abs(rna.stat)) %>% pull(gene_name) # 50
rna_protein_de_list = list("RNAseq: VCP vs CTRL" = untx_vcp_vs_ctrl$res, "Mass spec: VCP vs CTRL" = dep.vcp_vs_ctrl$res)
untx_vcp_vs_ctrl.rna.protein.scatter = plot_de_correlation(model_list1 = rna_protein_de_list, model_list2 = rna_protein_de_list, mutation1 = "RNAseq: VCP vs CTRL", mutation2 = "Mass spec: VCP vs CTRL", gene_labels = c(untx_vcp_vs_ctrl.rna.protein.overlapping, "CFAP410"), col = "log2FoldChange", join_by = "gene_name") + ylim(c(-4,4)) + xlim(c(-4,4)) +
    annotate(geom="text", x=2, y=4, label="RNA & Protein both Up", color="black", size = 3.2)  + annotate(geom="text", x=-2, y=-4, label="RNA & Protein both Down", color="black", size = 3.2)
untx_vcp_vs_ctrl.rna.protein.scatter

untx_vcp_vs_ctrl.rna.protein %>% filter(protein.padj < 0.05, rna.padj < 0.05)

```


## Merge


```{r}
fig2.merged <- plot_grid(
  plot_grid(untx_vcp_vs_ctrl.volcano, untx_vcp_vs_ctrl_dge.go_curated, ncol=2, rel_widths = c(1,0.8), labels = "AUTO"), 
  plot_grid(untx_vcp_vs_ctrl.GO_LYSOSOMAL_LUMEN.gseaplot, dep.vcp_vs_ctrl.volcano, ncol = 2, rel_widths = c(0.8, 1), labels = c("C","D")), 
  plot_grid(vcp_vs_ctrl_mass_spec.go_curated, untx_vcp_vs_ctrl.rna.protein.scatter, ncol = 2, rel_widths = c(0.8, 1), labels = c("E","F")), 
  nrow = 3, rel_heights = c(1,1,1))
# fig2.merged
ggsave(fig2.merged, filename = here(proj_path, "expression/deseq2/fig2.merged.untx_vcp_vs_ctrl.png"), height = 11, width = 10)
ggsave(fig2.merged, filename = here(proj_path, "expression/deseq2/fig2.merged.untx_vcp_vs_ctrl.pdf"), height = 11, width = 10)


```


# Fig 3: Lysotracker VCP microglia


## Lysotracker & Capthepsin

F - Lysotracker representative image VCP microglia
G - Lysotracker quantification size VCP microglia 
H - Lysotracker quantification intensity VCP microglia 
I - Lysotracker quantification number VCP microglia 
J - Cytokine/Chemokine heatmap VCP microglia 
K - IL-6 barplot VCP microglia 
L - CXCL10 barplot VCP microglia 

A: Replace image with “fig.2A_rebuttal” in “rebuttal” folder on dropbox.
B: Lysosensor mean intensity data from tab "Figure 2H-M”
C: Lysotracker mean intensity data from tab "Figure 2H-M”
D: Lysotracker spot area data from tab "Figure 2H-M”
E: Lysotracker spot number per area data from tab "Figure 2H-M”
F: Add image “fig.3A_rebuttal” folder on dropbox.
G: Pepstatin A (cathepsin D) activity data from tab “Rebuttal_figure_3”.
H: Magic Red (cathepsin B) activity data from tab “Rebuttal_figure_3”.


```{r}
# https://www.dropbox.com/work/PataniR/Ben/Microglia%20figures

# import with magick 
Lyso_image.png = magick::image_read(here(proj_path,"ben-figures/Fig.2A_rebuttal.png"))
Lyso_image.gg = ggdraw() + draw_image(Lyso_image.png)
# Lyso_image.gg

pepstatin_magic_red.png = magick::image_read(here(proj_path,"ben-figures/Fig.3A_rebuttal.png"))
pepstatin_magic_red.gg = ggdraw() + draw_image(pepstatin_magic_red.png)
# pepstatin_magic_red.gg

lysotracker_data = read_excel(here(proj_path, "expression/lysotracker/Rebuttal_data_for Oli.xlsx"), sheet = "Figure_2H-M") %>% clean_names()
# lysotracker_data = read_excel(here(proj_path, "expression/lysotracker/lysotracker_tmrm.xlsx"), sheet = "assays") %>%
  # filter(grepl("Lysotracker",assay))
lysotracker_data %>% glimpse

filter(lysotracker_data, !is.na(lysosensor_mean_intensity)) %>% my_summarise(lysosensor_mean_intensity, genotype) %>% print(n=Inf)
# for stats, adjust for experiment
lysosensor_mean_intensity.fitg = glm(lysosensor_mean_intensity ~ genotype + experiment, data= filter(lysotracker_data, !is.na(lysosensor_mean_intensity)), family = gaussian)
summ(lysosensor_mean_intensity.fitg, digits = 5)$coeftable 
#                     Est.       S.E.     t val.            p
# (Intercept)  1.068747261 0.21341270  5.0078897 5.883147e-05
# genotypeVCP  0.235479619 0.04437919  5.3060818 2.920037e-05
lysosensor_mean_intensity.violin = violin_plot(data = filter(lysotracker_data, !is.na(lysosensor_mean_intensity)), color_by = "genotype", continuous = "lysosensor_mean_intensity", cols = brewer.pal(6, "Paired"), ylabel = "Lysosensor mean intensity\nFold change", ylims = c(0.7,1.5), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 1.5, tip_length = 0.01)
lysosensor_mean_intensity.violin

filter(lysotracker_data, !is.na(lysotracker_mean_intensity)) %>% my_summarise(lysotracker_mean_intensity, genotype) %>% print(n=Inf)
# for stats, adjust for experiment
lysotracker_mean_intensity.fitg = glm(lysotracker_mean_intensity ~ genotype + experiment, data= filter(lysotracker_data, !is.na(lysotracker_mean_intensity)), family = gaussian)
summ(lysotracker_mean_intensity.fitg, digits = 5)$coeftable 
#                    Est.       S.E.     t val.            p
# (Intercept) 0.987816370 0.07295243 13.5405544 1.373180e-20
# genotypeVCP 0.179552307 0.06395454  2.8074989 6.583878e-03
lysotracker_mean_intensity.violin = violin_plot(data = filter(lysotracker_data, !is.na(lysotracker_mean_intensity)), color_by = "genotype", continuous = "lysotracker_mean_intensity", cols = brewer.pal(6, "Paired"), ylabel = "Lysotracker mean intensity\nFold change", ylims = c(0,2), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 2, tip_length = 0.01)
lysotracker_mean_intensity.violin

filter(lysotracker_data, !is.na(lysotracker_spot_area)) %>% my_summarise(lysotracker_spot_area, genotype) %>% print(n=Inf)
lysotracker_spot_area.fitg = glm(lysotracker_spot_area ~ genotype + experiment, data= filter(lysotracker_data, !is.na(lysotracker_spot_area)), family = gaussian)
summ(lysotracker_spot_area.fitg, digits = 5)$coeftable 
#                   Est.       S.E.    t val.            p
# (Intercept)  1.20411091 0.05925510 20.320799 7.323904e-30
# genotypeVCP -0.08583059 0.05194663 -1.652284 1.032996e-01
lysotracker_spot_area.violin = violin_plot(data = filter(lysotracker_data, !is.na(lysotracker_spot_area)), color_by = "genotype", continuous = "lysotracker_spot_area", cols = brewer.pal(6, "Paired"), ylabel = "Lysotracker spot area (um2)", ylims = c(0,2), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 2, tip_length = 0.01)
lysotracker_spot_area.violin

filter(lysotracker_data, !is.na(lysotracker_spot_number_per_area)) %>% my_summarise(lysotracker_spot_number_per_area, genotype) %>% print(n=Inf)
lysotracker_spot_number_per_area.fitg = glm(lysotracker_spot_number_per_area ~ genotype + experiment, data= filter(lysotracker_data, !is.na(lysotracker_spot_number_per_area)), family = gaussian)
summ(lysotracker_spot_number_per_area.fitg, digits = 5)$coeftable 
#                    Est.       S.E.    t val.            p
# (Intercept)  1.029075294 0.03513652 29.287911 3.657225e-39
# genotypeVCP -0.048993708 0.03080281 -1.590559 1.165611e-01
lysotracker_spot_number_per_area.violin = violin_plot(data = filter(lysotracker_data, !is.na(lysotracker_spot_number_per_area)), color_by = "genotype", continuous = "lysotracker_spot_number_per_area", cols = brewer.pal(6, "Paired"), ylabel = "Lysotracker spot number / area", ylims = c(0,1.5), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 1.5, tip_length = 0.01)
lysotracker_spot_number_per_area.violin

# Pepstatin A (Cathepsin D)
filter(lysotracker_data, !is.na(cathepsin_d_activity)) %>% my_summarise(cathepsin_d_activity, genotype) %>% print(n=Inf)
cathepsin_d_activity.fitg = glm(cathepsin_d_activity ~ genotype + experiment, data= filter(lysotracker_data, !is.na(cathepsin_d_activity)), family = gaussian)
summ(cathepsin_d_activity.fitg, digits = 5)$coeftable 
#                    Est.       S.E.     t val.           p
# (Intercept)  1.13323915 0.33673899  3.3653339 0.002926592
# genotypeVCP  0.14813101 0.07002491  2.1154047 0.046517158
cathepsin_d_activity.violin = violin_plot(data = filter(lysotracker_data, !is.na(cathepsin_d_activity)), color_by = "genotype", continuous = "cathepsin_d_activity", cols = brewer.pal(6, "Paired"), ylabel = "Pepstatin A intensity\nFold change from CTRL", ylims = c(0.5,1.7), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 1.7, tip_length = 0.01)
cathepsin_d_activity.violin

# Magic Red (Cathepsin B)
filter(lysotracker_data, !is.na(cathepsin_b_activity)) %>% my_summarise(cathepsin_b_activity, genotype) %>% print(n=Inf)
cathepsin_b_activity.fitg = glm(cathepsin_b_activity ~ genotype + experiment, data= filter(lysotracker_data, !is.na(cathepsin_b_activity)), family = gaussian)
summ(cathepsin_b_activity.fitg, digits = 5)$coeftable 
#                   Est.       S.E.    t val.          p
# (Intercept) 0.56632637 0.31165494 1.8171583 0.08349236
# genotypeVCP 0.04180214 0.06480868 0.6450084 0.52590634
cathepsin_b_activity.violin = violin_plot(data = filter(lysotracker_data, !is.na(cathepsin_b_activity)), color_by = "genotype", continuous = "cathepsin_b_activity", cols = brewer.pal(6, "Paired"), ylabel = "Magic Red intensity\nFold change from CTRL", ylims = c(0.5,1.5), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 1.5, tip_length = 0.01)
cathepsin_b_activity.violin

# fig3.merged <- plot_grid(
#   plot_grid(Fig.2A_rebuttal.gg, pepstatin_magic_red.gg, nrow = 2, labels = c("A","F")),
#   plot_grid(lysosensor_mean_intensity.violin, lysotracker_mean_intensity.violin, lysotracker_spot_area.violin, lysotracker_spot_number_per_area.violin, cathepsin_d_activity.violin, cathepsin_b_activity.violin, nrow = 3, ncol = 2, labels = c("B","C","D","E","G","H")),
#             ncol = 2)
# fig3.merged
# ggsave(fig3.merged, filename = here(proj_path, "expression/deseq2/fig3.merged.lysotracker_cathepsin.png"), height = 8, width = 9.5)
# ggsave(fig3.merged, filename = here(proj_path, "expression/deseq2/fig3.merged.lysotracker_cathepsin.pdf"), height = 8, width = 9.5)


```

## Luminex heatmap

```{r}
luminex_data = read_excel(here(proj_path, "expression/luminex/luminex_data.xlsx"), sheet = "Untreated") %>%
  group_by(batch, cytokine) %>% mutate(mean_batch_intensity = mean(intensity), intensity_batch_normalised = intensity / mean_batch_intensity, intensity_normalised =  intensity_batch_normalised / confluency, intensity_normalised.transformed = intensity_normalised * 1 / mean(intensity_normalised)) %>% ungroup() %>%
  filter(cytokine %in% c("IL1b", "IL10","IL6","IL12","CCL11", "CCL3", "CCL4", "CCL2", "IFNa", "IL1RA","TNFa", "CXCL10","IL8"))   
luminex_data %>% glimpse

luminex_data %>% my_summarise(intensity_normalised.transformed, cytokine, mutation) %>% print(n=Inf)

luminex_data.wide = luminex_data %>% group_by(cytokine, mutation) %>% summarise(intensity = mean(intensity_normalised.transformed)) %>% pivot_wider(names_from = "mutation", values_from = "intensity") %>% #mutate(log2FoldChange = log2(VCP) - log2(CTRL)) %>% 
  ungroup() 
luminex_data.lfc.mat <- make_matrix(select(luminex_data.wide,-cytokine), pull(luminex_data.wide,cytokine))


col_fun = colorRamp2(c(0, 1, 2), c("blue", "white", "red"))
lgd = Legend(col_fun = colorRamp2(c(0, 1, 2), c("blue", "white", "red")))

luminex.heatmap <- as.ggplot(Heatmap(t(luminex_data.lfc.mat), 
          heatmap_legend_param = list(title = "", at = c(0,1,2)),
          col = colorRamp2(c(0,1,2), c("blue", "white", "red")),
          cluster_columns = FALSE, cluster_column_slices = FALSE, cluster_rows = FALSE, border = TRUE, 
          # row_order = myeloid.markers.df$gene_name,
          column_names_gp = gpar(fontsize = 8), row_names_gp = gpar(fontsize = 8), #, #, col = c(rep("red", 10), rep("blue", 8)))
          bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = "white"), labels = "Cytokines", labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          column_title_gp = gpar(fontsize = 8), row_title = NULL, row_names_rot = 0, row_names_centered = TRUE, row_split =  c(1,2)))
luminex.heatmap

# luminex_cytokine.violin = violin_plot(data = luminex_data, color_by = "mutation", continuous = "intensity_normalised.transformed", ylims = c(0.05,3.5), cols = brewer.pal(6, "Paired"), ylabel = "% Positive", facet_by = "cytokine", plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 3, tip_length = 0.003) + theme(axis.text.x = element_text(angle = 30, vjust = 0.7, hjust=0.7))
# luminex_cytokine.violin

luminex_cytokine_cxcl10_il6.violin = violin_plot(data = filter(luminex_data, cytokine %in% c("IL6", "CXCL10")), color_by = "mutation", continuous = "intensity_normalised.transformed", ylims = c(0,3), cols = brewer.pal(6, "Paired"), ylabel = "Median intensity\nnormalised to confluency", facet_by = "cytokine", plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 3, tip_length = 0.003)
luminex_cytokine_cxcl10_il6.violin

```


## Merge


A - Lysotracker representative image VCP microglia
B - Lysotracker quantification size VCP microglia 
C - Lysotracker quantification intensity VCP microglia 
D - Lysotracker quantification number VCP microglia 
E - Cytokine/Chemokine heatmap VCP microglia 
F - IL-6 barplot VCP microglia 
G - CXCL10 barplot VCP microglia 


```{r}

fig3.merged <- plot_grid(
  plot_grid(
    plot_grid(Lyso_image.gg, pepstatin_magic_red.gg, nrow = 2, labels = c("A","F")),
    plot_grid(lysosensor_mean_intensity.violin, lysotracker_mean_intensity.violin, lysotracker_spot_area.violin, lysotracker_spot_number_per_area.violin, cathepsin_d_activity.violin, cathepsin_b_activity.violin,  nrow = 3, ncol = 2, labels = c("B","C","D","E","G","H")), ncol = 2),
  plot_grid(NULL, luminex.heatmap, luminex_cytokine_cxcl10_il6.violin, ncol = 3, rel_widths = c(0.1,1,1), labels = c("I","","J")),
  nrow = 2, rel_heights = c(3,1))
# fig3.merged
ggsave(fig3.merged, filename = here(proj_path, "expression/deseq2/fig3.merged.lysotracker_cathepsin.png"), height = 11, width = 9.5)
ggsave(fig3.merged, filename = here(proj_path, "expression/deseq2/fig3.merged.lysotracker_cathepsin.pdf"), height = 11, width = 9.5)

```

### Rebuttal Lysosensor & Lystoracker 

```{r}
# import the immuno images with magick 
Fig.2A_rebuttal.png = magick::image_read(here(proj_path,"ben-figures/Fig.2A_rebuttal.png"))
Fig.2A_rebuttal.gg = ggdraw() + draw_image(Fig.2A_rebuttal.png)
Fig.2A_rebuttal.gg

Fig.2F_rebuttal.png = magick::image_read(here(proj_path,"ben-figures/Fig.2F_rebuttal.png"))
Fig.2F_rebuttal.gg = ggdraw() + draw_image(Fig.2F_rebuttal.png)
Fig.2F_rebuttal.gg

lysotracker_data = read_excel(here(proj_path, "expression/lysotracker/Rebuttal_data_for Oli.xlsx"), sheet = "Rebuttal_Figure_2") %>% clean_names() #%>%  filter(grepl("Lysotracker",assay))
lysotracker_data %>% glimpse

lysotracker_data %>% my_summarise(lysosensor_intensity, experiment, cell_type) %>% print(n=Inf)

# Microglia
# for stats, adjust for experiment
lysosensor_intensity.mg.fitg = glm(lysosensor_intensity ~ genotype + experiment, data= filter(lysotracker_data, cell_type == "Microglia"), family = gaussian)
summ(lysosensor_intensity.mg.fitg, digits = 5)$coeftable 
#                     Est.       S.E.     t val.            p
# (Intercept)  1.017186815 0.06325008 16.0819848 2.787807e-13
# genotypeVCP  0.235479619 0.04437919  5.3060818 2.920037e-05
lysosensor_intensity.mg.violin = violin_plot(data = filter(lysotracker_data, cell_type == "Microglia"), color_by = "genotype", continuous = "lysosensor_intensity", ylabel = "Lysosensor intensity\nFC from CTRL", ylims = c(0,1.5), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 1.5, tip_length = 0.05)
lysosensor_intensity.mg.violin


lysotracker_intensity.mg.fitg = glm(lysotracker_intensity ~ genotype + experiment, data= filter(lysotracker_data, cell_type == "Microglia"), family = gaussian)
summ(lysotracker_intensity.mg.fitg, digits = 5)$coeftable 
#                   Est.       S.E.    t val.            p
# (Intercept) 0.94427574 0.13438826 7.0264747 6.173757e-07
# genotypeVCP 0.21276791 0.09429304 2.2564540 3.482705e-02
lysotracker_intensity.mg.violin = violin_plot(data = filter(lysotracker_data, cell_type == "Microglia"), color_by = "genotype", continuous = "lysotracker_intensity", ylabel = "Lysotracker intensity\nFC from CTRL", ylims = c(0,2), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 2, tip_length = 0.05)
lysotracker_intensity.mg.violin


lysotracker_spot_area.mg.fitg = glm(lysotracker_spot_area ~ genotype + experiment, data= filter(lysotracker_data, cell_type == "Microglia"), family = gaussian)
summ(lysotracker_spot_area.mg.fitg, digits = 5)$coeftable 
#                   Est.      S.E.    t val.            p
# (Intercept)  0.87744192 0.1452487  6.040963 5.387458e-06
# genotypeVCP -0.33507709 0.1019132 -3.287867 3.507772e-03
lysotracker_spot_area.mg.violin = violin_plot(data = filter(lysotracker_data, cell_type == "Microglia"), color_by = "genotype", continuous = "lysotracker_spot_area", ylabel = "Lysotracker spot area\nFC from CTRL", ylims = c(0,2), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 2, tip_length = 0.05)
lysotracker_spot_area.mg.violin


lysotracker_spot_number_per_area.mg.fitg = glm(lysotracker_spot_number_per_area ~ genotype + experiment, data= filter(lysotracker_data, cell_type == "Microglia"), family = gaussian)
summ(lysotracker_spot_number_per_area.mg.fitg, digits = 5)$coeftable 
#                    Est.       S.E.     t val.            p
# (Intercept)  0.96924591 0.05166780 18.7591851 1.344698e-14
# genotypeVCP -0.05165983 0.03625253 -1.4249993 1.688511e-01
lysotracker_spot_number_per_area.mg.violin = violin_plot(data = filter(lysotracker_data, cell_type == "Microglia"), color_by = "genotype", continuous = "lysotracker_spot_number_per_area", ylabel = "Lysotracker spot number / area\nFC from CTRL", ylims = c(0.5,1.5), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 1.5, tip_length = 0.05)
lysotracker_spot_number_per_area.mg.violin

# Microglia precursors
# for stats, adjust for experiment
lysosensor_intensity.pmg.fitg = glm(lysosensor_intensity ~ genotype + experiment, data= filter(lysotracker_data, cell_type == "Macrophage precursors"), family = gaussian)
summ(lysosensor_intensity.pmg.fitg, digits = 5)$coeftable 
#                   Est.       S.E.    t val.           p
# (Intercept)  0.93624303 0.07446499 12.572930 3.06631e-11
# genotypeVCP -0.07554258 0.05224809 -1.445844 1.62981e-01
lysosensor_intensity.pmg.violin = violin_plot(data = filter(lysotracker_data, cell_type == "Macrophage precursors"), color_by = "genotype", continuous = "lysosensor_intensity", ylabel = "Lysosensor intensity\nFC from CTRL", ylims = c(0,1.5), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 1.5, tip_length = 0.05)
lysosensor_intensity.pmg.violin


lysotracker_intensity.pmg.fitg = glm(lysotracker_intensity ~ genotype + experiment, data= filter(lysotracker_data, cell_type == "Macrophage precursors"), family = gaussian)
summ(lysotracker_intensity.pmg.fitg, digits = 5)$coeftable 
#                    Est.       S.E.    t val.            p
# (Intercept)  0.85244791 0.13388741  6.366901 2.596080e-06
# genotypeVCP -0.13733231 0.09394162 -1.461890 1.585756e-01
lysotracker_intensity.pmg.violin = violin_plot(data = filter(lysotracker_data, cell_type == "Macrophage precursors"), color_by = "genotype", continuous = "lysotracker_intensity", ylabel = "Lysotracker intensity\nFC from CTRL", ylims = c(0,1.5), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 1.5, tip_length = 0.05)
lysotracker_intensity.pmg.violin


lysotracker_spot_area.pmg.fitg = glm(lysotracker_spot_area ~ genotype + experiment, data= filter(lysotracker_data, cell_type == "Macrophage precursors"), family = gaussian)
summ(lysotracker_spot_area.pmg.fitg, digits = 5)$coeftable 
#                   Est.       S.E.    t val.            p
# (Intercept) 0.90799585 0.05170800 17.560065 4.971725e-14
# genotypeVCP 0.05554275 0.03628073  1.530916 1.407162e-01
lysotracker_spot_area.pmg.violin = violin_plot(data = filter(lysotracker_data, cell_type == "Macrophage precursors"), color_by = "genotype", continuous = "lysotracker_spot_area", ylabel = "Lysotracker spot area\nFC from CTRL", ylims = c(0,1.5), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 1.5, tip_length = 0.05)
lysotracker_spot_area.pmg.violin


lysotracker_spot_number_per_area.pmg.fitg = glm(lysotracker_spot_number_per_area ~ genotype + experiment, data= filter(lysotracker_data, cell_type == "Macrophage precursors"), family = gaussian)
summ(lysotracker_spot_number_per_area.pmg.fitg, digits = 5)$coeftable 
#                     Est.       S.E.     t val.            p
# (Intercept)  1.009606616 0.05023343 20.0983033 3.400433e-15
# genotypeVCP -0.072248274 0.03524610 -2.0498230 5.307583e-02
lysotracker_spot_number_per_area.pmg.violin = violin_plot(data = filter(lysotracker_data, cell_type == "Macrophage precursors"), color_by = "genotype", continuous = "lysotracker_spot_number_per_area", ylabel = "Lysotracker spot number / area\nFC from CTRL", ylims = c(0.5,1.5), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 1.5, tip_length = 0.05)
lysotracker_spot_number_per_area.pmg.violin


rebuttal_fig2.merged <- plot_grid(
  plot_grid(Fig.2A_rebuttal.gg,
            plot_grid(lysosensor_intensity.mg.violin, lysotracker_intensity.mg.violin, lysotracker_spot_area.mg.violin, lysotracker_spot_number_per_area.mg.violin, nrow = 2, ncol = 2, labels = c("B","C","D","E")),
            ncol = 2),
  plot_grid(Fig.2F_rebuttal.gg,
            plot_grid(lysosensor_intensity.pmg.violin, lysotracker_intensity.pmg.violin, lysotracker_spot_area.pmg.violin, lysotracker_spot_number_per_area.pmg.violin, nrow = 2, ncol = 2, labels = c("G","H","I","J")),
            ncol = 2),
  nrow = 2)
# rebuttal_fig2.merged
ggsave(rebuttal_fig2.merged, filename = here(proj_path, "expression/deseq2/rebuttal_fig2.merged.png"), height = 9, width = 9)


```

### Rebuttal Cathepsin D & B

```{r}
rebuttal_fig3.merged = plot_grid(pepstatin_magic_red.gg, cathepsin_d_activity.violin, cathepsin_b_activity.violin, ncol = 3, rel_widths = c(2,1,1))
ggsave(rebuttal_fig3.merged, filename = here(proj_path, "expression/deseq2/rebuttal_fig3.merged.png"), height = 3, width = 6)

```

# Fig S2 Phagocytosis & membrane potential

A - GSEA phagocytosis VCP microglia 
B - GSEA mitochondria VCP microglia 
C - Progeny VCP microglia 
D - Dorothea VCP microglia 
E - Volcano sensitivity analysis no iso vs all
F - Venn sensitivity analysis no iso vs all

## GSEA phagocytosis & mitochondria

```{r}
# GSEA phagocytosis RNAseq
phagocytosis_genes.list <- list("phagocytosis" = untx_vcp_vs_ctrl$res$gene_name[untx_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS])
untx_vcp_vs_ctrl.phagocytosis_fgseaRes <- fgsea(pathways = phagocytosis_genes.list, stats = untx_vcp_vs_ctrl.geneList)
untx_vcp_vs_ctrl.phagocytosis_fgseaRes
# pathway       pval       padj   log2err         ES       NES size                         leadingEdge
# 1: phagocytosis 0.00961924 0.00961924 0.3807304 -0.3016557 -1.366053  265 FCGR1A,CCL2,PLD4,IL1B,TNF,ARPC2,...
untx_vcp_vs_ctrl.phagocytosis.gseaplot = plotEnrichment(phagocytosis_genes.list[["phagocytosis"]], untx_vcp_vs_ctrl.geneList) + 
  theme(axis.text=element_text(size=8), axis.title=element_text(size=10)) + ggtitle("Phagocytosis GSEA") + xlab(expression(VCP~vs~CTRL)) +
  annotate("text", x = 100, y = -0.15, label = paste0("NES = ", round(untx_vcp_vs_ctrl.phagocytosis_fgseaRes$NES, digits = 3),"\nP = ", round(untx_vcp_vs_ctrl.phagocytosis_fgseaRes$pval, digits = 3)), size = 3, hjust = 0) + theme_oz()
# untx_vcp_vs_ctrl.phagocytosis.gseaplot

# GSEA mitochondrial RNAseq
mitochondrial.go = unique(c(go.pathways.msigdb$GO_POSITIVE_REGULATION_OF_MITOCHONDRIAL_MEMBRANE_POTENTIAL, go.pathways.msigdb$GO_MITOCHONDRIAL_GENE_EXPRESSION, go.pathways.msigdb$GO_REGULATION_OF_MITOCHONDRIAL_ATP_SYNTHESIS_COUPLED_ELECTRON_TRANSPORT))
mitochondrial_genes.list <- list("mitochondrial" = untx_vcp_vs_ctrl$res$gene_name[untx_vcp_vs_ctrl$res$gene_name %in% mitochondrial.go])
untx_vcp_vs_ctrl.mitochondrial_fgseaRes <- fgsea(pathways = mitochondrial_genes.list, stats = untx_vcp_vs_ctrl.geneList)
untx_vcp_vs_ctrl.mitochondrial_fgseaRes
#          pathway  pval  padj log2err         ES       NES size                              leadingEdge
# 1: mitochondrial 1e-10 1e-10      NA -0.5214601 -2.244648  181 VCP,SLC25A33,CCNB1,HARS1,MRPS11,GFM1,...
untx_vcp_vs_ctrl.mitochondrial.gseaplot = plotEnrichment(mitochondrial_genes.list[["mitochondrial"]], untx_vcp_vs_ctrl.geneList) + 
  theme(axis.text=element_text(size=8), axis.title=element_text(size=10)) + ggtitle("Mitochondrial GSEA") + xlab(expression(VCP~vs~CTRL)) +
  annotate("text", x = 1000, y = -0.35, label = paste0("NES = ", round(untx_vcp_vs_ctrl.mitochondrial_fgseaRes$NES, digits = 3),"\nP = ", round(untx_vcp_vs_ctrl.mitochondrial_fgseaRes$pval, digits = 3)), size = 3, hjust = 0) + theme_oz()
# untx_vcp_vs_ctrl.mitochondrial.gseaplot

```


## PROGENy

<https://github.com/saezlab/transcriptutorial/blob/master/scripts/03_Pathway_activity_with_Progeny.md> <https://saezlab.github.io/progeny/>

Pathway RespOnsive GENes (PROGENy) estimates signaling pathway activity

```{r}
# net_progeny = readRDS(here(nemo_path, "projects/ipsc-mn-als-meta/scripts/ipsc_mn_als_meta/ipsn_als_meta/net_progeny.rds"))
net_progeny <- get_progeny(organism = 'human', top = 100)
untx_vcp_vs_ctrl.res.matrix <- untx_vcp_vs_ctrl$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
untx_vcp_vs_ctrl_progeny.contrast_acts <- run_wmean(mat=untx_vcp_vs_ctrl.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "***", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
untx_vcp_vs_ctrl_progeny.pathwayActivity <- ggplot(untx_vcp_vs_ctrl_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "Signaling Pathways", x = "", y = "VCP vs CTRL Normalised enrichment") +
    stat_pvalue_manual(untx_vcp_vs_ctrl_progeny.contrast_acts, label = "p.signif", y.position = 6, xmin = "source", xmax = NULL, size = 3, hide.ns = TRUE) 
# untx_vcp_vs_ctrl_progeny.pathwayActivity



# untx_vcp_vs_ctrl.PathwayActivity_zscore_df.meanNES <- untx.vcp_vs_ctrl.PathwayActivity_zscore_df %>% group_by(Pathway) %>% summarise(NES = mean(NES,na.rm = TRUE)) %>% arrange(NES)
# untx.vcp_vs_ctrl.PathwayActivity_zscore_df$Pathway <- factor(untx.vcp_vs_ctrl.PathwayActivity_zscore_df$Pathway, levels = untx_vcp_vs_ctrl.PathwayActivity_zscore_df.meanNES$Pathway)
# untx_vcp_vs_ctrl.progeny.ordered.pathwayActivity <- ggplot(untx.vcp_vs_ctrl.PathwayActivity_zscore_df, aes(x=Pathway, y = NES, fill = Pathway)) + theme_classic() +
#   geom_bar(stat="identity", position=position_dodge(), width = 0.7) + 
#   theme(legend.position = "none", legend.title = element_blank(), axis.title = element_text(size = 12), axis.text.x = element_text(angle = 45, hjust = 1, size = 8), axis.text.y = element_text(size =8)) +
#   geom_hline(yintercept = 0, linetype = 2, colour = "darkgrey") + 
#   # scale_fill_manual( values = c("dodgerblue2","firebrick2")) +  scale_fill_npg() +
#   xlab("") + ylab(expression(Normalised~enrichment~VCP:CTRL))
# untx_vcp_vs_ctrl.progeny.ordered.pathwayActivity
# ggsave(untx_vcp_vs_ctrl.progeny.ordered.pathwayActivity, filename = here(proj_path, "expression/deseq2/untx_vcp_vs_ctrl.progeny.pathwayActivity.png")

```

## DoRothEA

```{r}
# net_dorothea = readRDS(here(nemo_path, "projects/ipsc-mn-als-meta/scripts/ipsc_mn_als_meta/ipsn_als_meta/net_dorothea.rds"))
net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
net_dorothea %>% distinct(source) # 429
net_dorothea %>% distinct(mor)
untx_vcp_vs_ctrl.dorothea <- run_wmean(mat=untx_vcp_vs_ctrl.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

untx_vcp_vs_ctrl.dorothea.labels.up = untx_vcp_vs_ctrl.dorothea %>% top_n(score, n = 5) %>% pull(source)
untx_vcp_vs_ctrl.dorothea.labels.down = untx_vcp_vs_ctrl.dorothea %>% top_n(-score, n = 5) %>% pull(source)

# Volcano of TFs score vs p_value
untx_vcp_vs_ctrl.dorothea.volcano = ggplot(untx_vcp_vs_ctrl.dorothea, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment VCP vs CTRL", title = "Transcription Factors") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  # scale_y_continuous(expand = c(0,0), limits = c(0,ymax)) + scale_x_continuous(limits = c(-xmax,xmax))
  geom_text_repel(fontface = "italic", data = filter(untx_vcp_vs_ctrl.dorothea, source %in% c(untx_vcp_vs_ctrl.dorothea.labels.up, untx_vcp_vs_ctrl.dorothea.labels.down)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.5) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
# untx_vcp_vs_ctrl.dorothea.volcano

untx_vcp_vs_ctrl.dorothea %>% arrange(-score)
untx_vcp_vs_ctrl.dorothea %>% arrange(score)

# dorothea.tf.activities.mat <- make_matrix(select(dorothea.tf.activities,-gene_name), pull(dorothea.tf.activities,gene_name))
# # dorothea.tf.activities.mat.scale <- scale(dorothea.tf.activities.mat, center = FALSE) # make variance equal between datasets
# # dorothea.tf.activities.mat.scale[dorothea.tf.activities.mat.scale > 6] <- 6 # crop outliers to improve colour distribution
# # dorothea.tf.activities.mat.scale = t(scale(t(dorothea.tf.activities.mat.scale))) # row scale
# # dorothea.tf.activities.mat.scale <- dorothea.tf.activities.mat.scale[!is.infinite(rowSums(dorothea.tf.activities.mat.scale)),]
# 
# # dorothea.tf.activitiess.filt <- astrocyte.reactivity.markers %>% filter(gene_name %in% dorothea.tf.activities$gene_name)
# col_ramp_reactivity.markers = colorRamp2(c(-3, 0, 3), c("blue", "white", "red")) # set colour limits to -3 to 3 to improve colour distribution, avoid outlier distortion
# 
# dorothea.tf.activities.heatmap <- Heatmap(dorothea.tf.activities.mat,
#           show_heatmap_legend = FALSE, col = col_ramp_reactivity.markers, 
#           cluster_rows = FALSE, cluster_columns = FALSE, cluster_row_slices = FALSE,          
#           row_names_gp = gpar(fontsize = 6), column_names_gp = gpar(fontsize = 8), border = TRUE, #, #, col = c(rep("red", 10), rep("blue", 8)))
#           # row_order = dorothea.tf.activitiess.filt$gene_name,
#           # row_split = factor(dorothea.tf.activitiess.filt$group, levels = unique(dorothea.tf.activitiess.filt$group)),
#           # right_annotation = rowAnnotation(markers = anno_block(gp = gpar(fill = c("white", "white", "white")), labels = c("Pan\nreactive", "A1", "A2"), labels_gp = gpar(fontsize = 8), labels_rot = 0)),
#           # bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = c("white", "white")), labels = c("Untreated", "LPS treated"), labels_gp = gpar(fontsize = 8), labels_rot = 0)),
#           column_title = NULL, column_names_rot = 0, column_names_centered = TRUE)
# dorothea.tf.activities.heatmap
# ggsave(as.ggplot(dorothea.tf.activities.heatmap), filename = here(proj_path, "expression/deseq2/lps_untx_vcp_vs_ctrl.dorothea.tfactivities.png")


```



## No isogenics scatter & venn

```{r}
# # Volcano
# untx_vcp_vs_ctrl.noiso.labels <- untx_vcp_vs_ctrl.noiso$res  %>% filter(-log10(padj) > 4, gene_name %in% all.reactivity.go ) %>% pull(gene_name)
# untx_vcp_vs_ctrl.noiso.volcano <- volcano_plot(untx_vcp_vs_ctrl.noiso$res, gene_labels = untx_vcp_vs_ctrl.noiso.labels, xmax = 5, ymax = 10, title = "No Isogenics", numerator = "VCP", denomenator = "CTRL")
# untx_vcp_vs_ctrl.noiso.volcano

untx_vcp_vs_ctrl.noiso$res %>% filter(padj < 0.05, gene_name %in% all.microglia.go)
untx_vcp_vs_ctrl.noiso$res %>% filter(padj < 0.05, gene_name %in% microglia_signature)
untx_vcp_vs_ctrl.noiso$res %>% filter(padj < 0.05, gene_name %in% RNA_BINDING_PROTEINS)

# Scatter
all.untx_vcp_vs_ctrl.join = select(untx_vcp_vs_ctrl$res, gene_name, gene_id, nygc.stat = stat, nygc.padj = padj) %>% left_join(select(untx_vcp_vs_ctrl.noiso$res, gene_name, gene_id, ipsm.stat = stat, ipsm.padj = padj))
untx_vcp_vs_ctrl.all.noiso.overlapping = all.untx_vcp_vs_ctrl.join %>% filter(nygc.padj < 0.05, ipsm.padj < 0.05, (nygc.stat > 0 & ipsm.stat > 0) | (nygc.stat < 0 & ipsm.stat < 0)) %>% pull(gene_name) # 42
untx_vcp_vs_ctrl.all.noiso_list = list("All samples" = untx_vcp_vs_ctrl$res, "No isogenics" = untx_vcp_vs_ctrl.noiso$res)
untx_vcp_vs_ctrl.all.noiso_list_t_stat_plot <- 
plot_de_correlation(model_list1 = untx_vcp_vs_ctrl.all.noiso_list, model_list2 = untx_vcp_vs_ctrl.all.noiso_list, mutation1 = "All samples", mutation2 = "No isogenics", gene_labels = untx_vcp_vs_ctrl.all.noiso.overlapping, plot_p_value = TRUE) +  xlim(-8,8) & ylim(-8,8)
untx_vcp_vs_ctrl.all.noiso_list_t_stat_plot

## Venn overlap
untx_vcp_vs_ctrl.res.dge.up <- untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
untx_vcp_vs_ctrl.res.dge.down <- untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
untx_vcp_vs_ctrl.noiso.res.dge.up <- untx_vcp_vs_ctrl.noiso$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
untx_vcp_vs_ctrl.noiso.res.dge.down <- untx_vcp_vs_ctrl.noiso$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
# untx_vcp_vs_ctrl.iso.res.dge.up <- untx_vcp_vs_ctrl.iso$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
# untx_vcp_vs_ctrl.iso.res.dge.down <- untx_vcp_vs_ctrl.iso$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)

# 4 way overlap
untx_vcp_vs_ctrl_all_noiso_venn = ggvenn(list(`no iso\nVCP\ndown` = untx_vcp_vs_ctrl.noiso.res.dge.down, `no isogenics\nVCP up` = untx_vcp_vs_ctrl.noiso.res.dge.up, `all samples\nVCP up` = untx_vcp_vs_ctrl.res.dge.up, `all\nVCP\ndown` = untx_vcp_vs_ctrl.res.dge.down), fill_color = c("dodgerblue3", "firebrick3", "firebrick1", "dodgerblue1"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.7)
untx_vcp_vs_ctrl_all_noiso_venn

# # GO terms
# untx_vcp_vs_ctrl.noiso.gost <- untx_vcp_vs_ctrl.noiso$res %>% filter(padj < 0.05) %>% select(gene_name, gene_id, log2FoldChange, padj, pvalue) %>% 
#   mutate(deg = case_when(padj < 0.05 & log2FoldChange > 0 ~ "VCP Up", padj < 0.05 & log2FoldChange < 0 ~ "VCP Down", TRUE ~ "None")) %>% 
#   group_split(deg) %>% map("gene_id") %>% gost()
#          
# untx_vcp_vs_ctrl.noiso.gost_filt <- untx_vcp_vs_ctrl.noiso.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM|WP")) %>% arrange( -log10(p_value) ) %>% 
#   mutate(query = case_when(query == "query_1" ~ "VCP\n Down", query == "query_2" ~ "VCP\n Up"), query = factor(query, levels = c("VCP\n Up", "VCP\n Down")))
# table(untx_vcp_vs_ctrl.noiso.gost_filt$query)
# #    VCP\n Up VCP\n Down 
# #        646        274 
# untx_vcp_vs_ctrl.noiso.dge_gprofiles_down <- untx_vcp_vs_ctrl.noiso.gost_filt %>% filter(query == "VCP\n Down")
# untx_vcp_vs_ctrl.noiso.dge_gprofiles_up <- untx_vcp_vs_ctrl.noiso.gost_filt %>% filter(query == "VCP\n Up")
# paste('"',unique(untx_vcp_vs_ctrl.noiso.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
# down_list = c(
# "Tacc1-chTOG-AuroraA  complex",
# "Neutrophil extracellular trap formation",
# "Staphylococcus aureus infection",
# "Osteoclast differentiation",
# "Leishmaniasis",
# "IgG binding")
# untx_vcp_vs_ctrl.noiso.dge_gprofiles_filt_down <- untx_vcp_vs_ctrl.noiso.gost_filt %>% filter(query == "VCP\n Down") %>% filter(term_name %in% down_list)
# # untx_vcp_vs_ctrl.noiso.dge_gprofiles_filt_down$term_name = gsub("Viral protein interaction with cytokine and cytokine receptor", "Viral interaction with cytokine receptor", untx_vcp_vs_ctrl.noiso.dge_gprofiles_filt_down$term_name) %>% gsub("E. coli ", "", .) %>% gsub("Absence of bactericidal oxidative respiratory burst in phagocytes", "Absent respiratory burst in phagocytes", .) %>% gsub("Cross-presentation of particulate exogenous antigens [(]phagosomes[)]", "Phagosome antigen cross-presentation", .)
# paste('"',unique(untx_vcp_vs_ctrl.noiso.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
# up_list = c(
# "activation of phospholipase C activity",
# "NAD+ binding")
# untx_vcp_vs_ctrl.noiso.dge_gprofiles_filt_up <- untx_vcp_vs_ctrl.noiso.gost_filt %>% filter(query == "VCP\n Up")  %>% filter(term_name %in% up_list)
# untx_vcp_vs_ctrl.noiso.dge_gprofiles_filtered = bind_rows(untx_vcp_vs_ctrl.noiso.dge_gprofiles_filt_up, untx_vcp_vs_ctrl.noiso.dge_gprofiles_filt_down) %>% mutate(query = fct_relevel(query,"VCP\n Up"))
# untx_vcp_vs_ctrl.noiso_dge.go_curated <- curated_profiles(untx_vcp_vs_ctrl.noiso.dge_gprofiles_filtered, gprofiles =, cols = c("firebrick2", "dodgerblue2")) + theme(strip.text.y.right = element_text(angle = 0))# + theme(legend.position = "none", axis.text = element_text(size = 10))
# untx_vcp_vs_ctrl.noiso_dge.go_curated

## Scatter correlation
# no isogenics vs all samples
untx_vcp_vs_ctrl.all_noiso <- full_join(select(untx_vcp_vs_ctrl.noiso$res, gene_name, gene_id, noiso.stat = stat, noiso.padj = padj), 
                                        select(untx_vcp_vs_ctrl$res, gene_name, gene_id, all.stat = stat, all.padj = padj), by = c("gene_name", "gene_id")) %>% 
  mutate(overlap = case_when(noiso.padj < 0.05 & all.padj < 0.05 ~ "overlapping", noiso.padj < 0.05 ~ "noiso specific", all.padj < 0.05 ~ "all specific", TRUE ~ "None"),
         overlap_direction = case_when(noiso.padj < 0.05 & noiso.stat > 0 & all.padj < 0.05 & all.stat > 0 ~ "overlapping up", noiso.padj < 0.05 & noiso.stat < 0 & all.padj < 0.05 & all.stat < 0 ~ "overlapping down",
                                       noiso.padj < 0.05 & noiso.stat < 0 & all.padj < 0.05 & all.stat > 0 ~ "noiso down, all up", noiso.padj < 0.05 & noiso.stat > 0 & all.padj < 0.05 & all.stat < 0 ~ "noiso up, all down",
                                       noiso.padj < 0.05 & noiso.stat > 0 ~ "noiso specific up", noiso.padj < 0.05 & noiso.stat < 0 ~ "noiso specific down",
                                       all.padj < 0.05 & all.stat > 0 ~ "all specific up", all.padj < 0.05 & all.stat < 0 ~ "all specific down", TRUE ~ "None"),
         overlap_direction = factor(overlap_direction, levels = c("overlapping up", "overlapping down", "noiso down, all up", "noiso up, all down", "noiso specific up", "noiso specific down", "all specific up", "all specific down", "None")),
         overlap = factor(overlap, levels = c("overlapping",  "noiso specific", "all specific", "None"))) %>% arrange(overlap_direction)
table(untx_vcp_vs_ctrl.all_noiso$overlap)
table(untx_vcp_vs_ctrl.all_noiso$overlap_direction)
# overlapping up    overlapping down  noiso down, all up  noiso up, all down   noiso specific up noiso specific down 
#                  21                  21                   0                   0                  99                 121 
#     all specific up   all specific down                None 
#                  89                  52               60273 
cor.test(x = untx_vcp_vs_ctrl.all_noiso$noiso.stat, y = untx_vcp_vs_ctrl.all_noiso$all.stat) # 0.70
t.test( x = untx_vcp_vs_ctrl.all_noiso$noiso.stat, y = untx_vcp_vs_ctrl.all_noiso$all.stat, paired = TRUE )

untx_vcp_vs_ctrl.all_noiso_scatter <- ggscatter(untx_vcp_vs_ctrl.all_noiso, x = "noiso.stat", y = "all.stat", color = "overlap_direction", palette = c("firebrick2", "dodgerblue2", "forestgreen", "gold2", "grey", "grey", "grey", "grey", "grey"), 
                                        cor.coef = TRUE, cor.method = "pearson", size = 0.5, cor.coeff.args = list(label.x = -5.5, label.y = 4, label.sep="\n"),
          xlab = expression(z-score~noiso~VCP/CTRL), ylab = expression(z~score~all~VCP/CTRL)) +
          geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") +  
    coord_cartesian(xlim = c(-6,6), ylim = c(-5,5)) +
    geom_smooth(data = untx_vcp_vs_ctrl.all_noiso, aes(x = noiso.stat, y = all.stat), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 2) + theme(legend.position = "none") +
    geom_text_repel(data = filter(untx_vcp_vs_ctrl.all_noiso, overlap == "overlapping", gene_name %in% all.reactivity.go, abs(all.stat) > 0.5, abs(noiso.stat) > 1), aes(x = noiso.stat, y = all.stat, label = gene_name), 
                    fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.8, max.overlaps = 30) +  
  annotate(geom="text", x=3, y=5, label="noiso & all both Up", color="firebrick2", size = 3.2)  + annotate(geom="text", x=-3, y=-5, label="noiso & all both Down", color="dodgerblue2", size = 3.2)
untx_vcp_vs_ctrl.all_noiso_scatter


# fisher overlap
# Overlap tested using Fisher's exact test (alternative=greater)
untx_vcp_vs_ctrl.noiso.deg.genes <- untx_vcp_vs_ctrl.noiso$res %>% filter(padj < 0.05) %>% pull(gene_name)
untx_vcp_vs_ctrl.deg.genes <- untx_vcp_vs_ctrl$res %>% filter(padj < 0.05) %>% pull(gene_name)
genome.size <- ctrl_lps_vs_untx$res %>% distinct(gene_name) %>% nrow
newGeneOverlap(untx_vcp_vs_ctrl.deg.genes, untx_vcp_vs_ctrl.noiso.deg.genes,genome.size=genome.size) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=183
# listB size=262
# Intersection size=42
# Overlapping p-value=4.3e-60
# Jaccard Index=0.1

# 
# untx_vcp_vs_ctrl.all_noiso <- full_join(select(untx_vcp_vs_ctrl.noiso$lfcshrink, gene_name, gene_id, noiso.log2FoldChange = log2FoldChange, noiso.padj = padj), 
#                                         select(untx_vcp_vs_ctrl$lfcshrink, gene_name, gene_id, all.log2FoldChange = log2FoldChange, all.padj = padj), by = c("gene_name", "gene_id")) %>% 
#   mutate(overlap = case_when(noiso.padj < 0.05 & all.padj < 0.05 ~ "overlapping", noiso.padj < 0.05 ~ "noiso specific", all.padj < 0.05 ~ "all specific", TRUE ~ "None"),
#          overlap_direction = case_when(noiso.padj < 0.05 & noiso.log2FoldChange > 0 & all.padj < 0.05 & all.log2FoldChange > 0 ~ "overlapping up", noiso.padj < 0.05 & noiso.log2FoldChange < 0 & all.padj < 0.05 & all.log2FoldChange < 0 ~ "overlapping down",
#                                        noiso.padj < 0.05 & noiso.log2FoldChange < 0 & all.padj < 0.05 & all.log2FoldChange > 0 ~ "noiso down, all up", noiso.padj < 0.05 & noiso.log2FoldChange > 0 & all.padj < 0.05 & all.log2FoldChange < 0 ~ "noiso up, all down",
#                                        noiso.padj < 0.05 & noiso.log2FoldChange > 0 ~ "noiso specific up", noiso.padj < 0.05 & noiso.log2FoldChange < 0 ~ "noiso specific down",
#                                        all.padj < 0.05 & all.log2FoldChange > 0 ~ "all specific up", all.padj < 0.05 & all.log2FoldChange < 0 ~ "all specific down", TRUE ~ "None"),
#          overlap_direction = factor(overlap_direction, levels = c("overlapping up", "overlapping down", "noiso down, all up", "noiso up, all down", "noiso specific up", "noiso specific down", "all specific up", "all specific down", "None")),
#          overlap = factor(overlap, levels = c("overlapping",  "noiso specific", "all specific", "None"))) %>% arrange(overlap_direction)
# table(untx_vcp_vs_ctrl.all_noiso$overlap_direction)
# 
# cor.test(x = untx_vcp_vs_ctrl.all_noiso$noiso.log2FoldChange, y = untx_vcp_vs_ctrl.all_noiso$all.log2FoldChange) # 0.32
# 
# 
# untx_vcp_vs_ctrl.all_noiso <- full_join(select(untx_vcp_vs_ctrl.noiso$res, gene_name, gene_id, noiso.log2FoldChange = log2FoldChange, noiso.padj = padj), 
#                                         select(untx_vcp_vs_ctrl$res, gene_name, gene_id, all.log2FoldChange = log2FoldChange, all.padj = padj), by = c("gene_name", "gene_id")) %>% 
#   mutate(overlap = case_when(noiso.padj < 0.05 & all.padj < 0.05 ~ "overlapping", noiso.padj < 0.05 ~ "noiso specific", all.padj < 0.05 ~ "all specific", TRUE ~ "None"),
#          overlap_direction = case_when(noiso.padj < 0.05 & noiso.log2FoldChange > 0 & all.padj < 0.05 & all.log2FoldChange > 0 ~ "overlapping up", noiso.padj < 0.05 & noiso.log2FoldChange < 0 & all.padj < 0.05 & all.log2FoldChange < 0 ~ "overlapping down",
#                                        noiso.padj < 0.05 & noiso.log2FoldChange < 0 & all.padj < 0.05 & all.log2FoldChange > 0 ~ "noiso down, all up", noiso.padj < 0.05 & noiso.log2FoldChange > 0 & all.padj < 0.05 & all.log2FoldChange < 0 ~ "noiso up, all down",
#                                        noiso.padj < 0.05 & noiso.log2FoldChange > 0 ~ "noiso specific up", noiso.padj < 0.05 & noiso.log2FoldChange < 0 ~ "noiso specific down",
#                                        all.padj < 0.05 & all.log2FoldChange > 0 ~ "all specific up", all.padj < 0.05 & all.log2FoldChange < 0 ~ "all specific down", TRUE ~ "None"),
#          overlap_direction = factor(overlap_direction, levels = c("overlapping up", "overlapping down", "noiso down, all up", "noiso up, all down", "noiso specific up", "noiso specific down", "all specific up", "all specific down", "None")),
#          overlap = factor(overlap, levels = c("overlapping",  "noiso specific", "all specific", "None"))) %>% arrange(overlap_direction)
# table(untx_vcp_vs_ctrl.all_noiso$overlap)
# table(untx_vcp_vs_ctrl.all_noiso$overlap_direction)
# 
# cor.test(x = untx_vcp_vs_ctrl.all_noiso$noiso.log2FoldChange, y = untx_vcp_vs_ctrl.all_noiso$all.log2FoldChange) # 0.68
# 
# 
# 
# 
# # Isogenics only vs all samples
# untx_vcp_vs_ctrl.all_iso <- full_join(select(untx_vcp_vs_ctrl.iso$res, gene_name, gene_id, iso.log2FoldChange = log2FoldChange, iso.padj = padj), select(untx_vcp_vs_ctrl$res, gene_name, gene_id, all.log2FoldChange = log2FoldChange, all.padj = padj), by = c("gene_name", "gene_id")) %>% 
#   mutate(overlap = case_when(iso.padj < 0.05 & all.padj < 0.05 ~ "overlapping", iso.padj < 0.05 ~ "iso specific", all.padj < 0.05 ~ "all specific", TRUE ~ "None"),
#          overlap_direction = case_when(iso.padj < 0.05 & iso.log2FoldChange > 0 & all.padj < 0.05 & all.log2FoldChange > 0 ~ "overlapping up", iso.padj < 0.05 & iso.log2FoldChange < 0 & all.padj < 0.05 & all.log2FoldChange < 0 ~ "overlapping down",
#                                        iso.padj < 0.05 & iso.log2FoldChange < 0 & all.padj < 0.05 & all.log2FoldChange > 0 ~ "iso down, all up", iso.padj < 0.05 & iso.log2FoldChange > 0 & all.padj < 0.05 & all.log2FoldChange < 0 ~ "iso up, all down",
#                                        iso.padj < 0.05 & iso.log2FoldChange > 0 ~ "iso specific up", iso.padj < 0.05 & iso.log2FoldChange < 0 ~ "iso specific down",
#                                        all.padj < 0.05 & all.log2FoldChange > 0 ~ "all specific up", all.padj < 0.05 & all.log2FoldChange < 0 ~ "all specific down", TRUE ~ "None"),
#          overlap_direction = factor(overlap_direction, levels = c("overlapping up", "overlapping down", "iso down, all up", "iso up, all down", "iso specific up", "iso specific down", "all specific up", "all specific down", "None")),
#          overlap = factor(overlap, levels = c("overlapping",  "iso specific", "all specific", "None"))) %>% arrange(overlap_direction)
# table(untx_vcp_vs_ctrl.all_iso$overlap)
# table(untx_vcp_vs_ctrl.all_iso$overlap_direction)
# 
# cor.test(x = untx_vcp_vs_ctrl.all_iso$iso.log2FoldChange, y = untx_vcp_vs_ctrl.all_iso$all.log2FoldChange) # 0.70
# t.test( x = untx_vcp_vs_ctrl.all_iso$iso.log2FoldChange, y = untx_vcp_vs_ctrl.all_iso$all.log2FoldChange, paired = TRUE )
# 
# untx_vcp_vs_ctrl.all_iso_scatter <- ggscatter(untx_vcp_vs_ctrl.all_iso, x = "iso.log2FoldChange", y = "all.log2FoldChange", color = "overlap_direction", palette = c("firebrick2", "dodgerblue2", "forestgreen", "gold2", "grey", "grey", "grey", "grey", "grey"), 
#                                         cor.coef = TRUE, cor.method = "pearson",
#           xlab = expression(z-score~iso~VCP/CTRL), ylab = expression(z~score~all~VCP/CTRL), size = 0.5, cor.coeff.args = list(label.x = -5.5, label.y = 4, label.sep="\n")) +
#           geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") +  
#     coord_cartesian(xlim = c(-6,6), ylim = c(-5,5)) +
#     geom_smooth(data = untx_vcp_vs_ctrl.all_iso, aes(x = iso.log2FoldChange, y = all.log2FoldChange), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 2) + theme(legend.position = "none") +
#     geom_text_repel(data = filter(untx_vcp_vs_ctrl.all_iso, overlap == "overlapping", gene_name %in% all.reactivity.go, abs(all.log2FoldChange) > 0.5, abs(iso.log2FoldChange) > 1), aes(x = iso.log2FoldChange, y = all.log2FoldChange, label = gene_name), 
#                     fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.8, max.overlaps = 30) +  
#   annotate(geom="text", x=3, y=5, label="iso & all both Up", color="firebrick2", size = 3.2)  + annotate(geom="text", x=-3, y=-5, label="iso & all both Down", color="dodgerblue2", size = 3.2)
# untx_vcp_vs_ctrl.all_iso_scatter
# 
# untx_vcp_vs_ctrl.all_iso <- full_join(select(untx_vcp_vs_ctrl.iso$res, gene_name, gene_id, iso.stat = stat, iso.padj = padj), select(untx_vcp_vs_ctrl$res, gene_name, gene_id, all.stat = stat, all.padj = padj), by = c("gene_name", "gene_id")) %>% 
#   mutate(overlap = case_when(iso.padj < 0.05 & all.padj < 0.05 ~ "overlapping", iso.padj < 0.05 ~ "iso specific", all.padj < 0.05 ~ "all specific", TRUE ~ "None"),
#          overlap_direction = case_when(iso.padj < 0.05 & iso.stat > 0 & all.padj < 0.05 & all.stat > 0 ~ "overlapping up", iso.padj < 0.05 & iso.stat < 0 & all.padj < 0.05 & all.stat < 0 ~ "overlapping down",
#                                        iso.padj < 0.05 & iso.stat < 0 & all.padj < 0.05 & all.stat > 0 ~ "iso down, all up", iso.padj < 0.05 & iso.stat > 0 & all.padj < 0.05 & all.stat < 0 ~ "iso up, all down",
#                                        iso.padj < 0.05 & iso.stat > 0 ~ "iso specific up", iso.padj < 0.05 & iso.stat < 0 ~ "iso specific down",
#                                        all.padj < 0.05 & all.stat > 0 ~ "all specific up", all.padj < 0.05 & all.stat < 0 ~ "all specific down", TRUE ~ "None"),
#          overlap_direction = factor(overlap_direction, levels = c("overlapping up", "overlapping down", "iso down, all up", "iso up, all down", "iso specific up", "iso specific down", "all specific up", "all specific down", "None")),
#          overlap = factor(overlap, levels = c("overlapping",  "iso specific", "all specific", "None"))) %>% arrange(overlap_direction)
# table(untx_vcp_vs_ctrl.all_iso$overlap)
# table(untx_vcp_vs_ctrl.all_iso$overlap_direction)
# 
# cor.test(x = untx_vcp_vs_ctrl.all_iso$iso.stat, y = untx_vcp_vs_ctrl.all_iso$all.stat) # 0.54
# 
# untx_vcp_vs_ctrl.all_iso <- full_join(select(untx_vcp_vs_ctrl.iso$lfcshrink, gene_name, gene_id, iso.log2FoldChange = log2FoldChange, iso.padj = padj), select(untx_vcp_vs_ctrl$lfcshrink, gene_name, gene_id, all.log2FoldChange = log2FoldChange, all.padj = padj), by = c("gene_name", "gene_id")) %>% 
#   mutate(overlap = case_when(iso.padj < 0.05 & all.padj < 0.05 ~ "overlapping", iso.padj < 0.05 ~ "iso specific", all.padj < 0.05 ~ "all specific", TRUE ~ "None"),
#          overlap_direction = case_when(iso.padj < 0.05 & iso.log2FoldChange > 0 & all.padj < 0.05 & all.log2FoldChange > 0 ~ "overlapping up", iso.padj < 0.05 & iso.log2FoldChange < 0 & all.padj < 0.05 & all.log2FoldChange < 0 ~ "overlapping down",
#                                        iso.padj < 0.05 & iso.log2FoldChange < 0 & all.padj < 0.05 & all.log2FoldChange > 0 ~ "iso down, all up", iso.padj < 0.05 & iso.log2FoldChange > 0 & all.padj < 0.05 & all.log2FoldChange < 0 ~ "iso up, all down",
#                                        iso.padj < 0.05 & iso.log2FoldChange > 0 ~ "iso specific up", iso.padj < 0.05 & iso.log2FoldChange < 0 ~ "iso specific down",
#                                        all.padj < 0.05 & all.log2FoldChange > 0 ~ "all specific up", all.padj < 0.05 & all.log2FoldChange < 0 ~ "all specific down", TRUE ~ "None"),
#          overlap_direction = factor(overlap_direction, levels = c("overlapping up", "overlapping down", "iso down, all up", "iso up, all down", "iso specific up", "iso specific down", "all specific up", "all specific down", "None")),
#          overlap = factor(overlap, levels = c("overlapping",  "iso specific", "all specific", "None"))) %>% arrange(overlap_direction)
# table(untx_vcp_vs_ctrl.all_iso$overlap_direction)
# cor.test(x = untx_vcp_vs_ctrl.all_iso$iso.log2FoldChange, y = untx_vcp_vs_ctrl.all_iso$all.log2FoldChange) # 0.53
# 
# # No isogenics vs Isogenics only
# untx_vcp_vs_ctrl.noiso_iso <- full_join(select(untx_vcp_vs_ctrl.iso$res, gene_name, gene_id, iso.stat = stat, iso.padj = padj), select(untx_vcp_vs_ctrl.noiso$res, gene_name, gene_id, noiso.stat = stat, noiso.padj = padj), by = c("gene_name", "gene_id")) %>% 
#   mutate(overlap = case_when(iso.padj < 0.05 & noiso.padj < 0.05 ~ "overlapping", iso.padj < 0.05 ~ "iso specific", noiso.padj < 0.05 ~ "noiso specific", TRUE ~ "None"),
#          overlap_direction = case_when(iso.padj < 0.05 & iso.stat > 0 & noiso.padj < 0.05 & noiso.stat > 0 ~ "overlapping up", iso.padj < 0.05 & iso.stat < 0 & noiso.padj < 0.05 & noiso.stat < 0 ~ "overlapping down",
#                                        iso.padj < 0.05 & iso.stat < 0 & noiso.padj < 0.05 & noiso.stat > 0 ~ "iso down, noiso up", iso.padj < 0.05 & iso.stat > 0 & noiso.padj < 0.05 & noiso.stat < 0 ~ "iso up, noiso down",
#                                        iso.padj < 0.05 & iso.stat > 0 ~ "iso specific up", iso.padj < 0.05 & iso.stat < 0 ~ "iso specific down",
#                                        noiso.padj < 0.05 & noiso.stat > 0 ~ "noiso specific up", noiso.padj < 0.05 & noiso.stat < 0 ~ "noiso specific down", TRUE ~ "None"),
#          overlap_direction = factor(overlap_direction, levels = c("overlapping up", "overlapping down", "iso down, noiso up", "iso up, noiso down", "iso specific up", "iso specific down", "noiso specific up", "noiso specific down", "None")),
#          overlap = factor(overlap, levels = c("overlapping",  "iso specific", "noiso specific", "None"))) %>% arrange(overlap_direction)
# table(untx_vcp_vs_ctrl.noiso_iso$overlap)
# table(untx_vcp_vs_ctrl.noiso_iso$overlap_direction)
# 
# cor.test(x = untx_vcp_vs_ctrl.noiso_iso$iso.stat, y = untx_vcp_vs_ctrl.noiso_iso$noiso.stat) # 0.10
# t.test( x = untx_vcp_vs_ctrl.noiso_iso$iso.stat, y = untx_vcp_vs_ctrl.noiso_iso$noiso.stat, paired = TRUE )
# 
# untx_vcp_vs_ctrl.noiso_iso_scatter <- ggscatter(untx_vcp_vs_ctrl.noiso_iso, x = "iso.stat", y = "noiso.stat", color = "overlap_direction", palette = c("firebrick2", "dodgerblue2", "forestgreen", "gold2", "grey", "grey", "grey", "grey", "grey"), 
#                                         cor.coef = TRUE, cor.method = "pearson",
#           xlab = expression(z-score~iso~VCP/CTRL), ylab = expression(z~score~noiso~VCP/CTRL), size = 0.5, cor.coeff.args = list(label.x = -5.5, label.y = 4, label.sep="\n")) +
#           geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") +  
#     coord_cartesian(xlim = c(-6,6), ylim = c(-5,5)) +
#     geom_smooth(data = untx_vcp_vs_ctrl.noiso_iso, aes(x = iso.stat, y = noiso.stat), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 2) + theme(legend.position = "none") +
#     geom_text_repel(data = filter(untx_vcp_vs_ctrl.noiso_iso, overlap == "overlapping", gene_name %in% all.reactivity.go, abs(noiso.stat) > 0.5, abs(iso.stat) > 1), aes(x = iso.stat, y = noiso.stat, label = gene_name), 
#                     fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.8, max.overlaps = 30) +  
#   annotate(geom="text", x=3, y=5, label="iso & noiso both Up", color="firebrick2", size = 3.2)  + annotate(geom="text", x=-3, y=-5, label="iso & noiso both Down", color="dodgerblue2", size = 3.2)
# untx_vcp_vs_ctrl.noiso_iso_scatter
# 
# untx_vcp_vs_ctrl.noiso_iso <- full_join(select(untx_vcp_vs_ctrl.iso$res, gene_name, gene_id, iso.log2FoldChange = log2FoldChange, iso.padj = padj), select(untx_vcp_vs_ctrl.noiso$res, gene_name, gene_id, noiso.log2FoldChange = log2FoldChange, noiso.padj = padj), by = c("gene_name", "gene_id")) %>% 
#   mutate(overlap = case_when(iso.padj < 0.05 & noiso.padj < 0.05 ~ "overlapping", iso.padj < 0.05 ~ "iso specific", noiso.padj < 0.05 ~ "noiso specific", TRUE ~ "None"),
#          overlap_direction = case_when(iso.padj < 0.05 & iso.log2FoldChange > 0 & noiso.padj < 0.05 & noiso.log2FoldChange > 0 ~ "overlapping up", iso.padj < 0.05 & iso.log2FoldChange < 0 & noiso.padj < 0.05 & noiso.log2FoldChange < 0 ~ "overlapping down",
#                                        iso.padj < 0.05 & iso.log2FoldChange < 0 & noiso.padj < 0.05 & noiso.log2FoldChange > 0 ~ "iso down, noiso up", iso.padj < 0.05 & iso.log2FoldChange > 0 & noiso.padj < 0.05 & noiso.log2FoldChange < 0 ~ "iso up, noiso down",
#                                        iso.padj < 0.05 & iso.log2FoldChange > 0 ~ "iso specific up", iso.padj < 0.05 & iso.log2FoldChange < 0 ~ "iso specific down",
#                                        noiso.padj < 0.05 & noiso.log2FoldChange > 0 ~ "noiso specific up", noiso.padj < 0.05 & noiso.log2FoldChange < 0 ~ "noiso specific down", TRUE ~ "None"),
#          overlap_direction = factor(overlap_direction, levels = c("overlapping up", "overlapping down", "iso down, noiso up", "iso up, noiso down", "iso specific up", "iso specific down", "noiso specific up", "noiso specific down", "None")),
#          overlap = factor(overlap, levels = c("overlapping",  "iso specific", "noiso specific", "None"))) %>% arrange(overlap_direction)
# table(untx_vcp_vs_ctrl.noiso_iso$overlap)
# table(untx_vcp_vs_ctrl.noiso_iso$overlap_direction)
# cor.test(x = untx_vcp_vs_ctrl.noiso_iso$iso.log2FoldChange, y = untx_vcp_vs_ctrl.noiso_iso$noiso.log2FoldChange) # 0.19
# 
# 
# untx_vcp_vs_ctrl.noiso_iso <- full_join(select(untx_vcp_vs_ctrl.iso$lfcshrink, gene_name, gene_id, iso.log2FoldChange = log2FoldChange, iso.padj = padj), select(untx_vcp_vs_ctrl.noiso$lfcshrink, gene_name, gene_id, noiso.log2FoldChange = log2FoldChange, noiso.padj = padj), by = c("gene_name", "gene_id")) %>% 
#   mutate(overlap = case_when(iso.padj < 0.05 & noiso.padj < 0.05 ~ "overlapping", iso.padj < 0.05 ~ "iso specific", noiso.padj < 0.05 ~ "noiso specific", TRUE ~ "None"),
#          overlap_direction = case_when(iso.padj < 0.05 & iso.log2FoldChange > 0 & noiso.padj < 0.05 & noiso.log2FoldChange > 0 ~ "overlapping up", iso.padj < 0.05 & iso.log2FoldChange < 0 & noiso.padj < 0.05 & noiso.log2FoldChange < 0 ~ "overlapping down",
#                                        iso.padj < 0.05 & iso.log2FoldChange < 0 & noiso.padj < 0.05 & noiso.log2FoldChange > 0 ~ "iso down, noiso up", iso.padj < 0.05 & iso.log2FoldChange > 0 & noiso.padj < 0.05 & noiso.log2FoldChange < 0 ~ "iso up, noiso down",
#                                        iso.padj < 0.05 & iso.log2FoldChange > 0 ~ "iso specific up", iso.padj < 0.05 & iso.log2FoldChange < 0 ~ "iso specific down",
#                                        noiso.padj < 0.05 & noiso.log2FoldChange > 0 ~ "noiso specific up", noiso.padj < 0.05 & noiso.log2FoldChange < 0 ~ "noiso specific down", TRUE ~ "None"),
#          overlap_direction = factor(overlap_direction, levels = c("overlapping up", "overlapping down", "iso down, noiso up", "iso up, noiso down", "iso specific up", "iso specific down", "noiso specific up", "noiso specific down", "None")),
#          overlap = factor(overlap, levels = c("overlapping",  "iso specific", "noiso specific", "None"))) %>% arrange(overlap_direction)
# table(untx_vcp_vs_ctrl.noiso_iso$overlap)
# table(untx_vcp_vs_ctrl.noiso_iso$overlap_direction)
# cor.test(x = untx_vcp_vs_ctrl.noiso_iso$iso.log2FoldChange, y = untx_vcp_vs_ctrl.noiso_iso$noiso.log2FoldChange) # 0.01

```


Isogenics only

```{r}
# # Volcano
# untx_vcp_vs_ctrl.iso.labels <- untx_vcp_vs_ctrl.iso$res  %>% filter(-log10(padj) > 2, gene_name %in% all.reactivity.go ) %>% pull(gene_name)
# untx_vcp_vs_ctrl.iso.volcano <- volcano_plot(data = untx_vcp_vs_ctrl.iso$res, gene_labels = untx_vcp_vs_ctrl.iso.labels) + 
#   xlab(expression(RNA~seq~log[2]~FC~VCP:CTRL)) + 
#   coord_cartesian(xlim = c(-5,5), ylim = c(0,10)) +
#   geom_segment(aes(x = 2, xend = 5, y= 9, yend= 9), arrow=arrow(length=unit(0.3,"cm")), color = "firebrick2" ) +  geom_segment(aes(x = -2, xend = -5, y= 9, yend=9),arrow=arrow(length=unit(0.3,"cm")), color = "dodgerblue2") +
#   annotate("text", x = 3.5, y = 9.5, label = "VCP up", color = "firebrick2", size = 2.8) + 
#   annotate("text", x = -3.5, y = 9.5, label = "VCP down", color = "dodgerblue2", size = 2.8)
# untx_vcp_vs_ctrl.iso.volcano
# 
# untx_vcp_vs_ctrl.iso$res %>% filter(padj < 0.05, gene_name %in% all.microglia.go)
# untx_vcp_vs_ctrl.iso$res %>% filter(padj < 0.05, gene_name %in% microglia_signature)
# untx_vcp_vs_ctrl.iso$res %>% filter(padj < 0.05, gene_name %in% RNA_BINDING_PROTEINS)
# 
# # GO terms
# untx_vcp_vs_ctrl.iso.gost <- untx_vcp_vs_ctrl.iso$res %>% filter(padj < 0.05) %>% select(gene_name, gene_id, log2FoldChange, padj, pvalue) %>% 
#   mutate(deg = case_when(padj < 0.05 & log2FoldChange > 0 ~ "VCP Up", padj < 0.05 & log2FoldChange < 0 ~ "VCP Down", TRUE ~ "None")) %>% 
#   group_split(deg) %>% map("gene_id") %>% gost()
#          
# untx_vcp_vs_ctrl.iso.gost_filt <- untx_vcp_vs_ctrl.iso.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM|WP")) %>% arrange( -log10(p_value) ) %>% 
#   mutate(query = case_when(query == "query_1" ~ "VCP\n Down", query == "query_2" ~ "VCP\n Up"), query = factor(query, levels = c("VCP\n Up", "VCP\n Down")))
# table(untx_vcp_vs_ctrl.iso.gost_filt$query)
# #    VCP\n Up VCP\n Down 
# #        646        274 
# untx_vcp_vs_ctrl.iso.dge_gprofiles_down <- untx_vcp_vs_ctrl.iso.gost_filt %>% filter(query == "VCP\n Down")
# untx_vcp_vs_ctrl.iso.dge_gprofiles_up <- untx_vcp_vs_ctrl.iso.gost_filt %>% filter(query == "VCP\n Up")
# 
# untx_vcp_vs_ctrl.iso.gost = gprofiler2rrvgo(untx_vcp_vs_ctrl.iso$res, query1 = "VCP Down", query2 = "VCP Up", GO_cat = "BP")
# untx_vcp_vs_ctrl.iso.rrvgo = rrvgo_plot(gprofiler2rrvgo = untx_vcp_vs_ctrl.iso.gost, n_terms = 10, remove_terms = NULL, colours = 4, cols =  c("firebrick2", "dodgerblue2", "forestgreen", "gold2"))
# untx_vcp_vs_ctrl.iso.rrvgo
#   
#   
# paste('"',unique(untx_vcp_vs_ctrl.iso.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
# down_list = c(
# "gliogenesis",
# "regulation of catalytic activity",
# "regulation of cell death",
# "positive regulation of gliogenesis",
# "NGF-stimulated transcription",
# "regulation of signaling",
# "regulation of response to wounding",
# "regulation of mononuclear cell migration",
# "defense response to other organism",
# "monocyte chemotaxis",
# "lymphocyte differentiation",
# "developmental process",
# "TNF related weak inducer of apoptosis (TWEAK) Signaling Pathway",
# "T cell migration",
# "positive regulation of mononuclear cell migration",
# "response to organonitrogen compound",
# "regulation of biological quality",
# "positive regulation of NF-kappaB transcription factor activity",
# "regulation of interleukin-1 beta production",
# "interleukin-1 beta production",
# "Hematopoietic cell lineage",
# "positive regulation of cytokine production involved in immune response",
# "Folate Metabolism",
# "positive regulation of interleukin-8 production",
# "natural killer cell chemotaxis",
# "regulation of lymphocyte migration",
# "positive regulation of tumor necrosis factor superfamily cytokine production",
# "positive regulation of T cell activation",
# "myeloid leukocyte differentiation",
# "myeloid leukocyte migration",
# "regulation of leukocyte differentiation",
# "negative regulation of response to stimulus",
# "system development",
# "positive regulation of smooth muscle cell proliferation",
# "neutrophil chemotaxis",
# "regulation of anatomical structure morphogenesis",
# "Yersinia infection",
# "Kaposi sarcoma-associated herpesvirus infection",
# "positive regulation of tumor necrosis factor production",
# "cellular response to organic cyclic compound",
# "cell surface receptor signaling pathway",
# "animal organ development",
# "Nuclear Receptors Meta-Pathway",
# "Chemokine signaling pathway",
# "lipopolysaccharide-mediated signaling pathway",
# "regulation of lymphocyte activation",
# "granulocyte migration",
# "transport",
# "Salmonella infection",
# "guanosine phosphorylase activity",
# "regulation of hydrolase activity",
# "positive regulation of DNA-binding transcription factor activity",
# "establishment of localization",
# "membrane microdomain",
# "membrane raft",
# "regulation of hemopoiesis",
# "Human Complement System",
# "granulocyte activation",
# "regulation of natural killer cell chemotaxis",
# "positive regulation of inflammatory response",
# "T cell differentiation",
# "response to interferon-gamma",
# "Pertussis",
# "positive regulation of GTPase activity",
# "neuroinflammatory response",
# "external side of plasma membrane",
# "Coronavirus disease - COVID-19",
# "regulation of GTPase activity",
# "positive regulation of cell population proliferation",
# "eosinophil migration",
# "Influenza A",
# "signaling receptor binding",
# "African trypanosomiasis",
# "epithelial cell apoptotic process",
# "regulation of neuroinflammatory response",
# "Vitamin B12 Metabolism",
# "response to hormone",
# "Human T-cell leukemia virus 1 infection",
# "plasma membrane",
# "Cytokines and Inflammatory Response",
# "granulocyte chemotaxis",
# "positive regulation of molecular function",
# "regulation of cell population proliferation",
# "extracellular space",
# "C-type lectin receptor signaling pathway",
# "cell periphery",
# "leukocyte cell-cell adhesion",
# "regulation of DNA-binding transcription factor activity",
# "extracellular region",
# "regulation of cellular component movement",
# "positive regulation of neuroinflammatory response",
# "eosinophil chemotaxis",
# "interleukin-6 production",
# "regulation of interleukin-6 production",
# "signaling receptor activator activity",
# "leukocyte chemotaxis",
# "positive regulation of lymphocyte activation",
# "positive regulation of production of molecular mediator of immune response",
# "lymphocyte migration",
# "regulation of response to external stimulus",
# "receptor ligand activity",
# "mononuclear cell differentiation",
# "regulation of cell migration",
# "positive regulation of interleukin-1 production",
# "Hepatitis B infection",
# "Cytokine Signaling in Immune system",
# "positive regulation of cell differentiation",
# "positive regulation of cell activation",
# "intrinsic component of plasma membrane",
# "regulation of leukocyte mediated immunity",
# "Non-genomic actions of 1,25 dihydroxyvitamin D3",
# "regulation of epithelial cell apoptotic process",
# "regulation of leukocyte cell-cell adhesion",
# "regulation of molecular function",
# "intracellular signal transduction",
# "cellular developmental process",
# "positive regulation of leukocyte activation",
# "Overview of nanoparticle effects",
# "regulation of cell motility",
# "positive regulation of cellular process",
# "regulation of cell adhesion",
# "cytokine production",
# "regulation of cytokine production",
# "regulation of response to stimulus",
# "Human cytomegalovirus infection",
# "positive regulation of interleukin-1 beta production",
# "miRNAs involvement in the immune response in sepsis",
# "cell differentiation",
# "cellular response to chemokine",
# "response to chemokine",
# "Cells and Molecules involved in local acute inflammatory response ",
# "Immune System",
# "integral component of plasma membrane",
# "chemokine receptor binding",
# "myeloid cell activation involved in immune response",
# "mononuclear cell migration",
# "regulation of cytokine production involved in immune response",
# "cytokine production involved in immune response",
# "Hepatitis B",
# "positive regulation of leukocyte migration",
# "positive regulation of developmental process",
# "cellular response to oxygen-containing compound",
# "Chemokine receptors bind chemokines",
# "leukocyte degranulation",
# "positive regulation of leukocyte cell-cell adhesion",
# "regulation of locomotion",
# "regulation of cell differentiation",
# "regulation of leukocyte activation",
# "cytokine receptor binding",
# "regulation of cell-cell adhesion",
# "cell-cell adhesion",
# "signal transduction",
# "chemokine-mediated signaling pathway",
# "positive regulation of cell adhesion",
# "positive regulation of cellular component movement",
# "Signaling by Interleukins",
# "lymphocyte activation",
# "positive regulation of cell motility",
# "Amoebiasis",
# "positive regulation of cell-cell adhesion",
# "biological adhesion",
# "cell adhesion",
# "regulation of immune effector process",
# "T cell activation",
# "response to endogenous stimulus",
# "regulation of production of molecular mediator of immune response",
# "response to interleukin-1",
# "positive regulation of cell migration",
# "response to bacterium",
# "regulation of leukocyte migration",
# "myeloid leukocyte mediated immunity",
# "regulation of multicellular organismal development",
# "response to tumor necrosis factor",
# "Legionellosis",
# "cytokine activity",
# "SARS-CoV-2 Innate Immunity Evasion and Cell-specific immune response",
# "cell surface",
# "chemokine activity",
# "response to biotic stimulus",
# "positive regulation of locomotion",
# "cellular response to tumor necrosis factor",
# "response to oxygen-containing compound",
# "production of molecular mediator of immune response",
# "response to stress",
# "cellular response to interleukin-1",
# "cellular response to stimulus",
# "response to lipid",
# "TNF signaling pathway",
# "response to external biotic stimulus",
# "response to other organism",
# "response to molecule of bacterial origin",
# "LTF danger signal response pathway",
# "myeloid leukocyte activation",
# "positive regulation of interleukin-6 production",
# "regulation of cell activation",
# "taxis",
# "chemotaxis",
# "Photodynamic therapy-induced NF-kB survival signaling",
# "positive regulation of immune effector process",
# "localization",
# "NF-kappa B signaling pathway",
# "Chagas disease",
# "positive regulation of metabolic process",
# "response to organic cyclic compound",
# "AGE-RAGE signaling pathway in diabetic complications",
# "response to lipopolysaccharide",
# "Fluid shear stress and atherosclerosis",
# "biological process involved in interspecies interaction between organisms",
# "Platelet-mediated interactions with vascular and circulating cells",
# "positive regulation of biological process",
# "cellular response to molecule of bacterial origin",
# "Regulation of toll-like receptor signaling pathway",
# "hematopoietic or lymphoid organ development",
# "Interleukin-4 and Interleukin-13 signaling",
# "cellular response to lipid",
# "regulation of developmental process",
# "cell communication",
# "cellular response to lipopolysaccharide",
# "movement of cell or subcellular component",
# "hemopoiesis",
# "cell chemotaxis",
# "immune system development",
# "positive regulation of macromolecule metabolic process",
# "cellular response to biotic stimulus",
# "positive regulation of cytokine production",
# "positive regulation of gene expression",
# "response to stimulus",
# "IL1 and megakaryocytes in obesity",
# "chemokine production",
# "regulation of chemokine production",
# "IL-18 signaling pathway",
# "signaling",
# "Viral protein interaction with cytokine and cytokine receptor",
# "cell migration",
# "Lipid and atherosclerosis",
# "Toll-like Receptor Signaling Pathway",
# "IL-17 signaling pathway",
# "regulation of immune system process",
# "cell motility",
# "localization of cell",
# "positive regulation of multicellular organismal process",
# "leukocyte differentiation",
# "positive regulation of immune system process",
# "leukocyte mediated immunity",
# "Cytokine-cytokine receptor interaction",
# "response to organic substance",
# "positive regulation of chemokine production",
# "COVID-19 Adverse Outcome Pathway",
# "cytokine-mediated signaling pathway",
# "leukocyte migration",
# "response to chemical",
# "immune response",
# "locomotion",
# "response to cytokine",
# "regulation of multicellular organismal process",
# "defense response",
# "Rheumatoid arthritis",
# "response to external stimulus",
# "cellular response to organic substance",
# "Spinal Cord Injury",
# "cell activation involved in immune response",
# "leukocyte activation involved in immune response",
# "cellular response to cytokine stimulus",
# "cellular response to chemical stimulus",
# "Lung fibrosis",
# "Toll-like receptor signaling pathway",
# "Interleukin-10 signaling",
# "Malaria",
# "immune effector process",
# "immune system process",
# "leukocyte activation",
# "cell activation",
# "inflammatory response")
# untx_vcp_vs_ctrl.iso.dge_gprofiles_filt_down <- untx_vcp_vs_ctrl.iso.gost_filt %>% filter(query == "VCP\n Down") %>% filter(term_name %in% down_list)
# # untx_vcp_vs_ctrl.iso.dge_gprofiles_filt_down$term_name = gsub("Viral protein interaction with cytokine and cytokine receptor", "Viral interaction with cytokine receptor", untx_vcp_vs_ctrl.iso.dge_gprofiles_filt_down$term_name) %>% gsub("E. coli ", "", .) %>% gsub("Absence of bactericidal oxidative respiratory burst in phagocytes", "Absent respiratory burst in phagocytes", .) %>%  gsub("Cross-presentation of particulate exogenous antigens [(]phagosomes[)]", "Phagosome antigen cross-presentation", .)
# paste('"',unique(untx_vcp_vs_ctrl.iso.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
# up_list = c(
# "localization of cell",
# "cell motility",
# "locomotion",
# "blood vessel morphogenesis",
# "collagen fibril organization",
# "cellular developmental process",
# "cell differentiation",
# "cell surface receptor signaling pathway",
# "regulation of multicellular organismal development",
# "cell junction",
# "structural molecule activity",
# "movement of cell or subcellular component",
# "anchoring junction",
# "cell migration",
# "cell adhesion molecule binding",
# "regulation of multicellular organismal process",
# "blood vessel development",
# "enzyme linked receptor protein signaling pathway",
# "regulation of developmental process",
# "extracellular space",
# "tube morphogenesis",
# "vasculature development",
# "animal organ development",
# "extracellular region",
# "multicellular organismal process",
# "anatomical structure formation involved in morphogenesis",
# "tube development",
# "tissue development",
# "animal organ morphogenesis",
# "cell periphery",
# "developmental process",
# "circulatory system development",
# "Extracellular matrix organization",
# "anatomical structure development",
# "multicellular organism development",
# "system development",
# "extracellular matrix structural constituent",
# "biological adhesion",
# "cell adhesion",
# "external encapsulating structure organization",
# "extracellular structure organization",
# "extracellular matrix organization",
# "collagen-containing extracellular matrix",
# "anatomical structure morphogenesis",
# "external encapsulating structure",
# "extracellular matrix")
# untx_vcp_vs_ctrl.iso.dge_gprofiles_filt_up <- untx_vcp_vs_ctrl.iso.gost_filt %>% filter(query == "VCP\n Up")  %>% filter(term_name %in% up_list)
# untx_vcp_vs_ctrl.iso.dge_gprofiles_filtered = bind_rows(untx_vcp_vs_ctrl.iso.dge_gprofiles_filt_up, untx_vcp_vs_ctrl.iso.dge_gprofiles_filt_down) %>% mutate(query = fct_relevel(query,"VCP\n Up"))
# untx_vcp_vs_ctrl.iso_dge.go_curated <- curated_profiles(untx_vcp_vs_ctrl.iso.dge_gprofiles_filtered, gprofiles =, cols = c("firebrick2", "dodgerblue2")) + theme(strip.text.y.right = element_text(angle = 0))# + theme(legend.position = "none", axis.text = element_text(size = 10))
# untx_vcp_vs_ctrl.iso_dge.go_curated
# 
# # untx_vcp_vs_ctrl.iso_gprofiler2rrvgo = gprofiler2rrvgo(deseq.res = untx_vcp_vs_ctrl.iso$res, query1 = "VCP\n Down", query2 = "VCP\n Up", GO_cat = "BP")
# # # untx_vcp_vs_ctrl.iso_gprofiler2rrvgo$parentTerm <- gsub("SRP-dependent cotranslational ", "", untx_vcp_vs_ctrl.iso_gprofiler2rrvgo$parentTerm) %>% gsub("nuclear-transcribed mRNA catabolic process, ", "", .)
# # untx_vcp_vs_ctrl.iso.go_rrvgo = rrvgo_plot(gprofiler2rrvgo = untx_vcp_vs_ctrl.iso_gprofiler2rrvgo, n_terms = 10, remove_terms = NULL) + 
# #   theme(axis.text = element_text(size=10), strip.text.y.right = element_text(angle = 0))
# # untx_vcp_vs_ctrl.iso.go_rrvgo

```


## Phagocytosis

Phagocytosis functional assay images
Phagocytosis functional assay quantification
TMRM representative images
TMRM quantification
mitochondrial mass quantification

```{r}
# https://www.dropbox.com/work/PataniR/Ben/Microglia%20figures

# import with magick 
FigS2G_Phag_image.png = magick::image_read(here(proj_path,"ben-figures/FigS2G_Phag_image.png"))
FigS2G_Phag_image.gg = ggdraw() + draw_image(FigS2G_Phag_image.png)
# FigS2G_Phag_image.gg

FigS2H_Phag_quant.png = magick::image_read(here(proj_path,"ben-figures/FigS2H_Phag_quant.png"))
FigS2H_Phag_quant.gg = ggdraw() + draw_image(FigS2H_Phag_quant.png)
# FigS2H_Phag_quant.gg

FigS2I_TMRM_image.png = magick::image_read(here(proj_path,"ben-figures/FigS2I_TMRM_image.png"))
FigS2I_TMRM_image.gg = ggdraw() + draw_image(FigS2I_TMRM_image.png)
# FigS2I_TMRM_image.gg

# FigS2J_TMRM_intensity.png = magick::image_read(here(proj_path,"ben-figures/FigS2J_TMRM_intensity.png"))
# FigS2J_TMRM_intensity.gg = ggdraw() + draw_image(FigS2J_TMRM_intensity.png)
# # FigS2J_TMRM_intensity.gg
# 
# FigS2K_TMRM_mass.png = magick::image_read(here(proj_path,"ben-figures/FigS2K_TMRM_mass.png"))
# FigS2K_TMRM_mass.gg = ggdraw() + draw_image(FigS2K_TMRM_mass.png)
# # FigS2K_TMRM_mass.gg


## TMRM
tmrm_data = read_excel(here(proj_path, "expression/lysotracker/lysotracker_tmrm.xlsx"), sheet = "assays") %>%
  filter(grepl("TMRM|Mitochondrial",assay))
tmrm_data %>% glimpse
tmrm_data %>% my_summarise(value, assay,  group) %>% print(n=Inf)

tmrm.mean_intensity.violin = violin_plot(data = filter(tmrm_data, assay == "TMRM mean intensity fold change"), color_by = "group", continuous = "value", cols = brewer.pal(6, "Paired"), ylabel = "TMRM mean intensity\nFold change", ylims = c(0.78,1.4), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 1.4, tip_length = 0.01)
tmrm.mean_intensity.violin

mito.mass.violin = violin_plot(data = filter(tmrm_data, assay == "Mitochondrial mass (% cytoplasmic area)"), color_by = "group", continuous = "value", cols = brewer.pal(6, "Paired"), ylabel = "Mitochondrial mass\n(% cytoplasmic area)", ylims = c(32,39), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 38.5, tip_length = 0.01)
mito.mass.violin


```

PROGENy VCP vs CTRL
DoroThEA VCP vs CTRL
GSEA phagocytosis
pHrodo image
pHrodo quantification
GSEA mitochondria
TMRM image
TMRM intensity
TMRM mito mass
Isogenic correlation
Isogenic Venn

## Merge

```{r}
figure.s2.merged = plot_grid(
  plot_grid(untx_vcp_vs_ctrl.all.noiso_list_t_stat_plot, untx_vcp_vs_ctrl_all_noiso_venn, ncol = 2, labels = LETTERS[1:2]), 
  plot_grid(untx_vcp_vs_ctrl_progeny.pathwayActivity, untx_vcp_vs_ctrl.dorothea.volcano, ncol = 2, labels = LETTERS[3:4]), 
  plot_grid(untx_vcp_vs_ctrl.phagocytosis.gseaplot, FigS2G_Phag_image.gg, FigS2H_Phag_quant.gg, ncol = 3, rel_widths = c(0.7,1,1.2),labels = LETTERS[5:7]),
  plot_grid(untx_vcp_vs_ctrl.mitochondrial.gseaplot, FigS2I_TMRM_image.gg,tmrm.mean_intensity.violin, mito.mass.violin, ncol=4, rel_widths = c(0.9,2,0.5,0.5), labels = LETTERS[8:11]),
  nrow = 4)
# figure.s2.merged
ggsave(figure.s2.merged, filename = here(proj_path, "expression/deseq2/fig.s2.merged.phagocytosis_mito.png"), height = 12.5, width = 9)
ggsave(figure.s2.merged, filename = here(proj_path, "expression/deseq2/fig.s2.merged.phagocytosis_mito.pdf"), height = 12.5, width = 9)

```


# Fig S3 iPSC & pMG Lysosensor & Lysotracker

```{r}
lysotracker_data = read_excel(here(proj_path, "expression/lysotracker/Rebuttal_2_Data_for_Oli.xlsx"), sheet = "Lysosensor and Lysotracker") %>% clean_names() %>% mutate(
  celltype = case_when(cell_type == "pMG" ~ "Macrophage Precursors",TRUE~cell_type), celltype = factor(celltype, levels = c("iPSC","Macrophage Precursors")), condition = paste0(cell_type,"\n",genotype), condition = factor(condition, levels = c("iPSC\nCTRL", "iPSC\nVCP", "pMG\nCTRL", "pMG\nVCP")))
lysotracker_data %>% glimpse

# A - iPSC & pMG Lysosensor mean intensity quantification
filter(lysotracker_data, !is.na(lysosensor_mean_intensity)) %>% my_summarise(lysosensor_mean_intensity, genotype) %>% print(n=Inf)
# for stats, adjust for experiment
lysosensor_mean_intensity.pMG.fitg = glm(lysosensor_mean_intensity ~ genotype + experiment, data= filter(lysotracker_data, cell_type == "pMG"), family = gaussian)
summ(lysosensor_mean_intensity.pMG.fitg, digits = 5)$coeftable 
#                    Est.       S.E.    t val.           p
# (Intercept)  0.93624303 0.07446499 12.572930 3.06631e-11
# lysosensor_mean_intensity.ipsc.fitg = glm(lysosensor_mean_intensity ~ genotype + experiment, data= filter(lysotracker_data, cell_type == "iPSC"), family = gaussian)
summ(lysosensor_mean_intensity.ipsc.fitg, digits = 5)$coeftable 
#                    Est.       S.E.     t val.            p
# (Intercept)  0.91201880 0.17092028  5.3359309 6.845488e-06
# genotypeVCP -0.05850725 0.05439075 -1.0756839 2.898706e-01

lysosensor_mean_intensity.violin = violin_plot(data = lysotracker_data, color_by = "genotype", continuous = "lysosensor_mean_intensity", ylims = c(0.7,1.5), cols = c("firebrick2", "dodgerblue2"), ylabel = "Lysosensor mean intensity\nFold change", facet_by = "celltype", plot_violin = FALSE, arrow_labels = FALSE, dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif")
lysosensor_mean_intensity.violin

# lysosensor_mean_intensity.violin = violin_plot(data = filter(lysotracker_data, !is.na(lysosensor_mean_intensity)), color_by = "genotype", continuous = "lysosensor_mean_intensity", cols = brewer.pal(6, "Paired"), ylabel = "Lysosensor mean intensity\nFold change", ylims = c(0.7,1.5), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 1.5, tip_length = 0.01)
# lysosensor_mean_intensity.violin


# B - iPSC & pMG Lysotracker mean intensity quantification
filter(lysotracker_data, !is.na(lysotracker_mean_intensity)) %>% my_summarise(lysotracker_mean_intensity, genotype) %>% print(n=Inf)
# for stats, adjust for experiment
lysotracker_mean_intensity.pMG.fitg = glm(lysotracker_mean_intensity ~ genotype + experiment, data= filter(lysotracker_data, cell_type == "pMG"), family = gaussian)
summ(lysotracker_mean_intensity.pMG.fitg, digits = 5)$coeftable 
#                  Est.       S.E.    t val.            p
# (Intercept)  0.85244791 0.13388741  6.366901 2.596080e-06
# genotypeVCP -0.13733231 0.09394162 -1.461890 1.585756e-01
lysotracker_mean_intensity.ipsc.fitg = glm(lysotracker_mean_intensity ~ genotype + experiment, data= filter(lysotracker_data, cell_type == "iPSC"), family = gaussian)
summ(lysotracker_mean_intensity.ipsc.fitg, digits = 5)$coeftable 
#                    Est.       S.E.    t val.            p
# (Intercept) 0.952752356 0.17918799 5.3170549 7.238055e-06
# genotypeVCP 0.033498525 0.05702172 0.5874695 5.608870e-01

lysotracker_mean_intensity.violin = violin_plot(data = lysotracker_data, color_by = "genotype", continuous = "lysotracker_mean_intensity", ylims = c(0,2), cols = c("firebrick2", "dodgerblue2"), ylabel = "Lysotracker mean intensity\nFold change", facet_by = "celltype", plot_violin = FALSE, arrow_labels = FALSE, dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 2, tip_length = 0.01)
lysotracker_mean_intensity.violin

# lysotracker_mean_intensity.violin = violin_plot(data = filter(lysotracker_data, !is.na(lysotracker_mean_intensity)), color_by = "genotype", continuous = "lysotracker_mean_intensity", cols = brewer.pal(6, "Paired"), ylabel = "Lysotracker mean intensity\nFold change", ylims = c(0,2), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 2, tip_length = 0.01)
# lysotracker_mean_intensity.violin


# C - pMG Lysotracker spot area quantification
filter(lysotracker_data, !is.na(lysotracker_spot_area)) %>% my_summarise(lysotracker_spot_area, genotype) %>% print(n=Inf)
lysotracker_spot_area.pMG.fitg = glm(lysotracker_spot_area ~ genotype + experiment, data= filter(lysotracker_data, cell_type == "pMG"), family = gaussian)
summ(lysotracker_spot_area.pMG.fitg, digits = 5)$coeftable 
#                   Est.       S.E.    t val.            p
# (Intercept) 0.90799585 0.05170800 17.560065 4.971725e-14
# genotypeVCP 0.05554275 0.03628073  1.530916 1.407162e-01
lysotracker_spot_area.ipsc.fitg = glm(lysotracker_spot_area ~ genotype + experiment, data= filter(lysotracker_data, cell_type == "iPSC"), family = gaussian)
summ(lysotracker_spot_area.ipsc.fitg, digits = 5)$coeftable 
#                    Est.       S.E.     t val.            p
# (Intercept)  1.06357307 0.19080999  5.5739904 3.388053e-06
# genotypeVCP -0.03187031 0.06072011 -0.5248723 6.031806e-01

lysotracker_spot_area.violin = violin_plot(data = lysotracker_data, color_by = "genotype", continuous = "lysotracker_spot_area", ylims = c(0,2), cols = c("firebrick2", "dodgerblue2"), ylabel = "Lysotracker spot area (um2)", facet_by = "celltype", plot_violin = FALSE, arrow_labels = FALSE, dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 2, tip_length = 0.01)
lysotracker_spot_area.violin

# lysotracker_spot_area.violin = violin_plot(data = filter(lysotracker_data, !is.na(lysotracker_spot_area)), color_by = "genotype", continuous = "lysotracker_spot_area", cols = brewer.pal(6, "Paired"), ylabel = "Lysotracker spot area (um2)", ylims = c(0,2), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 2, tip_length = 0.01)
# lysotracker_spot_area.violin


# D - pMG Lysotracker spot number per area cytoplasm quantification
filter(lysotracker_data, !is.na(lysotracker_spot_number_per_area_cytoplasm)) %>% my_summarise(lysotracker_spot_number_per_area_cytoplasm, genotype) %>% print(n=Inf)
lysotracker_spot_number_per_area_cytoplasm.pMG.fitg = glm(lysotracker_spot_number_per_area_cytoplasm ~ genotype + experiment, data= filter(lysotracker_data, cell_type == "pMG"), family = gaussian)
summ(lysotracker_spot_number_per_area_cytoplasm.pMG.fitg, digits = 5)$coeftable 
#                     Est.       S.E.     t val.            p
# (Intercept)  1.009606616 0.05023343 20.0983033 3.400433e-15
# genotypeVCP -0.072248274 0.03524610 -2.0498230 5.307583e-02
lysotracker_spot_number_per_area_cytoplasm.ipsc.fitg = glm(lysotracker_spot_number_per_area_cytoplasm ~ genotype + experiment, data= filter(lysotracker_data, cell_type == "iPSC"), family = gaussian)
summ(lysotracker_spot_number_per_area_cytoplasm.ipsc.fitg, digits = 5)$coeftable 
#                     Est.       S.E.     t val.            p
# (Intercept)  0.977328467 0.18157244  5.3825817 5.964129e-06
# genotypeVCP -0.008170962 0.05778051 -0.1414138 8.884028e-01

lysotracker_spot_number_per_area_cytoplasm.violin = violin_plot(data = lysotracker_data, color_by = "genotype", continuous = "lysotracker_spot_number_per_area_cytoplasm", ylims = c(0.4,1.5), cols = c("firebrick2", "dodgerblue2"), ylabel = "Lysotracker spot number / area", facet_by = "celltype", plot_violin = FALSE, arrow_labels = FALSE, dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 1.5, tip_length = 0.01)
lysotracker_spot_number_per_area_cytoplasm.violin

# lysotracker_spot_number_per_area_cytoplasm.violin = violin_plot(data = filter(lysotracker_data, !is.na(lysotracker_spot_number_per_area_cytoplasm)), color_by = "genotype", continuous = "lysotracker_spot_number_per_area_cytoplasm", cols = brewer.pal(6, "Paired"), ylabel = "Lysotracker spot number / area", ylims = c(0,1.5), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 1.5, tip_length = 0.01)
# lysotracker_spot_number_per_area_cytoplasm.violin


# E - iPSC Lysosensor mean intensity quantification
# F - iPSC Lysotracker mean intensity quantification
# G - iPSC Lysotracker spot area quantification
# H - iPSC Lysotracker spot number per area cytoplasm quantification


figS3.merged <- plot_grid(
  lysosensor_mean_intensity.violin, lysotracker_mean_intensity.violin, lysotracker_spot_area.violin, lysotracker_spot_number_per_area_cytoplasm.violin,  nrow = 2, ncol = 2, labels = "AUTO")
figS3.merged
ggsave(figS3.merged, filename = here(proj_path, "expression/deseq2/figS3.merged.lysotracker_iPSC_pMG.png"), height = 6, width = 8.25)
ggsave(figS3.merged, filename = here(proj_path, "expression/deseq2/figS3.merged.lysotracker_iPSC_pMG.pdf"), height = 6, width = 8.25)



```





# Fig 4: LPS & VCP comparison 

A - Schematic VCP and LPS comparison
B - Global correlation LPS vs VCP related gene expression changes
C - Venn diagram DEGs LPS vs VCP related gene expression changes
D - GO LPS vs VCP related gene expression changes
E - Global correlation LPS vs VCP related protein expression changes
F - Volcano plot VCP+LPS 
G - Progeny VCP+LPS
H - Volcano plot hypoxia pathway progeny
I - Volcano plot Dorothea
J - Volcano plot HIF1A regulon
K -  Bar plots of any significantly changed LPS treated in CTRL vs VCP microglia


ctrl_lps_vs_untx & untx_vcp_vs_ctrl

```{r}
# https://www.dropbox.com/work/PataniR/Ben/Microglia%20figures

# import with magick 
Fig3A_schematic.png = magick::image_read(here(proj_path,"ben-figures/Fig3A_schematic.png"))
Fig3A_schematic.gg = ggdraw() + draw_image(Fig3A_schematic.png)
# Fig3A_schematic.gg
```

## RNAseq Scatter & Venn

```{r}
ctrl_lps_vs_untx$res %>% filter(padj < 0.05) # 8097
ctrl_lps_vs_untx$res %>% filter(padj< 0.05, stat > 0) # 4169
ctrl_lps_vs_untx$res %>% filter(padj< 0.05, stat < 0) # 3928

untx_vcp_vs_ctrl_ctrl_lps_vs_untx.labels = select(ctrl_lps_vs_untx$res, gene_id, gene_name, ctrl_lps_vs_untx.log2FoldChange = log2FoldChange) %>% left_join(select(untx_vcp_vs_ctrl$res, gene_id, gene_name, untx_vcp_vs_ctrl.log2FoldChange = log2FoldChange)) %>% filter((untx_vcp_vs_ctrl.log2FoldChange > 2.5 & ctrl_lps_vs_untx.log2FoldChange > 2.5) | (untx_vcp_vs_ctrl.log2FoldChange < -2.5 & ctrl_lps_vs_untx.log2FoldChange < -2.5), gene_name %in% all.reactivity.go)
untx_vcp_vs_ctrl_ctrl_lps_vs_untx.list = list("LPS vs Untreated" = ctrl_lps_vs_untx$res, "VCP vs CTRL" = untx_vcp_vs_ctrl$res)
untx_vcp_vs_ctrl_ctrl_lps_vs_untx_log2FoldChange.corr = plot_de_correlation(model_list1 = untx_vcp_vs_ctrl_ctrl_lps_vs_untx.list, model_list2 = untx_vcp_vs_ctrl_ctrl_lps_vs_untx.list, mutation1 = "VCP vs CTRL", mutation2 = "LPS vs Untreated", gene_labels = pull(untx_vcp_vs_ctrl_ctrl_lps_vs_untx.labels,gene_name), plot_corr_line = TRUE, title = "RNA sequencing", col = "log2FoldChange") +ylim(c(-8,8)) + xlim(c(-6,6)) +
  annotate(geom="text", x=3, y=8, label="LPS & VCP both Up", color="black", size = 3.2)  + annotate(geom="text", x=-3, y=-8, label="LPS & VCP both Down", color="black", size = 3.2)
# untx_vcp_vs_ctrl_ctrl_lps_vs_untx_log2FoldChange.corr


ctrl_lps_vs_untx.res.dge.up <- ctrl_lps_vs_untx$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
ctrl_lps_vs_untx.res.dge.down <- ctrl_lps_vs_untx$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
untx_vcp_vs_ctrl.res.dge.up <- untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
untx_vcp_vs_ctrl.res.dge.down <- untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)

# 4 way overlap
untx_vcp_vs_ctrl_ctrl_lps_vs_untx_venn = ggvenn(list(`LPS\ndown` = ctrl_lps_vs_untx.res.dge.down, `LPS up` = ctrl_lps_vs_untx.res.dge.up, `VCP up` = untx_vcp_vs_ctrl.res.dge.up, `VCP\ndown` = untx_vcp_vs_ctrl.res.dge.down),
                                                fill_color = c("dodgerblue3", "firebrick3", "firebrick1", "dodgerblue1"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.7) + ggtitle("RNA sequencing") + theme(plot.title = element_text(hjust = 0.5))# + theme_oz()
# untx_vcp_vs_ctrl_ctrl_lps_vs_untx_venn

# # 2 way overlap VCP & LPS
# # ggvenn
# ctrl_lps_vs_untx.untx_vcp_vs_ctrl.VCP.up_genes <- list("LPS" = ctrl_lps_vs_untx.res.dge.up, "VCP" = untx_vcp_vs_ctrl.res.dge.up)
# ctrl_lps_vs_untx.untx_vcp_vs_ctrl.VCP.down_genes <- list("LPS" = ctrl_lps_vs_untx.res.dge.down, " VCP" = untx_vcp_vs_ctrl.res.dge.down)
# untx_vcp_vs_ctrl_ctrl_lps_vs_untx_venn_up <- ggvenn(ctrl_lps_vs_untx.untx_vcp_vs_ctrl.VCP.up_genes, fill_color = c("dodgerblue3", "firebrick1", "forestgreen"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.7) + labs(subtitle = "Overlap Up")  + theme(plot.subtitle = element_text(hjust = 0.5), plot.margin = unit(c(0, -1, -1, -1), "lines"))
# untx_vcp_vs_ctrl_ctrl_lps_vs_untx_venn_down <- ggvenn(ctrl_lps_vs_untx.untx_vcp_vs_ctrl.VCP.down_genes, fill_color = c("dodgerblue3", "firebrick1", "forestgreen"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.7) + labs(subtitle = "Overlap Down") + theme(plot.subtitle = element_text(hjust = 0.5), plot.margin = unit(c(0, -1, -1, -1), "lines"))
# untx_vcp_vs_ctrl_ctrl_lps_vs_untx_venn = ggarrange(untx_vcp_vs_ctrl_ctrl_lps_vs_untx_venn_up, untx_vcp_vs_ctrl_ctrl_lps_vs_untx_venn_down, nrow = 2)
# untx_vcp_vs_ctrl_ctrl_lps_vs_untx_venn



ctrl_lps_vs_untx.dge <- ctrl_lps_vs_untx$res %>% select(gene_name, gene_id, log2FC.LPS = log2FoldChange, FDR.LPS = padj)
untx_vcp_vs_ctrl.dge <- untx_vcp_vs_ctrl$res %>% select(gene_name, gene_id, log2FC.VCP = log2FoldChange, FDR.VCP = padj)
ctrl_lps_vs_untx_untx_vcp_vs_ctrl.dge <- full_join(ctrl_lps_vs_untx.dge, untx_vcp_vs_ctrl.dge, by = c("gene_name", "gene_id")) %>%
  mutate(overlap = case_when(FDR.LPS < 0.05 & FDR.VCP < 0.05 ~ "overlapping", # down in LPS
                                                        FDR.LPS < 0.05 ~ "LPS specific",
                                                        FDR.VCP < 0.05 ~ "VCP specific", TRUE ~ "None"),
                                    overlap_direction = case_when(FDR.LPS < 0.05 & log2FC.LPS > 0 & FDR.VCP < 0.05 & log2FC.VCP > 0 ~ "overlapping up", # down in LPS
                                                                  FDR.LPS < 0.05 & log2FC.LPS < 0 & FDR.VCP < 0.05 & log2FC.VCP < 0 ~ "overlapping down",
                                                                  FDR.LPS < 0.05 & log2FC.LPS < 0 & FDR.VCP < 0.05 & log2FC.VCP > 0 ~ "LPS down, VCP up", # down in LPS
                                                                  FDR.LPS < 0.05 & log2FC.LPS > 0 & FDR.VCP < 0.05 & log2FC.VCP < 0 ~ "LPS up, VCP down",
                                                                  FDR.LPS < 0.05 & log2FC.LPS > 0 ~ "LPS specific up",
                                                                  FDR.LPS < 0.05 & log2FC.LPS < 0 ~ "LPS specific down",
                                                                  FDR.VCP < 0.05 & log2FC.VCP > 0 ~ "VCP specific up",
                                                                  FDR.VCP < 0.05 & log2FC.VCP < 0 ~ "VCP specific down",
                                                                  TRUE ~ "None"),
                                    overlap_direction = factor(overlap_direction, levels = c("overlapping up", "overlapping down", "LPS down, VCP up", "LPS up, VCP down", "LPS specific up", "LPS specific down", "VCP specific up", "VCP specific down", "None")),
                                    overlap = factor(overlap, levels = c("overlapping",  "LPS specific", "VCP specific", "None"))) %>%
  arrange(overlap_direction)
table(ctrl_lps_vs_untx_untx_vcp_vs_ctrl.dge$overlap)
table(ctrl_lps_vs_untx_untx_vcp_vs_ctrl.dge$overlap_direction)
# ctrl_lps_vs_untx_untx_vcp_vs_ctrl.dge

cor.test(x = ctrl_lps_vs_untx_untx_vcp_vs_ctrl.dge$log2FC.LPS, y = ctrl_lps_vs_untx_untx_vcp_vs_ctrl.dge$log2FC.VCP) # 0.094
t.test( x = ctrl_lps_vs_untx_untx_vcp_vs_ctrl.dge$log2FC.LPS, y = ctrl_lps_vs_untx_untx_vcp_vs_ctrl.dge$log2FC.VCP, paired = TRUE )
# 
# untx_vcp_vs_ctrl.VCP_LPS_scatter <- ggscatter(ctrl_lps_vs_untx_untx_vcp_vs_ctrl.dge, x = "log2FC.LPS", y = "log2FC.VCP", color = "overlap_direction", palette = c("firebrick2", "dodgerblue2", "forestgreen", "gold2", "grey", "grey", "grey", "grey", "grey"),  cor.coef = TRUE, cor.method = "pearson", xlab = expression(RNA~seq~Log[2]~Fold~Change~LPS:Untreated), ylab = expression(RNA~seq~Log[2]~Fold~Change~VCP:CTRL), size = 0.5, cor.coeff.args = list(label.x = -5.5, label.y = 4, label.sep="\n")) +
#           geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") +  
#     coord_cartesian(xlim = c(-6,6), ylim = c(-5,5)) +
#     geom_smooth(data = ctrl_lps_vs_untx_untx_vcp_vs_ctrl.dge, aes(x = log2FC.LPS, y = log2FC.VCP), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 2) + theme(legend.position = "none") +
#     geom_text_repel(data = filter(ctrl_lps_vs_untx_untx_vcp_vs_ctrl.dge, overlap == "overlapping", gene_name %in% all.reactivity.go, abs(log2FC.VCP) > 0.5, abs(log2FC.LPS) > 1), aes(x = log2FC.LPS, y = log2FC.VCP, label = gene_name), 
#                     fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.8, max.overlaps = 30) +  
#   annotate(geom="text", x=3, y=5, label="LPS & VCP both Up", color="firebrick2", size = 3.2)  + annotate(geom="text", x=-3, y=-5, label="LPS & VCP both Down", color="dodgerblue2", size = 3.2)
# untx_vcp_vs_ctrl.VCP_LPS_scatter



# Overlap tested using Fisher's exact test (alternative=greater)
LPS.DGE.up.genes <- ctrl_lps_vs_untx$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
VCP.DGE.up.genes <- untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
genome.size <- ctrl_lps_vs_untx$res %>% distinct(gene_name) %>% nrow
newGeneOverlap(LPS.DGE.up.genes,  VCP.DGE.up.genes,genome.size=genome.size) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=4169
# listB size=110
# Intersection size=15
# Overlapping p-value=8.4e-03
# Jaccard Index=0.0

LPS.DGE.down.genes <- ctrl_lps_vs_untx$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
VCP.DGE.down.genes <- untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
genome.size <- ctrl_lps_vs_untx$res %>% distinct(gene_name) %>% nrow
newGeneOverlap(LPS.DGE.down.genes, VCP.DGE.down.genes,genome.size=genome.size) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=3927
# listB size=73
# Intersection size=21
# Overlapping p-value=4.2e-09
# Jaccard Index=0.0
```


GO terms

```{r}
ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost = ctrl_lps_vs_untx_untx_vcp_vs_ctrl.dge %>% filter(overlap == "overlapping") %>% split(.$overlap_direction) %>%  map("gene_id") %>% gost()
ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt <- ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% 
  mutate(query = case_when(query == "overlapping down" ~ "LPS down\n VCP down", query == "overlapping up" ~ "LPS up\n VCP up", TRUE ~ query), query = gsub(",", "\n", query), term_name = as.factor(term_name)) 
ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gprofiles_down <- ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt %>% filter(query == "LPS down\n VCP down")
ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gprofiles_up <- ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt %>% filter(query == "LPS up\n VCP up")
# ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gprofiles_updown <- ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt %>% filter(query == "LPS up\n VCP down")
# ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gprofiles_downup <- ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt %>% filter(query == "LPS down\n VCP up")

paste('"',unique(ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
LPS.VCP.overlap.up_curated_list <- c(
"IL10-IL10RA complex",
"regulation of cell death",
"mitochondrial outer membrane",
# "negative regulation of cellular process",
"lymphocyte apoptotic process",
"regulation of programmed cell death",
# "regulation of establishment of protein localization",
"regulation of apoptotic process",
"regulation of protein transport",
# "regulation of lymphocyte apoptotic process",
"leukocyte apoptotic process")
# "regulation of leukocyte apoptotic process"
ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt.up <- ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt %>% filter(term_name %in% LPS.VCP.overlap.up_curated_list)
ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt.up$term_name <- gsub("regulation of ", "", ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt.up$term_name)
paste('"',unique(ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
LPS.VCP.overlap.down_curated_list <- c(
"nucleus")
ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt.down <- ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt %>% filter(term_name %in% LPS.VCP.overlap.down_curated_list)
# paste('"',unique(ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gprofiles_updown$term_name),'",', sep = "", collapse = '\n') %>% cat()
# LPSup.VCPdown.overlap_curated_list <- c("Toll-like receptor signaling pathway",
# "lymphocyte migration",
# "Rheumatoid arthritis",
# "response to chemokine",
# "Chemokine signaling pathway",
# "regulation of lymphocyte chemotaxis",
# # "chemokine-mediated signaling pathway",
# "monocyte chemotaxis",
# # "Viral protein interaction with cytokine and cytokine receptor",
# # "chemokine receptor binding",
# "natural killer cell chemotaxis",
# "Interleukin-10 signaling",
# "chemokine activity")
# ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt.updown <- ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt %>% filter(term_name %in% LPSup.VCPdown.overlap_curated_list)
paste('"',unique(ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt.up$term_name),'",', sep = "", collapse = '\n') %>% cat()
LPSdown.VCPup.overlap_curated_list = c("Insulin signaling pathway",
"phosphatidylinositol 3-kinase binding",
# "Insulin resistance",
"regulation of protein autophosphorylation")
ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt.downup <- ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt %>% filter(term_name %in% LPSdown.VCPup.overlap_curated_list)
LPS.VCP.overlap.up_gprofiles_filt.up_down <- bind_rows(ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt.up, ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt.down)  %>%#, ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt.updown, ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt.downup) %>% 
  mutate(query = factor(query, levels = c("LPS up\n VCP up", "LPS down\n VCP down")), term_name = gsub("regulation of ", "", term_name))#, "LPS up\n VCP down", "LPS down\n VCP up"
ctrl_lps_vs_untx_VCP.go_curated <- curated_profiles(gprofiles = LPS.VCP.overlap.up_gprofiles_filt.up_down, colours = 2) + theme(axis.text = element_text(size=8), strip.text.y.right = element_text(angle = 0))
# ctrl_lps_vs_untx_VCP.go_curated


# ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost = ctrl_lps_vs_untx_untx_vcp_vs_ctrl.dge %>% filter(overlap_direction %in% c("overlapping up", "overlapping down")) %>% split(.$overlap_direction) %>%  map("gene_id") %>% gost()
# ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt <- ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% 
#   mutate(query = case_when(query == "overlapping down" ~ " VCP Down\n LPS Down ", query == "overlapping up" ~ " VCP Up\n LPS Up "), term_name = as.factor(term_name)) 
# ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gprofiles_down <- ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt %>% filter(query == " VCP Down\n LPS Down ")
# ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gprofiles_up <- ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt %>% filter(query == " VCP Up\n LPS Up ")
# paste('"',unique(ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
# LPS.VCP.overlap.up_curated_list <- c(
# "IL10-IL10RA complex",
# "regulation of leukocyte apoptotic process",
# "lymphocyte apoptotic process",
# "regulation of lymphocyte apoptotic process")
# ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt.up <- ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt %>% filter(term_name %in% LPS.VCP.overlap.up_curated_list)
# ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt.up$term_name <- gsub("regulation of ", "", ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt.up$term_name)
# paste('"',unique(ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
# LPS.VCP.overlap.down_curated_list <- c(
# "nucleus")
# ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt.down <- ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt %>% filter(term_name %in% LPS.VCP.overlap.down_curated_list)
# LPS.VCP.overlap.up_gprofiles_filt.up_down <- bind_rows(ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt.up, ctrl_lps_vs_untx_untx_vcp_vs_ctrl.gost_filt.down) %>% mutate(query = factor(query, levels = c(" VCP Up\n LPS Up ", " VCP Down\n LPS Down ")))
# ctrl_lps_vs_untx_VCP.go_curated <- curated_profiles(gprofiles = LPS.VCP.overlap.up_gprofiles_filt.up_down, colours = 2) + theme(axis.text = element_text(size=8), strip.text.y.right = element_text(angle = 0))
# ctrl_lps_vs_untx_VCP.go_curated

```

## Mass spec Scatter & Venn

```{r}
dep.lps_vs_untx$res %>% filter(padj < 0.05) # 552
dep.lps_vs_untx$res %>% filter(padj < 0.05, stat>0) # 291
dep.lps_vs_untx$res %>% filter(padj < 0.05, stat<0) # 261


dep.untx_vcp_vs_ctrl_dep.lps_vs_untx.labels = select(dep.lps_vs_untx$res,gene_name, dep.lps_vs_untx.log2FoldChange = log2FoldChange) %>% left_join(select(dep.vcp_vs_ctrl$res, gene_name, dep.vcp_vs_ctrl.log2FoldChange = log2FoldChange)) %>% filter((dep.vcp_vs_ctrl.log2FoldChange > 2.5 & dep.lps_vs_untx.log2FoldChange > 2.5) | (dep.vcp_vs_ctrl.log2FoldChange < -2.5 & dep.lps_vs_untx.log2FoldChange < -2.5), gene_name %in% all.reactivity.go)
dep.untx_vcp_vs_ctrl_dep.lps_vs_untx.list = list("LPS vs Untreated" = dep.lps_vs_untx$res, "VCP vs CTRL" = dep.vcp_vs_ctrl$res)
dep.untx_vcp_vs_ctrl_dep.lps_vs_untx_log2FoldChange.corr = plot_de_correlation(model_list1 = dep.untx_vcp_vs_ctrl_dep.lps_vs_untx.list, model_list2 = dep.untx_vcp_vs_ctrl_dep.lps_vs_untx.list, mutation1 = "VCP vs CTRL", mutation2 = "LPS vs Untreated", gene_labels = pull(dep.untx_vcp_vs_ctrl_dep.lps_vs_untx.labels,gene_name), plot_corr_line = TRUE, join_by = "gene_name", title = "Proteomics", col = "log2FoldChange") + ylim(c(-7,7)) + xlim(c(-5,5)) +
  annotate(geom="text", x=3, y=7, label="LPS & VCP both Up", color="black", size = 3.2)  + annotate(geom="text", x=-3, y=-7, label="LPS & VCP both Down", color="black", size = 3.2)
# dep.untx_vcp_vs_ctrl_dep.lps_vs_untx_log2FoldChange.corr


dep.lps_vs_untx.res.dge.up <- dep.lps_vs_untx$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
dep.lps_vs_untx.res.dge.down <- dep.lps_vs_untx$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
dep.vcp_vs_ctrl.res.dge.up <- dep.vcp_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
dep.vcp_vs_ctrl.res.dge.down <- dep.vcp_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)

# 4 way overlap
dep.vcp_vs_ctrl_dep.lps_vs_untx_venn = ggvenn(list(`LPS\ndown` = dep.lps_vs_untx.res.dge.down, `LPS up` = dep.lps_vs_untx.res.dge.up, `VCP up` = dep.vcp_vs_ctrl.res.dge.up, `VCP\ndown` = dep.vcp_vs_ctrl.res.dge.down),
                                                fill_color = c("dodgerblue3", "firebrick3", "firebrick1", "dodgerblue1"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.7) + ggtitle("Proteomics") + theme(plot.title = element_text(hjust = 0.5))
# dep.vcp_vs_ctrl_dep.lps_vs_untx_venn


dep.lps_vs_untx.dge <- dep.lps_vs_untx$res %>% select(gene_name, log2FC.LPS = log2FoldChange, FDR.LPS = padj)
dep.vcp_vs_ctrl.dge <- dep.vcp_vs_ctrl$res %>% select(gene_name, log2FC.VCP = log2FoldChange, FDR.VCP = padj)
dep.lps_vs_untx_dep.vcp_vs_ctrl.dge <- full_join(dep.lps_vs_untx.dge, dep.vcp_vs_ctrl.dge, by = "gene_name")

dep.lps_vs_untx_dep.vcp_vs_ctrl.dge <- dep.lps_vs_untx_dep.vcp_vs_ctrl.dge %>% mutate(overlap = case_when(FDR.LPS < 0.05 & FDR.VCP < 0.05 ~ "overlapping", # down in LPS
                                                        FDR.LPS < 0.05 ~ "LPS specific",
                                                        FDR.VCP < 0.05 ~ "VCP specific",
                                                        TRUE ~ "None"),
                                    overlap_direction = case_when(FDR.LPS < 0.05 & log2FC.LPS > 0 & FDR.VCP < 0.05 & log2FC.VCP > 0 ~ "overlapping up", # down in LPS
                                                                  FDR.LPS < 0.05 & log2FC.LPS < 0 & FDR.VCP < 0.05 & log2FC.VCP < 0 ~ "overlapping down",
                                                                  FDR.LPS < 0.05 & log2FC.LPS > 0 & FDR.VCP < 0.05 & log2FC.VCP < 0 ~ "LPS up, VCP down", # down in LPS
                                                                  FDR.LPS < 0.05 & log2FC.LPS < 0 & FDR.VCP < 0.05 & log2FC.VCP > 0 ~ "LPS down, VCP up",
                                                                  FDR.LPS < 0.05 & log2FC.LPS > 0 ~ "LPS specific up",
                                                                  FDR.LPS < 0.05 & log2FC.LPS < 0 ~ "LPS specific down",
                                                                  FDR.VCP < 0.05 & log2FC.VCP > 0 ~ "VCP specific up",
                                                                  FDR.VCP < 0.05 & log2FC.VCP < 0 ~ "VCP specific down",
                                                                  TRUE ~ "None"),
                                    overlap_direction = factor(overlap_direction, levels = c("overlapping up", "overlapping down", "LPS up, VCP down", "LPS down, VCP up", "LPS specific up", "LPS specific down", "VCP specific up", "VCP specific down", "None")),
                                    overlap = factor(overlap, levels = c("overlapping",  "LPS specific", "VCP specific", "None"))) %>%
  arrange(overlap_direction)
table(dep.lps_vs_untx_dep.vcp_vs_ctrl.dge$overlap)
dep.lps_vs_untx_dep.vcp_vs_ctrl.dge %>% count(overlap_direction)
dep.lps_vs_untx_dep.vcp_vs_ctrl.dge

cor.test(x = dep.lps_vs_untx_dep.vcp_vs_ctrl.dge$log2FC.LPS, y = dep.lps_vs_untx_dep.vcp_vs_ctrl.dge$log2FC.VCP) # 0.094
t.test( x = dep.lps_vs_untx_dep.vcp_vs_ctrl.dge$log2FC.LPS, y = dep.lps_vs_untx_dep.vcp_vs_ctrl.dge$log2FC.VCP, paired = TRUE )

dep.vcp_vs_ctrl.VCP_LPS_scatter <- ggscatter(dep.lps_vs_untx_dep.vcp_vs_ctrl.dge, x = "log2FC.LPS", y = "log2FC.VCP", color = "overlap_direction", palette = c("firebrick2", "dodgerblue2", "grey", "grey", "grey", "grey", "grey"), 
                                        cor.coef = TRUE, cor.method = "pearson",
          xlab = expression(Mass~spec~Log[2]~Fold~Change~LPS:Untreated), ylab = expression(Mass~spec~Log[2]~Fold~Change~VCP:CTRL), size = 0.5, cor.coeff.args = list(label.x = -4, label.y = 2.5, label.sep="\n")) +
          geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") +  
    coord_cartesian(xlim = c(-4,5.5), ylim = c(-3,3)) +
    geom_smooth(data = dep.lps_vs_untx_dep.vcp_vs_ctrl.dge, aes(x = log2FC.LPS, y = log2FC.VCP), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 2) + theme(legend.position = "none") +
    geom_text_repel(data = filter(dep.lps_vs_untx_dep.vcp_vs_ctrl.dge, overlap_direction %in%  c("overlapping up", "overlapping down"), gene_name %in% all.reactivity.go, log2FC.VCP < 9), aes(x = log2FC.LPS, y = log2FC.VCP, label = gene_name), 
                    fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.8, max.overlaps = 30) +  
  annotate(geom="text", x=3, y=3, label="LPS & VCP both Up", color="firebrick2", size = 3.2)  + annotate(geom="text", x=-2, y=-3, label="LPS & VCP both Down", color="dodgerblue2", size = 3.2)
# dep.vcp_vs_ctrl.VCP_LPS_scatter

# Overlap tested using Fisher's exact test (alternative=greater)
LPS.DGE.up.genes <- dep.lps_vs_untx$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
VCP.DGE.up.genes <- dep.vcp_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
genome.size <- dep.lps_vs_untx$res %>% distinct(gene_name) %>% nrow
newGeneOverlap(LPS.DGE.up.genes,  VCP.DGE.up.genes,genome.size=genome.size) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=4169
# listB size=110
# Intersection size=15
# Overlapping p-value=8.4e-03
# Jaccard Index=0.0

LPS.DGE.down.genes <- dep.lps_vs_untx$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
VCP.DGE.down.genes <- dep.vcp_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
genome.size <- dep.lps_vs_untx$res %>% distinct(gene_name) %>% nrow
newGeneOverlap(LPS.DGE.down.genes, VCP.DGE.down.genes,genome.size=genome.size) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=3927
# listB size=73
# Intersection size=21
# Overlapping p-value=4.2e-09
# Jaccard Index=0.0
```

## LPS treated VCP vs CTRL


```{r}
lps_vcp_vs_ctrl$res  %>% filter(padj < 0.05) #30
lps_vcp_vs_ctrl$res  %>% filter(padj < 0.05, stat > 0) #16

# Volcano
lps_vcp_vs_ctrl.labels <- lps_vcp_vs_ctrl$res  %>% filter(padj < 0.05, !grepl("^AL|AC",gene_name)) %>% pull(gene_name)
lps_vcp_vs_ctrl.volcano <- volcano_plot(lps_vcp_vs_ctrl$res, gene_labels = lps_vcp_vs_ctrl.labels, ymax = 10, xmax = 4, title = "LPS treatment", numerator = "VCP", denomenator = "CTRL")
# lps_vcp_vs_ctrl.volcano

lps_vcp_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange > 0, gene_name %in% all.microglia.go)
lps_vcp_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange > 0, gene_name %in% RNA_BINDING_PROTEINS)

# # ## GO terms
# lps_vcp_vs_ctrl.res.gost <- lps_vcp_vs_ctrl$res %>% filter(padj < 0.05) %>% mutate(deg = case_when(padj < 0.05 & log2FoldChange > 0 ~ "VCP up", padj < 0.05 & log2FoldChange < 0 ~ "VCP down", TRUE ~ "None")) %>% group_split(deg) %>% map("gene_id") %>% gost()
# lps_vcp_vs_ctrl.res.gost_filt <- lps_vcp_vs_ctrl.res.gost$result %>% filter(!str_detect(source, "MIRNA")) %>% arrange( -log10(p_value) ) %>%
#   mutate(query = case_when(query == "query_1" ~ "VCP down", query == "query_2" ~ "VCP up"), query = factor(query, levels = c("VCP up", "VCP down")))
# table(lps_vcp_vs_ctrl.res.gost_filt$query)
# lps_vcp_vs_ctrl_gprofiles_curated.barchart <- curated_profiles(gprofiles = lps_vcp_vs_ctrl.res.gost_filt, cols = "dodgerblue2")
# lps_vcp_vs_ctrl_gprofiles_curated.barchart


# # GSEA
# lps_vcp_vs_ctrl.geneList <- lps_vcp_vs_ctrl$res %>% drop_na(log2FoldChange) %>% pull(log2FoldChange)
# names(lps_vcp_vs_ctrl.geneList) <- lps_vcp_vs_ctrl$res %>% drop_na(log2FoldChange) %>% pull(gene_name) %>% as.character()
# lps_vcp_vs_ctrl.geneList = sort(lps_vcp_vs_ctrl.geneList, decreasing = TRUE)
# 
# 
# multi_gsea_list <- list("GO_PHAGOCYTOSIS" = lps_vcp_vs_ctrl$res$gene_name[lps_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS],
#                         "GO_NEGATIVE_REGULATION_OF_PHAGOCYTOSIS" = lps_vcp_vs_ctrl$res$gene_name[lps_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_NEGATIVE_REGULATION_OF_PHAGOCYTOSIS],
#                         "GO_POSITIVE_REGULATION_OF_PHAGOCYTOSIS" = lps_vcp_vs_ctrl$res$gene_name[lps_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_POSITIVE_REGULATION_OF_PHAGOCYTOSIS],
#                         "GO_POSITIVE_REGULATION_OF_MITOCHONDRIAL_MEMBRANE_POTENTIAL" = lps_vcp_vs_ctrl$res$gene_name[lps_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_POSITIVE_REGULATION_OF_MITOCHONDRIAL_MEMBRANE_POTENTIAL],
#                         "GO_MITOCHONDRIAL_GENE_EXPRESSION" = lps_vcp_vs_ctrl$res$gene_name[lps_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_MITOCHONDRIAL_GENE_EXPRESSION],
#                         "GO_REGULATION_OF_MITOCHONDRIAL_ATP_SYNTHESIS_COUPLED_ELECTRON_TRANSPORT" = lps_vcp_vs_ctrl$res$gene_name[lps_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_REGULATION_OF_MITOCHONDRIAL_ATP_SYNTHESIS_COUPLED_ELECTRON_TRANSPORT],
#                         "GO_CHAPERONE_MEDIATED_AUTOPHAGY" = lps_vcp_vs_ctrl$res$gene_name[lps_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_CHAPERONE_MEDIATED_AUTOPHAGY],
#                         "GO_MACROAUTOPHAGY" = lps_vcp_vs_ctrl$res$gene_name[lps_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_MACROAUTOPHAGY],
#                         "GO_REGULATION_OF_AUTOPHAGY" = lps_vcp_vs_ctrl$res$gene_name[lps_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_REGULATION_OF_AUTOPHAGY],
#                         "GO_MICROGLIAL_CELL_MIGRATION" = lps_vcp_vs_ctrl$res$gene_name[lps_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_MICROGLIAL_CELL_MIGRATION],
#                         "GO_MICROGLIAL_CELL_ACTIVATION" = lps_vcp_vs_ctrl$res$gene_name[lps_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_MICROGLIAL_CELL_ACTIVATION],
#                         "GO_REGULATION_OF_MICROGLIAL_CELL_ACTIVATION" = lps_vcp_vs_ctrl$res$gene_name[lps_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_REGULATION_OF_MICROGLIAL_CELL_ACTIVATION])
# lps_vcp_vs_ctrl.multi_fgseaRes <- fgsea(pathways = multi_gsea_list, stats = lps_vcp_vs_ctrl.geneList)
# lps_vcp_vs_ctrl.multi_fgseaRes
# #                                                                    pathway       pval      padj    log2err         ES        NES size
# #  1:                                         GO_CHAPERONE_MEDIATED_AUTOPHAGY 0.77611940 0.8537313 0.05412006  0.3408339  0.7884772   16
# #  2:                                                       GO_MACROAUTOPHAGY 0.96343179 0.9634318 0.03099008  0.2221326  0.8045442  291
# #  3:                                            GO_MICROGLIAL_CELL_MIGRATION 0.37701613 0.6911962 0.09528798 -0.5471282 -1.0863347    8
# #  4:                                        GO_MITOCHONDRIAL_GENE_EXPRESSION 0.12951807 0.4582026 0.21654284 -0.3077560 -1.1560048  165
# #  5:                                  GO_NEGATIVE_REGULATION_OF_PHAGOCYTOSIS 0.45168539 0.7097913 0.09110731 -0.4059531 -1.0005397   18
# #  6:                                                         GO_PHAGOCYTOSIS 0.20827389 0.4582026 0.11056472  0.3117608  1.1155811  251
# #  7:              GO_POSITIVE_REGULATION_OF_MITOCHONDRIAL_MEMBRANE_POTENTIAL 0.73588710 0.8537313 0.06024841 -0.3927842 -0.8066926    9
# #  8:                                  GO_POSITIVE_REGULATION_OF_PHAGOCYTOSIS 0.17676768 0.4582026 0.13355495  0.3987261  1.2024393   63
# #  9:                                              GO_REGULATION_OF_AUTOPHAGY 0.05745554 0.4582026 0.21925035  0.3400770  1.2443469  322
# # 10:                             GO_REGULATION_OF_MICROGLIAL_CELL_ACTIVATION 0.73967684 0.8537313 0.05456806  0.3382441  0.8151990   18
# # 11: GO_REGULATION_OF_MITOCHONDRIAL_ATP_SYNTHESIS_COUPLED_ELECTRON_TRANSPORT 0.19367589 0.4582026 0.13880511  0.6376474  1.2420702    8
# 
# # GSEA phagocytosis
# phagocytosis_genes.list <- list("phagocytosis" = lps_vcp_vs_ctrl$res$gene_name[lps_vcp_vs_ctrl$res$gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS])
# lps_vcp_vs_ctrl.phagocytosis_fgseaRes <- fgsea(pathways = phagocytosis_genes.list, stats = lps_vcp_vs_ctrl.geneList)
# lps_vcp_vs_ctrl.phagocytosis_fgseaRes
# #         pathway      pval      padj    log2err         ES        NES size
# # 1: phagocytosis 0.5581395 0.5581395 0.05785298 -0.2561417 -0.9540701  265
# lps_vcp_vs_ctrl.phagocytosis.gseaplot = plotEnrichment(phagocytosis_genes.list[["phagocytosis"]], lps_vcp_vs_ctrl.geneList) + 
#   theme(axis.text=element_text(size=8), axis.title=element_text(size=10)) + ylab(expression(Phagocytosis)) + xlab(expression(LPS~treated~VCP~vs~CTRL)) +
#   annotate("text", x = 100, y = -0.15, label = paste0("NES = ", round(lps_vcp_vs_ctrl.phagocytosis_fgseaRes$NES, digits = 3),"\nP = ", round(lps_vcp_vs_ctrl.phagocytosis_fgseaRes$pval, digits = 3)), size = 3, hjust = 0)
# lps_vcp_vs_ctrl.phagocytosis.gseaplot

```

## Progeny & Dorothea

Pathway RespOnsive GENes (PROGENy) estimates signaling pathway activity

```{r}
net_progeny = readRDS(here(nemo_path, "projects/ipsc-mn-als-meta/scripts/ipsc_mn_als_meta/ipsn_als_meta/net_progeny.rds"))
# net_progeny <- get_progeny(organism = 'human', top = 100)
lps_vcp_vs_ctrl.res.matrix <- lps_vcp_vs_ctrl$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
lps_vcp_vs_ctrl_progeny.contrast_acts <- run_wmean(mat=lps_vcp_vs_ctrl.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "***", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
lps_vcp_vs_ctrl_progeny.pathwayActivity <- ggplot(lps_vcp_vs_ctrl_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "Signaling Pathways", x = "", y = "LPS VCP vs LPS CTRL NES") +
    stat_pvalue_manual(lps_vcp_vs_ctrl_progeny.contrast_acts, label = "p.signif", y.position = 14, xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE) 
# lps_vcp_vs_ctrl_progeny.pathwayActivity


# Hypoxia weights
lps_vcp_vs_ctrl_progeny.Hypoxiaweights = net_progeny %>% filter(source == "Hypoxia", target %in% lps_vcp_vs_ctrl$res$gene_name) %>% left_join(select(lps_vcp_vs_ctrl$res, target=gene_name,stat)) %>% 
  mutate(color = case_when(weight > 0 & stat > 0 ~ '1', weight > 0 & stat < 0 ~ '2', weight < 0 & stat > 0 ~ '2', weight < 0 & stat < 0 ~ '1', TRUE ~ "3"))
lps_vcp_vs_ctrl_progeny.Hypoxiaweights.plot = ggplot(lps_vcp_vs_ctrl_progeny.Hypoxiaweights, aes(x = weight, y = stat, color = color)) + geom_point(size = 0.5) +
  scale_colour_manual(values = c("red","royalblue3","grey")) +
  geom_text_repel(fontface = "italic", data = filter(lps_vcp_vs_ctrl_progeny.Hypoxiaweights, weight > 2 & stat > 2.3 | weight > 14 & stat > 1.8), aes(label = target), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.1) +
  theme_oz() + theme(legend.position = "none", plot.title = element_text(hjust = 0.5)) +
  labs(title = "Hypoxia pathway weights", x = "Hypoxia pathway gene weight", y = "LPS VCP vs LPS CTRL test statistic") +
  geom_vline(xintercept = 0, linetype = 'dotted') +  geom_hline(yintercept = 0, linetype = 'dotted') #+
# lps_vcp_vs_ctrl_progeny.Hypoxiaweights.plot

```

https://www.bioconductor.org/packages/release/bioc/vignettes/decoupleR/inst/doc/tf_bk.html
https://github.com/saezlab/transcriptutorial/blob/master/scripts/04_TranscriptionFactor_activity_with_Dorothea.md

```{r}
net_dorothea = readRDS(here(nemo_path, "projects/ipsc-mn-als-meta/scripts/ipsc_mn_als_meta/ipsn_als_meta/net_dorothea.rds"))
lps_vcp_vs_ctrl.dorothea <- run_wmean(mat=lps_vcp_vs_ctrl.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

lps_vcp_vs_ctrl.dorothea.labels.up = lps_vcp_vs_ctrl.dorothea %>% top_n(score, n = 5) %>% pull(source)
lps_vcp_vs_ctrl.dorothea.labels.down = lps_vcp_vs_ctrl.dorothea %>% top_n(-score, n = 5) %>% pull(source)

# Volcano of TFs score vs p_value
lps_vcp_vs_ctrl.dorothea.volcano = ggplot(lps_vcp_vs_ctrl.dorothea, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment LPS VCP vs LPS CTRL", title = "Transcription Factors") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  geom_text_repel(fontface = "italic", data = filter(lps_vcp_vs_ctrl.dorothea, source %in% c(lps_vcp_vs_ctrl.dorothea.labels.up, lps_vcp_vs_ctrl.dorothea.labels.down)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.5) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
# lps_vcp_vs_ctrl.dorothea.volcano



# # HIF1A target genes
lps_vcp_vs_ctrl_dorothea.HIF1A <- net_dorothea %>%  filter(source == "HIF1A") %>% mutate(gene_name = target) %>% left_join(select(lps_vcp_vs_ctrl$res, gene_name,stat,log2FoldChange,pvalue, padj)) %>%
  mutate(color = case_when(mor > 0 & stat > 0 ~ '1', mor > 0 & stat < 0~'2',mor < 0 & stat > 0~'2',mor < 0 & stat < 0~'1',TRUE ~ "3"))
lps_vcp_vs_ctrl_dorothea.HIF1A.labels = pull(filter(lps_vcp_vs_ctrl_dorothea.HIF1A, abs(stat) > 2.5 | abs(log2FoldChange > 3)), gene_name)
lps_vcp_vs_ctrl_dorothea.HIF1A.volcano = volcano_plot(lps_vcp_vs_ctrl_dorothea.HIF1A, title = "LPS treatment: HIF1a regulon", gene_labels = lps_vcp_vs_ctrl_dorothea.HIF1A.labels, ymax = 3, xmax = 4, numerator = "VCP", denomenator = "CTRL", significance_threshold = "pvalue")
# lps_vcp_vs_ctrl_dorothea.HIF1A.volcano
# lps_vcp_vs_ctrl_dorothea.HIF1A.volcano = ggplot(lps_vcp_vs_ctrl_dorothea.HIF1A, aes(x = log2FoldChange, y = -log10(pvalue), color = color, size=abs(mor))) + geom_point() + scale_size(range = c(0, 3)) +
#   scale_colour_manual(values = c("red","royalblue3","grey")) +
#   geom_text_repel(data = filter(lps_vcp_vs_ctrl_dorothea.HIF1A, abs(stat) > 2.5), aes(label = gene_name, size=1), fontface = "italic", max.overlaps = Inf, size = 2.3) +
#   theme_oz() + theme(legend.position = "none", plot.title = element_text(hjust = 0.5)) +  geom_vline(xintercept = 0, linetype = 'dotted') + geom_hline(yintercept = 0, linetype = 'dotted') +
#     labs(title = "HIF1A regulon", x = "log2 fold change (VCP vs  CTRL)", y = expression(-log[10]~P~value)) #+ ylim(c(0,2)) +   

```

## Luminex

```{r}
luminex_lps_data = read_excel(here(proj_path, "expression/luminex/luminex_data.xlsx"), sheet = "LPS") %>%
  pivot_longer(!c(batch,treatment, induction, mutation, sample, confluency), names_to = "cytokine", values_to = "intensity") %>%
  filter(treatment == "LPS treated") %>%
  group_by(batch, cytokine) %>% mutate(mean_batch_intensity = mean(intensity), intensity_batch_normalised = intensity / mean_batch_intensity, intensity_normalised =  intensity_batch_normalised / confluency, intensity_normalised.transformed = intensity_normalised * 1 / mean(intensity_normalised)) %>% ungroup()
luminex_lps_data


luminex_lps_data %>% my_summarise(intensity_normalised.transformed, cytokine, mutation) %>% print(n=Inf)


luminex_lps_cytokine.violin = violin_plot(data = luminex_lps_data, color_by = "mutation", continuous = "intensity_normalised.transformed", ylims = c(0.05,3.5), cols = brewer.pal(6, "Paired"), ylabel = "% Positive", facet_by = "cytokine", plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 3, tip_length = 0.003) + theme(axis.text.x = element_text(angle = 30, vjust = 0.7, hjust=0.7))
# luminex_lps_cytokine.violin

luminex_lps_data.wide = luminex_lps_data %>% group_by(cytokine, mutation) %>% summarise(intensity = mean(intensity_normalised.transformed)) %>% pivot_wider(names_from = "mutation", values_from = "intensity") %>% #mutate(log2FoldChange = log2(VCP) - log2(CTRL)) %>% 
  ungroup() 
luminex_lps_data.lfc.mat <- make_matrix(select(luminex_lps_data.wide,-cytokine), pull(luminex_lps_data.wide,cytokine))


col_fun = colorRamp2(c(0, 1, 2), c("blue", "white", "red"))
lgd = Legend(col_fun = colorRamp2(c(0, 1, 2), c("blue", "white", "red")))

luminex_lps.massspec.heatmap <- as.ggplot(Heatmap(t(luminex_lps_data.lfc.mat), 
          heatmap_legend_param = list(title = "", at = c(0,0.6,1.3)),
          col = colorRamp2(c(0,1,1.3), c("blue", "white", "red")),
          cluster_columns = FALSE, cluster_column_slices = FALSE, cluster_rows = FALSE, border = TRUE, 
          # row_order = myeloid.markers.df$gene_name,
          column_names_gp = gpar(fontsize = 8), row_names_gp = gpar(fontsize = 8), #, #, col = c(rep("red", 10), rep("blue", 8)))
          bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = "white"), labels = "Cytokines", labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          column_title_gp = gpar(fontsize = 8), row_title = NULL, row_names_rot = 0, row_names_centered = TRUE, row_split =  c(1,2)))
luminex_lps.massspec.heatmap



luminex_lps_cytokine_il1b.violin = violin_plot(data = filter(luminex_lps_data, cytokine %in% c("IL1b")), color_by = "mutation", continuous = "intensity_normalised.transformed", ylims = c(0,2.5), cols = brewer.pal(6, "Paired"), ylabel = "Median intensity\nnormalised to confluency", facet_by = "cytokine", plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 2.5, tip_length = 0.003)
# luminex_lps_cytokine_il1b.violin


```


## Merge

A - Schematic VCP and LPS comparison
B - Global correlation LPS vs VCP related gene expression changes
C - Venn diagram DEGs LPS vs VCP related gene expression changes
D - GO LPS vs VCP related gene expression changes
E - Global correlation LPS vs VCP related protein expression changes
F - Volcano plot VCP+LPS 
G - Progeny VCP+LPS
H - Volcano plot hypoxia pathway progeny
I - Volcano plot Dorothea
J - Volcano plot HIF1A regulon
K -  Bar plots of any significantly changed LPS treated in CTRL vs VCP microglia

```{r}
fig3.merged <- plot_grid(
  Fig3A_schematic.gg,
  plot_grid(untx_vcp_vs_ctrl_ctrl_lps_vs_untx_log2FoldChange.corr, dep.untx_vcp_vs_ctrl_dep.lps_vs_untx_log2FoldChange.corr, ncol = 2, labels = c("B", "D")),
  plot_grid(untx_vcp_vs_ctrl_ctrl_lps_vs_untx_venn, dep.vcp_vs_ctrl_dep.lps_vs_untx_venn, lps_vcp_vs_ctrl.volcano, ncol = 3, rel_widths = c(0.8,0.8,1.2), labels = c("C", "E","F")),
  plot_grid(lps_vcp_vs_ctrl_progeny.pathwayActivity, lps_vcp_vs_ctrl_progeny.Hypoxiaweights.plot, lps_vcp_vs_ctrl.dorothea.volcano, ncol = 3, rel_widths = c(1,0.9,1), labels = c("G","H","I")),
  plot_grid(NULL,luminex_lps.massspec.heatmap, luminex_lps_cytokine_il1b.violin, ncol = 3, rel_widths = c(0.1,2,1), labels = c("J","", "K")),
  nrow = 5, rel_heights = c(0.8,1.2,1,1,0.6), labels = c("A","","","",""))
# fig3.merged
ggsave(fig3.merged, filename = here(proj_path, "expression/deseq2/fig3.merged.png"), height = 14, width = 10)
ggsave(fig3.merged, filename = here(proj_path, "expression/deseq2/fig3.merged.pdf"), height = 14, width = 10)

```

# Table S5-6 LPS and VCP overlap

```{r}
ctrl_lps_vs_untx.dge <- ctrl_lps_vs_untx$res %>% select(gene_name, gene_id, log2FC.LPS = log2FoldChange, FDR.LPS = padj)
untx_vcp_vs_ctrl.dge <- untx_vcp_vs_ctrl$res %>% select(gene_name, gene_id, log2FC.VCP = log2FoldChange, FDR.VCP = padj)
ctrl_lps_vs_untx_untx_vcp_vs_ctrl.dge <- full_join(ctrl_lps_vs_untx.dge, untx_vcp_vs_ctrl.dge, by = c("gene_name", "gene_id")) %>%
  mutate(overlap = case_when(FDR.LPS < 0.05 & FDR.VCP < 0.05 ~ "overlapping", # down in LPS
                                                        FDR.LPS < 0.05 ~ "LPS specific",
                                                        FDR.VCP < 0.05 ~ "VCP specific", TRUE ~ "None"),
                                    overlap_direction = case_when(FDR.LPS < 0.05 & log2FC.LPS > 0 & FDR.VCP < 0.05 & log2FC.VCP > 0 ~ "overlapping up", # down in LPS
                                                                  FDR.LPS < 0.05 & log2FC.LPS < 0 & FDR.VCP < 0.05 & log2FC.VCP < 0 ~ "overlapping down",
                                                                  FDR.LPS < 0.05 & log2FC.LPS < 0 & FDR.VCP < 0.05 & log2FC.VCP > 0 ~ "LPS down, VCP up", # down in LPS
                                                                  FDR.LPS < 0.05 & log2FC.LPS > 0 & FDR.VCP < 0.05 & log2FC.VCP < 0 ~ "LPS up, VCP down",
                                                                  FDR.LPS < 0.05 & log2FC.LPS > 0 ~ "LPS specific up",
                                                                  FDR.LPS < 0.05 & log2FC.LPS < 0 ~ "LPS specific down",
                                                                  FDR.VCP < 0.05 & log2FC.VCP > 0 ~ "VCP specific up",
                                                                  FDR.VCP < 0.05 & log2FC.VCP < 0 ~ "VCP specific down",
                                                                  TRUE ~ "None"),
                                    overlap_direction = factor(overlap_direction, levels = c("overlapping up", "overlapping down", "LPS down, VCP up", "LPS up, VCP down", "LPS specific up", "LPS specific down", "VCP specific up", "VCP specific down", "None")),
                                    overlap = factor(overlap, levels = c("overlapping",  "LPS specific", "VCP specific", "None"))) %>%
  arrange(overlap_direction)
ctrl_lps_vs_untx_untx_vcp_vs_ctrl.dge.filt = ctrl_lps_vs_untx_untx_vcp_vs_ctrl.dge %>% filter(overlap == "overlapping") %>% select(-overlap)
write_csv(ctrl_lps_vs_untx_untx_vcp_vs_ctrl.dge.filt, here(proj_path, "expression/deseq2/table.S5_rnaseq_lps_vs_untx_vcp_vs_ctrl.csv"))
library(googlesheets4)
gs4_browse("https://docs.google.com/spreadsheets/d/1CVDodcgGdB2rDqnDtfGWMUMyb-BdNFmfzyAX0F-bQqY/edit#gid=0")
write_sheet(ctrl_lps_vs_untx_untx_vcp_vs_ctrl.dge.filt, ss = "https://docs.google.com/spreadsheets/d/1CVDodcgGdB2rDqnDtfGWMUMyb-BdNFmfzyAX0F-bQqY/edit#gid=0", sheet = "S5. LPS RNAseq")
sheet_properties("https://docs.google.com/spreadsheets/d/1CVDodcgGdB2rDqnDtfGWMUMyb-BdNFmfzyAX0F-bQqY/edit#gid=0")



dep.lps_vs_untx.dpe <- dep.lps_vs_untx$res %>% select(gene_name, log2FC.LPS = log2FoldChange, FDR.LPS = padj)
dep.vcp_vs_ctrl.dpe <- dep.vcp_vs_ctrl$res %>% select(gene_name, log2FC.VCP = log2FoldChange, FDR.VCP = padj)
dep.lps_vs_untx_dep.vcp_vs_ctrl.dpe <- full_join(dep.lps_vs_untx.dpe, dep.vcp_vs_ctrl.dpe, by = "gene_name") %>% 
  mutate(overlap = case_when(FDR.LPS < 0.05 & FDR.VCP < 0.05 ~ "overlapping", # down in LPS
                                                        FDR.LPS < 0.05 ~ "LPS specific",
                                                        FDR.VCP < 0.05 ~ "VCP specific",
                                                        TRUE ~ "None"),
                                    overlap_direction = case_when(FDR.LPS < 0.05 & log2FC.LPS > 0 & FDR.VCP < 0.05 & log2FC.VCP > 0 ~ "overlapping up", # down in LPS
                                                                  FDR.LPS < 0.05 & log2FC.LPS < 0 & FDR.VCP < 0.05 & log2FC.VCP < 0 ~ "overlapping down",
                                                                  FDR.LPS < 0.05 & log2FC.LPS > 0 & FDR.VCP < 0.05 & log2FC.VCP < 0 ~ "LPS up, VCP down", # down in LPS
                                                                  FDR.LPS < 0.05 & log2FC.LPS < 0 & FDR.VCP < 0.05 & log2FC.VCP > 0 ~ "LPS down, VCP up",
                                                                  FDR.LPS < 0.05 & log2FC.LPS > 0 ~ "LPS specific up",
                                                                  FDR.LPS < 0.05 & log2FC.LPS < 0 ~ "LPS specific down",
                                                                  FDR.VCP < 0.05 & log2FC.VCP > 0 ~ "VCP specific up",
                                                                  FDR.VCP < 0.05 & log2FC.VCP < 0 ~ "VCP specific down",
                                                                  TRUE ~ "None"),
                                    overlap_direction = factor(overlap_direction, levels = c("overlapping up", "overlapping down", "LPS up, VCP down", "LPS down, VCP up", "LPS specific up", "LPS specific down", "VCP specific up", "VCP specific down", "None")),
                                    overlap = factor(overlap, levels = c("overlapping",  "LPS specific", "VCP specific", "None"))) %>%
  arrange(overlap_direction)
dep.lps_vs_untx_dep.vcp_vs_ctrl.dpe.filt = dep.lps_vs_untx_dep.vcp_vs_ctrl.dpe %>% filter(overlap == "overlapping") %>% select(-overlap)
write_csv(dep.lps_vs_untx_dep.vcp_vs_ctrl.dpe.filt, here(proj_path, "expression/deseq2/dep.lps_vs_untx_dep.vcp_vs_ctrl.dpe.filt.csv"))
write_sheet(dep.lps_vs_untx_dep.vcp_vs_ctrl.dpe.filt, ss = "https://docs.google.com/spreadsheets/d/1CVDodcgGdB2rDqnDtfGWMUMyb-BdNFmfzyAX0F-bQqY/edit#gid=0", sheet = "S6. LPS mass spec")


```


# Fig S4 LPS vs untreated

## PCA

```{r}
clarke.pcaData <- plotPCA(clarke$vsd, intgroup=c("mutation", "group", "sample", "cellline", "genotype", "treatment", "replicate"), returnData=TRUE)
clarke.percentVar <- round(100 * attr(clarke.pcaData, "percentVar"))
clarke.pcaData = clarke.pcaData %>% mutate(cellline = toupper(cellline), name = toupper(paste(mutation, replicate)), mutation = toupper(mutation), treatment = gsub("untx", "Untreated", treatment), treatment = gsub("lps", "LPS Treated", treatment), treatment_name = paste(treatment, "\n", name)) %>% arrange(treatment, mutation, replicate)
clarke.pca <- ggplot(clarke.pcaData, aes(PC1, PC2, color=treatment)) +   geom_point(size=1) + 
  geom_text_repel(aes(label=name), size = 2.5) + 
 scale_shape_manual(values=c(1, 16)) + scale_colour_manual( values = c("firebrick2", "dodgerblue2") ) +  scale_fill_manual( values = c("firebrick2", "dodgerblue2") ) +  
    theme_oz() + theme(legend.text=element_text(size=8), legend.title=element_blank(), legend.position = "none", legend.spacing = unit(0.01, 'cm')) + 
 # guides(color=guide_legend(title="Treatment", keywidth=0.1, keyheight=0.15, default.unit="inch", reverse = TRUE, nrow = 1)) +
  xlab(paste0("PC1: ",clarke.percentVar[1],"% variance")) +  ylab(paste0("PC2: ",clarke.percentVar[2],"% variance")) + #coord_fixed()  + 
  ggforce::geom_mark_ellipse(inherit.aes = FALSE, aes(PC1, PC2, fill = treatment, label = treatment, con.cap = 0), show_guide=FALSE)
# clarke.pca

# ggsave(pca, filename = here(proj_path, "expression/deseq2/pca.png")


# ## Dendrogram
# clarke.sampleDists <- dist(t(assay(clarke$vsd)))
# clarke.hc <- hclust(clarke.sampleDists, method = "ward.D2")
# clarke.hc$labels = clarke.pcaData$treatment_name
# clarke.a <- c(rep("firebrick2", 7), rep("dodgerblue2", 7))
# clarke.dendrogram <- ggdendrogram(clarke.hc, rotate = FALSE, theme_dendro = FALSE) +  theme_classic() + theme(axis.line=element_blank(), axis.title=element_blank(), axis.text.y=element_blank(), axis.ticks.y=element_blank(), axis.text.x= element_text(colour = clarke.a, size = 8, angle = 90), strip.background = element_rect(colour="black", fill="white"), panel.grid = element_blank())
# clarke.dendrogram
# # ggsave(dendrogram, filename = here(proj_path, "expression/deseq2/dendrogram.png")

# ## LPS regulated marker Heatmap
# # Table S4 from <https://www.pnas.org/content/113/12/E1738/tab-figures-data>
# # import Table S4
# # conda install -c conda-forge poppler
# # conda install -c conda-forge pkg-config
# # install.packages("pdftools")
# library(pdftools)
# txt <- pdf_text(here(proj_path, "expression/bennett.microglia.mouse.human.pnas.2016.pdf")
# cat(txt[13])
# # paste and edit in Atom
# lps.responsive.genes.mus = c("Saa3","Slc39a14","Lcn2","Fxyd5","Srgn","Ehd1","Gpr84","Map3k8","Mt2","Ch25h","Ccl5","Fmnl2","Msr1","Spp1","Ifitm3","Il2rg","Marco","Il1b","AW112010","Cd300lf","Igsf6","Mocs1","Ms4a6d","Slc7a11","Sdc4","Isg15","Mt1","Gbp2","Tlr2","Slc13a3","Il1rn","Pilrb2","Ms4a6c","Lilrb3","Lilrb4","Postn","Pilra","Fgr","Fcgr4","Cd274","Aoah","Dctd","Cybb","Cd44","Bcl2a1d","Vcam1","Fpr1","Cd244","Ifitm2","Rrs1","Rnf149","Irak3","Sod2","1600029D21Rik","Ifi204","Slfn5","Cxcl10","Ezr","Slfn2","Adamtsl4","Tnf","Pyhin1","Cpd","Plaur","Clec4a1",
#                              "P2ry12","Gm11428",
#                              "Slc2a5","Gpr65","Atf3","Cxcl16","Bmyc","Irg1","Gpr165","Mpdu1","Susd3","Tspo","Upk1b","AI607873","Gm10790","Fpr2","D18Ertd653e","Rtp4","Slc46a3","Pilrb1","Csmd3","Zbp1","Plxdc1","Il1r2","Fbxw4","Niacr1","Gm6277","Il15ra","Gabbr1","Cxcl2","2010015L04Rik","Cp","St6gal1","Slc7a5","Tal1","Nfil3","Tppp","Edn1","Prkca","Mmp14")
# lps.downregulated.genes.mus = c("P2ry12", "Slc2a5","Atf3","Bmyc","Gpr165","Susd3","Upk1b","Gm10790","D18Ertd653e","Slc46a3","Csmd3","Plxdc1","Fbxw4","Gm6277","Gabbr1","2010015L04Rik","St6gal1","Tal1","Tppp","Prkca")
# 
# # convertMouseGeneList <- function(x){
# #   require("biomaRt")
# #   human = useMart("ensembl", dataset = "hsapiens_gene_ensembl")
# #   mouse = useMart("ensembl", dataset = "mmusculus_gene_ensembl")
# #   genesV2 = getLDS(attributes = c("mgi_symbol"), filters = "mgi_symbol", values = mus_musculus.gtf$gene_name, mart = mouse, attributesL = c("hgnc_symbol"), martL = human, uniqueRows=T)
# #   humanx <- unique(genesV2[, 2])
# #   # Print the first 6 genes found to the screen
# #   print(head(humanx))
# #   return(humanx)
# # }
# # Sys.setenv("http_proxy" = "http://my.proxy.org:9999") # Configure biomaRt. NB this will stop gprofiler2 working
# # Sys.unsetenv("http_proxy") # unset proxy to fix gprofiler2
# # mus_musculus.gtf <- readRDS("/nemo/home/ziffo/home/genomes/ensembl/Mus_musculus.GRCm38.101.gtf.RDS")
# library(myfunctions)
# lps.responsive.genes = myfunctions::convertMouseGeneList(lps.responsive.genes.mus) # 89 / 105
# lps.downregulated.genes = myfunctions::convertMouseGeneList(lps.downregulated.genes.mus) 
# lps.responsive.genes.df = tibble("gene_name" = lps.responsive.genes) %>% filter(gene_name %in% lps.responsive.genes) %>% mutate(group = case_when(gene_name %in% lps.downregulated.genes ~ "LPS downregulated", TRUE ~ "LPS upregulated")) %>% arrange(group)
# 
# lps.responsive.genes.marker <- clarke$vsd.counts %>% distinct(gene_name, .keep_all = TRUE) %>% 
#   filter(gene_name %in% lps.responsive.genes) %>% #, rowSums(across(where(is.numeric))) > 162) %>% 
#   dplyr::mutate_if(is.numeric, funs((. - 3.165)))  %>% # subtract from all values so start from 0,log2 transform
#   arrange(factor(gene_name, levels = lps.responsive.genes.df$gene_name)) %>%
#   select(gene_name, untx_ctrl3, untx_ctrl4,untx_ncrm1,untx_vcpf10,untx_cb1d,untx_glib,untx_ncrme6, lps_ctrl3,lps_ctrl4,lps_ncrm1,lps_vcpf10,lps_cb1d,lps_glib,lps_ncrme6) %>%  
#   rename_at(vars(starts_with(c("untx", "lps"))), ~ str_replace_all(., setNames(clarke.pcaData$treatment_name, clarke.pcaData$sample)))
# lps.responsive.genes.marker.mat <- make_matrix(select(lps.responsive.genes.marker,-gene_name), pull(lps.responsive.genes.marker,gene_name))
# # lps.responsive.genes.marker.mat[lps.responsive.genes.marker.mat > 6] <- 6 # crop outliers to improve colour distribution
# lps.responsive.genes.marker.scaled_mat = t(scale(t(lps.responsive.genes.marker.mat))) # row scale
# # lps.responsive.genes.marker.scaled_mat <- lps.responsive.genes.marker.scaled_mat[!is.infinite(rowSums(lps.responsive.genes.marker.scaled_mat)),]
# 
# lps.responsive.genes.marker.heatmap <- as.ggplot(
#   Heatmap(lps.responsive.genes.marker.scaled_mat, #lps.responsive.genes.marker.mat, 
#           heatmap_legend_param = list(title = ""), 
#           cluster_rows = FALSE, cluster_columns = FALSE,
#           row_split = factor(lps.responsive.genes.df$group, levels = unique(lps.responsive.genes.df$group)),
#           row_order = lps.responsive.genes.df$gene_name,
#           row_names_gp = gpar(fontsize = 5), column_names_gp = gpar(fontsize = 10), border = TRUE, #, #, col = c(rep("red", 10), rep("blue", 8)))
#           right_annotation = rowAnnotation(markers = anno_block(gp = gpar(fill = c("white", "white")), labels = c("LPS downregulated", "LPS upregulated"), 
#                                                                 labels_gp = gpar(fontsize = 6))), row_title = NULL, cluster_row_slices = FALSE,
#           column_names_rot = 90, #column_names_centered = TRUE,
#           column_split = factor(clarke.pcaData$treatment, levels = unique(clarke.pcaData$treatment)),
#           column_title = NULL,
#           bottom_annotation = HeatmapAnnotation(markers = anno_block(gp = gpar(fill = c("white", "white")), labels = c("Untreated", "LPS Treated"), labels_gp = gpar(fontsize = 10))), border_gp = gpar(col = "black")
#           ))
# lps.responsive.genes.marker.heatmap
# # ggsave(lps.responsive.genes.marker.heatmap, filename = here(proj_path, "expression/deseq2/lps.responsive.genes.marker.heatmap.png", height = 11.75, width = 8.25)

```

## Volcano

```{r}
ctrl_lps_vs_untx.labels <- ctrl_lps_vs_untx$res  %>% filter(-log10(padj) > 5 & gene_name %in% all.microglia.go | -log10(padj) > 60 & gene_name %in% all.inflammation.go) %>% pull(gene_name)
ctrl_lps_vs_untx.volcano <- volcano_plot(ctrl_lps_vs_untx$res, gene_labels = ctrl_lps_vs_untx.labels, numerator = "LPS", denomenator = "Untreated", xmax = 11, ymax = 150)
# ctrl_lps_vs_untx.volcano

# # # histogram of differentially expressed genes
# ctrl_lps_vs_untx.dge_numbers <- c(sum(ctrl_lps_vs_untx$res$padj <= 0.05 & ctrl_lps_vs_untx$res$log2FoldChange >= 0, na.rm=TRUE), sum(ctrl_lps_vs_untx$res$padj <= 0.05 & ctrl_lps_vs_untx$res$log2FoldChange <= 0, na.rm=TRUE))
# ctrl_lps_vs_untx.dge_summary  <- as.data.frame(matrix(data = ctrl_lps_vs_untx.dge_numbers , nrow = 2, ncol = 1))
# colnames(ctrl_lps_vs_untx.dge_summary) <- "de_genes"
# ctrl_lps_vs_untx.dge_summary$direction <- c("LPS Up","LPS Down")
# ctrl_lps_vs_untx.dge_summary$sample <- "ctrl_lps_vs_untx"
# ctrl_lps_vs_untx.dge_histogram <- ggbarplot(ctrl_lps_vs_untx.dge_summary, x="direction" , y="de_genes", fill="direction", label = TRUE, lab.pos = "in", position = position_dodge(0.75), xlab = "") +
#   scale_fill_manual( values = c("firebrick2", "dodgerblue2") ) +  theme_bw() +  theme(panel.grid = element_blank(), legend.title = element_blank(), legend.position = "none") + ylab(expression(LPS:Untreated~No.~DEGs~FDR<0.05))
# ctrl_lps_vs_untx.dge_histogram


```

## GO terms

```{r}
ctrl_lps_vs_untx.dge_genes <- ctrl_lps_vs_untx$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
ctrl_lps_vs_untx.dge_gprofiles_filt <- ctrl_lps_vs_untx.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM")) %>% arrange( -log10(p_value) ) %>% 
  mutate(query = case_when(query == "query_1" ~ "LPS\n Down", query == "query_2" ~ "LPS\n Up"), term_name = as.factor(term_name)) 
ctrl_lps_vs_untx.dge_gprofiles_down <- ctrl_lps_vs_untx.dge_gprofiles_filt %>% filter(query == "LPS\n Down")
ctrl_lps_vs_untx.dge_gprofiles_up <- ctrl_lps_vs_untx.dge_gprofiles_filt %>% filter(query == "LPS\n Up")
paste('"',unique(ctrl_lps_vs_untx.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
"rRNA processing",
"Nonsense-Mediated Decay (NMD)",
"Influenza Infection",
"Cellular response to starvation",
"protein targeting to ER",
"oxidative phosphorylation",
"aerobic respiration",
"Translation",
"Viral mRNA Translation",
"ribosome",
"mitochondrial membrane")
ctrl_lps_vs_untx.dge_gprofiles_filt_down <- ctrl_lps_vs_untx.dge_gprofiles_filt %>% filter(query == "LPS\n Down")  %>% filter(term_name %in% down_list)
paste('"',unique(ctrl_lps_vs_untx.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
"response to lipopolysaccharide",
"TNF signaling pathway",
"regulation of metabolic process",
"immune system process",
"response to stress",
"regulation of response to stimulus",
"regulation of signaling",
"protein binding",
"cytokine-mediated signaling pathway",
"cellular response to cytokine stimulus")
ctrl_lps_vs_untx.dge_gprofiles_filt_up <- ctrl_lps_vs_untx.dge_gprofiles_filt %>% filter(query == "LPS\n Up")  %>% filter(term_name %in% up_list)
ctrl_lps_vs_untx.dge_gprofiles_filt_combined <- bind_rows(ctrl_lps_vs_untx.dge_gprofiles_filt_down, ctrl_lps_vs_untx.dge_gprofiles_filt_up) %>% mutate(query = factor(query, levels = c("LPS\n Up", "LPS\n Down")))
ctrl_lps_vs_untx.dge_gprofiles_filt_combined$term_name <- gsub("signaling pathway", "signaling", ctrl_lps_vs_untx.dge_gprofiles_filt_combined$term_name) %>% gsub("cellular response to ", "response to ", .)
ctrl_lps_vs_untx.go_curated <- curated_profiles(gprofiles = ctrl_lps_vs_untx.dge_gprofiles_filt_combined) + theme(strip.text.y.right = element_text(angle = 0))
# ctrl_lps_vs_untx.go_curated

# ctrl_lps_vs_untx_gprofiler2rrvgo = gprofiler2rrvgo(deseq.res = ctrl_lps_vs_untx$res, query1 = "LPS\n Down", query2 = "LPS\n Up", GO_cat = "BP")
# ctrl_lps_vs_untx_gprofiler2rrvgo$parentTerm <- gsub("SRP-dependent cotranslational ", "", ctrl_lps_vs_untx_gprofiler2rrvgo$parentTerm) %>% gsub("nuclear-transcribed mRNA catabolic process, ", "", .)
# ctrl_lps_vs_untx.go_rrvgo = rrvgo_plot(gprofiler2rrvgo = ctrl_lps_vs_untx_gprofiler2rrvgo, n_terms = 10, remove_terms = c("interspecies interaction between organisms")) + 
#   theme(axis.text = element_text(size=10), strip.text.y.right = element_text(angle = 0))
# ctrl_lps_vs_untx.go_rrvgo

# which genes are driving the terms
ctrl_lps_vs_untx$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_GABA_RECEPTOR_COMPLEX)
ctrl_lps_vs_untx$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_PLASMA_MEMBRANE_REGION)
ctrl_lps_vs_untx$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_GAP_JUNCTION)
ctrl_lps_vs_untx$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_EXTRACELLULAR_MATRIX_STRUCTURAL_CONSTITUENT_CONFERRING_COMPRESSION_RESISTANCE)
ctrl_lps_vs_untx$res %>% filter(padj < 0.05, log2FoldChange < 0, gene_name %in% gprofiler_database$`Translation`) %>% arrange(stat)%>% print(n=20) %>% pull(gene_name)

```

## Progeny & Dorothea

Pathway RespOnsive GENes (PROGENy) estimates signaling pathway activity

```{r}
# net_progeny = readRDS(here(nemo_path, "projects/ipsc-mn-als-meta/scripts/ipsc_mn_als_meta/ipsn_als_meta/net_progeny.rds"))
# net_progeny <- get_progeny(organism = 'human', top = 100)
ctrl_lps_vs_untx.res.matrix <- ctrl_lps_vs_untx$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
ctrl_lps_vs_untx_progeny.contrast_acts <- run_wmean(mat=ctrl_lps_vs_untx.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "***", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
ctrl_lps_vs_untx_progeny.pathwayActivity <- ggplot(ctrl_lps_vs_untx_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "Signaling Pathways", x = "", y = "LPS vs Untreated Normalised enrichment") +
    stat_pvalue_manual(ctrl_lps_vs_untx_progeny.contrast_acts, label = "p.signif", y.position = 24, xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE) 
# ctrl_lps_vs_untx_progeny.pathwayActivity

```

https://www.bioconductor.org/packages/release/bioc/vignettes/decoupleR/inst/doc/tf_bk.html
https://github.com/saezlab/transcriptutorial/blob/master/scripts/04_TranscriptionFactor_activity_with_Dorothea.md

```{r}
# net_dorothea = readRDS(here(nemo_path, "projects/ipsc-mn-als-meta/scripts/ipsc_mn_als_meta/ipsn_als_meta/net_dorothea.rds"))
ctrl_lps_vs_untx.dorothea <- run_wmean(mat=ctrl_lps_vs_untx.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

ctrl_lps_vs_untx.dorothea.labels.up = ctrl_lps_vs_untx.dorothea %>% top_n(score, n = 5) %>% pull(source)
ctrl_lps_vs_untx.dorothea.labels.down = ctrl_lps_vs_untx.dorothea %>% top_n(-score, n = 5) %>% pull(source)

# Volcano of TFs score vs p_value
ctrl_lps_vs_untx.dorothea.volcano = ggplot(ctrl_lps_vs_untx.dorothea, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment VCP vs CTRL", title = "Transcription Factors") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  # scale_y_continuous(expand = c(0,0), limits = c(0,ymax)) + scale_x_continuous(limits = c(-xmax,xmax))
  geom_text_repel(fontface = "italic", data = filter(ctrl_lps_vs_untx.dorothea, source %in% c(ctrl_lps_vs_untx.dorothea.labels.up, ctrl_lps_vs_untx.dorothea.labels.down)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.5) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
# ctrl_lps_vs_untx.dorothea.volcano

ctrl_lps_vs_untx_dorothea.contrast_acts %>% arrange(-score)
ctrl_lps_vs_untx_dorothea.contrast_acts %>% arrange(score)


```


```{r}
ctrl_lps_vs_untx.res.matrix <- ctrl_lps_vs_untx$res %>% select(gene_name, log2FoldChange) %>%  drop_na(log2FoldChange) %>% distinct(gene_name, .keep_all = TRUE) %>% column_to_rownames(var = "gene_name") %>% as.matrix()
ctrl_lps_vs_untx.PathwayActivity_zscore <- progeny(ctrl_lps_vs_untx.res.matrix, scale=TRUE, organism="Human", top = 100, perm = 10000, z_scores = TRUE) %>%  t()# enrichment analysis of each pathway activity using competitive permutation approach
colnames(ctrl_lps_vs_untx.PathwayActivity_zscore) <- "NES"
ctrl_lps_vs_untx.PathwayActivity_zscore_df <- as.data.frame(ctrl_lps_vs_untx.PathwayActivity_zscore) %>% rownames_to_column(var = "Pathway") %>% arrange(NES) %>% mutate(Pathway = factor(Pathway))

# barchart
ctrl_lps_vs_untx.untx_vcp_vs_ctrl_progeny.pathwayActivity <- ggplot(ctrl_lps_vs_untx.PathwayActivity_zscore_df,aes(x = reorder(Pathway, NES), y = NES)) + 
    geom_bar(aes(fill = NES), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_minimal() + theme(axis.title.x = element_text(face = "bold", size = 12), axis.title.y=element_text(size=9), axis.text.x = element_text(angle = 90, hjust = 1, size =10), axis.text.y = element_text(size =10),
                            panel.grid.major = element_blank(), panel.grid.minor = element_blank(), legend.position = "none") +
    xlab("Signalling Pathways") + ylab(expression(NES~LPS:Untreated)) +
    geom_hline(yintercept = 0, linetype = 2, colour = "darkgrey")
# ctrl_lps_vs_untx.untx_vcp_vs_ctrl_progeny.pathwayActivity


# data(dorothea_hs, package = "dorothea")
# regulons <- dorothea_hs %>%  filter(confidence %in% c("A", "B","C")) # 271 TFs
# regulons <- dorothea_hs %>%  filter(confidence %in% c("A", "B")) # 118 TFs
ctrl_lps_vs_untx.tf_activities_stat <- dorothea::run_viper(ctrl_lps_vs_untx$res.matrix, regulons, options =  list(minsize = 5, eset.filter = FALSE, cores = 4, verbose = TRUE, nes = TRUE)) %>% as_tibble(rownames = "gene_name")
# regulons <- dorothea_hs %>%  filter(confidence %in% c("A")) # 96 TFs
ctrl_lps_vs_untx.res.matrix <- ctrl_lps_vs_untx$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% column_to_rownames(var = "gene_name") %>% as.matrix()
ctrl_lps_vs_untx.tf_activities_stat <- dorothea::run_viper(ctrl_lps_vs_untx.res.matrix, regulons, options =  list(minsize = 5, eset.filter = FALSE, cores = 4, verbose = TRUE, nes = TRUE)) %>% as_tibble(rownames = "gene_name")
ctrl_lps_vs_untx.dorothea.tf.activities <- ctrl_lps_vs_untx.tf_activities_stat %>% top_n(25, wt = abs(stat)) %>% arrange(stat) %>% select(gene_name, "NES" = stat)
ctrl_lps_vs_untx.dorothea.tf.activities.barplot <- ggplot(ctrl_lps_vs_untx.dorothea.tf.activities, aes(x = reorder(gene_name, NES), y = NES)) + 
    geom_bar(aes(fill = NES), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_minimal() + theme(axis.title.x = element_text(face = "bold", size = 12), axis.title.y=element_text(size=9), axis.text.x = element_text(angle = 90, hjust = 1, size = 10), axis.text.y = element_text(size =10), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), legend.position = "none") +
    xlab("Transcription Factors") + ylab(expression(NES~LPS:Untreated)) +
    geom_hline(yintercept = 0, linetype = 2, colour = "darkgrey") 
ctrl_lps_vs_untx.dorothea.tf.activities.barplot


```

## Mass spec

```{r}
# Volcano
dep.lps_vs_untx.volcano <- volcano_plot(dep.lps_vs_untx$res, title = "Proteomics", numerator = "LPS", denomenator = "CTRL", xmax = 7, ymax = 10, gene_labels = pull(filter(dep.lps_vs_untx$res, padj < 0.05, gene_name %in% all.microglia.go | -log10(padj) > 4 & gene_name %in% all.inflammation.go), gene_name)) 
# dep.lps_vs_untx.volcano

dep.lps_vs_untx$res$sig.direction %>% table
# downregulated     unchanged   upregulated 
#            44          2727            72 

# GO terms mass spec
patani.mass_spec.dge_genes <- dep.lps_vs_untx$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_name") %>% gost() # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
patani.mass_spec.dge_gprofiles_filt <- patani.mass_spec.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM|HPA")) %>% arrange( -log10(p_value) ) %>% 
  mutate(query = case_when(query == "query_1" ~ "LPS\n Down", query == "query_2" ~ "LPS\n Up"), term_name = as.factor(term_name)) 
patani.mass_spec.dge_gprofiles_down <- patani.mass_spec.dge_gprofiles_filt %>% filter(query == "LPS\n Down")
patani.mass_spec.dge_gprofiles_up <- patani.mass_spec.dge_gprofiles_filt %>% filter(query == "LPS\n Up")
paste('"',unique(patani.mass_spec.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list = c("secretion by cell",
"regulation of localization",
"exocytosis",
"mitochondrion",
"secretory granule",
"leukocyte degranulation",
"myeloid leukocyte activation",
"lysosomal lumen",
"vesicle-mediated transport",
"vacuole",
"lysosome")
patani.mass_spec.dge_gprofiles_filt_down <- patani.mass_spec.dge_gprofiles_filt %>% filter(query == "LPS\n Down") %>% filter(term_name %in% down_list)
paste('"',unique(patani.mass_spec.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list = c(
# "exocytosis",
# "cell activation",
# "secretion by cell",
# "Interferon Signaling",
# "cytokine-mediated signaling pathway",
"response to stress",
"extracellular space",
# "defense response",
# "extracellular region",
# "innate immune response",
# "response to external stimulus",
"response to chemical",
"immune response",
"response to cytokine",
"vesicle",
"extracellular exosome",
"protein binding",
"cell-substrate junction",
"focal adhesion")
patani.mass_spec.dge_gprofiles_filt_up <- patani.mass_spec.dge_gprofiles_filt %>% filter(query == "LPS\n Up")  %>% filter(term_name %in% up_list) %>% mutate(term_name = gsub("myeloid cell activation involved in immune response", "myeloid activation in immune response", term_name))
patani.mass_spec.dge_gprofiles_filtered = bind_rows(patani.mass_spec.dge_gprofiles_filt_up, patani.mass_spec.dge_gprofiles_filt_down) %>% mutate(query = fct_relevel(query,"LPS\n Up"))
lps_vs_untx_mass_spec.go_curated <- curated_profiles(patani.mass_spec.dge_gprofiles_filtered, gprofiles =, cols = c("firebrick2", "dodgerblue2")) + theme(strip.text.y.right = element_text(angle = 0))# + theme(legend.position = "none", axis.text = element_text(size = 10))
# lps_vs_untx_mass_spec.go_curated


# GSEA phagocytosis mass spec
massspec_lps_vs_untx.geneList <- dep.lps_vs_untx$res  %>% drop_na(log2FoldChange) %>% pull(log2FoldChange)
names(massspec_lps_vs_untx.geneList) <- dep.lps_vs_untx$res  %>% drop_na(log2FoldChange) %>% pull(gene_name) %>% as.character()
massspec_lps_vs_untx.geneList = sort(massspec_lps_vs_untx.geneList, decreasing = TRUE)
phagocytosis_genes.list <- list("phagocytosis" = dep.lps_vs_untx$res $gene_name[dep.lps_vs_untx$res $gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS])
massspec_lps_vs_untx.phagocytosis_fgseaRes <- fgsea(pathways = phagocytosis_genes.list, stats = massspec_lps_vs_untx.geneList)
massspec_lps_vs_untx.phagocytosis_fgseaRes
#         pathway      pval      padj    log2err         ES        NES size                                leadingEdge
# 1: phagocytosis 0.6089965 0.6089965 0.06197627 -0.2769121 -0.9342016  107 COLEC12,ITGAL,SCARB1,CRK,CDC42SE2,MSR1,...
massspec_lps_vs_untx.phagocytosis.gseaplot = plotEnrichment(phagocytosis_genes.list[["phagocytosis"]], massspec_lps_vs_untx.geneList) +
  theme(axis.text=element_text(size=8), axis.title=element_text(size=10)) + ylab(expression(Phagocytosis)) + xlab(expression(Untreated~LPS~vs~CTRL)) +
  annotate("text", x = 100, y = -0.15, label = paste0("NES = ", round(massspec_lps_vs_untx.phagocytosis_fgseaRes$NES, digits = 3),"\nP = ", round(massspec_lps_vs_untx.phagocytosis_fgseaRes$pval, digits = 3)), size = 3, hjust = 0)
massspec_lps_vs_untx.phagocytosis.gseaplot

multi_gsea_list <- list("GO_PHAGOCYTOSIS" = dep.lps_vs_untx$res$gene_name[dep.lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS],
                        "GO_POSITIVE_REGULATION_OF_MITOCHONDRIAL_MEMBRANE_POTENTIAL" = dep.lps_vs_untx$res$gene_name[dep.lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_POSITIVE_REGULATION_OF_MITOCHONDRIAL_MEMBRANE_POTENTIAL],
                        "GO_MITOCHONDRIAL_GENE_EXPRESSION" = dep.lps_vs_untx$res$gene_name[dep.lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_MITOCHONDRIAL_GENE_EXPRESSION],
                        "GO_REGULATION_OF_MITOCHONDRIAL_ATP_SYNTHESIS_COUPLED_ELECTRON_TRANSPORT" = dep.lps_vs_untx$res$gene_name[dep.lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_REGULATION_OF_MITOCHONDRIAL_ATP_SYNTHESIS_COUPLED_ELECTRON_TRANSPORT],
                        "GO_CHAPERONE_MEDIATED_AUTOPHAGY" = dep.lps_vs_untx$res$gene_name[dep.lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_CHAPERONE_MEDIATED_AUTOPHAGY],
                        "GO_MACROAUTOPHAGY" = dep.lps_vs_untx$res$gene_name[dep.lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_MACROAUTOPHAGY],
                        "GO_REGULATION_OF_AUTOPHAGY" = dep.lps_vs_untx$res$gene_name[dep.lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_REGULATION_OF_AUTOPHAGY],
                        "GO_MICROGLIAL_CELL_MIGRATION" = dep.lps_vs_untx$res$gene_name[dep.lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_MICROGLIAL_CELL_MIGRATION],
                        "GO_MICROGLIAL_CELL_ACTIVATION" = dep.lps_vs_untx$res$gene_name[dep.lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_MICROGLIAL_CELL_ACTIVATION],
                        "GO_REGULATION_OF_MICROGLIAL_CELL_ACTIVATION" = dep.lps_vs_untx$res$gene_name[dep.lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_REGULATION_OF_MICROGLIAL_CELL_ACTIVATION])
dep.lps_vs_untx.multi_fgseaRes <- fgsea(pathways = multi_gsea_list, stats = massspec_lps_vs_untx.geneList)
dep.lps_vs_untx.multi_fgseaRes

dep.lps_vs_untx$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS) %>% print(n=Inf) # 7
dep.lps_vs_untx$res %>% filter(gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS) # 104
dep.lps_vs_untx$res %>% filter(gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS, log2FoldChange < 0) # 53 49.5%

dep.lps_vs_untx$res %>% filter(gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS) %>% mutate(phagocytosis_regulator = case_when(gene_name %in% go.pathways.msigdb$GO_POSITIVE_REGULATION_OF_PHAGOCYTOSIS ~ "positive", 
                                                                                             gene_name %in% gene_name %in% go.pathways.msigdb$GO_NEGATIVE_REGULATION_OF_PHAGOCYTOSIS ~ "negative", TRUE ~ "")) %>%
  write_csv(here(proj_path, "mass-spectrometry/mass_spec.lps_vs_untx.phagocytosis_proteins.csv")

```


## Cytokine heatmap

```{r}
# https://www.dropbox.com/work/PataniR/Ben/Microglia%20figures

# # import with magick 
# FigS3H_cytokine_LPS_ctrl.png = magick::image_read(here(proj_path,"ben-figures/FigS3H_cytokine_heatmap.tiff"))
# FigS3I_cytokine_LPS_ctrl.gg = ggdraw() + draw_image(FigS3H_cytokine_LPS_ctrl.png)
# FigS3I_cytokine_LPS_ctrl.gg

lps_cytokine_data = read_excel(here(proj_path, "expression/luminex/cytokine.lps_treatment.xlsx")) %>% 
  pivot_longer(!Treatment, names_to = "cytokine", values_to = "intensity") %>%
  group_by(Treatment, cytokine) %>% mutate(mean_intensity = mean(intensity)) %>% ungroup() #%>% filter(cytokine %in% c("IL1b", "IL10","IL6","IL12","CCL11", "CCL3", "CCL4", "CCL2", "IFNa", "IL1RA","TNFa", "CXCL10","IL8"))   
lps_cytokine_data %>% glimpse

lps_cytokine_data %>% my_summarise(mean_intensity, cytokine, Treatment) %>% print(n=Inf)

lps_cytokine_data.wide = lps_cytokine_data %>% group_by(cytokine, Treatment) %>% summarise(intensity = mean(intensity)) %>% pivot_wider(names_from = "Treatment", values_from = "intensity") %>% #mutate(log2FoldChange = log2(VCP) - log2(CTRL)) %>% 
  ungroup() 
lps_cytokine_data.lfc.mat <- make_matrix(select(lps_cytokine_data.wide,-cytokine), pull(lps_cytokine_data.wide,cytokine))

col_fun = colorRamp2(c(-0.623, 1, 6.68), c("blue", "white", "red"))
lgd = Legend(col_fun = colorRamp2(c(-0.623, 1, 6.68), c("blue", "white", "red")))

lps.cytokine.luminex.heatmap <- as.ggplot(Heatmap(lps_cytokine_data.lfc.mat, 
          heatmap_legend_param = list(title = "", at = c(0, 3, 6)),
          col = colorRamp2(c(-0.623, 1, 6.68), c("blue", "white", "red")),
          cluster_rows = FALSE, cluster_row_slices = FALSE, cluster_columns = FALSE, border = TRUE, 
          # row_order = myeloid.markers.df$gene_name,
          row_names_gp = gpar(fontsize = 8), column_names_gp = gpar(fontsize = 8), #, #, col = c(rep("red", 10), rep("blue", 8)))
          # right_annotation = rowAnnotation(markers = anno_block(gp = gpar(fill = "white"), labels = "Cytokines", labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          row_title_gp = gpar(fontsize = 8), column_title = NULL, column_names_rot = 0, column_names_centered = TRUE, column_split =  c(1,2)))
lps.cytokine.luminex.heatmap

```


## Merge

A - PCA LPS vs untreated microglia
B - Dendrogram LPS vs untreated microglia
C - Volcano plot gene expression LPS vs untreated control microglia
D - GO terms gene expression LPS vs untreated control microglia
E - Progeny LPS vs untreated control microglia
F - Dorothea LPS vs untreated control microglia
G - Volcano plot proteomics LPS vs untreated control microglia
H - GO terms proteomics LPS vs untreated control microglia
I - Heat map of cytokines and chemokines LPS vs untreated control microglia

```{r}
fig.s3.merged <- plot_grid(
  clarke.pca,  
  plot_grid(ctrl_lps_vs_untx.volcano, ctrl_lps_vs_untx.go_curated, ncol = 2, rel_widths = c(1.2,1), labels = c("B", "C")),
  plot_grid(ctrl_lps_vs_untx_progeny.pathwayActivity, ctrl_lps_vs_untx.dorothea.volcano, ncol = 2, rel_widths = c(1,1.2), labels = c("D", "E")),
  plot_grid(dep.lps_vs_untx.volcano, lps_vs_untx_mass_spec.go_curated, NULL, lps.cytokine.luminex.heatmap, ncol = 4, rel_widths = c(1.2,1,0.07,0.8), labels = c("F", "G","H","")), #
  nrow = 4, rel_heights = c(0.7,1,0.9,1), labels = c("A","","",""))
# fig.s3.merged
ggsave(fig.s3.merged, filename = here(proj_path, "expression/deseq2/fig.s3.merged.png"), height = 13, width = 9)
ggsave(fig.s3.merged, filename = here(proj_path, "expression/deseq2/fig.s3.merged.pdf"), height = 13, width = 9)


# plot_grid(clarke.pca, clarke.dendrogram, nrow = 2, rel_heights = c(0.7,0.6), labels = c("A", "B")),
       

```



### GSEA

```{r}
ctrl_lps_vs_untx.geneList <- ctrl_lps_vs_untx$res %>% drop_na(stat) %>% pull(stat)
names(ctrl_lps_vs_untx.geneList) <- ctrl_lps_vs_untx$res %>% drop_na(stat) %>% pull(gene_name) %>% as.character()
ctrl_lps_vs_untx.geneList = sort(ctrl_lps_vs_untx.geneList, decreasing = TRUE)

ctrl_lps_vs_untx.phagocytosis_genes.list <- list("phagocytosis" = ctrl_lps_vs_untx$res$gene_name[ctrl_lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS]) # gprofiler_database$`glutamatergic synapse`
ctrl_lps_vs_untx.phagocytosis_fgseaRes <- fgsea(pathways = ctrl_lps_vs_untx.phagocytosis_genes.list, stats = ctrl_lps_vs_untx.geneList)
ctrl_lps_vs_untx.phagocytosis_fgseaRes
#         pathway        pval        padj   log2err        ES      NES size
# 1: phagocytosis 0.003233999 0.003233999 0.4317077 0.3415305 1.450964  263
ctrl_lps_vs_untx.phagocytosis.gseaplot = plotEnrichment(ctrl_lps_vs_untx.phagocytosis_genes.list[["phagocytosis"]], ctrl_lps_vs_untx.geneList) + 
  theme(axis.text=element_text(size=8), axis.title=element_text(size=10)) + ylab(expression(Phagocytosis~GSEA)) + xlab(expression(LPS~vs~Untreated))+
  annotate("text", x = 23000, y = 0.25, label = paste0("NES = ", round(ctrl_lps_vs_untx.phagocytosis_fgseaRes$NES, digits = 3),"\nP = ", round(ctrl_lps_vs_untx.phagocytosis_fgseaRes$pval, digits = 4)), size = 3, hjust = 0)
ctrl_lps_vs_untx.phagocytosis.gseaplot

# GSEA autophagy RNAseq
autophagy.go = unique(c(go.pathways.msigdb$GO_REGULATION_OF_AUTOPHAGY, go.pathways.msigdb$GO_MACROAUTOPHAGY, go.pathways.msigdb$GO_CHAPERONE_MEDIATED_AUTOPHAGY))
autophagy_genes.list <- list("autophagy" = ctrl_lps_vs_untx$res$gene_name[ctrl_lps_vs_untx$res$gene_name %in% autophagy.go])
ctrl_lps_vs_untx.autophagy_fgseaRes <- fgsea(pathways = autophagy_genes.list, stats = ctrl_lps_vs_untx.geneList)
ctrl_lps_vs_untx.autophagy_fgseaRes
#     pathway     pval     padj    log2err        ES      NES size                                    leadingEdge
# 1: autophagy 0.895122 0.895122 0.02751829 0.1939684 0.855609  439 LAMP3,MEFV,NAMPT,MAP1LC3A,ZC3H12A,ATP6V1B1,...
ctrl_lps_vs_untx.autophagy.gseaplot = plotEnrichment(autophagy_genes.list[["autophagy"]], ctrl_lps_vs_untx.geneList) + 
  theme(axis.text=element_text(size=8), axis.title=element_text(size=10)) + ylab(expression(Autophagy~GSEA)) + xlab(expression(LPS~vs~Untreated)) +
  annotate("text", x = 1000, y = -0.04, label = paste0("NES = ", round(ctrl_lps_vs_untx.autophagy_fgseaRes$NES, digits = 3),"\nP = ", round(ctrl_lps_vs_untx.autophagy_fgseaRes$pval, digits = 3)), size = 3, hjust = 0)
ctrl_lps_vs_untx.autophagy.gseaplot

# GSEA mitochondrial RNAseq
mitochondrial.go = unique(c(go.pathways.msigdb$GO_POSITIVE_REGULATION_OF_MITOCHONDRIAL_MEMBRANE_POTENTIAL, go.pathways.msigdb$GO_MITOCHONDRIAL_GENE_EXPRESSION, go.pathways.msigdb$GO_REGULATION_OF_MITOCHONDRIAL_ATP_SYNTHESIS_COUPLED_ELECTRON_TRANSPORT))
mitochondrial_genes.list <- list("mitochondrial" = ctrl_lps_vs_untx$res$gene_name[ctrl_lps_vs_untx$res$gene_name %in% mitochondrial.go])
ctrl_lps_vs_untx.mitochondrial_fgseaRes <- fgsea(pathways = mitochondrial_genes.list, stats = ctrl_lps_vs_untx.geneList)
ctrl_lps_vs_untx.mitochondrial_fgseaRes
#         pathway         pval         padj   log2err         ES       NES size                                leadingEdge
# 1: mitochondrial 4.330731e-05 4.330731e-05 0.5573322 -0.3836533 -1.729287  181 CDK1,CHCHD10,PPARGC1B,CCNB1,MRPL24,NNT,...
ctrl_lps_vs_untx.mitochondrial.gseaplot = plotEnrichment(mitochondrial_genes.list[["mitochondrial"]], ctrl_lps_vs_untx.geneList) + 
  theme(axis.text=element_text(size=8), axis.title=element_text(size=10)) + ylab(expression(Mitochondrial~GSEA)) + xlab(expression(LPS~vs~Untreated)) +
  annotate("text", x = 1000, y = -0.35, label = paste0("NES = ", round(ctrl_lps_vs_untx.mitochondrial_fgseaRes$NES, digits = 3),"\nP = ", round(ctrl_lps_vs_untx.mitochondrial_fgseaRes$pval, digits = 3)), size = 3, hjust = 0)
ctrl_lps_vs_untx.mitochondrial.gseaplot

multi_gsea_list <- list("GO_ATF6_MEDIATED_UNFOLDED_PROTEIN_RESPONSE" = ctrl_lps_vs_untx$res$gene_name[ctrl_lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_ATF6_MEDIATED_UNFOLDED_PROTEIN_RESPONSE],
                        "GO_CELLULAR_RESPONSE_TO_UNFOLDED_PROTEIN" = ctrl_lps_vs_untx$res$gene_name[ctrl_lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_CELLULAR_RESPONSE_TO_UNFOLDED_PROTEIN],
                        "GO_ENDOPLASMIC_RETICULUM_UNFOLDED_PROTEIN_RESPONSE" = ctrl_lps_vs_untx$res$gene_name[ctrl_lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_ENDOPLASMIC_RETICULUM_UNFOLDED_PROTEIN_RESPONSE],
                        "GO_IRE1_MEDIATED_UNFOLDED_PROTEIN_RESPONSE" = ctrl_lps_vs_untx$res$gene_name[ctrl_lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_IRE1_MEDIATED_UNFOLDED_PROTEIN_RESPONSE],
                        "GO_PERK_MEDIATED_UNFOLDED_PROTEIN_RESPONSE" = ctrl_lps_vs_untx$res$gene_name[ctrl_lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_PERK_MEDIATED_UNFOLDED_PROTEIN_RESPONSE],
                        "GO_POSITIVE_REGULATION_OF_ENDOPLASMIC_RETICULUM_UNFOLDED_PROTEIN_RESPONSE" = ctrl_lps_vs_untx$res$gene_name[ctrl_lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_POSITIVE_REGULATION_OF_ENDOPLASMIC_RETICULUM_UNFOLDED_PROTEIN_RESPONSE],
                        "GO_POSITIVE_REGULATION_OF_IRE1_MEDIATED_UNFOLDED_PROTEIN_RESPONSE" = ctrl_lps_vs_untx$res$gene_name[ctrl_lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_POSITIVE_REGULATION_OF_IRE1_MEDIATED_UNFOLDED_PROTEIN_RESPONSE])
ctrl_lps_vs_untx.multi_fgseaRes <- fgsea(pathways = multi_gsea_list, stats = ctrl_lps_vs_untx.geneList)
ctrl_lps_vs_untx.multi_fgseaRes


multi_gsea_list <- list("GO_PHAGOCYTOSIS" = ctrl_lps_vs_untx$res$gene_name[ctrl_lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS],
                        "GO_POSITIVE_REGULATION_OF_PHAGOCYTOSIS" = ctrl_lps_vs_untx$res$gene_name[ctrl_lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_POSITIVE_REGULATION_OF_PHAGOCYTOSIS],
                        "GO_NEGATIVE_REGULATION_OF_PHAGOCYTOSIS" = ctrl_lps_vs_untx$res$gene_name[ctrl_lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_NEGATIVE_REGULATION_OF_PHAGOCYTOSIS],
                        "GO_POSITIVE_REGULATION_OF_MITOCHONDRIAL_MEMBRANE_POTENTIAL" = ctrl_lps_vs_untx$res$gene_name[ctrl_lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_POSITIVE_REGULATION_OF_MITOCHONDRIAL_MEMBRANE_POTENTIAL],
                        "GO_MITOCHONDRIAL_GENE_EXPRESSION" = ctrl_lps_vs_untx$res$gene_name[ctrl_lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_MITOCHONDRIAL_GENE_EXPRESSION],
                        "GO_REGULATION_OF_MITOCHONDRIAL_ATP_SYNTHESIS_COUPLED_ELECTRON_TRANSPORT" = ctrl_lps_vs_untx$res$gene_name[ctrl_lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_REGULATION_OF_MITOCHONDRIAL_ATP_SYNTHESIS_COUPLED_ELECTRON_TRANSPORT],
                        "GO_CHAPERONE_MEDIATED_AUTOPHAGY" = ctrl_lps_vs_untx$res$gene_name[ctrl_lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_CHAPERONE_MEDIATED_AUTOPHAGY],
                        "GO_MACROAUTOPHAGY" = ctrl_lps_vs_untx$res$gene_name[ctrl_lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_MACROAUTOPHAGY],
                        "GO_REGULATION_OF_AUTOPHAGY" = ctrl_lps_vs_untx$res$gene_name[ctrl_lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_REGULATION_OF_AUTOPHAGY],
                        "GO_MICROGLIAL_CELL_MIGRATION" = ctrl_lps_vs_untx$res$gene_name[ctrl_lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_MICROGLIAL_CELL_MIGRATION],
                        "GO_MICROGLIAL_CELL_ACTIVATION" = ctrl_lps_vs_untx$res$gene_name[ctrl_lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_MICROGLIAL_CELL_ACTIVATION],
                        "GO_REGULATION_OF_MICROGLIAL_CELL_ACTIVATION" = ctrl_lps_vs_untx$res$gene_name[ctrl_lps_vs_untx$res$gene_name %in% go.pathways.msigdb$GO_REGULATION_OF_MICROGLIAL_CELL_ACTIVATION])
ctrl_lps_vs_untx.multi_fgseaRes <- fgsea(pathways = multi_gsea_list, stats = ctrl_lps_vs_untx.geneList)
ctrl_lps_vs_untx.multi_fgseaRes

# pathway         pval        padj    log2err         ES        NES size
#  1:                                         GO_CHAPERONE_MEDIATED_AUTOPHAGY 0.0520646320 0.118279570 0.26635066  0.5922658  1.5251693   16
#  2:                                                       GO_MACROAUTOPHAGY 0.9934810952 0.993481095 0.02555230  0.1743307  0.7467814  287
#  3:                                            GO_MICROGLIAL_CELL_MIGRATION 0.1652173913 0.263275991 0.15964670 -0.6169197 -1.3508770    7
#  4:                                        GO_MITOCHONDRIAL_GENE_EXPRESSION 0.0001018806 0.001120686 0.53843410 -0.3873336 -1.6968096  157
#  5:                                  GO_NEGATIVE_REGULATION_OF_PHAGOCYTOSIS 0.5854922280 0.715601612 0.06378454  0.3391349  0.8922882   18
#  6:                                                         GO_PHAGOCYTOSIS 0.0085656589 0.047111124 0.38073040  0.3234753  1.3681726  255
#  7:              GO_POSITIVE_REGULATION_OF_MITOCHONDRIAL_MEMBRANE_POTENTIAL 0.3934065934 0.540934066 0.09787733 -0.4550977 -1.0630189    9
#  8:                                  GO_POSITIVE_REGULATION_OF_PHAGOCYTOSIS 0.0223941717 0.082111963 0.35248786  0.4238859  1.4703645   63
#  9:                                              GO_REGULATION_OF_AUTOPHAGY 0.7424836601 0.816732026 0.03984069  0.2103614  0.9057966  309
# 10:                             GO_REGULATION_OF_MICROGLIAL_CELL_ACTIVATION 0.1675392670 0.263275991 0.14040624  0.4735097  1.2705942   20
# 11: GO_REGULATION_OF_MITOCHONDRIAL_ATP_SYNTHESIS_COUPLED_ELECTRON_TRANSPORT 0.0537634409 0.118279570 0.28785712 -0.7465641 -1.5541144    6

filter(ctrl_lps_vs_untx.res, gene_name %in% go.pathways.msigdb$GO_PHAGOCYTOSIS, padj < 0.05, log2FoldChange < 0) %>% arrange(log2FoldChange)
filter(ctrl_lps_vs_untx.res, gene_name %in% gprofiler_database$`glutamatergic synapse`, padj < 0.05, log2FoldChange < 0) %>% arrange(log2FoldChange)
filter(ctrl_lps_vs_untx.res, gene_name %in% go.pathways.msigdb$GO_GLUTAMATE_METABOLIC_PROCESS, padj < 0.05, log2FoldChange < 0) %>% arrange(log2FoldChange)
filter(ctrl_lps_vs_untx.res, gene_name %in% c("SLC1A2", "TMEM259"), padj < 0.05, log2FoldChange < 0) %>% arrange(log2FoldChange)

# plot_grid(ctrl_lps_vs_untx.phagocytosis.gseaplot, ctrl_lps_vs_untx.autophagy.gseaplot, ctrl_lps_vs_untx.mitochondrial.gseaplot, labels = c("C", "D", "E"), ncol = 3),

```



# Table S9 SOD1 mouse & ipsc microglia overlap DGE

```{r}
mouse_sod1_ipsc_overlap.res = 
  mutate(select(chiu_end_stage_untx_sod1_vs_ctrl$res, gene_name, padj.mouse_sod1 = padj, stat.mouse_sod1 = stat, lfc.mouse_sod1 = log2FoldChange), direction.mouse_sod1 = case_when(padj.mouse_sod1 < 0.05 & lfc.mouse_sod1 > 0 ~ "up", padj.mouse_sod1 < 0.05 & lfc.mouse_sod1 < 0 ~ "down", TRUE ~ "non-significant")) %>%
  full_join(mutate(select(untx_vcp_vs_ctrl$res, gene_name, padj.ipsm = padj, stat.ipsm = stat, lfc.ipsm = log2FoldChange), direction.ipsm = case_when(padj.ipsm < 0.05 & lfc.ipsm > 0 ~ "up", padj.ipsm < 0.05 & lfc.ipsm < 0 ~ "down", TRUE ~ "non-significant"))) %>%
  select(gene_name, everything()) %>% 
  filter(padj.ipsm < 0.05, padj.mouse_sod1 < 0.05, ((stat.ipsm > 0 & stat.mouse_sod1 > 0) | (stat.ipsm < 0 & stat.mouse_sod1 < 0)))
mouse_sod1_ipsc_overlap.res
write_csv(mouse_sod1_ipsc_overlap.res, here(proj_path,"expression/deseq2/Table.mouse_sod1_ipsc_overlap.csv"))

library(googlesheets4)
gs4_browse("https://docs.google.com/spreadsheets/d/1CVDodcgGdB2rDqnDtfGWMUMyb-BdNFmfzyAX0F-bQqY/edit#gid=0")
write_sheet(mouse_sod1_ipsc_overlap.res, ss = "https://docs.google.com/spreadsheets/d/1CVDodcgGdB2rDqnDtfGWMUMyb-BdNFmfzyAX0F-bQqY/edit#gid=0", sheet = "S9. Mouse SOD1")
sheet_properties("https://docs.google.com/spreadsheets/d/1CVDodcgGdB2rDqnDtfGWMUMyb-BdNFmfzyAX0F-bQqY/edit#gid=0")



```

# Fig 5 Postmortem & SOD1 mouse correlation

A - Global correlation PM total vs VCP microglia
B - M17 module vs VCP microglia
C - M37 module vs VCP microglia
D - Venn PM total vs VCP microglia
E - GO terms PM total vs VCP microglia

## SOD1 mutant microglia

Chiu et al

chiu_end_stage_untx_sod1_vs_ctrl

```{r}
# Scatter
untx_vcp_vs_ctrl_chiu_end_stage_untx_sod1_vs_ctrl.labels = select(chiu_end_stage_untx_sod1_vs_ctrl$res, gene_name, untx_vcp_vs_ctrl.stat = stat) %>% 
  left_join(select(untx_vcp_vs_ctrl$res, gene_name, chiu_end_stage_untx_sod1_vs_ctrl.stat = stat)) %>% filter((untx_vcp_vs_ctrl.stat > 2.5 & chiu_end_stage_untx_sod1_vs_ctrl.stat > 2.5) | (untx_vcp_vs_ctrl.stat < -2.5 & chiu_end_stage_untx_sod1_vs_ctrl.stat < -2.5), gene_name %in% all.reactivity.go)
untx_vcp_vs_ctrl_chiu_end_stage_untx_sod1_vs_ctrl.list = list("VCP vs CTRL" = untx_vcp_vs_ctrl$res, "SOD1 vs CTRL" = chiu_end_stage_untx_sod1_vs_ctrl$res)
untx_vcp_vs_ctrl_chiu_end_stage_untx_sod1_vs_ctrl_stat.corr = plot_de_correlation(model_list1 = untx_vcp_vs_ctrl_chiu_end_stage_untx_sod1_vs_ctrl.list, model_list2 = untx_vcp_vs_ctrl_chiu_end_stage_untx_sod1_vs_ctrl.list, mutation1 = "VCP vs CTRL", mutation2 = "SOD1 vs CTRL", gene_labels = pull(untx_vcp_vs_ctrl_chiu_end_stage_untx_sod1_vs_ctrl.labels,gene_name), plot_corr_line = TRUE, join_by = "gene_name", col = "log2FoldChange") + ylim(c(-8,8)) + xlim(c(-5,5)) +
  annotate(geom="text", x=3, y=8, label="SOD1 & VCP both Up", color="black", size = 3.2)  + annotate(geom="text", x=-3, y=-8, label="SOD1 & VCP both Down", color="black", size = 3.2)
untx_vcp_vs_ctrl_chiu_end_stage_untx_sod1_vs_ctrl_stat.corr

chiu_end_stage_untx_sod1_vs_ctrl.dge <- chiu_end_stage_untx_sod1_vs_ctrl$res %>% select(gene_name, stat.LPS = stat, FDR.LPS = padj)
untx_vcp_vs_ctrl.dge <- untx_vcp_vs_ctrl$res %>% select(gene_name, gene_id, stat.VCP = stat, FDR.VCP = padj)
chiu_end_stage_untx_sod1_vs_ctrl_untx_vcp_vs_ctrl.dge <- full_join(chiu_end_stage_untx_sod1_vs_ctrl.dge, untx_vcp_vs_ctrl.dge, by = "gene_name") %>%
  mutate(overlap = case_when(FDR.LPS < 0.05 & FDR.VCP < 0.05 ~ "overlapping", # down in LPS
                                                        FDR.LPS < 0.05 ~ "LPS specific",
                                                        FDR.VCP < 0.05 ~ "VCP specific",
                                                        TRUE ~ "None"),
                                    overlap_direction = case_when(FDR.LPS < 0.05 & stat.LPS > 0 & FDR.VCP < 0.05 & stat.VCP > 0 ~ "overlapping up", # down in LPS
                                                                  FDR.LPS < 0.05 & stat.LPS < 0 & FDR.VCP < 0.05 & stat.VCP < 0 ~ "overlapping down",
                                                                  FDR.LPS < 0.05 & stat.LPS < 0 & FDR.VCP < 0.05 & stat.VCP > 0 ~ "LPS down, VCP up", # down in LPS
                                                                  FDR.LPS < 0.05 & stat.LPS > 0 & FDR.VCP < 0.05 & stat.VCP < 0 ~ "LPS up, VCP down",
                                                                  FDR.LPS < 0.05 & stat.LPS > 0 ~ "LPS specific up",
                                                                  FDR.LPS < 0.05 & stat.LPS < 0 ~ "LPS specific down",
                                                                  FDR.VCP < 0.05 & stat.VCP > 0 ~ "VCP specific up",
                                                                  FDR.VCP < 0.05 & stat.VCP < 0 ~ "VCP specific down",
                                                                  TRUE ~ "None"),
                                    overlap_direction = factor(overlap_direction, levels = c("overlapping up", "overlapping down", "LPS down, VCP up", "LPS up, VCP down", "LPS specific up", "LPS specific down", "VCP specific up", "VCP specific down", "None")),
                                    overlap = factor(overlap, levels = c("overlapping",  "LPS specific", "VCP specific", "None"))) %>%
  arrange(overlap_direction)
table(chiu_end_stage_untx_sod1_vs_ctrl_untx_vcp_vs_ctrl.dge$overlap)
table(chiu_end_stage_untx_sod1_vs_ctrl_untx_vcp_vs_ctrl.dge$overlap_direction)
# overlapping up  overlapping down  LPS down, VCP up  LPS up, VCP down   LPS specific up LPS specific down   VCP specific up VCP specific down              None
#                16                 1                11                 7              1596              1181                84                65             96436
cor.test(x = chiu_end_stage_untx_sod1_vs_ctrl_untx_vcp_vs_ctrl.dge$stat.LPS, y = chiu_end_stage_untx_sod1_vs_ctrl_untx_vcp_vs_ctrl.dge$stat.VCP) # -0.04
t.test( x = chiu_end_stage_untx_sod1_vs_ctrl_untx_vcp_vs_ctrl.dge$stat.LPS, y = chiu_end_stage_untx_sod1_vs_ctrl_untx_vcp_vs_ctrl.dge$stat.VCP, paired = TRUE )

# untx_vcp_vs_ctrl.VCP_LPS_scatter <- ggscatter(chiu_end_stage_untx_sod1_vs_ctrl_untx_vcp_vs_ctrl.dge, x = "stat.LPS", y = "stat.VCP", color = "overlap_direction", palette = c("firebrick2", "dodgerblue2", "forestgreen", "gold2", "grey", "grey", "grey", "grey", "grey"), size = 0.7,
#                                         cor.coef = FALSE, #cor.method = "pearson", cor.coeff.args = list(label.x = -5.5, label.y = 4, label.sep="\n"),
#           xlab = "z-statistic SOD1 / CTRL", ylab = "z-statistic VCP / CTRL") +
#           geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") + geom_vline(xintercept = 0, linetype = 3, colour = "darkgrey") +
#     coord_cartesian(xlim = c(-5,15), ylim = c(-6,7)) +
#     theme(axis.text=element_text(size=8), axis.title=element_text(size=10), legend.position = "none") +
#     # geom_smooth(data = chiu_end_stage_untx_sod1_vs_ctrl_untx_vcp_vs_ctrl.dge, aes(x = stat.LPS, y = stat.VCP), method = "lm", se = FALSE, show.legend = TRUE, colour = "black", linetype = 2) + theme(legend.position = "none") +
#     geom_text_repel(data = filter(chiu_end_stage_untx_sod1_vs_ctrl_untx_vcp_vs_ctrl.dge, overlap == "overlapping"), aes(x = stat.LPS, y = stat.VCP, label = gene_name),  # , gene_name %in% all.reactivity.go, abs(stat.VCP) > 0.5, abs(stat.LPS) > 1
#                     fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.8, max.overlaps = 30) +
#   annotate(geom="text", x=9, y=7, label="SOD1 & VCP both Up", color="firebrick2", size = 3.2)  + annotate(geom="text", x=-3, y=-6, label="SOD1 & VCP both Down", color="dodgerblue2", size = 3.2)
# untx_vcp_vs_ctrl.VCP_LPS_scatter

chiu_end_stage_untx_sod1_vs_ctrl_untx_vcp_vs_ctrl.dge %>% filter(overlap_direction == "overlapping up") %>% pull("gene_id") %>% gost() 


# Venn
untx_vcp_vs_ctrl.res.dge.up <- untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, stat > 0) %>% pull(gene_name)
untx_vcp_vs_ctrl.res.dge.down <- untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, stat < 0) %>% pull(gene_name)
chiu_end_stage_untx_sod1_vs_ctrl.res.dge.up <- chiu_end_stage_untx_sod1_vs_ctrl$res %>% filter(padj < 0.05, stat > 0) %>% pull(gene_name)
chiu_end_stage_untx_sod1_vs_ctrl.res.dge.down <- chiu_end_stage_untx_sod1_vs_ctrl$res %>% filter(padj < 0.05, stat < 0) %>% pull(gene_name)

# 4 way overlap
untx_vcp_vs_ctrl_chiu_sod1_degs_venn = ggvenn(list(`mouse\nSOD1\ndown` = chiu_end_stage_untx_sod1_vs_ctrl.res.dge.down, `mouse\nSOD1 up` = chiu_end_stage_untx_sod1_vs_ctrl.res.dge.up, `iPS-MG\nVCP up` = untx_vcp_vs_ctrl.res.dge.up, `iPS-MG\nVCP\ndown` = untx_vcp_vs_ctrl.res.dge.down),
                                                fill_color = c("dodgerblue3", "firebrick3", "firebrick1", "dodgerblue1"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.5)
untx_vcp_vs_ctrl_chiu_sod1_degs_venn


# # Merge
# figureS6.merged = plot_grid(untx_vcp_vs_ctrl_chiu_end_stage_untx_sod1_vs_ctrl_stat.corr, untx_vcp_vs_ctrl_chiu_sod1_degs_venn, ncol = 2, rel_widths = c(1.5,1.1), labels = c("A", "B"))
# ggsave(figureS6.merged, filename = here(proj_path, "expression/deseq2/figureS6.merged.chiu.overlap.png"), height = 5, width = 10)
# ggsave(figureS6.merged, filename = here(proj_path, "expression/deseq2/figureS6.merged.chiu.overlap.pdf"), height = 5, width = 10)

```

Overlap DAM genes from Keren-Shaul Cell 2017 TREM2 DAMs & ALS DAMs

Chiu Cell reports 2013 SOD1 DEGs

```{r}
trem2_dam_genes =  read_excel(here(proj_path, "expression/keren_shaul.cell.2017.tableS2.dam_trem2.xlsx") %>% clean_names() %>% 
  mutate(gene_name = toupper(x1)) %>% filter(gene_name %in% gtf$gene_name) %>% # 300 / 500
  # left_join(mouse2human, by = c("x1"="mouse")) %>% drop_na(human) %>% distinct(human, .keep_all = TRUE) %>% rename(gene_name = human) %>% # 304 / 500
  rename(direction = up_down)
trem2_dam_genes.up = trem2_dam_genes %>% filter(direction == 1) %>% pull(gene_name)
trem2_dam_genes.down = trem2_dam_genes %>% filter(direction == -1) %>% pull(gene_name)

# GSEA
untx_vcp_vs_ctrl.geneList <- untx_vcp_vs_ctrl$res %>% drop_na(stat) %>% pull(stat)
names(untx_vcp_vs_ctrl.geneList) <- untx_vcp_vs_ctrl$res %>% drop_na(stat) %>% pull(gene_name) %>% as.character()
untx_vcp_vs_ctrl.geneList = sort(untx_vcp_vs_ctrl.geneList, decreasing = TRUE)
trem2_DAM_genes.list <- list("trem2 DAM up" = untx_vcp_vs_ctrl$res$gene_name[untx_vcp_vs_ctrl$res$gene_name %in% trem2_dam_genes.up], "trem2 DAM down" = untx_vcp_vs_ctrl$res$gene_name[untx_vcp_vs_ctrl$res$gene_name %in% trem2_dam_genes.down])
untx_vcp_vs_ctrl.trem2_DAM_fgseaRes <- fgsea(pathways = trem2_DAM_genes.list, stats = untx_vcp_vs_ctrl.geneList)
untx_vcp_vs_ctrl.trem2_DAM_fgseaRes
#            pathway        pval        padj   log2err        ES      NES size
# 1: trem2 DAM down 0.054687500 0.054687500 0.2712886 0.4807985 1.441888   24
# 2:   trem2 DAM up 0.002380393 0.004760786 0.4317077 0.3162215 1.391449  273

# Venn
untx_vcp_vs_ctrl.res.dge.up <- untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
untx_vcp_vs_ctrl.res.dge.down <- untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)

# 4 way overlap
untx_vcp_vs_ctrl_trem2_dam_venn = ggvenn(list(`DAM down` = trem2_dam_genes.down, `DAM up` = trem2_dam_genes.up, `VCP up` = untx_vcp_vs_ctrl.res.dge.up, `VCP down` = untx_vcp_vs_ctrl.res.dge.down),
                                                fill_color = c("dodgerblue3", "firebrick3", "firebrick1", "dodgerblue1"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.7)
untx_vcp_vs_ctrl_trem2_dam_venn

untx_vcp_vs_ctrl.res.dge.up[untx_vcp_vs_ctrl.res.dge.up %in% trem2_dam_genes.up] # CXCR4, GPNMB

# ALS mice DAMs Table S6
als_dam_genes = read_excel(here(proj_path, "expression/keren_shaul.cell.2017.tableS6.dam_als.xlsx") %>% clean_names() %>% 
  mutate(gene_name = toupper(name)) %>% filter(gene_name %in% gtf$gene_name) %>% # 14,854 / 34,016
  # left_join(mouse2human, by = c("name"="mouse")) %>% drop_na(human) %>% distinct(human, .keep_all = TRUE) %>% rename(gene_name = human) %>% # 15,952 / 34,016
  filter(log10_p_value > 1.30103) %>% drop_na(fold_change_dam_to_homeostatic) %>% arrange(log10_p_value) %>% mutate(direction = case_when(fold_change_dam_to_homeostatic > 0 ~ "up", fold_change_dam_to_homeostatic < 0 ~ "down"))
als_dam_genes.up = als_dam_genes %>% filter(direction == "up") %>% pull(gene_name)
als_dam_genes.down = als_dam_genes %>% filter(direction == "down") %>% pull(gene_name)

# GSEA
untx_vcp_vs_ctrl.geneList <- untx_vcp_vs_ctrl$res %>% drop_na(stat) %>% pull(stat)
names(untx_vcp_vs_ctrl.geneList) <- untx_vcp_vs_ctrl$res %>% drop_na(stat) %>% pull(gene_name) %>% as.character()
untx_vcp_vs_ctrl.geneList = sort(untx_vcp_vs_ctrl.geneList, decreasing = TRUE)
ALS_DAM_genes.list <- list("ALS DAM up" = untx_vcp_vs_ctrl$res$gene_name[untx_vcp_vs_ctrl$res$gene_name %in% als_dam_genes.up], "ALS DAM down" = untx_vcp_vs_ctrl$res$gene_name[untx_vcp_vs_ctrl$res$gene_name %in% als_dam_genes.down])
untx_vcp_vs_ctrl.ALS_DAM_fgseaRes <- fgsea(pathways = ALS_DAM_genes.list, stats = untx_vcp_vs_ctrl.geneList)
untx_vcp_vs_ctrl.ALS_DAM_fgseaRes
#         pathway  pval  padj log2err         ES       NES size                            leadingEdge
# 1: ALS DAM down 1e-10 1e-10      NA -0.3703959 -1.954390 2926 NEDD9,TUBB6,CDK6,CCL4,FFAR2,ATP1A1,...
# 2:   ALS DAM up 1e-10 1e-10      NA -0.3254524 -1.628651  929      PNP,TCF4,EGR1,SRA1,PSMC3,PLD4,...

# Venn
untx_vcp_vs_ctrl.res.dge.up <- untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
untx_vcp_vs_ctrl.res.dge.down <- untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)

# 4 way overlap
untx_vcp_vs_ctrl_als_dam_venn = ggvenn(list(`DAM down` = als_dam_genes.down, `DAM up` = als_dam_genes.up, `VCP up` = untx_vcp_vs_ctrl.res.dge.up, `VCP down` = untx_vcp_vs_ctrl.res.dge.down),
                                                fill_color = c("dodgerblue3", "firebrick3", "firebrick1", "dodgerblue1"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.7)
untx_vcp_vs_ctrl_als_dam_venn

untx_vcp_vs_ctrl.res.dge.up[untx_vcp_vs_ctrl.res.dge.up %in% als_dam_genes.up] # "CCR5"   "ASAH1"  "MARCKS" "BNIP3L" "CALM3" 
```

chiu.cellreports.2013.tableS3.sod1_dge

End Stage

```{r}
chiu_sod1_degs = read_excel(here(proj_path, "expression/chiu.cellreports.2013.tableS3.sod1_dge.xlsx", skip = 3, sheet = "End-stage Regulated Genes") %>% clean_names() #%>%

Sys.setenv("http_proxy" = "http://my.proxy.org:9999") # Configure biomaRt. NB this will stop gprofiler2 working
Sys.unsetenv("http_proxy") # unset proxy to fix gprofiler2
httr::set_config(httr::config(ssl_verifypeer = FALSE)) # httr::set_config(httr::config(ssl_cipher_list = "DEFAULT@SECLEVEL=1"))
mouse.mart<- useDataset("mmusculus_gene_ensembl", useMart("ensembl"))
genbank2hgnc = getBM(attributes=c("refseq_mrna","mgi_symbol"), filters="refseq_mrna", values=chiu_sod1_degs$genbank_accession, mart=mouse.mart, useCache = FALSE)

chiu_sod1_degs = chiu_sod1_degs %>% left_join(genbank2hgnc, by = c("genbank_accession"="refseq_mrna"))
chiu_sod1_degs = chiu_sod1_degs %>%  mutate(gene_name = toupper(mgi_symbol)) %>% filter(gene_name %in% gtf$gene_name) %>% # 12,859 / 15,585
  # left_join(mouse2human, by = c("name"="mouse")) %>% drop_na(human) %>% distinct(human, .keep_all = TRUE) %>% rename(gene_name = human) %>% # 15,952 / 34,016
  filter(q_value < 0.05) %>% drop_na(fold_change_mutant_control) %>% arrange(p_value) %>% mutate(direction = case_when(z_score > 0 ~ "up", z_score < 1 ~ "down"))
chiu_sod1_degs.up = chiu_sod1_degs %>% filter(direction == "up") %>% pull(gene_name)
chiu_sod1_degs.down = chiu_sod1_degs %>% filter(direction == "down") %>% pull(gene_name)

# GSEA
untx_vcp_vs_ctrl.geneList <- untx_vcp_vs_ctrl$res %>% drop_na(stat) %>% pull(stat)
names(untx_vcp_vs_ctrl.geneList) <- untx_vcp_vs_ctrl$res %>% drop_na(stat) %>% pull(gene_name) %>% as.character()
untx_vcp_vs_ctrl.geneList = sort(untx_vcp_vs_ctrl.geneList, decreasing = TRUE)
chiu_sod1_degs.list <- list("SOD1 up" = untx_vcp_vs_ctrl$res$gene_name[untx_vcp_vs_ctrl$res$gene_name %in% chiu_sod1_degs.up], "SOD1 down" = untx_vcp_vs_ctrl$res$gene_name[untx_vcp_vs_ctrl$res$gene_name %in% chiu_sod1_degs.down])
untx_vcp_vs_ctrl.chiu_sod1_degs_fgseaRes <- fgsea(pathways = chiu_sod1_degs.list, stats = untx_vcp_vs_ctrl.geneList)
untx_vcp_vs_ctrl.chiu_sod1_degs_fgseaRes
#     pathway         pval         padj   log2err         ES       NES size                           leadingEdge
# 1: SOD1 down 3.273455e-02 3.273455e-02 0.3217759  0.3078741  1.296549  174   ABCC5,BMF,PER3,CCR5,TLR5,MBOAT1,...
# 2:   SOD1 up 1.601258e-09 3.202516e-09 0.7881868 -0.3261426 -1.613094  802 PPM1F,CCL3,IQGAP3,GPR84,ASPH,TCF4,...

# Venn
untx_vcp_vs_ctrl.res.dge.up <- untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
untx_vcp_vs_ctrl.res.dge.down <- untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)

# 4 way overlap
untx_vcp_vs_ctrl_chiu_sod1_degs_venn = ggvenn(list(`SOD1 down` = chiu_sod1_degs.down, `SOD1 up` = chiu_sod1_degs.up, `VCP up` = untx_vcp_vs_ctrl.res.dge.up, `VCP down` = untx_vcp_vs_ctrl.res.dge.down),
                                                fill_color = c("dodgerblue3", "firebrick3", "firebrick1", "dodgerblue1"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.7)
untx_vcp_vs_ctrl_chiu_sod1_degs_venn
```

Day 100

```{r}
chiu_sod1_degs.d100 = read_excel(here(proj_path, "expression/chiu.cellreports.2013.tableS3.sod1_dge.xlsx", skip = 3, sheet = "D100 Regulated Genes") %>% clean_names() #%>%
genbank2hgnc.d100 = getBM(attributes=c("refseq_mrna","mgi_symbol"), filters="refseq_mrna", values=chiu_sod1_degs.d100$genbank_accession, mart=mouse.mart, useCache = FALSE)
chiu_sod1_degs.d100 = chiu_sod1_degs.d100 %>% left_join(genbank2hgnc.d100, by = c("genbank_accession"="refseq_mrna"))
chiu_sod1_degs.d100 = chiu_sod1_degs.d100 %>%  mutate(gene_name = toupper(mgi_symbol)) %>% filter(gene_name %in% gtf$gene_name) %>% # 12,859 / 15,585
  filter(q_value < 0.05) %>% drop_na(fold_change_mutant_control) %>% arrange(p_value) %>% mutate(direction = case_when(z_score > 0 ~ "up", z_score < 1 ~ "down"))
chiu_sod1_degs.d100.up = chiu_sod1_degs.d100 %>% filter(direction == "up") %>% pull(gene_name)
chiu_sod1_degs.d100.down = chiu_sod1_degs.d100 %>% filter(direction == "down") %>% pull(gene_name)

# GSEA
chiu_sod1_degs.d100.list <- list("SOD1 up" = untx_vcp_vs_ctrl$res$gene_name[untx_vcp_vs_ctrl$res$gene_name %in% chiu_sod1_degs.d100.up], "SOD1 down" = untx_vcp_vs_ctrl$res$gene_name[untx_vcp_vs_ctrl$res$gene_name %in% chiu_sod1_degs.d100.down])
untx_vcp_vs_ctrl.chiu_sod1_degs.d100_fgseaRes <- fgsea(pathways = chiu_sod1_degs.d100.list, stats = untx_vcp_vs_ctrl.geneList)
untx_vcp_vs_ctrl.chiu_sod1_degs.d100_fgseaRes
#     pathway         pval         padj   log2err         ES       NES size                                leadingEdge
# 1: SOD1 down 0.0834990060 0.0834990060 0.2192503 -0.4635730 -1.359220   22   DUSP7,CYCS,CRIM1,SH2B3,GSK3B,TMEM86B,...
# 2:   SOD1 up 0.0000000001 0.0000000002        NA -0.4023443 -1.935581  578 TUBB6,PPM1F,CCL3,ATP13A3,ASPH,SLC25A25,...

# Venn
untx_vcp_vs_ctrl.res.dge.up <- untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
untx_vcp_vs_ctrl.res.dge.down <- untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
untx_vcp_vs_ctrl_chiu_sod1_degs.d100_venn = ggvenn(list(`SOD1 down` = chiu_sod1_degs.d100.down, `SOD1 up` = chiu_sod1_degs.d100.up, `VCP up` = untx_vcp_vs_ctrl.res.dge.up, `VCP down` = untx_vcp_vs_ctrl.res.dge.down),
                                                fill_color = c("dodgerblue3", "firebrick3", "firebrick1", "dodgerblue1"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.7)
untx_vcp_vs_ctrl_chiu_sod1_degs.d100_venn
```

Day 65 = 0 DEGs FDR \< 0.05

```{r}
chiu_sod1_degs.d65 = read_excel(here(proj_path, "expression/chiu.cellreports.2013.tableS3.sod1_dge.xlsx", skip = 3, sheet = "D65 Regulated Genes") %>% clean_names() #%>%
genbank2hgnc.d65 = getBM(attributes=c("refseq_mrna","mgi_symbol"), filters="refseq_mrna", values=chiu_sod1_degs.d65$genbank_accession, mart=mouse.mart, useCache = FALSE)
chiu_sod1_degs.d65 = chiu_sod1_degs.d65 %>% left_join(genbank2hgnc.d65, by = c("genbank_accession"="refseq_mrna"))
chiu_sod1_degs.d65 = chiu_sod1_degs.d65 %>%  mutate(gene_name = toupper(mgi_symbol)) %>% filter(gene_name %in% gtf$gene_name) %>% # 12,859 / 15,585
  filter(q_value < 0.05) %>% drop_na(fold_change_mutant_control) %>% arrange(p_value) %>% mutate(direction = case_when(z_score > 0 ~ "up", z_score < 1 ~ "down"))
chiu_sod1_degs.d65.up = chiu_sod1_degs.d65 %>% filter(direction == "up") %>% pull(gene_name)
chiu_sod1_degs.d65.down = chiu_sod1_degs.d65 %>% filter(direction == "down") %>% pull(gene_name)

# GSEA
chiu_sod1_degs.d65.list <- list("SOD1 up" = untx_vcp_vs_ctrl$res$gene_name[untx_vcp_vs_ctrl$res$gene_name %in% chiu_sod1_degs.d65.up], "SOD1 down" = untx_vcp_vs_ctrl$res$gene_name[untx_vcp_vs_ctrl$res$gene_name %in% chiu_sod1_degs.d65.down])
untx_vcp_vs_ctrl.chiu_sod1_degs.d65_fgseaRes <- fgsea(pathways = chiu_sod1_degs.d65.list, stats = untx_vcp_vs_ctrl.geneList)
untx_vcp_vs_ctrl.chiu_sod1_degs.d65_fgseaRes
#       pathway        pval       padj   log2err        ES      NES size                                     leadingEdge
# 1: SOD1 down 0.006347936 0.01269587 0.4070179 0.2961797 1.348697  346 ABCC5,PPP2R5A,ASAH1,SLC29A3,FBXL20,TMEM229B,...
# 2:   SOD1 up 0.280318091 0.28031809 0.1128434 0.3539554 1.119643   33         CERS4,TMEM37,CD63,ARNT2,IGF2R,PTPRD,...
# Venn
untx_vcp_vs_ctrl.res.dge.up <- untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
untx_vcp_vs_ctrl.res.dge.down <- untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
untx_vcp_vs_ctrl_chiu_sod1_degs.d65_venn = ggvenn(list(`SOD1 down` = chiu_sod1_degs.d65.down, `SOD1 up` = chiu_sod1_degs.d65.up, `VCP up` = untx_vcp_vs_ctrl.res.dge.up, `VCP down` = untx_vcp_vs_ctrl.res.dge.down),
                                                fill_color = c("dodgerblue3", "firebrick3", "firebrick1", "dodgerblue1"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.7)
untx_vcp_vs_ctrl_chiu_sod1_degs.d65_venn
```





## Scatter postmortem against iPSN

```{r}
load(file = here(nemo_path, "projects/ipsc-mn-als-meta/expression/deseq2/nygc_postmortem_spinal_cord.RData"))

# correlate to meta
postmortem_ipsm.als_vs_ctrl = select(nygc_postmortem_spinal_cord$res, gene_name, gene_id, nygc.stat = stat, nygc.padj = padj) %>% 
  left_join(select(untx_vcp_vs_ctrl$res, gene_name, gene_id, ipsm.stat = stat, ipsm.padj = padj))
postmortem_ipsm.als_vs_ctrl.overlapping = postmortem_ipsm.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsm.padj < 0.05, (nygc.stat > 0 & ipsm.stat > 0) | (nygc.stat < 0 & ipsm.stat < 0)) %>% top_n(n = 20, wt = abs(nygc.stat)) %>% pull(gene_name) # 50
postmortem_ipsn_de_list = list("Postmortem spinal cord: ALS vs CTRL" = nygc_postmortem_spinal_cord$res, "iPSC-microglia: VCP vs CTRL" = untx_vcp_vs_ctrl$res)
postmortem_ipsm.als_vs_ctrl.scatter = plot_de_correlation(model_list1 = postmortem_ipsn_de_list, model_list2 = postmortem_ipsn_de_list, mutation1 = "iPSC-microglia: VCP vs CTRL", mutation2 = "Postmortem spinal cord: ALS vs CTRL", gene_labels = c(postmortem_ipsm.als_vs_ctrl.overlapping, "CFAP410"), col = "log2FoldChange") + ylim(c(-3,4)) + xlim(c(-8,8)) +
    annotate(geom="text", x=4, y=4, label="ALS PM & VCP both Up", color="black", size = 3.2)  + annotate(geom="text", x=-4, y=-3, label="ALS PM & VCP both Down", color="black", size = 3.2)
postmortem_ipsm.als_vs_ctrl.scatter



postmortem_ipsm.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsm.padj < 0.05, (nygc.stat > 0 & ipsm.stat > 0) | (nygc.stat < 0 & ipsm.stat < 0)) %>% top_n(n = 20, wt = abs(nygc.stat))
postmortem_ipsm.als_vs_ctrl.overlapping # 
postmortem_ipsm.als_vs_ctrl.overlapping[postmortem_ipsm.als_vs_ctrl.overlapping %in% ALS_genes] # 0
postmortem_ipsm.als_vs_ctrl.overlapping[postmortem_ipsm.als_vs_ctrl.overlapping %in% RNA_BINDING_PROTEINS] # "RNASEL" 
postmortem_ipsm.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsm.padj < 0.05, (nygc.stat > 0 & ipsm.stat > 0) | (nygc.stat < 0 & ipsm.stat < 0), gene_name %in% RNA_BINDING_PROTEINS) 
postmortem_ipsm.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsm.padj < 0.05, (nygc.stat > 0 & ipsm.stat > 0) | (nygc.stat < 0 & ipsm.stat < 0), gene_name %in% gprofiler_database$`DNA Damage Response`) 
postmortem_ipsm.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsm.padj < 0.05, (nygc.stat > 0 & ipsm.stat > 0) | (nygc.stat < 0 & ipsm.stat < 0), gene_name %in% RNA_BINDING_PROTEINS  ) %>% arrange(-nygc.stat + ipsm.stat)
postmortem_ipsm.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsm.padj < 0.05, (nygc.stat > 0 & ipsm.stat > 0) | (nygc.stat < 0 & ipsm.stat < 0)) %>% group_split(nygc.stat > 0)


nygc_postmortem_spinal_cord.res.dge.up <- nygc_postmortem_spinal_cord$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
nygc_postmortem_spinal_cord.res.dge.down <- nygc_postmortem_spinal_cord$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
untx_vcp_vs_ctrl.res.dge.up <- untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
untx_vcp_vs_ctrl.res.dge.down <- untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
newGeneOverlap(c(untx_vcp_vs_ctrl.res.dge.up,untx_vcp_vs_ctrl.res.dge.down),
                         c(nygc_postmortem_spinal_cord.res.dge.up,nygc_postmortem_spinal_cord.res.dge.down),
                         genome.size=nrow(nygc_postmortem_spinal_cord$res)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=183
# listB size=14058
# Intersection size=111
# Overlapping p-value=2.1e-27
# Jaccard Index=0.0

# Fisher exact overlap test
newGeneOverlap(untx_vcp_vs_ctrl.res.dge.up,
                         nygc_postmortem_spinal_cord.res.dge.up,
                         genome.size=nrow(nygc_postmortem_spinal_cord$res)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=110
# listB size=6574
# Intersection size=52
# Overlapping p-value=7e-22
# Jaccard Index=0.0
newGeneOverlap(untx_vcp_vs_ctrl.res.dge.down,
                         nygc_postmortem_spinal_cord.res.dge.down,
                         genome.size=nrow(nygc_postmortem_spinal_cord$res)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=73
# listB size=7487
# Intersection size=20
# Overlapping p-value=4e-04
# Jaccard Index=0.0

# hypergeometric: .  Sorry to be pedantic, but an overlap of 15 feels small given the prevalence of differential genes post-mortem (14529) against iPSN (37), and the discrepancy between these two vastly different rates might draw attention to this 
phyper(22-1, 14058, nrow(nygc_postmortem_spinal_cord$res)-14058, 43, lower.tail=FALSE)
# 0.000158402
```

## Genetic subgroups correlation heatmap
for rebuttal

```{r}
humphrey_als_spinal_cord_modules = read_excel(here(proj_path, "sample-details/humphrey_natneuro_als_spinal_cord_modules.xlsx"), sheet = "ST5", skip = 1)
humphrey_als_spinal_cord_modules
humphrey_als_spinal_cord_M17 = humphrey_als_spinal_cord_modules %>% filter(module_name == "M17")

# correlate M17 genes
nygc_postmortem.M17_spinal_cord.als_vs_ctrl = nygc_postmortem_spinal_cord$res %>% filter(gene_name %in% humphrey_als_spinal_cord_M17$genename)
postmortem.M17_ipsm.als_vs_ctrl = select(nygc_postmortem.M17_spinal_cord.als_vs_ctrl, gene_name, gene_id, nygc.stat = stat, nygc.padj = padj) %>% 
  left_join(select(untx_vcp_vs_ctrl$res, gene_name, gene_id, ipsm.stat = stat, ipsm.padj = padj))
postmortem.M17_ipsm.als_vs_ctrl.overlapping = postmortem.M17_ipsm.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsm.padj < 0.05, (nygc.stat > 0 & ipsm.stat > 0) | (nygc.stat < 0 & ipsm.stat < 0)) %>% 
  top_n(n = 20, wt = abs(nygc.stat)) %>% pull(gene_name) # 14

postmortem.M17_ipsn_de_list = list("postmortem spinal cord M17: ALS vs CTRL" = nygc_postmortem.M17_spinal_cord.als_vs_ctrl, "iPSC-microglia: VCP vs CTRL" = untx_vcp_vs_ctrl$res)
postmortem.M17_ipsm.als_vs_ctrl.scatter = plot_de_correlation(model_list1 = postmortem.M17_ipsn_de_list, model_list2 = postmortem.M17_ipsn_de_list, mutation1 = "iPSC-microglia: VCP vs CTRL", mutation2 = "postmortem spinal cord M17: ALS vs CTRL", gene_labels = c(postmortem.M17_ipsm.als_vs_ctrl.overlapping), col = "stat", plot_p_value = TRUE) + ggtitle("Microglia module M17") #+ ylim(c(-1,2.5)) + xlim(c(-2,3))
postmortem.M17_ipsm.als_vs_ctrl.scatter
ggsave(postmortem.M17_ipsm.als_vs_ctrl.scatter, filename = here(proj_path, "expression/deseq2/postmortem.M17_ipsm.als_vs_ctrl.scatter.pdf"))


# heatmap of correlation coefficients
postmortem_mutations_stat = select(nygc_postmortem_spinal_cord.sporadic$res, gene_id, PM_Sporadic = stat) %>% 
  left_join(select(nygc_postmortem_spinal_cord.c9orf72$res, gene_id, PM_C9orf72 = stat)) %>%
  left_join(select(nygc_postmortem_spinal_cord.sod1$res, gene_id, PM_SOD1 = stat)) %>%
  left_join(select(nygc_postmortem_spinal_cord.fus$res, gene_id, PM_FUS = stat)) %>%
  left_join(select(untx_vcp_vs_ctrl$res, gene_id, MG_VCP = stat)) %>%
  drop_na() %>%
  filter(gene_id %in% humphrey_als_spinal_cord_M17$ensemblid) %>% # filter to only M17 module
  column_to_rownames("gene_id")
cormat <- round(cor(postmortem_mutations_stat),2)
head(cormat)
melted_cormat <- melt(cormat)
head(melted_cormat)
get_lower_tri<-function(cormat){ # Get lower triangle of the correlation matrix
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
get_upper_tri <- function(cormat){ # Get upper triangle of the correlation matrix
  cormat[lower.tri(cormat)]<- NA
  return(cormat)
  }
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE)
reorder_cormat <- function(cormat){ # Use correlation between variables as distance
  dd <- as.dist((1-cormat)/2)
  hc <- hclust(dd)
  cormat <-cormat[hc$order, hc$order]
}
cormat <- reorder_cormat(cormat)
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE) # Melt the correlation matrix
postmortem.dge_correlation_ggheatmap <- ggplot(melted_cormat, aes(Var2, Var1, fill = value))+
  geom_tile(color = "white") +  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, limit = c(-1,1), space = "Lab", name="Pearson\nCorrelation") + 
  geom_text(aes(Var2, Var1, label = value), color = "black", size = 3)  +
  theme_minimal() + theme(axis.title.x = element_blank(), axis.title.y = element_blank(), panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
                          legend.position = "top", legend.direction = "horizontal", legend.title = element_text(size = 8), axis.text=element_text(size=7.5)) + # coord_fixed()+ 
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1, title.position = "top", title.hjust = 0.5))
postmortem.dge_correlation_ggheatmap

```



## Venn

```{r}
nygc_postmortem_spinal_cord.res.dge.up <- nygc_postmortem_spinal_cord$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_id)
nygc_postmortem_spinal_cord.res.dge.down <- nygc_postmortem_spinal_cord$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_id)
untx_vcp_vs_ctrl.res.dge.up <- untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_id)
untx_vcp_vs_ctrl.res.dge.down <- untx_vcp_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_id)

postmortem_ipsc_microglia_genes_up <- list("Postmortem ALS" = nygc_postmortem_spinal_cord.res.dge.up, "Microglia VCP" = untx_vcp_vs_ctrl.res.dge.up)
postmortem_ipsc_microglia_genes_up_venn <- ggvenn(postmortem_ipsc_microglia_genes_up, fill_color = c("firebrick2", "dodgerblue2"), set_name_color = c("firebrick2", "dodgerblue2"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.5) + labs(subtitle = "Upregulated")  + 
  theme(plot.subtitle = element_text(hjust = 0.5, size = 9), plot.margin = unit(c(0, -1, 0, -1), "lines")) #+ theme_oz()
postmortem_ipsc_microglia_genes_up_venn

postmortem_ipsc_microglia_genes_down <- list("Postmortem ALS" = nygc_postmortem_spinal_cord.res.dge.down, "Microglia VCP" = untx_vcp_vs_ctrl.res.dge.down)
postmortem_ipsc_microglia_genes_down_venn <- ggvenn(postmortem_ipsc_microglia_genes_down, fill_color = c("firebrick2", "dodgerblue2"), set_name_color = c("firebrick2", "dodgerblue2"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.5) + labs(subtitle = "Downregulated")  + 
  theme(plot.subtitle = element_text(hjust = 0.5, size = 9), plot.margin = unit(c(0, -1, 0, -1), "lines")) #+ theme_oz()
postmortem_ipsc_microglia_genes_down_venn

plot_grid(postmortem_ipsc_microglia_genes_up_venn, postmortem_ipsc_microglia_genes_down_venn, ncol = 2)

# 4 way overlap
untx_vcp_vs_ctrl_postmortem_venn = ggvenn(list(`Post\nmortem\ndown` = nygc_postmortem_spinal_cord.res.dge.down, `ALS\nPostmortem up` = nygc_postmortem_spinal_cord.res.dge.up, 
                                               `iPS-MG\nVCP up` = untx_vcp_vs_ctrl.res.dge.up, `iPS-MG\nVCP\ndown` = untx_vcp_vs_ctrl.res.dge.down),
                                                fill_color = c("dodgerblue3", "firebrick3", "firebrick1", "dodgerblue1"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.7)
untx_vcp_vs_ctrl_postmortem_venn

nygc_postmortem_spinal_cord.res.dge.down[nygc_postmortem_spinal_cord.res.dge.down %in% nygc_postmortem_spinal_cord.res.dge.up]

nygc_postmortem_spinal_cord$res %>% filter(padj < 0.05, gene_name %in% c("ABCF2", "SOD2", "MATR3"))

# fisher overlap
# Overlap tested using Fisher's exact test (alternative=greater)
newGeneOverlap(nygc_postmortem_spinal_cord.res.dge.up, untx_vcp_vs_ctrl.res.dge.up,genome.size=nrow(untx_vcp_vs_ctrl$res)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=6574
# listB size=110
# Intersection size=52
# Overlapping p-value=7e-22
# Jaccard Index=0.0

newGeneOverlap(nygc_postmortem_spinal_cord.res.dge.down, untx_vcp_vs_ctrl.res.dge.down,genome.size=nrow(untx_vcp_vs_ctrl$res)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=7487
# listB size=73
# Intersection size=20
# Overlapping p-value=4e-04
# Jaccard Index=0.0

```

## GO

```{r}
postmortem_ipsc_overlap.gost = mutate(select(nygc_postmortem_spinal_cord$res, gene_name, gene_id, padj.postmortem = padj, stat.postmortem = stat, lfc.postmortem = log2FoldChange), direction.postmortem = case_when(padj.postmortem < 0.05 & lfc.postmortem > 0 ~ "up", padj.postmortem < 0.05 & lfc.postmortem < 0 ~ "down", TRUE ~ "non-significant")) %>%
  full_join(mutate(select(untx_vcp_vs_ctrl$res, gene_name, gene_id, padj.ipsm = padj, stat.ipsm = stat, lfc.ipsm = log2FoldChange), direction.ipsm = case_when(padj.ipsm < 0.05 & lfc.ipsm > 0 ~ "up", padj.ipsm < 0.05 & lfc.ipsm < 0 ~ "down", TRUE ~ "non-significant"))) %>%
  filter(padj.ipsm < 0.05, padj.postmortem < 0.05, ((stat.ipsm > 0 & stat.postmortem > 0) | (stat.ipsm < 0 & stat.postmortem < 0))) %>% group_split(stat.ipsm > 0) %>%  map("gene_id") %>% gost()

postmortem_ipsc_overlap.gprofiles <- postmortem_ipsc_overlap.gost$result %>% filter(!str_detect(source, "TF|CORUM|HPA|MIRNA")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ " PM & IPSC-MG Down ", query == "query_2" ~ " PM & IPSC-MG Up "), term_name = as.factor(term_name))
postmortem_ipsc_overlap.gprofiles_down <- postmortem_ipsc_overlap.gprofiles %>% filter(query == " PM & IPSC-MG Down ")
postmortem_ipsc_overlap.gprofiles_up <- postmortem_ipsc_overlap.gprofiles %>% filter(query == " PM & IPSC-MG Up ")
paste('"',unique(postmortem_ipsc_overlap.gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
)
postmortem_ipsc_overlap.gprofiles_filt_down <- postmortem_ipsc_overlap.gprofiles %>% filter(query == " PM & IPSC-MG Down ")  %>% filter(term_name %in% down_list)
paste('"',unique(postmortem_ipsc_overlap.gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
# "positive regulation of programmed cell death",
"FOXO-mediated transcription",
# "positive regulation of apoptotic process",
# "programmed cell death",
"regulation of cellular response to stress",
# "Binding and entry of HIV virion",
"regulation of apoptotic process",
"regulation of autophagy",
"cell death",
# "FOXO-mediated transcription of cell cycle genes",
"regulation of programmed cell death",
"regulation of cell death")
postmortem_ipsc_overlap.gprofiles_filt_up <- postmortem_ipsc_overlap.gprofiles %>% filter(query == " PM & IPSC-MG Up ")  %>% filter(term_name %in% up_list)
postmortem_ipsc_overlap.gprofiles_filt_combined <- bind_rows(postmortem_ipsc_overlap.gprofiles_filt_down, postmortem_ipsc_overlap.gprofiles_filt_up) %>% mutate(query = factor(query, levels = c(" PM & IPSC-MG Up ", " PM & IPSC-MG Down ")))
postmortem_ipsc_overlap.gprofiles_filt_combined = postmortem_ipsc_overlap.gprofiles_filt_combined %>% mutate(term_name = gsub("regulation of ", "", term_name), term_name = gsub("and","&",term_name), term_name = gsub("cellular ","",term_name))
postmortem_ipsc_overlap.go_curated <- curated_profiles(gprofiles = postmortem_ipsc_overlap.gprofiles_filt_combined)# + theme_oz() #theme(legend.position = "none", axis.text = element_text(size=10))
postmortem_ipsc_overlap.go_curated

```

## Humphrey module correlation heatmap

M17 module from https://www.nature.com/articles/s41593-022-01205-3 

```{r}
humphrey_als_spinal_cord_modules = read_excel(here(proj_path, "sample-details/humphrey_natneuro_als_spinal_cord_modules.xlsx"), sheet = "ST5", skip = 1)
humphrey_als_spinal_cord_modules
nygc_postmortem_spinal_cord.modules = select(nygc_postmortem_spinal_cord$res, gene_id, gene_name, pm.stat = stat) %>% 
  left_join(select(untx_vcp_vs_ctrl$res, gene_id, gene_name, vcp.stat = stat)) %>%
  left_join(select(humphrey_als_spinal_cord_modules, module=module_name, gene_name = genename, gene_id = ensemblid)) %>% mutate(module = as.numeric(gsub("^M","",module)))

nygc_postmortem.M17_spinal_cord.als_vs_ctrl = nygc_postmortem_spinal_cord$res %>% filter(gene_name %in% humphrey_als_spinal_cord_M17$genename)

nygc_postmortem_spinal_cord.modules %>% drop_na() %>%
  group_by(module) %>% 
  summarize(cor=cor(vcp.stat, pm.stat)) %>% arrange(-cor) %>% print(n=Inf)


# heatmap of correlation coefficients
untreated.als_vs_ctrl.nuc_vs_cyt.stat = select(untx_vcp_vs_ctrl$res, gene_id, `VCP Microglia` = stat) %>% 
  left_join(select(nygc_postmortem_spinal_cord.modules, -gene_name, -baseMean, -log2FoldChange, -lfcSE, -pvalue, -padj)) %>%
  drop_na(`VCP Microglia`) %>% distinct(gene_id, .keep_all = TRUE) %>% column_to_rownames("gene_id")
cormat <- round(cor(untreated.als_vs_ctrl.nuc_vs_cyt.stat),2)
head(cormat)
melted_cormat <- melt(untreated.als_vs_ctrl.nuc_vs_cyt.stat)
head(melted_cormat)
get_lower_tri<-function(cormat){ # Get lower triangle of the correlation matrix
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
get_upper_tri <- function(cormat){ # Get upper triangle of the correlation matrix
  cormat[lower.tri(cormat)]<- NA
  return(cormat)
  }
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE)
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE) #%>% # Melt the correlation matrix
humphrey_modules.untx_vcp_vs_ctrl.dep_correlation_ggheatmap <- ggplot(melted_cormat, aes(Var2, Var1, fill = value))+
  geom_tile(color = "white") +  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, limit = c(-1,1), space = "Lab", name="Pearson\nCorrelation") + 
  geom_text(aes(Var2, Var1, label = value), color = "black", size = 3)  +
  theme_minimal() + theme(axis.title.x = element_blank(), axis.title.y = element_blank(), panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
                          legend.position = "top", legend.direction = "horizontal", legend.title = element_text(size = 8), axis.text=element_text(size=7.5)) + # coord_fixed()+ 
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1, title.position = "top", title.hjust = 0.5)) +
  scale_x_discrete(labels = scales::parse_format())  +  scale_y_discrete(labels = scales::parse_format()) 
humphrey_modules.untx_vcp_vs_ctrl.dep_correlation_ggheatmap


## Humphrey module M17. read in M17 module genes
humphrey_als_spinal_cord_modules = read_excel(here(proj_path, "sample-details/humphrey_natneuro_als_spinal_cord_modules.xlsx"), sheet = "ST5", skip = 1)
humphrey_als_spinal_cord_modules
humphrey_als_spinal_cord_M17 = humphrey_als_spinal_cord_modules %>% filter(module_name == "M17")

# correlate M17 genes
nygc_postmortem.M17_spinal_cord.als_vs_ctrl = nygc_postmortem_spinal_cord$res %>% filter(gene_name %in% humphrey_als_spinal_cord_M17$genename)
postmortem.M17_ipsm.als_vs_ctrl = select(nygc_postmortem.M17_spinal_cord.als_vs_ctrl, gene_name, gene_id, nygc.stat = stat, nygc.padj = padj) %>% 
  left_join(select(untx_vcp_vs_ctrl$res, gene_name, gene_id, ipsm.stat = stat, ipsm.padj = padj))
postmortem.M17_ipsm.als_vs_ctrl.overlapping = postmortem.M17_ipsm.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsm.padj < 0.05, (nygc.stat > 0 & ipsm.stat > 0) | (nygc.stat < 0 & ipsm.stat < 0)) %>% 
  top_n(n = 20, wt = abs(nygc.stat)) %>% pull(gene_name) # 14

postmortem.M17_ipsn_de_list = list("postmortem spinal cord M17: ALS vs CTRL" = nygc_postmortem.M17_spinal_cord.als_vs_ctrl, "iPSC-microglia: VCP vs CTRL" = untx_vcp_vs_ctrl$res)
postmortem.M17_ipsm.als_vs_ctrl.scatter = plot_de_correlation(model_list1 = postmortem.M17_ipsn_de_list, model_list2 = postmortem.M17_ipsn_de_list, mutation1 = "iPSC-microglia: VCP vs CTRL", mutation2 = "postmortem spinal cord M17: ALS vs CTRL", gene_labels = c(postmortem.M17_ipsm.als_vs_ctrl.overlapping), col = "stat", plot_p_value = TRUE) + ggtitle("Microglia module M17") #+ ylim(c(-1,2.5)) + xlim(c(-2,3))
postmortem.M17_ipsm.als_vs_ctrl.scatter
ggsave(postmortem.M17_ipsm.als_vs_ctrl.scatter, filename = here(proj_path, "expression/deseq2/postmortem.M17_ipsm.als_vs_ctrl.scatter.pdf"))


```



## GSEA

```{r}
nygc_postmortem_spinal_cord.geneList <- nygc_postmortem_spinal_cord$res %>% drop_na(stat) %>% pull(stat)
names(nygc_postmortem_spinal_cord.geneList) <- nygc_postmortem_spinal_cord$res %>% drop_na(stat) %>% pull(gene_name) %>% as.character()
nygc_postmortem_spinal_cord.geneList = sort(nygc_postmortem_spinal_cord.geneList, decreasing = TRUE)

# GSEA GO_LYSOSOMAL_LUMEN
GO_LYSOSOMAL_LUMEN_genes.list <- list("GO_LYSOSOMAL_LUMEN" = nygc_postmortem_spinal_cord$res$gene_name[nygc_postmortem_spinal_cord$res$gene_name %in% go.pathways.msigdb$GO_LYSOSOMAL_LUMEN])
nygc_postmortem_spinal_cord.GO_LYSOSOMAL_LUMEN_fgseaRes <- fgsea(pathways = GO_LYSOSOMAL_LUMEN_genes.list, stats = nygc_postmortem_spinal_cord.geneList)
nygc_postmortem_spinal_cord.GO_LYSOSOMAL_LUMEN_fgseaRes
# # pathway         pval         padj   log2err        ES      NES size                       leadingEdge
# 1: GO_LYSOSOMAL_LUMEN 7.365793e-09 7.365793e-09 0.7614608 0.6545977 2.103606   92 CTSS,ASAH1,HEXB,PSAP,CTSL,GNS,...
nygc_postmortem_spinal_cord.GO_LYSOSOMAL_LUMEN.gseaplot = plotEnrichment(GO_LYSOSOMAL_LUMEN_genes.list[["GO_LYSOSOMAL_LUMEN"]], nygc_postmortem_spinal_cord.geneList) +
  theme(axis.text=element_text(size=8), axis.title=element_text(size=10)) + ggtitle("Lysosomal Lumen GSEA") + xlab(expression(VCP~vs~CTRL)) +
  annotate("text", x = 25000, y = 0.45, label = paste0("NES = ", round(nygc_postmortem_spinal_cord.GO_LYSOSOMAL_LUMEN_fgseaRes$NES, digits = 3),"\nP = ", round(nygc_postmortem_spinal_cord.GO_LYSOSOMAL_LUMEN_fgseaRes$pval, digits = 3)), size = 3, hjust = 0) + theme_oz()
nygc_postmortem_spinal_cord.GO_LYSOSOMAL_LUMEN.gseaplot
```


## Merge

```{r}
postmortem_ipsc_microglia_comparison.merged = plot_grid(
  plot_grid(untx_vcp_vs_ctrl_chiu_end_stage_untx_sod1_vs_ctrl_stat.corr, untx_vcp_vs_ctrl_chiu_sod1_degs_venn, ncol = 2, labels = "AUTO"),
  plot_grid(postmortem_ipsm.als_vs_ctrl.scatter, postmortem.M17_ipsm.als_vs_ctrl.scatter, ncol = 2, labels = LETTERS[3:4]),
  plot_grid(postmortem.dge_correlation_ggheatmap, untx_vcp_vs_ctrl_postmortem_venn, ncol = 2, labels = LETTERS[5:6]),
  plot_grid(postmortem_ipsc_overlap.go_curated, nygc_postmortem_spinal_cord.GO_LYSOSOMAL_LUMEN.gseaplot, ncol = 2, labels = LETTERS[7:8]),
  nrow = 4)
# postmortem_ipsc_microglia_comparison.merged
ggsave(postmortem_ipsc_microglia_comparison.merged, filename = here(proj_path, "expression/deseq2/fig4.postmortem_ipsc_microglia_comparison.merged.png"), height = 11.75, width = 8.25)
ggsave(postmortem_ipsc_microglia_comparison.merged, filename = here(proj_path, "expression/deseq2/fig4.postmortem_ipsc_microglia_comparison.merged.pdf"), height = 11.75, width = 8.25)

untx_vcp_vs_ctrl_postmortem_venn

```


# Table S10 postmortem & ipsc microglia overlap DGE

```{r}
postmortem_ipsc_overlap.res = 
  mutate(select(nygc_postmortem_spinal_cord$res, gene_name, gene_id, padj.postmortem = padj, stat.postmortem = stat, lfc.postmortem = log2FoldChange), direction.postmortem = case_when(padj.postmortem < 0.05 & lfc.postmortem > 0 ~ "up", padj.postmortem < 0.05 & lfc.postmortem < 0 ~ "down", TRUE ~ "non-significant")) %>%
  full_join(mutate(select(untx_vcp_vs_ctrl$res, gene_name, gene_id, padj.ipsm = padj, stat.ipsm = stat, lfc.ipsm = log2FoldChange), direction.ipsm = case_when(padj.ipsm < 0.05 & lfc.ipsm > 0 ~ "up", padj.ipsm < 0.05 & lfc.ipsm < 0 ~ "down", TRUE ~ "non-significant"))) %>%
  select(gene_name, gene_id, everything()) %>% 
  filter(padj.ipsm < 0.05, padj.postmortem < 0.05, ((stat.ipsm > 0 & stat.postmortem > 0) | (stat.ipsm < 0 & stat.postmortem < 0)))
postmortem_ipsc_overlap.res
write_csv(postmortem_ipsc_overlap.res, here(proj_path,"expression/deseq2/Table.postmortem_ipsc_overlap.csv"))

library(googlesheets4)
gs4_browse("https://docs.google.com/spreadsheets/d/1CVDodcgGdB2rDqnDtfGWMUMyb-BdNFmfzyAX0F-bQqY/edit#gid=0")
write_sheet(postmortem_ipsc_overlap.res, ss = "https://docs.google.com/spreadsheets/d/1CVDodcgGdB2rDqnDtfGWMUMyb-BdNFmfzyAX0F-bQqY/edit#gid=0", sheet = "S10. postmortem")
sheet_properties("https://docs.google.com/spreadsheets/d/1CVDodcgGdB2rDqnDtfGWMUMyb-BdNFmfzyAX0F-bQqY/edit#gid=0")

```


# Fig 6: GPNMB knockdown


## GPNMB siRNA Western blot

```{r}
# import with magick 
Fig_5A_western_GBPNMB_siRNA.png = magick::image_read(here(proj_path,"ben-figures/Fig_5A_GPNMB_siRNA WB.png"))
Fig_5A_western_GBPNMB_siRNA.gg = ggdraw() + draw_image(Fig_5A_western_GBPNMB_siRNA.png)
Fig_5A_western_GBPNMB_siRNA.gg

# Quantification
Fig_5A_WBquantification.xlsx = read_excel(here(proj_path,"ben-figures/Fig_5A_WBquantification.xlsx")) %>% mutate(`siRNA treatment` = gsub("Scr", "Scr siRNA", `siRNA treatment`), `siRNA treatment` = gsub("GPNMB", "GPNMB siRNA", `siRNA treatment`), `siRNA treatment` = factor(`siRNA treatment`, levels = c("Scr siRNA", "GPNMB siRNA")))
Fig_5A_WBquantification.xlsx %>% glimpse
Fig_5A_WBquantification.xlsx %>% my_summarise(`FC in expression w excl values`, `siRNA treatment`)

Fig_5A_WBquantification.summarise = Fig_5A_WBquantification.xlsx %>% my_summarise(`FC in expression w excl values`, `siRNA treatment`) %>% 
  mutate(p.signif = case_when(`siRNA treatment` == "GPNMB" ~ "*", TRUE ~ "ns"), group1 = `siRNA treatment`, group2 = `siRNA treatment`) # add stat *
Fig_5A_WBquantification.summarise %>% print(n=Inf)

# for stats at each timepoint:
Fig_5A_WBquantification.ctrl.fitg = glm(`FC in expression w excl values` ~ `siRNA treatment` + Line, data= filter(Fig_5A_WBquantification.xlsx, Genotype == "CTRL"), family = gaussian)
summ(Fig_5A_WBquantification.ctrl.fitg, digits = 5)$coeftable 
#                            Est.      S.E.     t val.          p
# (Intercept)           0.1719116 0.2495850  0.6887897 0.54044352
# `siRNA treatment`Scr  0.7690270 0.2232356  3.4449122 0.04108861
Fig_5A_WBquantification.vcp.fitg = glm(`FC in expression w excl values` ~ `siRNA treatment` + Line, data= filter(Fig_5A_WBquantification.xlsx, Genotype == "VCP"), family = gaussian)
summ(Fig_5A_WBquantification.vcp.fitg, digits = 5)$coeftable 
#                            Est.       S.E.    t val.           p
# (Intercept)           1.0029545 0.10383985  9.658666 0.002356183
# `siRNA treatment`Scr  0.6396503 0.09287718  6.887055 0.006271218

Fig_5A_WBquantification.violin = violin_plot(data = Fig_5A_WBquantification.xlsx, color_by = "siRNA treatment", continuous = "FC in expression w excl values", ylims = c(0,1.8), ylabel = "GPNMB / GAPDH Fold change", facet_by = "Genotype", plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "t_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 1.7, tip_length = 0.03) + theme(axis.text.x = element_text(angle = 30, vjust = 0.7, hjust=0.7)) 
Fig_5A_WBquantification.violin
```

## Volcano GPNMB vcp vs ctrl

```{r}
gpnmb_kd_path = "/Volumes/lab-patanir/home/shared/patani-collab/microglia-cxcr4-gpnmb-sirna"
vcp.gpnmb_vs_scramble = readRDS(here(gpnmb_kd_path,"expression/deseq2/rds/vcp.gpnmb_vs_scramble.rds"))
scramble.vcp_vs_ctrl = readRDS(here(gpnmb_kd_path,"expression/deseq2/rds/scramble.vcp_vs_ctrl.rds"))

vcp.gpnmb_vs_scramble$res  %>% filter(padj < 0.05) #15
vcp.gpnmb_vs_scramble$res  %>% filter(padj < 0.05, stat > 0) # 8

vcp.gpnmb_vs_scramble.labels <- vcp.gpnmb_vs_scramble$res  %>% filter(padj < 0.05) %>% pull(gene_name)
vcp.gpnmb_vs_scramble.volcano <- volcano_plot(vcp.gpnmb_vs_scramble$res, gene_labels = vcp.gpnmb_vs_scramble.labels, ymax = 16, xmax = 3, title = "VCP mutant microglia: GPNMB knockdown", numerator = "GPNMB knockdown", denomenator = "Scramble")
vcp.gpnmb_vs_scramble.volcano

```

## PROGENy GPNMB vcp vs ctrl
DoroThEA GPNMB vcp vs ctrl

Pathway RespOnsive GENes (PROGENy) estimates signaling pathway activity

```{r}
# net_progeny <- get_progeny(organism = 'human', top = 100)
vcp.gpnmb_vs_scramble.res.matrix <- vcp.gpnmb_vs_scramble$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
vcp.gpnmb_vs_scramble_progeny.contrast_acts <- run_wmean(mat=vcp.gpnmb_vs_scramble.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "***", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
vcp.gpnmb_vs_scramble_progeny.pathwayActivity <- ggplot(vcp.gpnmb_vs_scramble_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "Signaling Pathways", x = "", y = "VCP microglia: GPNMB knockdown vs Scramble") +
    stat_pvalue_manual(vcp.gpnmb_vs_scramble_progeny.contrast_acts, label = "p.signif", y.position = 2, xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE) 
vcp.gpnmb_vs_scramble_progeny.pathwayActivity

```


```{r}
# net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
net_dorothea %>% distinct(source) # 429
net_dorothea %>% distinct(mor)
vcp.gpnmb_vs_scramble.dorothea <- run_wmean(mat=vcp.gpnmb_vs_scramble.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

vcp.gpnmb_vs_scramble.dorothea.labels.up = vcp.gpnmb_vs_scramble.dorothea %>% top_n(score, n = 5) %>% pull(source)
vcp.gpnmb_vs_scramble.dorothea.labels.down = vcp.gpnmb_vs_scramble.dorothea %>% top_n(-score, n = 5) %>% pull(source)

# Volcano of TFs score vs p_value
vcp.gpnmb_vs_scramble.dorothea.volcano = ggplot(vcp.gpnmb_vs_scramble.dorothea, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "VCP microglia: GPNMB knockdown vs Scramble", title = "Transcription Factors") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  # scale_y_continuous(expand = c(0,0), limits = c(0,ymax)) + scale_x_continuous(limits = c(-xmax,xmax))
  geom_text_repel(fontface = "italic", data = filter(vcp.gpnmb_vs_scramble.dorothea, source %in% c(vcp.gpnmb_vs_scramble.dorothea.labels.up, vcp.gpnmb_vs_scramble.dorothea.labels.down)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.5) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
vcp.gpnmb_vs_scramble.dorothea.volcano

vcp.gpnmb_vs_scramble.dorothea %>% arrange(-score)
vcp.gpnmb_vs_scramble.dorothea %>% arrange(score)

```

Old Correlation GPNMB vcp vs ctrl vs VCP vs CTRL - removed

```{r}
# Scatter
scramble.vcp_vs_ctrl_vcp.gpnmb_vs_scramble.labels = select(vcp.gpnmb_vs_scramble$res, gene_name, vcp.gpnmb_vs_scramble.stat = stat) %>% 
  left_join(select(scramble.vcp_vs_ctrl$res, gene_name, scramble.vcp_vs_ctrl.stat = stat)) %>% filter((scramble.vcp_vs_ctrl.stat > 2.5 & vcp.gpnmb_vs_scramble.stat > 2.5) | (scramble.vcp_vs_ctrl.stat < -2.5 & vcp.gpnmb_vs_scramble.stat < -2.5), gene_name %in% all.reactivity.go)
scramble.vcp_vs_ctrl_vcp.gpnmb_vs_scramble.list = list("Scramble-treated microglia: VCP vs CTRL" = scramble.vcp_vs_ctrl$res, "VCP microglia: GPNMB knockdown vs Scramble" = vcp.gpnmb_vs_scramble$res)
scramble.vcp_vs_ctrl_vcp.gpnmb_vs_scramble_stat.corr = plot_de_correlation(model_list1 = scramble.vcp_vs_ctrl_vcp.gpnmb_vs_scramble.list, model_list2 = scramble.vcp_vs_ctrl_vcp.gpnmb_vs_scramble.list, mutation1 = "Scramble-treated microglia: VCP vs CTRL", mutation2 = "VCP microglia: GPNMB knockdown vs Scramble", gene_labels = pull(scramble.vcp_vs_ctrl_vcp.gpnmb_vs_scramble.labels,gene_name), plot_corr_line = TRUE, join_by = "gene_name", col = "log2FoldChange") + ylim(c(-6,6)) + xlim(c(-5,5)) #+
  # annotate(geom="text", x=3, y=8, label="SOD1 & VCP both Up", color="black", size = 3.2)  + annotate(geom="text", x=-3, y=-8, label="SOD1 & VCP both Down", color="black", size = 3.2)
scramble.vcp_vs_ctrl_vcp.gpnmb_vs_scramble_stat.corr


# Venn
scramble.vcp_vs_ctrl.res.dge.up <- scramble.vcp_vs_ctrl$res %>% filter(padj < 0.05, stat > 0) %>% pull(gene_name)
scramble.vcp_vs_ctrl.res.dge.down <- scramble.vcp_vs_ctrl$res %>% filter(padj < 0.05, stat < 0) %>% pull(gene_name)
vcp.gpnmb_vs_scramble.res.dge.up <- vcp.gpnmb_vs_scramble$res %>% filter(padj < 0.05, stat > 0) %>% pull(gene_name)
vcp.gpnmb_vs_scramble.res.dge.down <- vcp.gpnmb_vs_scramble$res %>% filter(padj < 0.05, stat < 0) %>% pull(gene_name)

# 4 way overlap
scramble.vcp_vs_ctrl_gpnmb_kd_degs_venn = ggvenn(list(`GPNMB kd\ndown` = vcp.gpnmb_vs_scramble.res.dge.down, `GPNMB kd\n up` = vcp.gpnmb_vs_scramble.res.dge.up, `Untreated\nVCP up` = scramble.vcp_vs_ctrl.res.dge.up, `Untreated\nVCP\ndown` = scramble.vcp_vs_ctrl.res.dge.down),
                                                fill_color = c("dodgerblue3", "firebrick3", "firebrick1", "dodgerblue1"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.5)
scramble.vcp_vs_ctrl_gpnmb_kd_degs_venn

```


## CXCL10 secretion

```{r}
# Quantification
read_excel(here(proj_path,"ben-figures/Fig_5F_CXCL10_data.xlsx")) %>% count(Treatment)
Fig_5G_CXCL10quantification.xlsx = read_excel(here(proj_path,"ben-figures/Fig_5F_CXCL10_data.xlsx")) %>% mutate(Treatment = factor(Treatment, levels = c("Scrambled siRNA", "GPNMB siRNA")), Genotype = case_when(grepl("^C[0-9]", Line) ~ "CTRL", TRUE ~ "VCP"))
Fig_5G_CXCL10quantification.xlsx %>% glimpse
Fig_5G_CXCL10quantification.xlsx %>% print(n=Inf)
Fig_5G_CXCL10quantification.xlsx %>% my_summarise(`CXCL10 FC normalised`, Treatment)

### facet by Genotype
# for stats at each genotype:
Fig_5G_CXCL10quantification.ctrl.fitg = glm(`CXCL10 FC normalised` ~ Treatment + Experiment, data= filter(Fig_5G_CXCL10quantification.xlsx, Genotype == "CTRL"), family = gaussian)
summ(Fig_5G_CXCL10quantification.ctrl.fitg, digits = 5)$coeftable 
#                           Est.      S.E.     t val.         p
# (Intercept)           1.6412535 1.1942180  1.3743333 0.1871867
# TreatmentGPNMB siRNA  0.3033271 0.7490733  0.4049365 0.6905741
Fig_5G_CXCL10quantification.vcp.fitg = glm(`CXCL10 FC normalised` ~ Treatment + Experiment, data= filter(Fig_5G_CXCL10quantification.xlsx, Genotype == "VCP"), family = gaussian)
summ(Fig_5G_CXCL10quantification.vcp.fitg, digits = 5)$coeftable 
#                           Est.     S.E.     t val.         p
# (Intercept)           4.380010 5.975156  0.7330370 0.4765469
# TreatmentGPNMB siRNA -2.829758 3.603154 -0.7853558 0.4463228

Fig_5G_CXCL10quantification.violin = violin_plot(data = Fig_5G_CXCL10quantification.xlsx, color_by = "Treatment", continuous = "CXCL10 FC normalised", ylims = c(0,26), ylabel = "CXCL10 normalised Fold change", facet_by = "Genotype", plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "t_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 1.7, tip_length = 0.03) + theme(axis.text.x = element_text(angle = 30, vjust = 0.7, hjust=0.7)) 
Fig_5G_CXCL10quantification.violin

# ### facet by treatment
# # for stats at each treatment:
# Fig_5G_CXCL10quantification.scr.fitg = glm(`CXCL10 FC normalised` ~ Genotype + Experiment, data= filter(Fig_5G_CXCL10quantification.xlsx, Treatment == "Scrambled siRNA"), family = gaussian)
# summ(Fig_5G_CXCL10quantification.scr.fitg, digits = 5)$coeftable 
# #                 Est.     S.E.     t val.          p
# # (Intercept) -2.234781 4.287566 -0.5212238 0.60982057
# # GenotypeVCP  6.539489 2.767612  2.3628634 0.03206565
# Fig_5G_CXCL10quantification.gpnmb.fitg = glm(`CXCL10 FC normalised` ~ Genotype + Experiment, data= filter(Fig_5G_CXCL10quantification.xlsx, Treatment == "GPNMB siRNA"), family = gaussian)
# summ(Fig_5G_CXCL10quantification.gpnmb.fitg, digits = 5)$coeftable 
# #                 Est.     S.E.     t val.          p
# # (Intercept)  2.360841 2.781109  0.8488848 0.40928888
# # GenotypeVCP  3.712996 1.795198  2.0682932 0.05630856
# 
# Fig_5G_CXCL10quantification.violin = violin_plot(data = Fig_5G_CXCL10quantification.xlsx, color_by = "Genotype", continuous = "CXCL10 FC normalised", ylims = c(0,26), ylabel = "CXCL10 normalised Fold change", facet_by = "Treatment", plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 20, tip_length = 0.03) + theme(axis.text.x = element_text(angle = 30, vjust = 0.7, hjust=0.7)) 
# Fig_5G_CXCL10quantification.violin

```

## IL6 secretion

```{r}
# Quantification
read_excel(here(proj_path,"ben-figures/Fig_5G_il6_data.xlsx")) %>% count(Treatment)
Fig_5H_IL6quantification.xlsx = read_excel(here(proj_path,"ben-figures/Fig_5G_il6_data.xlsx")) %>% mutate(Treatment = factor(Treatment, levels = c("Scrambled siRNA", "GPNMB siRNA")), Genotype = case_when(grepl("^C[0-9]", Line) ~ "CTRL", TRUE ~ "VCP"))
Fig_5H_IL6quantification.xlsx %>% glimpse
Fig_5H_IL6quantification.xlsx %>% print(n=Inf)
Fig_5H_IL6quantification.xlsx %>% my_summarise(`IL6 FC normalised`, Treatment)

Fig_5H_IL6quantification.summarise = Fig_5H_IL6quantification.xlsx %>% my_summarise(`IL6 FC normalised`, Treatment) %>% 
  mutate(p.signif = "ns", group1 = Treatment, group2 = Treatment) # add stat *
Fig_5H_IL6quantification.summarise %>% print(n=Inf)

### facet by Genotype
# for stats at each genotype:
Fig_5H_IL6quantification.ctrl.fitg = glm(`IL6 FC normalised` ~ Treatment + Experiment, data= filter(Fig_5H_IL6quantification.xlsx, Genotype == "CTRL"), family = gaussian)
summ(Fig_5H_IL6quantification.ctrl.fitg, digits = 5)$coeftable 
#                            Est.      S.E.     t val.         p
# (Intercept)           1.9002977 1.3059028  1.4551601 0.1638423
# TreatmentGPNMB siRNA  0.1674604 0.8191276  0.2044376 0.8404391
Fig_5H_IL6quantification.vcp.fitg = glm(`IL6 FC normalised` ~ Treatment + Experiment, data= filter(Fig_5H_IL6quantification.xlsx, Genotype == "VCP"), family = gaussian)
summ(Fig_5H_IL6quantification.vcp.fitg, digits = 5)$coeftable 
#                            Est.      S.E.     t val.         p
# (Intercept)           0.1639055 1.1968619  0.1369461 0.8931716
# TreatmentGPNMB siRNA -0.2309487 0.7217349 -0.3199911 0.7540598

Fig_5H_IL6quantification.violin = violin_plot(data = Fig_5H_IL6quantification.xlsx, color_by = "Treatment", continuous = "IL6 FC normalised", ylims = c(0,7), ylabel = "IL6 normalised Fold change", facet_by = "Genotype", plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "t_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 1.7, tip_length = 0.03) + theme(axis.text.x = element_text(angle = 30, vjust = 0.7, hjust=0.7)) 
Fig_5H_IL6quantification.violin

# ### facet by treatment
# # for stats at each treatment:
# Fig_5H_IL6quantification.scr.fitg = glm(`IL6 FC normalised` ~ Genotype + Experiment, data= filter(Fig_5H_IL6quantification.xlsx, Treatment == "Scrambled siRNA"), family = gaussian)
# summ(Fig_5H_IL6quantification.scr.fitg, digits = 5)$coeftable 
# #                    Est.      S.E.      t val.         p
# # (Intercept) -0.01302586 1.2097741 -0.01076719 0.9915511
# # GenotypeVCP  0.54794457 0.7809058  0.70167819 0.4936239
# Fig_5H_IL6quantification.gpnmb.fitg = glm(`IL6 FC normalised` ~ Genotype + Experiment, data= filter(Fig_5H_IL6quantification.xlsx, Treatment == "GPNMB siRNA"), family = gaussian)
# summ(Fig_5H_IL6quantification.gpnmb.fitg, digits = 5)$coeftable 
# #                   Est.      S.E.     t val.         p
# # (Intercept)  1.9268976 1.2937462  1.4893938 0.1571075
# # GenotypeVCP  0.2761399 0.8351096  0.3306631 0.7454726
# 
# Fig_5H_IL6quantification.violin = violin_plot(data = Fig_5H_IL6quantification.xlsx, color_by = "Genotype", continuous = "IL6 FC normalised", ylims = c(0,7), ylabel = "IL6 normalised Fold change", facet_by = "Treatment", plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 20, tip_length = 0.03) + theme(axis.text.x = element_text(angle = 30, vjust = 0.7, hjust=0.7)) 
# Fig_5H_IL6quantification.violin

```

## Merge

```{r}
# gpnmb_kd.merged = plot_grid(
#   plot_grid(Fig_5A_western_GBPNMB_siRNA.gg, Fig_5A_WBquantification.violin, ncol = 2, labels = "AUTO"),
#   plot_grid(vcp.gpnmb_vs_scramble.volcano, vcp.gpnmb_vs_scramble_progeny.pathwayActivity, ncol = 2, labels = LETTERS[3:4]),
#   plot_grid(vcp.gpnmb_vs_scramble.dorothea.volcano, scramble.vcp_vs_ctrl_vcp.gpnmb_vs_scramble_stat.corr, ncol = 2, labels = LETTERS[5:6]),
#   plot_grid(Fig_5G_CXCL10quantification.violin, Fig_5H_IL6quantification.violin, ncol = 2, labels = LETTERS[7:8]),
#     nrow = 4)
gpnmb_kd.merged = plot_grid(
  plot_grid(Fig_5A_western_GBPNMB_siRNA.gg, Fig_5A_WBquantification.violin, vcp.gpnmb_vs_scramble.volcano, ncol = 3, rel_widths = c(1,1,2), labels = "AUTO"),
  plot_grid(vcp.gpnmb_vs_scramble_progeny.pathwayActivity, vcp.gpnmb_vs_scramble.dorothea.volcano, ncol = 2, labels = LETTERS[4:5]),
  plot_grid(Fig_5G_CXCL10quantification.violin, Fig_5H_IL6quantification.violin, ncol = 2, labels = LETTERS[6:7]),
    nrow = 3)
# gpnmb_kd.merged
ggsave(gpnmb_kd.merged, filename = here(proj_path, "expression/deseq2/fig.5.gpnmb_kd.merged.png"), height = 10, width = 9)
ggsave(gpnmb_kd.merged, filename = here(proj_path, "expression/deseq2/fig.5.gpnmb_kd.merged.pdf"), height = 10, width = 9)

```


# Fig S5: GPNMB knockdown 

## qPCR GPNMB knockdown

```{r}
# import with magick 
# Fig_S4A_GPNMBsiRNAqPCR.png = magick::image_read(here(proj_path,"ben-figures/Fig_S4A_GPNMBsiRNAqPCR.png"))
# Fig_S4A_GPNMBsiRNAqPCR.gg = ggdraw() + draw_image(Fig_S4A_GPNMBsiRNAqPCR.png)
# Fig_S4A_GPNMBsiRNAqPCR.gg

Fig_S4A_GPNMBsiRNAqPCR.xlsx = read_excel(here(proj_path,"ben-figures/FigS4A_forOli.xlsx"))# %>% mutate(condition = paste0(celltype,"\n",mutation), condition = factor(condition, levels = c("Microglia\nCTRL", "Microglia\nVCP", "Macrophage\nCTRL", "Macrophage\nVCP")))
Fig_S4A_GPNMBsiRNAqPCR.xlsx %>% glimpse

Fig_S4A_GPNMBsiRNAqPCR.violin = violin_plot(data = Fig_S4A_GPNMBsiRNAqPCR.xlsx, color_by = "Treatment", continuous = "Fold change", ylims = c(0,3), ylabel = "qPCR: Fold change in \nGPNMB expression", facet_by = "Genotype", plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 2.5, tip_length = 0.03) + theme(axis.text.x = element_text(angle = 30, vjust = 0.7, hjust=0.7))
Fig_S4A_GPNMBsiRNAqPCR.violin

```


## qPCR GPNMB expression

```{r}
# Plot qPCR data by cell type (iPSC, pMG and MG) and genotype (CTRL or VCP) and add to the penultimate comment in the rebuttal document.
gpnmb_qpcr_data = read_excel(here(proj_path, "expression/lysotracker/Rebuttal_2_Data_for_Oli.xlsx"), sheet = "qPCR data") %>% clean_names() %>% mutate(
  celltype = case_when(cell_type=="pMG"~"Macrophage Precursors", cell_type=="MG"~"Microglia",TRUE~cell_type),celltype = factor(celltype, levels = c("iPSC","Macrophage Precursors","Microglia")), condition = paste0(cell_type,"\n",genotype), condition = factor(condition, levels = c("iPSC\nCTRL", "iPSC\nVCP", "pMG\nCTRL", "pMG\nVCP", "MG\nCTRL", "MG\nVCP")))
gpnmb_qpcr_data %>% glimpse

gpnmb_qpcr_data %>% my_summarise(fc_gpnmb, condition)
# condition    mean_fc_gpnmb median_fc_gpnmb sd_fc_gpnmb Q1_fc_gpnmb Q3_fc_gpnmb min_fc_gpnmb max_fc_gpnmb sum_fc_gpnmb n_fc_gpnmb se_fc_gpnmb
#   <fct>                <dbl>           <dbl>       <dbl>       <dbl>       <dbl>        <dbl>        <dbl>        <dbl>      <int>       <dbl>
# 1 "iPSC\nCTRL"     0.0000140       0.0000125  0.00000582   0.0000121   0.0000192   0.00000601    0.0000202    0.0000701          5  0.00000260
# 2 "iPSC\nVCP"      0.0000195       0.0000208  0.0000167    0.0000107   0.0000210   0             0.0000451    0.0000976          5  0.00000747
# 3 "pMG\nCTRL"      0.658           0.608      0.398        0.530       0.832       0.120         1.20         3.29               5  0.178     
# 4 "pMG\nVCP"       0.488           0.110      0.770        0.101       0.497       0.0874        1.64         1.95               4  0.385     
# 5 "MG\nCTRL"       1               0.763      0.865        0.412       1.52        0.133         2.95        11                 11  0.261     
# 6 "MG\nVCP"        1.55            1.35       1.24         0.637       2.14        0.144         3.81        18.7               12  0.359     

# gpnmb_qpcr.violin = violin_plot(data = gpnmb_qpcr_data, color_by = "condition", continuous = "fc_gpnmb", ylims = c(0,4), cols = brewer.pal(6, "Paired"), ylabel = "pQCR intensity", 
#                                 # facet_by = "cell_type", 
#                                 plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif")#, stat_ypos = rep(c(0,3,3.5,3,0,0),3), tip_length = 0.003) + theme(axis.text.x = element_text(angle = 30, vjust = 0.7, hjust=0.7))
# gpnmb_qpcr.violin


gpnmb_qpcr.violin = violin_plot(data = gpnmb_qpcr_data, color_by = "genotype", continuous = "fc_gpnmb", ylims = c(0,4), cols = c("firebrick2", "dodgerblue2"), ylabel = "qPCR: fold change\nin GPNMB intensity", 
                                facet_by = "celltype", plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif")#, stat_ypos = rep(c(0,3,3.5,3,0,0),3), tip_length = 0.003) + theme(axis.text.x = element_text(angle = 30, vjust = 0.7, hjust=0.7))
gpnmb_qpcr.violin

```



## GPNMB IF localisation

```{r}
# import the immuno images with magick 
Fig.6A_rebuttal.png = magick::image_read(here(proj_path,"ben-figures/Fig.6A_rebuttal.png"))
Fig.6A_rebuttal.gg = ggdraw() + draw_image(Fig.6A_rebuttal.png)
Fig.6A_rebuttal.gg

rebuttal_figure_6_data = read_excel(here(proj_path, "expression/lysotracker/Rebuttal_data_for Oli.xlsx"), sheet = "Rebuttal_Figure_6") %>% clean_names() 
rebuttal_figure_6_data %>% glimpse

rebuttal_figure_6_data %>% my_summarise(lysosensor_intensity, experiment) %>% print(n=Inf)

# for stats, adjust for experiment
gpnmb_mean_cytoplasm_intensity.gpnmbkd.fitg = glm(gpnmb_mean_cytoplasm_intensity ~ genotype + experiment, data= rebuttal_figure_6_data, family = gaussian)
summ(gpnmb_mean_cytoplasm_intensity.gpnmbkd.fitg, digits = 6)$coeftable 

gpnmb_mean_cytoplasm_intensity.gpnmbkd.violin = violin_plot(data = rebuttal_figure_6_data, color_by = "genotype", continuous = "gpnmb_mean_cytoplasm_intensity", ylabel = "GPNMB mean cytoplasm\nintensity fold change", ylims = c(0,2), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 2, tip_length = 0.06)
gpnmb_mean_cytoplasm_intensity.gpnmbkd.violin


gpnmb_mean_membrane_intensity.gpnmbkd.fitg = glm(gpnmb_mean_membrane_intensity ~ genotype + experiment, data= rebuttal_figure_6_data, family = gaussian)
summ(gpnmb_mean_membrane_intensity.gpnmbkd.fitg, digits = 6)$coeftable 

gpnmb_mean_membrane_intensity.gpnmbkd.violin = violin_plot(data = rebuttal_figure_6_data, color_by = "genotype", continuous = "gpnmb_mean_membrane_intensity", ylabel = "GPNMB mean membrane\nintensity fold change", ylims = c(0,2), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 2, tip_length = 0.06)
gpnmb_mean_membrane_intensity.gpnmbkd.violin


gpnmb_puncta_area_per_cell.gpnmbkd.fitg = glm(gpnmb_puncta_area_per_cell ~ genotype + experiment, data= rebuttal_figure_6_data, family = gaussian)
summ(gpnmb_puncta_area_per_cell.gpnmbkd.fitg, digits = 6)$coeftable 

gpnmb_puncta_area_per_cell.gpnmbkd.violin = violin_plot(data = rebuttal_figure_6_data, color_by = "genotype", continuous = "gpnmb_puncta_area_per_cell", ylabel = "GPNMB puncta area \nper cell fold change", ylims = c(0,2), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 2, tip_length = 0.06)
gpnmb_puncta_area_per_cell.gpnmbkd.violin


gpnmb_puncta_per_area_cytoplasm.gpnmbkd.fitg = glm(gpnmb_puncta_per_area_cytoplasm ~ genotype + experiment, data= rebuttal_figure_6_data, family = gaussian)
summ(gpnmb_puncta_per_area_cytoplasm.gpnmbkd.fitg, digits = 6)$coeftable 

gpnmb_puncta_per_area_cytoplasm.gpnmbkd.violin = violin_plot(data = rebuttal_figure_6_data, color_by = "genotype", continuous = "gpnmb_puncta_per_area_cytoplasm", ylabel = "GPNMB puncta per area\ncytoplasm fold change", ylims = c(0,2), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 2, tip_length = 0.06)
gpnmb_puncta_per_area_cytoplasm.gpnmbkd.violin


rebuttal_fig6.merged <- plot_grid(
  Fig.6A_rebuttal.gg,
  plot_grid(gpnmb_mean_cytoplasm_intensity.gpnmbkd.violin, gpnmb_mean_membrane_intensity.gpnmbkd.violin, gpnmb_puncta_area_per_cell.gpnmbkd.violin, gpnmb_puncta_per_area_cytoplasm.gpnmbkd.violin, nrow = 2, ncol = 2, labels = LETTERS[2:5]),
  ncol = 2, labels = c("A",""))
rebuttal_fig6.merged
# ggsave(rebuttal_fig6.merged, filename = here(proj_path, "expression/deseq2/rebuttal_fig6.merged.png"), height = 4, width = 8.25)
# ggsave(rebuttal_fig6.merged, filename = here(proj_path, "expression/deseq2/rebuttal_fig6.merged.pdf"), height = 4, width = 8.25)


```

## RNAseq quantification of GPNMB knockdown

```{r}
# Check GPNMB knockdown with GPNMB TPM
gpnmb_sirna_gpnmb.tpm <- read_tsv(here(collab_path,"microglia-cxcr4-gpnmb-sirna/nfcore/star_salmon/salmon.merged.gene_tpm.tsv")) %>% filter(gene_name %in% c("GPNMB")) %>%  
  pivot_longer(!c(gene_name,gene_id), names_to = "sample", values_to = "value") %>%
  filter(grepl("gpnmb|scramble", sample)) %>% mutate(Treatment = case_when(grepl("gpnmb",sample)~"GPNMB siRNA", TRUE ~ "Scramble siRNA"), Genotype = case_when(grepl("ctrl|vcpf10",sample)~"CTRL", TRUE~"VCP"))
gpnmb_sirna_gpnmb.tpm

gpnmb_sirna_gpnmb.tpm %>% my_summarise(value, Treatment)

GPNMB_gene_counts.TPM.violin = violin_plot(data = gpnmb_sirna_gpnmb.tpm, color_by = "Treatment", continuous = "value", ylims = c(0,3600), ylabel = "RNAseq: GPNMB TPM", facet_by = "Genotype", plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 3500, tip_length = 0.03) + theme(axis.text.x = element_text(angle = 30, vjust = 0.7, hjust=0.7))
GPNMB_gene_counts.TPM.violin

```

## Volcano CTRL gpnmb vs scr

```{r}
gpnmb_kd_path = "/Volumes/lab-patanir/home/shared/patani-collab/microglia-cxcr4-gpnmb-sirna"
ctrl.gpnmb_vs_scramble = readRDS(here(gpnmb_kd_path,"expression/deseq2/rds/ctrl.gpnmb_vs_scramble.rds"))

ctrl.gpnmb_vs_scramble$res  %>% filter(padj < 0.05) #19
ctrl.gpnmb_vs_scramble$res  %>% filter(padj < 0.05, stat > 0) # 6

ctrl.gpnmb_vs_scramble.labels <- ctrl.gpnmb_vs_scramble$res  %>% filter(-log10(padj) > 2, gene_name %in% all.reactivity.go ) %>% pull(gene_name)
ctrl.gpnmb_vs_scramble.volcano <- volcano_plot(ctrl.gpnmb_vs_scramble$res, gene_labels = ctrl.gpnmb_vs_scramble.labels, ymax = 7, xmax = 5, title = "CTRL microglia", numerator = "GPNMB knockdown", denomenator = "Scramble")
# ctrl.gpnmb_vs_scramble.volcano

```

## PROGENy CTRL GPNMB vs scr
DoroThEA CTRL GPNMB vs scr

Pathway RespOnsive GENes (PROGENy) estimates signaling pathway activity

```{r}
net_progeny <- get_progeny(organism = 'human', top = 100)
ctrl.gpnmb_vs_scramble.res.matrix <- ctrl.gpnmb_vs_scramble$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
ctrl.gpnmb_vs_scramble_progeny.contrast_acts <- run_wmean(mat=ctrl.gpnmb_vs_scramble.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "***", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
ctrl.gpnmb_vs_scramble_progeny.pathwayActivity <- ggplot(ctrl.gpnmb_vs_scramble_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "Signaling Pathways", x = "", y = "GPNMB knockdown vs scramble") +
    stat_pvalue_manual(ctrl.gpnmb_vs_scramble_progeny.contrast_acts, label = "p.signif", y.position = 2, xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE) 
# ctrl.gpnmb_vs_scramble_progeny.pathwayActivity

```


```{r}
net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
net_dorothea %>% distinct(source) # 429
net_dorothea %>% distinct(mor)
ctrl.gpnmb_vs_scramble.dorothea <- run_wmean(mat=ctrl.gpnmb_vs_scramble.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

ctrl.gpnmb_vs_scramble.dorothea.labels.up = ctrl.gpnmb_vs_scramble.dorothea %>% top_n(score, n = 5) %>% pull(source)
ctrl.gpnmb_vs_scramble.dorothea.labels.down = ctrl.gpnmb_vs_scramble.dorothea %>% top_n(-score, n = 5) %>% pull(source)

# Volcano of TFs score vs p_value
ctrl.gpnmb_vs_scramble.dorothea.volcano = ggplot(ctrl.gpnmb_vs_scramble.dorothea, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "GPNMB knockdown vs scramble", title = "Transcription Factors") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  # scale_y_continuous(expand = c(0,0), limits = c(0,ymax)) + scale_x_continuous(limits = c(-xmax,xmax))
  geom_text_repel(fontface = "italic", data = filter(ctrl.gpnmb_vs_scramble.dorothea, source %in% c(ctrl.gpnmb_vs_scramble.dorothea.labels.up, ctrl.gpnmb_vs_scramble.dorothea.labels.down)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.5) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
ctrl.gpnmb_vs_scramble.dorothea.volcano

ctrl.gpnmb_vs_scramble.dorothea %>% arrange(-score)
ctrl.gpnmb_vs_scramble.dorothea %>% arrange(score)

```

## Phagocyte assay GPNMB vs Scr 

```{r}
# import with magick 
Fig_S4E_immunblot.png = magick::image_read(here(proj_path,"ben-figures/Fig_S4E.png"))
Fig_S4E_immunblot.gg = ggdraw() + draw_image(Fig_S4E_immunblot.png)
Fig_S4E_immunblot.gg

Fig_S4E_GPNMB_siRNA_phago_data.xlsx = read_excel(here(proj_path,"ben-figures/Fig_S4E_GPNMB_siRNA_phago_data.xlsx"))# %>% mutate(condition = paste0(celltype,"\n",mutation), condition = factor(condition, levels = c("Microglia\nCTRL", "Microglia\nVCP", "Macrophage\nCTRL", "Macrophage\nVCP")))
Fig_S4E_GPNMB_siRNA_phago_data.xlsx %>% glimpse
Fig_S4E_GPNMB_siRNA_phago_data.xlsx %>% my_summarise(Intensity, Treatment)

Fig_S4E_GPNMB_siRNA_phago_data.summarise = Fig_S4E_GPNMB_siRNA_phago_data.xlsx %>% my_summarise(Intensity, Treatment, Time) %>% 
  mutate(p.signif = case_when(Time %in% c(5,6) & Treatment == "GPNMB siRNA" ~ "*", TRUE ~ "ns"), group1 = Treatment, group2 = Treatment) # add stat *
Fig_S4E_GPNMB_siRNA_phago_data.summarise %>% print(n=Inf)

# for stats at each timepoint:
Fig_S4E_GPNMB_siRNA_phago_data.fitg = glm(Intensity ~ Treatment + Line + Experiment, data= filter(Fig_S4E_GPNMB_siRNA_phago_data.xlsx, Time == 6), family = gaussian)
summ(Fig_S4E_GPNMB_siRNA_phago_data.fitg, digits = 5)$coeftable 
#                           Est.       S.E.     t val.            p
# (Intercept)         0.49543216 0.14734423  3.3624129 0.0021839239
# TreatmentScrambled  0.28500416 0.10418811  2.7354769 0.0105163977
# TreatmentUntreated  0.24375270 0.10418811  2.3395443 0.0264021839
plot_summs(Fig_S4E_GPNMB_siRNA_phago_data.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction

Fig_S4E_GPNMB_siRNA_phago_data.fitg = glm(Intensity ~ Treatment + Line + Experiment, data= filter(Fig_S4E_GPNMB_siRNA_phago_data.xlsx, Time == 5), family = gaussian)
summ(Fig_S4E_GPNMB_siRNA_phago_data.fitg, digits = 5)$coeftable 
#                           Est.       S.E.     t val.            p
# (Intercept)         0.55489118 0.11742827  4.7253627 5.439544e-05
# TreatmentScrambled  0.24171434 0.08303433  2.9110170 6.857608e-03
# TreatmentUntreated  0.19148064 0.08303433  2.3060419 2.845247e-02

Fig_S4E_GPNMB_siRNA_phago_data.fitg = glm(Intensity ~ Treatment + Line + Experiment, data= filter(Fig_S4E_GPNMB_siRNA_phago_data.xlsx, Time == 4), family = gaussian)
summ(Fig_S4E_GPNMB_siRNA_phago_data.fitg, digits = 5)$coeftable 
#                           Est.       S.E.     t val.            p
# (Intercept)         0.54212145 0.09472079  5.7233629 3.410782e-06
# TreatmentScrambled  0.12605532 0.06697771  1.8820487 6.989986e-02
# TreatmentUntreated  0.08970315 0.06697771  1.3392986 1.908765e-01

Fig_S4E_GPNMB_siRNA_phago_data.fitg = glm(Intensity ~ Treatment + Line + Experiment, data= filter(Fig_S4E_GPNMB_siRNA_phago_data.xlsx, Time == 3), family = gaussian)
summ(Fig_S4E_GPNMB_siRNA_phago_data.fitg, digits = 5)$coeftable 
#                           Est.       S.E.     t val.            p
# (Intercept)         0.51876924 0.07795418  6.6547969 2.688760e-07
# TreatmentScrambled  0.08255977 0.05512193  1.4977663 1.450007e-01
# TreatmentUntreated  0.04480298 0.05512193  0.8127978 4.229568e-01

Fig_S4E_GPNMB_siRNA_phago_data.summarise.lineplot = ggplot(Fig_S4E_GPNMB_siRNA_phago_data.summarise, aes(x = Time, y = mean_Intensity, colour = Treatment)) +
  geom_errorbar(aes(ymin=mean_Intensity-se_Intensity, ymax=mean_Intensity+se_Intensity), width=.1) +
  geom_line() + geom_point() + theme_oz() + theme(legend.position = "top", legend.title=element_blank()) +
  ylab("Phagocytic index") + xlab("Time (hrs)") +  scale_colour_manual(values = c("firebrick2", "dodgerblue2", "forestgreen", "gold2")) + 
  stat_pvalue_manual(Fig_S4E_GPNMB_siRNA_phago_data.summarise, label = "p.signif", y.position = 1.2, xmin = "Time", xmax = NULL, size = 6, hide.ns = TRUE) 
Fig_S4E_GPNMB_siRNA_phago_data.summarise.lineplot

```


## Merge

```{r}
gpnmb_kd_vs_scramble.merged = plot_grid(
  plot_grid(Fig_S4A_GPNMBsiRNAqPCR.violin, GPNMB_gene_counts.TPM.violin, ncol = 2, labels = "AUTO"),
  gpnmb_qpcr.violin,
  plot_grid(gpnmb_mean_cytoplasm_intensity.gpnmbkd.violin, gpnmb_mean_membrane_intensity.gpnmbkd.violin, gpnmb_puncta_area_per_cell.gpnmbkd.violin, gpnmb_puncta_per_area_cytoplasm.gpnmbkd.violin, ncol = 4, labels = LETTERS[4:7]),
  plot_grid(Fig.6A_rebuttal.gg, ctrl.gpnmb_vs_scramble.volcano, ctrl.gpnmb_vs_scramble_progeny.pathwayActivity, ncol = 3, labels = LETTERS[8:10]),
  plot_grid(ctrl.gpnmb_vs_scramble.dorothea.volcano, Fig_S4E_immunblot.gg, Fig_S4E_GPNMB_siRNA_phago_data.summarise.lineplot, ncol = 3, rel_widths = c(1.9,1,1.6), labels = LETTERS[11:13]),
  nrow = 5, rel_heights = c(0.7,0.7,0.7,1,1), labels = c("","C","","",""))
# gpnmb_kd_vs_scramble.merged
ggsave(gpnmb_kd_vs_scramble.merged, filename = here(proj_path, "expression/deseq2/FigS5.gpnmb_kd_vs_scramble.merged.png"), height = 11.75, width = 9)
ggsave(gpnmb_kd_vs_scramble.merged, filename = here(proj_path, "expression/deseq2/FigS5.gpnmb_kd_vs_scramble.merged.pdf"), height = 11.75, width = 9)

```


# Fig S6: GPNMB knockdown lysotracker pepstatin

## GPNMB knocdown on lsyotracker, cathepsin

```{r}
# import the immuno images with magick 
Fig.5A_rebuttal.png = magick::image_read(here(proj_path,"ben-figures/Fig.5A_rebuttal.png"))
Fig.5A_rebuttal.gg = ggdraw() + draw_image(Fig.5A_rebuttal.png)
Fig.5A_rebuttal.gg

Fig.5C_rebuttal.png = magick::image_read(here(proj_path,"ben-figures/Fig.5C_rebuttal.png"))
Fig.5C_rebuttal.gg = ggdraw() + draw_image(Fig.5C_rebuttal.png)
Fig.5C_rebuttal.gg

Fig.5G_rebuttal.png = magick::image_read(here(proj_path,"ben-figures/Fig.5G_rebuttal.png"))
Fig.5G_rebuttal.gg = ggdraw() + draw_image(Fig.5G_rebuttal.png)
Fig.5G_rebuttal.gg

Fig.5I_rebuttal.png = magick::image_read(here(proj_path,"ben-figures/Fig.5I_rebuttal.png"))
Fig.5I_rebuttal.gg = ggdraw() + draw_image(Fig.5I_rebuttal.png)
Fig.5I_rebuttal.gg


rebuttal_figure_5_data = read_excel(here(proj_path, "expression/lysotracker/Rebuttal_data_for Oli.xlsx"), sheet = "Rebuttal_Figure_5") %>% clean_names() %>%
  mutate(si_rna_treatment = gsub("GPNMB", "GPNMB\nsiRNA", si_rna_treatment), si_rna_treatment = gsub("Scr","Scr\nsiRNA", si_rna_treatment), si_rna_treatment = factor(si_rna_treatment, levels = c("Scr\nsiRNA", "GPNMB\nsiRNA")))
rebuttal_figure_5_data %>% glimpse

rebuttal_figure_5_data %>% my_summarise(lysosensor_intensity, experiment) %>% print(n=Inf)

# for stats, adjust for experiment
lysosensor_intensity.gpnmbkd.fitg = glm(lysosensor_intensity ~ si_rna_treatment + experiment, data= filter(rebuttal_figure_5_data, genotype == "CTRL"), family = gaussian)
summ(lysosensor_intensity.gpnmbkd.fitg, digits = 5)$coeftable 

lysosensor_intensity.gpnmbkd.fitg = glm(lysosensor_intensity ~ si_rna_treatment + experiment, data= filter(rebuttal_figure_5_data, genotype == "VCP"), family = gaussian)
summ(lysosensor_intensity.gpnmbkd.fitg, digits = 5)$coeftable 

lysosensor_intensity.gpnmbkd.violin = violin_plot(data = rebuttal_figure_5_data, color_by = "si_rna_treatment", continuous = "lysosensor_intensity", facet_by = "genotype", ylabel = "Lysosensor intensity fold change", ylims = c(0,2), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 2, tip_length = 0.05)
lysosensor_intensity.gpnmbkd.violin


lysotracker_intensity.gpnmbkd.fitg = glm(lysotracker_intensity ~ si_rna_treatment + experiment, data= filter(rebuttal_figure_5_data, genotype == "CTRL"), family = gaussian)
summ(lysotracker_intensity.gpnmbkd.fitg, digits = 5)$coeftable 

lysotracker_intensity.gpnmbkd.fitg = glm(lysotracker_intensity ~ si_rna_treatment + experiment, data= filter(rebuttal_figure_5_data, genotype == "VCP"), family = gaussian)
summ(lysotracker_intensity.gpnmbkd.fitg, digits = 5)$coeftable 

lysotracker_intensity.gpnmbkd.violin = violin_plot(data = rebuttal_figure_5_data, color_by = "si_rna_treatment", continuous = "lysotracker_intensity", facet_by = "genotype", ylabel = "Lysotracker intensity fold change", ylims = c(0,2), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 2, tip_length = 0.05)
lysotracker_intensity.gpnmbkd.violin


lysotracker_spot_area.gpnmbkd.fitg = glm(lysotracker_spot_area ~ si_rna_treatment + experiment, data= filter(rebuttal_figure_5_data, genotype == "CTRL"), family = gaussian)
summ(lysotracker_spot_area.gpnmbkd.fitg, digits = 5)$coeftable 

lysotracker_spot_area.gpnmbkd.fitg = glm(lysotracker_spot_area ~ si_rna_treatment + experiment, data= filter(rebuttal_figure_5_data, genotype == "VCP"), family = gaussian)
summ(lysotracker_spot_area.gpnmbkd.fitg, digits = 5)$coeftable 

lysotracker_spot_area.gpnmbkd.violin = violin_plot(data = rebuttal_figure_5_data, color_by = "si_rna_treatment", continuous = "lysotracker_spot_area", facet_by = "genotype", ylabel = "Lysotracker spot area fold change", ylims = c(0,2), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 2, tip_length = 0.05)
lysotracker_spot_area.gpnmbkd.violin


lysotracker_spot_number_per_area.gpnmbkd.fitg = glm(lysotracker_spot_number_per_area ~ si_rna_treatment + experiment, data= filter(rebuttal_figure_5_data, genotype == "CTRL"), family = gaussian)
summ(lysotracker_spot_number_per_area.gpnmbkd.fitg, digits = 5)$coeftable 

lysotracker_spot_number_per_area.gpnmbkd.fitg = glm(lysotracker_spot_number_per_area ~ si_rna_treatment + experiment, data= filter(rebuttal_figure_5_data, genotype == "VCP"), family = gaussian)
summ(lysotracker_spot_number_per_area.gpnmbkd.fitg, digits = 5)$coeftable 

lysotracker_spot_number_per_area.gpnmbkd.violin = violin_plot(data = rebuttal_figure_5_data, color_by = "si_rna_treatment", continuous = "lysotracker_spot_number_per_area", facet_by = "genotype", ylabel = "Lysotracker spot number / area cytoplasm\nfold change", ylims = c(0.5,1.5), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 2, tip_length = 0.05)
lysotracker_spot_number_per_area.gpnmbkd.violin


pepstatin_a_intensity.gpnmbkd.fitg = glm(pepstatin_a_intensity ~ si_rna_treatment + experiment, data= filter(rebuttal_figure_5_data, genotype == "CTRL"), family = gaussian)
summ(pepstatin_a_intensity.gpnmbkd.fitg, digits = 5)$coeftable 

pepstatin_a_intensity.gpnmbkd.fitg = glm(pepstatin_a_intensity ~ si_rna_treatment + experiment, data= filter(rebuttal_figure_5_data, genotype == "VCP"), family = gaussian)
summ(pepstatin_a_intensity.gpnmbkd.fitg, digits = 5)$coeftable 

pepstatin_a_intensity.gpnmbkd.violin = violin_plot(data = rebuttal_figure_5_data, color_by = "si_rna_treatment", continuous = "pepstatin_a_intensity", facet_by = "genotype", ylabel = "Pepstatin A intensity fold change", ylims = c(0,2), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 2, tip_length = 0.05)
pepstatin_a_intensity.gpnmbkd.violin


magic_red_intensity.gpnmbkd.fitg = glm(magic_red_intensity ~ si_rna_treatment + experiment, data= filter(rebuttal_figure_5_data, genotype == "CTRL"), family = gaussian)
summ(magic_red_intensity.gpnmbkd.fitg, digits = 5)$coeftable 

magic_red_intensity.gpnmbkd.fitg = glm(magic_red_intensity ~ si_rna_treatment + experiment, data= filter(rebuttal_figure_5_data, genotype == "VCP"), family = gaussian)
summ(magic_red_intensity.gpnmbkd.fitg, digits = 5)$coeftable 

magic_red_intensity.gpnmbkd.violin = violin_plot(data = rebuttal_figure_5_data, color_by = "si_rna_treatment", continuous = "magic_red_intensity", facet_by = "genotype", ylabel = "Magic Red intensity fold change", ylims = c(0,2), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 2, tip_length = 0.05)
magic_red_intensity.gpnmbkd.violin


rebuttal_fig5.merged <- plot_grid(
  plot_grid(Fig.5A_rebuttal.gg, lysosensor_intensity.gpnmbkd.violin, Fig.5C_rebuttal.gg, ncol = 3, labels = c("A", "B","C")),
  plot_grid(lysotracker_intensity.gpnmbkd.violin, lysotracker_spot_area.gpnmbkd.violin, lysotracker_spot_number_per_area.gpnmbkd.violin, ncol = 3, labels = c("D","E","F")),
  plot_grid(Fig.5G_rebuttal.gg, pepstatin_a_intensity.gpnmbkd.violin, ncol = 2, labels = c("G","H")),
  plot_grid(Fig.5I_rebuttal.gg, magic_red_intensity.gpnmbkd.violin, ncol = 2, labels = c("I","J")),
  nrow = 4)
# rebuttal_fig5.merged
ggsave(rebuttal_fig5.merged, filename = here(proj_path, "expression/deseq2/figS6_gpnmb_kd_lyso.merged.png"), height = 11, width = 8.25)
ggsave(rebuttal_fig5.merged, filename = here(proj_path, "expression/deseq2/figS6_gpnmb_kd_lyso.merged.pdf"), height = 11, width = 8.25)

```

# Fig 7: MiCM on neurons and astrocytes
MiCM = microglia conditioned media

Volcano VCP MiCM vs CTRL MiCM on neurons and astrocytes
GO VCP MiCM vs CTRL MiCM on neurons and astrocytes
PROGENy VCP MiCM vs CTRL MiCM on neurons and astrocytes
DoroThEA VCP MiCM vs CTRL MiCM on neurons and astrocytes
Volcano LPS VCP MiCM vs LPS CTRL MiCM on neurons and astrocytes
PROGENy LPS VCP MiCM vs LPS CTRL MiCM on neurons and astrocytes
DoroThEA LPS VCP MiCM vs LPS CTRL MiCM on neurons and astrocytes


```{r}
# import with magick 
conditioned_media_schematic.png = magick::image_read(here(proj_path,"ben-figures/Microglia media transfer figure.png"))
conditioned_media_schematic.gg = ggdraw() + draw_image(conditioned_media_schematic.png)
conditioned_media_schematic.gg

# read in neuron data
micm_neuron_path = "/Volumes/lab-patanir/home/shared/patani-collab/motor-neuron-microglia-media"

mn.ctrl_untx_media.vcp_vs_ctrl = readRDS(here(micm_neuron_path,"expression/deseq2/rds/mn.ctrl_untx_media.vcp_vs_ctrl.rds"))
mn.ctrl_lps_media.vcp_vs_ctrl = readRDS(here(micm_neuron_path,"expression/deseq2/rds/mn.ctrl_lps_media.vcp_vs_ctrl.rds")) # Ctrl_lps_ctrl (7) vs Ctrl_lps_vcp (11) 

# # Ctrl_untx_ctrl (5) vs Vcp_untx_ctrl (6) – To see if there are any differences between ctrl and vcp motor neurons in the microglia media
# untx_media.ctrl_mn.vcp_vs_ctrl = readRDS(here(micm_neuron_path,"expression/deseq2/rds/untx_media.ctrl_mn.vcp_vs_ctrl.rds"))
# # Ctrl_untx_vcp (9) vs Vcp_untx_vcp (10) – to see whether VCP microglia conditioned media influences vcp motor neurons.
# untx_media.vcp_mn.vcp_vs_ctrl = readRDS(here(micm_neuron_path,"expression/deseq2/rds/untx_media.vcp_mn.vcp_vs_ctrl.rds"))
# # Ctrl_untx_ctrl (5) vs Ctrl_lps_ctrl (7) – To investigate the impact of LPS treated microglia on control motor neurons.
# mn.ctrl_media.ctrl_lps_vs_untx = readRDS(here(micm_neuron_path,"expression/deseq2/rds/mn.ctrl_media.ctrl_lps_vs_untx.rds"))
# # Ctrl_untx_fresh (1) vs Ctrl_lps_fresh (3) – as a control for the above comparison to see whether LPS itself has any effect on control motor neurons.
# mn.ctrl_media.fresh_lps_vs_untx = readRDS(here(micm_neuron_path,"expression/deseq2/rds/mn.ctrl_media.fresh_lps_vs_untx.rds"))
# # Ctrl_untx_fresh (1) vs Vcp_untx_fresh (2) – to allow us to elucidate whether the changes we saw in (5) Vs (6) are due to the media composition or the media’s exposure to microglia.
# untx_media.fresh_mn.vcp_vs_ctrl = readRDS(here(micm_neuron_path,"expression/deseq2/rds/untx_media.fresh_mn.vcp_vs_ctrl.rds"))

# read in astroctyte data
micm_astrocyte_path = here("/Volumes/lab-patanir/home/shared/patani-collab/astrocyte-microglia-media")

ac.ctrl_untx_media.vcp_vs_ctrl = readRDS(here(micm_astrocyte_path,"expression/deseq2/untx.vcp_vs_ctrl.rds"))
ac.ctrl_lps_media.vcp_vs_ctrl = readRDS(here(micm_astrocyte_path,"expression/deseq2/lps.vcp_vs_ctrl.rds"))
# ac.untx.ctrl_vs_nil = readRDS(here(micm_astrocyte_path,"expression/deseq2/untx.ctrl_vs_nil.rds"))
# ac.micm_nil.lps_vs_untx = readRDS(here(micm_astrocyte_path,"expression/deseq2/micm_nil.lps_vs_untx.rds"))
# ac.micm_ctrl.lps_vs_untx = readRDS(here(micm_astrocyte_path,"expression/deseq2/micm_ctrl.lps_vs_untx.rds"))
# ac.micm_vcp.lps_vs_untx = readRDS(here(micm_astrocyte_path,"expression/deseq2/micm_vcp.lps_vs_untx.rds"))

```

### Volcano

```{r}
mn.ctrl_untx_media.vcp_vs_ctrl.labels <- mn.ctrl_untx_media.vcp_vs_ctrl$res  %>% filter(-log10(pvalue) > 4) %>% pull(gene_name)
mn.ctrl_untx_media.vcp_vs_ctrl.volcano <- volcano_plot(mn.ctrl_untx_media.vcp_vs_ctrl$res, gene_labels = mn.ctrl_untx_media.vcp_vs_ctrl.labels, numerator = "VCP", denomenator = "CTRL", xmax = 3, ymax = 8, title = "Neurons", 
                                                       subtitle = "Microglia conditioned media")
# mn.ctrl_untx_media.vcp_vs_ctrl.volcano

ac.ctrl_untx_media.vcp_vs_ctrl.labels <- ac.ctrl_untx_media.vcp_vs_ctrl$res  %>% filter(-log10(pvalue) > 4) %>% pull(gene_name)
ac.ctrl_untx_media.vcp_vs_ctrl.volcano <- volcano_plot(ac.ctrl_untx_media.vcp_vs_ctrl$res, gene_labels = ac.ctrl_untx_media.vcp_vs_ctrl.labels, numerator = "VCP", denomenator = "CTRL", xmax = 3, ymax = 8, title = "Astrocytes", 
                                                       subtitle = "Microglia conditioned media")
# ac.ctrl_untx_media.vcp_vs_ctrl.volcano

ac_mn.ctrl_untx_media.vcp_vs_ctrl.volcano = plot_grid(mn.ctrl_untx_media.vcp_vs_ctrl.volcano, ac.ctrl_untx_media.vcp_vs_ctrl.volcano, ncol = 2)
ac_mn.ctrl_untx_media.vcp_vs_ctrl.volcano

```

### GO terms

```{r}
mn.ctrl_untx_media.vcp_vs_ctrl.dge_genes <- mn.ctrl_untx_media.vcp_vs_ctrl$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
mn.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_filt <- mn.ctrl_untx_media.vcp_vs_ctrl.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM")) %>% arrange( -log10(p_value) ) %>% 
  mutate(query = case_when(query == "query_1" ~ "VCP\n Down", query == "query_2" ~ "VCP\n Up"), term_name = as.factor(term_name)) 
mn.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_down <- mn.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_filt %>% filter(query == "VCP\n Down")
mn.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_up <- mn.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_filt %>% filter(query == "VCP\n Up")
paste('"',unique(mn.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
"mitochondrial membrane",
"Oxidative phosphorylation",
"NADH dehydrogenase activity",
"mitochondrial inner membrane")
mn.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_filt_down <- mn.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_filt %>% filter(query == "VCP\n Down")  %>% filter(term_name %in% down_list)
paste('"',unique(mn.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
)
mn.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_filt_up <- mn.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_filt %>% filter(query == "VCP\n Up")  %>% filter(term_name %in% up_list)
mn.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_filt_combined <- bind_rows(mn.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_filt_down, mn.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_filt_up) %>% mutate(query = factor(query, levels = c("VCP\n Up", "VCP\n Down")))
# mn.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_filt_combined$term_name <- gsub("signaling pathway", "signaling", mn.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_filt_combined$term_name) %>% gsub("cellular response to ", "response to ", .)
mn.ctrl_untx_media.vcp_vs_ctrl.go_curated <- curated_profiles(gprofiles = mn.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_filt_combined) + theme(strip.text.y.right = element_text(angle = 0))
# mn.ctrl_untx_media.vcp_vs_ctrl.go_curated


# ac.ctrl_untx_media.vcp_vs_ctrl.dge_genes <- ac.ctrl_untx_media.vcp_vs_ctrl$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
# ac.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_filt <- ac.ctrl_untx_media.vcp_vs_ctrl.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM")) %>% arrange( -log10(p_value) ) %>% 
#   mutate(query = case_when(query == "query_1" ~ "VCP\n Down", query == "query_2" ~ "VCP\n Up"), term_name = as.factor(term_name)) 
# ac.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_down <- ac.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_filt %>% filter(query == "VCP\n Down")
# ac.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_up <- ac.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_filt %>% filter(query == "VCP\n Up")
# paste('"',unique(ac.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
# down_list <- c(
# "mitochondrial membrane",
# "Oxidative phosphorylation",
# "NADH dehydrogenase activity",
# "mitochondrial inner membrane")
# ac.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_filt_down <- ac.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_filt %>% filter(query == "VCP\n Down")  %>% filter(term_name %in% down_list)
# paste('"',unique(ac.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
# up_list <- c(
# )
# ac.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_filt_up <- ac.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_filt %>% filter(query == "VCP\n Up")  %>% filter(term_name %in% up_list)
# ac.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_filt_combined <- bind_rows(ac.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_filt_down, ac.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_filt_up) %>% mutate(query = factor(query, levels = c("VCP\n Up", "VCP\n Down")))
# # ac.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_filt_combined$term_name <- gsub("signaling pathway", "signaling", ac.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_filt_combined$term_name) %>% gsub("cellular response to ", "response to ", .)
# ac.ctrl_untx_media.vcp_vs_ctrl.go_curated <- curated_profiles(gprofiles = ac.ctrl_untx_media.vcp_vs_ctrl.dge_gprofiles_filt_combined) + theme(strip.text.y.right = element_text(angle = 0))
# # ac.ctrl_untx_media.vcp_vs_ctrl.go_curated
```

### Progeny & Dorothea

Pathway RespOnsive GENes (PROGENy) estimates signaling pathway activity

```{r}
# net_progeny <- get_progeny(organism = 'human', top = 100)
mn.ctrl_untx_media.vcp_vs_ctrl.res.matrix <- mn.ctrl_untx_media.vcp_vs_ctrl$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
mn.ctrl_untx_media.vcp_vs_ctrl_progeny.contrast_acts <- run_wmean(mat=mn.ctrl_untx_media.vcp_vs_ctrl.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "***", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
mn.ctrl_untx_media.vcp_vs_ctrl_progeny.pathwayActivity <- ggplot(mn.ctrl_untx_media.vcp_vs_ctrl_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "Neurons", x = "", y = "Microglia conditioned media: VCP vs CTRL") +
    stat_pvalue_manual(mn.ctrl_untx_media.vcp_vs_ctrl_progeny.contrast_acts, label = "p.signif", y.position = 4.8, xmin = "source", xmax = NULL, size = 3, hide.ns = TRUE) 
# mn.ctrl_untx_media.vcp_vs_ctrl_progeny.pathwayActivity


ac.ctrl_untx_media.vcp_vs_ctrl.res.matrix <- ac.ctrl_untx_media.vcp_vs_ctrl$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
ac.ctrl_untx_media.vcp_vs_ctrl_progeny.contrast_acts <- run_wmean(mat=ac.ctrl_untx_media.vcp_vs_ctrl.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "***", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
ac.ctrl_untx_media.vcp_vs_ctrl_progeny.pathwayActivity <- ggplot(ac.ctrl_untx_media.vcp_vs_ctrl_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "Astrocytes", x = "", y = "Microglia conditioned media: VCP vs CTRL") +
    stat_pvalue_manual(ac.ctrl_untx_media.vcp_vs_ctrl_progeny.contrast_acts, label = "p.signif", y.position = 8.2, xmin = "source", xmax = NULL, size = 3, hide.ns = TRUE) 
# ac.ctrl_untx_media.vcp_vs_ctrl_progeny.pathwayActivity

ac_mn.ctrl_untx_media.vcp_vs_ctrl_progeny.pathwayActivity = plot_grid(mn.ctrl_untx_media.vcp_vs_ctrl_progeny.pathwayActivity, ac.ctrl_untx_media.vcp_vs_ctrl_progeny.pathwayActivity, ncol = 2)
# ac_mn.ctrl_untx_media.vcp_vs_ctrl_progeny.pathwayActivity
```

https://www.bioconductor.org/packages/release/bioc/vignettes/decoupleR/inst/doc/tf_bk.html
https://github.com/saezlab/transcriptutorial/blob/master/scripts/04_TranscriptionFactor_activity_with_Dorothea.md

```{r}
mn.ctrl_untx_media.vcp_vs_ctrl.dorothea <- run_wmean(mat=mn.ctrl_untx_media.vcp_vs_ctrl.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

mn.ctrl_untx_media.vcp_vs_ctrl.dorothea.labels.up = mn.ctrl_untx_media.vcp_vs_ctrl.dorothea %>% top_n(score, n = 5) %>% pull(source)
mn.ctrl_untx_media.vcp_vs_ctrl.dorothea.labels.down = mn.ctrl_untx_media.vcp_vs_ctrl.dorothea %>% top_n(-score, n = 5) %>% pull(source)

# Volcano of TFs score vs p_value
mn.ctrl_untx_media.vcp_vs_ctrl.dorothea.volcano = ggplot(mn.ctrl_untx_media.vcp_vs_ctrl.dorothea, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Microglia conditioned media: VCP vs CTRL", title = "TFs: Neurons") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  # scale_y_continuous(expand = c(0,0), limits = c(0,ymax)) + scale_x_continuous(limits = c(-xmax,xmax))
  geom_text_repel(fontface = "italic", data = filter(mn.ctrl_untx_media.vcp_vs_ctrl.dorothea, source %in% c(mn.ctrl_untx_media.vcp_vs_ctrl.dorothea.labels.up, mn.ctrl_untx_media.vcp_vs_ctrl.dorothea.labels.down)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.5) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
# mn.ctrl_untx_media.vcp_vs_ctrl.dorothea.volcano


ac.ctrl_untx_media.vcp_vs_ctrl.dorothea <- run_wmean(mat=ac.ctrl_untx_media.vcp_vs_ctrl.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

ac.ctrl_untx_media.vcp_vs_ctrl.dorothea.labels.up = ac.ctrl_untx_media.vcp_vs_ctrl.dorothea %>% top_n(score, n = 5) %>% pull(source)
ac.ctrl_untx_media.vcp_vs_ctrl.dorothea.labels.down = ac.ctrl_untx_media.vcp_vs_ctrl.dorothea %>% top_n(-score, n = 5) %>% pull(source)

# Volcano of TFs score vs p_value
ac.ctrl_untx_media.vcp_vs_ctrl.dorothea.volcano = ggplot(ac.ctrl_untx_media.vcp_vs_ctrl.dorothea, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Microglia conditioned media: VCP vs CTRL", title = "TFs: Astrocytes") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  # scale_y_continuous(expand = c(0,0), limits = c(0,ymax)) + scale_x_continuous(limits = c(-xmax,xmax))
  geom_text_repel(fontface = "italic", data = filter(ac.ctrl_untx_media.vcp_vs_ctrl.dorothea, source %in% c(ac.ctrl_untx_media.vcp_vs_ctrl.dorothea.labels.up, ac.ctrl_untx_media.vcp_vs_ctrl.dorothea.labels.down)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.5) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
ac.ctrl_untx_media.vcp_vs_ctrl.dorothea.volcano

ac_mn.ctrl_untx_media.vcp_vs_ctrl.dorothea.volcano = plot_grid(mn.ctrl_untx_media.vcp_vs_ctrl.dorothea.volcano, ac.ctrl_untx_media.vcp_vs_ctrl.dorothea.volcano, ncol = 2)
# ac_mn.ctrl_untx_media.vcp_vs_ctrl.dorothea.volcano

```

### LPS Volcano

```{r}
mn.ctrl_lps_media.vcp_vs_ctrl.labels <- mn.ctrl_lps_media.vcp_vs_ctrl$res  %>% filter(-log10(pvalue) > 4) %>% pull(gene_name)
mn.ctrl_lps_media.vcp_vs_ctrl.volcano <- volcano_plot(mn.ctrl_lps_media.vcp_vs_ctrl$res, gene_labels = mn.ctrl_lps_media.vcp_vs_ctrl.labels, numerator = "VCP", denomenator = "CTRL", xmax = 3, ymax = 6, title = "Neurons", 
                                                      subtitle = "LPS treated microglia media")
# mn.ctrl_lps_media.vcp_vs_ctrl.volcano

ac.ctrl_lps_media.vcp_vs_ctrl.labels <- ac.ctrl_lps_media.vcp_vs_ctrl$res  %>% filter(-log10(pvalue) > 4) %>% pull(gene_name)
ac.ctrl_lps_media.vcp_vs_ctrl.volcano <- volcano_plot(ac.ctrl_lps_media.vcp_vs_ctrl$res, gene_labels = ac.ctrl_lps_media.vcp_vs_ctrl.labels, numerator = "VCP", denomenator = "CTRL", xmax = 3, ymax = 6, title = "Astrocytes", 
                                                      subtitle = "LPS treated microglia media")
# ac.ctrl_lps_media.vcp_vs_ctrl.volcano

ac_mn.ctrl_lps_media.vcp_vs_ctrl.volcano = plot_grid(mn.ctrl_lps_media.vcp_vs_ctrl.volcano, ac.ctrl_lps_media.vcp_vs_ctrl.volcano, ncol = 2)
# ac_mn.ctrl_lps_media.vcp_vs_ctrl.volcano
```


### LPS Progeny & Dorothea

Pathway RespOnsive GENes (PROGENy) estimates signaling pathway activity

```{r}
# net_progeny <- get_progeny(organism = 'human', top = 100)
mn.ctrl_lps_media.vcp_vs_ctrl.res.matrix <- mn.ctrl_lps_media.vcp_vs_ctrl$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
mn.ctrl_lps_media.vcp_vs_ctrl_progeny.contrast_acts <- run_wmean(mat=mn.ctrl_lps_media.vcp_vs_ctrl.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "***", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
mn.ctrl_lps_media.vcp_vs_ctrl_progeny.pathwayActivity <- ggplot(mn.ctrl_lps_media.vcp_vs_ctrl_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "Neurons", x = "", y = "LPS treated microglia media: VCP vs CTRL") +
    stat_pvalue_manual(mn.ctrl_lps_media.vcp_vs_ctrl_progeny.contrast_acts, label = "p.signif", y.position = 1.5, xmin = "source", xmax = NULL, size = 3, hide.ns = TRUE) 
# mn.ctrl_lps_media.vcp_vs_ctrl_progeny.pathwayActivity


ac.ctrl_lps_media.vcp_vs_ctrl.res.matrix <- ac.ctrl_lps_media.vcp_vs_ctrl$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
ac.ctrl_lps_media.vcp_vs_ctrl_progeny.contrast_acts <- run_wmean(mat=ac.ctrl_lps_media.vcp_vs_ctrl.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "***", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
ac.ctrl_lps_media.vcp_vs_ctrl_progeny.pathwayActivity <- ggplot(ac.ctrl_lps_media.vcp_vs_ctrl_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "Astrocytes", x = "", y = "LPS treated microglia media: VCP vs CTRL") +
    stat_pvalue_manual(ac.ctrl_lps_media.vcp_vs_ctrl_progeny.contrast_acts, label = "p.signif", y.position = 1.5, xmin = "source", xmax = NULL, size = 3, hide.ns = TRUE) 
# ac.ctrl_lps_media.vcp_vs_ctrl_progeny.pathwayActivity

ac_mn.ctrl_lps_media.vcp_vs_ctrl_progeny.pathwayActivity = plot_grid(mn.ctrl_lps_media.vcp_vs_ctrl_progeny.pathwayActivity, ac.ctrl_lps_media.vcp_vs_ctrl_progeny.pathwayActivity, ncol = 2)
# ac_mn.ctrl_lps_media.vcp_vs_ctrl_progeny.pathwayActivity
```

https://www.bioconductor.org/packages/release/bioc/vignettes/decoupleR/inst/doc/tf_bk.html
https://github.com/saezlab/transcriptutorial/blob/master/scripts/04_TranscriptionFactor_activity_with_Dorothea.md

```{r}
mn.ctrl_lps_media.vcp_vs_ctrl.dorothea <- run_wmean(mat=mn.ctrl_lps_media.vcp_vs_ctrl.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

mn.ctrl_lps_media.vcp_vs_ctrl.dorothea.labels.up = mn.ctrl_lps_media.vcp_vs_ctrl.dorothea %>% top_n(score, n = 5) %>% pull(source)
mn.ctrl_lps_media.vcp_vs_ctrl.dorothea.labels.down = mn.ctrl_lps_media.vcp_vs_ctrl.dorothea %>% top_n(-score, n = 5) %>% pull(source)

# Volcano of TFs score vs p_value
mn.ctrl_lps_media.vcp_vs_ctrl.dorothea.volcano = ggplot(mn.ctrl_lps_media.vcp_vs_ctrl.dorothea, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "LPS treated microglia media: VCP vs CTRL", title = "TFs: Neurons") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  # scale_y_continuous(expand = c(0,0), limits = c(0,ymax)) + scale_x_continuous(limits = c(-xmax,xmax))
  geom_text_repel(fontface = "italic", data = filter(mn.ctrl_lps_media.vcp_vs_ctrl.dorothea, source %in% c(mn.ctrl_lps_media.vcp_vs_ctrl.dorothea.labels.up, mn.ctrl_lps_media.vcp_vs_ctrl.dorothea.labels.down)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.5) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
# mn.ctrl_lps_media.vcp_vs_ctrl.dorothea.volcano


ac.ctrl_lps_media.vcp_vs_ctrl.dorothea <- run_wmean(mat=ac.ctrl_lps_media.vcp_vs_ctrl.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

ac.ctrl_lps_media.vcp_vs_ctrl.dorothea.labels.up = ac.ctrl_lps_media.vcp_vs_ctrl.dorothea %>% top_n(score, n = 5) %>% pull(source)
ac.ctrl_lps_media.vcp_vs_ctrl.dorothea.labels.down = ac.ctrl_lps_media.vcp_vs_ctrl.dorothea %>% top_n(-score, n = 5) %>% pull(source)

# Volcano of TFs score vs p_value
ac.ctrl_lps_media.vcp_vs_ctrl.dorothea.volcano = ggplot(ac.ctrl_lps_media.vcp_vs_ctrl.dorothea, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "LPS treated microglia media: VCP vs CTRL", title = "TFs: Astrocytes") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  # scale_y_continuous(expand = c(0,0), limits = c(0,ymax)) + scale_x_continuous(limits = c(-xmax,xmax))
  geom_text_repel(fontface = "italic", data = filter(ac.ctrl_lps_media.vcp_vs_ctrl.dorothea, source %in% c(ac.ctrl_lps_media.vcp_vs_ctrl.dorothea.labels.up, ac.ctrl_lps_media.vcp_vs_ctrl.dorothea.labels.down)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.5) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
# ac.ctrl_lps_media.vcp_vs_ctrl.dorothea.volcano

ac_mn.ctrl_lps_media.vcp_vs_ctrl.dorothea.volcano = plot_grid(mn.ctrl_lps_media.vcp_vs_ctrl.dorothea.volcano, ac.ctrl_lps_media.vcp_vs_ctrl.dorothea.volcano, ncol = 2)
# ac_mn.ctrl_lps_media.vcp_vs_ctrl.dorothea.volcano
```


## Merge

```{r}
micm_ac_mn.merged = plot_grid(conditioned_media_schematic.gg,
                              plot_grid(
  ac_mn.ctrl_untx_media.vcp_vs_ctrl.volcano, 
  ac_mn.ctrl_untx_media.vcp_vs_ctrl_progeny.pathwayActivity,
  ac_mn.ctrl_untx_media.vcp_vs_ctrl.dorothea.volcano, 
  ac_mn.ctrl_lps_media.vcp_vs_ctrl.volcano,
  ac_mn.ctrl_lps_media.vcp_vs_ctrl_progeny.pathwayActivity, 
  ac_mn.ctrl_lps_media.vcp_vs_ctrl.dorothea.volcano,
  nrow = 3, ncol = 2, labels = c("B","E","C","F","D","G"), byrow = FALSE),
  nrow = 2, labels = c("A",""), rel_heights = c(1,3))
# micm_ac_mn.merged
ggsave(micm_ac_mn.merged, filename = here(proj_path, "expression/deseq2/Fig6.micm_ac_mn.merged.png"), height = 12.5, width = 10)
ggsave(micm_ac_mn.merged, filename = here(proj_path, "expression/deseq2/Fig6.micm_ac_mn.merged.pdf"), height = 12.5, width = 10)

```


# Figure S7

Caspase 3 images
Caspase 3 quantification

                                                                                                                              
```{r}
# import with magick 
Fig_S5A_caspase_image.png = magick::image_read(here(proj_path,"ben-figures/FigS5A.png"))
Fig_S5A_caspase_image.gg = ggdraw() + draw_image(Fig_S5A_caspase_image.png)
Fig_S5A_caspase_image.gg

Fig_S5B_caspase_quantification.png = magick::image_read(here(proj_path,"ben-figures/FigS5B.png"))
Fig_S5B_caspase_quantification.gg = ggdraw() + draw_image(Fig_S5B_caspase_quantification.png)
Fig_S5B_caspase_quantification.gg

```

## LPS only PROGENy & Dorothea on neurons and astrocytes

micm_nil.lps_vs_untx

```{r}
micm_neuron_path = "/Volumes/lab-patanir/home/shared/patani-collab/motor-neuron-microglia-media"
mn.micm_nil.lps_vs_untx = readRDS(here(micm_neuron_path,"expression/deseq2/rds/mn.ctrl_media.fresh_lps_vs_untx.rds"))

# read in astroctyte data
micm_astrocyte_path = here("/Volumes/lab-patanir/home/shared/patani-collab/astrocyte-microglia-media")
ac.micm_nil.lps_vs_untx = readRDS(here(micm_astrocyte_path,"expression/deseq2/micm_nil.lps_vs_untx.rds"))


# net_progeny <- get_progeny(organism = 'human', top = 100)
mn.micm_nil.lps_vs_untx.res.matrix <- mn.micm_nil.lps_vs_untx$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
mn.micm_nil.lps_vs_untx_progeny.contrast_acts <- run_wmean(mat=mn.micm_nil.lps_vs_untx.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "***", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
mn.micm_nil.lps_vs_untx_progeny.pathwayActivity <- ggplot(mn.micm_nil.lps_vs_untx_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "LPS-stimulated Neurons", x = "", y = "Neurons:\nLPS stimulated vs no LPS") +
    stat_pvalue_manual(mn.micm_nil.lps_vs_untx_progeny.contrast_acts, label = "p.signif", y.position = 2, xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE) 
mn.micm_nil.lps_vs_untx_progeny.pathwayActivity


ac.micm_nil.lps_vs_untx.res.matrix <- ac.micm_nil.lps_vs_untx$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
ac.micm_nil.lps_vs_untx_progeny.contrast_acts <- run_wmean(mat=ac.micm_nil.lps_vs_untx.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "***", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
ac.micm_nil.lps_vs_untx_progeny.pathwayActivity <- ggplot(ac.micm_nil.lps_vs_untx_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "LPS-stimulated Astrocytes", x = "", y = "Astrocytes:\nLLPS stimulated vs no LPS") +
    stat_pvalue_manual(ac.micm_nil.lps_vs_untx_progeny.contrast_acts, label = "p.signif", y.position = 3, xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE) 
ac.micm_nil.lps_vs_untx_progeny.pathwayActivity

```


```{r}
# net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
mn.micm_nil.lps_vs_untx.dorothea <- run_wmean(mat=mn.micm_nil.lps_vs_untx.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))
mn.micm_nil.lps_vs_untx.dorothea.labels.up = mn.micm_nil.lps_vs_untx.dorothea %>% top_n(score, n = 5) %>% pull(source)
mn.micm_nil.lps_vs_untx.dorothea.labels.down = mn.micm_nil.lps_vs_untx.dorothea %>% top_n(-score, n = 5) %>% pull(source)

mn.micm_nil.lps_vs_untx.dorothea.volcano = ggplot(mn.micm_nil.lps_vs_untx.dorothea, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Neurons: LPS stimulated vs no LPS", title = "LPS-stimulated Neurons") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  geom_text_repel(fontface = "italic", data = filter(mn.micm_nil.lps_vs_untx.dorothea, source %in% c(mn.micm_nil.lps_vs_untx.dorothea.labels.up, mn.micm_nil.lps_vs_untx.dorothea.labels.down)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.5) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
mn.micm_nil.lps_vs_untx.dorothea.volcano


ac.micm_nil.lps_vs_untx.dorothea <- run_wmean(mat=ac.micm_nil.lps_vs_untx.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))
ac.micm_nil.lps_vs_untx.dorothea.labels.up = ac.micm_nil.lps_vs_untx.dorothea %>% top_n(score, n = 5) %>% pull(source)
ac.micm_nil.lps_vs_untx.dorothea.labels.down = ac.micm_nil.lps_vs_untx.dorothea %>% top_n(-score, n = 5) %>% pull(source)

ac.micm_nil.lps_vs_untx.dorothea.volcano = ggplot(ac.micm_nil.lps_vs_untx.dorothea, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Astrocytes: LPS stimulated vs no LPS", title = "LPS-stimulated Astrocytes") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  geom_text_repel(fontface = "italic", data = filter(ac.micm_nil.lps_vs_untx.dorothea, source %in% c(ac.micm_nil.lps_vs_untx.dorothea.labels.up, ac.micm_nil.lps_vs_untx.dorothea.labels.down)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.5) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
ac.micm_nil.lps_vs_untx.dorothea.volcano

```


## Volcano LPS MiCM vs untx MiCM on neurons and astrocytes

```{r}
micm_neuron_path = "/Volumes/lab-patanir/home/shared/patani-collab/motor-neuron-microglia-media"
mn.micm_ctrl.lps_vs_untx = readRDS(here(micm_neuron_path,"expression/deseq2/rds/mn.ctrl_media.ctrl_lps_vs_untx.rds"))

# read in astroctyte data
micm_astrocyte_path = here("/Volumes/lab-patanir/home/shared/patani-collab/astrocyte-microglia-media")
ac.micm_ctrl.lps_vs_untx = readRDS(here(micm_astrocyte_path,"expression/deseq2/micm_ctrl.lps_vs_untx.rds"))


mn.micm_ctrl.lps_vs_untx.labels <- mn.micm_ctrl.lps_vs_untx$res  %>% filter(-log10(pvalue) > 4, !grepl("^AL", gene_name)) %>% pull(gene_name)
mn.micm_ctrl.lps_vs_untx.volcano <- volcano_plot(mn.micm_ctrl.lps_vs_untx$res, gene_labels = mn.micm_ctrl.lps_vs_untx.labels, numerator = "LPS", denomenator = "no LPS", xmax = 3, ymax = 7, title = "LPS-stimulated microglia media on Neurons")
mn.micm_ctrl.lps_vs_untx.volcano

ac.micm_ctrl.lps_vs_untx.labels <- ac.micm_ctrl.lps_vs_untx$res  %>% filter(-log10(pvalue) > 5, !grepl("^AL", gene_name)) %>% pull(gene_name)
ac.micm_ctrl.lps_vs_untx.volcano <- volcano_plot(ac.micm_ctrl.lps_vs_untx$res, gene_labels = ac.micm_ctrl.lps_vs_untx.labels, numerator = "LPS", denomenator = "no LPS", xmax = 4, ymax = 8, title = "LPS-stimulated microglia media on Astrocytes")
ac.micm_ctrl.lps_vs_untx.volcano

# ac_mn.micm_ctrl.lps_vs_untx.volcano = plot_grid(mn.micm_ctrl.lps_vs_untx.volcano, ac.micm_ctrl.lps_vs_untx.volcano, ncol = 2)
# ac_mn.micm_ctrl.lps_vs_untx.volcano

```

## PROGENy & Dorothea LPS MiCM vs untx MiCM on neurons and astrocytes

```{r}
# net_progeny <- get_progeny(organism = 'human', top = 100)
mn.micm_ctrl.lps_vs_untx.res.matrix <- mn.micm_ctrl.lps_vs_untx$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
mn.micm_ctrl.lps_vs_untx_progeny.contrast_acts <- run_wmean(mat=mn.micm_ctrl.lps_vs_untx.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "***", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
mn.micm_ctrl.lps_vs_untx_progeny.pathwayActivity <- ggplot(mn.micm_ctrl.lps_vs_untx_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "LPS-stimulated microglia media\non Neurons", x = "", y = "Microglia media on Neurons:\nLPS stimulated vs no LPS") +
    stat_pvalue_manual(mn.micm_ctrl.lps_vs_untx_progeny.contrast_acts, label = "p.signif", y.position = 20, xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE) 
mn.micm_ctrl.lps_vs_untx_progeny.pathwayActivity


ac.micm_ctrl.lps_vs_untx.res.matrix <- ac.micm_ctrl.lps_vs_untx$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
ac.micm_ctrl.lps_vs_untx_progeny.contrast_acts <- run_wmean(mat=ac.micm_ctrl.lps_vs_untx.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "***", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
ac.micm_ctrl.lps_vs_untx_progeny.pathwayActivity <- ggplot(ac.micm_ctrl.lps_vs_untx_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "LPS-stimulated microglia media\non Astrocytes", x = "", y = "Microglia media on Astrocytes:\nLPS stimulated vs no LPS") +
    stat_pvalue_manual(ac.micm_ctrl.lps_vs_untx_progeny.contrast_acts, label = "p.signif", y.position = 27, xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE) 
ac.micm_ctrl.lps_vs_untx_progeny.pathwayActivity

```


```{r}
# net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
mn.micm_ctrl.lps_vs_untx.dorothea <- run_wmean(mat=mn.micm_ctrl.lps_vs_untx.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))
mn.micm_ctrl.lps_vs_untx.dorothea.labels.up = mn.micm_ctrl.lps_vs_untx.dorothea %>% top_n(score, n = 5) %>% pull(source)
mn.micm_ctrl.lps_vs_untx.dorothea.labels.down = mn.micm_ctrl.lps_vs_untx.dorothea %>% top_n(-score, n = 5) %>% pull(source)

mn.micm_ctrl.lps_vs_untx.dorothea.volcano = ggplot(mn.micm_ctrl.lps_vs_untx.dorothea, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Microglia media on Neurons: LPS stimulated vs no LPS", title = "LPS-stimulated microglia media on Neurons") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  geom_text_repel(fontface = "italic", data = filter(mn.micm_ctrl.lps_vs_untx.dorothea, source %in% c(mn.micm_ctrl.lps_vs_untx.dorothea.labels.up, mn.micm_ctrl.lps_vs_untx.dorothea.labels.down)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.5) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
mn.micm_ctrl.lps_vs_untx.dorothea.volcano


ac.micm_ctrl.lps_vs_untx.dorothea <- run_wmean(mat=ac.micm_ctrl.lps_vs_untx.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))
ac.micm_ctrl.lps_vs_untx.dorothea.labels.up = ac.micm_ctrl.lps_vs_untx.dorothea %>% top_n(score, n = 5) %>% pull(source)
ac.micm_ctrl.lps_vs_untx.dorothea.labels.down = ac.micm_ctrl.lps_vs_untx.dorothea %>% top_n(-score, n = 5) %>% pull(source)

ac.micm_ctrl.lps_vs_untx.dorothea.volcano = ggplot(ac.micm_ctrl.lps_vs_untx.dorothea, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Microglia media on Astrocytes: LPS stimulated vs no LPS", title = "LPS-stimulated microglia media on Astrocytes") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  geom_text_repel(fontface = "italic", data = filter(ac.micm_ctrl.lps_vs_untx.dorothea, source %in% c(ac.micm_ctrl.lps_vs_untx.dorothea.labels.up, ac.micm_ctrl.lps_vs_untx.dorothea.labels.down)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.5) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
ac.micm_ctrl.lps_vs_untx.dorothea.volcano


```


## Merge

```{r}
lps_on_ac_mn_merged = plot_grid(
plot_grid(Fig_S5A_caspase_image.gg, Fig_S5B_caspase_quantification.gg, ncol = 2, rel_widths = c(1,1), labels = "AUTO"),
plot_grid(mn.micm_nil.lps_vs_untx_progeny.pathwayActivity,mn.micm_nil.lps_vs_untx.dorothea.volcano, ac.micm_nil.lps_vs_untx_progeny.pathwayActivity, ac.micm_nil.lps_vs_untx.dorothea.volcano, ncol = 4, rel_widths = c(1,1.4,1,1.4), labels = LETTERS[3:6]),
plot_grid(mn.micm_ctrl.lps_vs_untx.volcano, ac.micm_ctrl.lps_vs_untx.volcano, ncol = 2, labels = LETTERS[7:8]),
plot_grid(mn.micm_ctrl.lps_vs_untx_progeny.pathwayActivity, mn.micm_ctrl.lps_vs_untx.dorothea.volcano, ac.micm_ctrl.lps_vs_untx_progeny.pathwayActivity, ac.micm_ctrl.lps_vs_untx.dorothea.volcano, ncol = 4, rel_widths = c(1,1.4,1,1.4), labels = LETTERS[9:12]),
nrow = 4, rel_heights = c(1,1,0.8,1))
# lps_on_ac_mn_merged
ggsave(lps_on_ac_mn_merged, filename = here(proj_path, "expression/deseq2/lps_on_ac_mn_merged.png"), height = 12, width = 11)
ggsave(lps_on_ac_mn_merged, filename = here(proj_path, "expression/deseq2/lps_on_ac_mn_merged.pdf"), height = 12, width = 11)


```

# Fig 8

## pSTAT2 MiCM on AC

Add pSTAT2 MiCM on ACs images and quantification. I have provided the N/C ratio and the nuclear intensity values - I think it would be preferable to plot the N/C values. 

```{r}
# import with magick 
Fig_7A_pSTAT2images.png = magick::image_read(here(proj_path,"ben-figures/Fig_7A_pSTAT2images.png"))
Fig_7A_pSTAT2images.gg = ggdraw() + draw_image(Fig_7A_pSTAT2images.png)
Fig_7A_pSTAT2images.gg

# Quantification
read_excel(here(proj_path,"ben-figures/Fig_7B_210923_pSTAT2_MiCM_quantification.xlsx"))
read_excel(here(proj_path,"ben-figures/Fig_7B_210923_pSTAT2_MiCM_quantification.xlsx")) %>% glimpse
read_excel(here(proj_path,"ben-figures/Fig_7B_210923_pSTAT2_MiCM_quantification.xlsx")) %>% print(n=Inf)
read_excel(here(proj_path,"ben-figures/Fig_7B_210923_pSTAT2_MiCM_quantification.xlsx")) %>% count(Treatment)
Fig_7B_210923_pSTAT2_MiCM_quantification.xlsx = read_excel(here(proj_path,"ben-figures/Fig_7B_210923_pSTAT2_MiCM_quantification.xlsx")) %>% mutate(Treatment = factor(Treatment, levels = c("Untreated","CTRL MiCM Scr","CTRL MiCM GPNMB","VCP MiCM Scr","VCP MiCM GPNMB")))
Fig_7B_210923_pSTAT2_MiCM_quantification.xlsx %>% glimpse
Fig_7B_210923_pSTAT2_MiCM_quantification.xlsx %>% my_summarise(`Nuclear intensity raw FC from av Un`, Treatment)

Fig_7B_210923_pSTAT2_MiCM_quantification.summarise = Fig_7B_210923_pSTAT2_MiCM_quantification.xlsx %>% my_summarise(`Nuclear intensity raw FC from av Un`, Treatment) %>% 
  mutate(p.signif = case_when(Treatment == "GPNMB" ~ "*", TRUE ~ "ns"), group1 = Treatment, group2 = Treatment) # add stat *
Fig_7B_210923_pSTAT2_MiCM_quantification.summarise %>% print(n=Inf)

# for stats:
# VCP vs CTRL Scramble
Fig_7B_210923_pSTAT2_MiCM_quantification.ctrl.fitg = glm(`Nuclear intensity raw FC from av Un` ~ Treatment + Experiment + Line, data= filter(Fig_7B_210923_pSTAT2_MiCM_quantification.xlsx, Treatment %in% c("CTRL MiCM Scr", "VCP MiCM Scr")), family = gaussian)
summ(Fig_7B_210923_pSTAT2_MiCM_quantification.ctrl.fitg, digits = 5)$coeftable 
#                            Est.      S.E.    t val.           p
# (Intercept)           1.5429322 0.2556111 6.0362484 0.000125886
# TreatmentVCP MiCM Scr 0.2378694 0.3614887 0.6580273 0.525378576

Fig_7B_210923_pSTAT2_MiCM_quantification.ctrl.fitg = glm(`Nuclear intensity raw FC from av Un` ~ Treatment + Experiment + Line, data= filter(Fig_7B_210923_pSTAT2_MiCM_quantification.xlsx, Treatment %in% c("CTRL MiCM Scr", "CTRL MiCM GPNMB")), family = gaussian)
summ(Fig_7B_210923_pSTAT2_MiCM_quantification.ctrl.fitg, digits = 5)$coeftable 
#                                 Est.       S.E.     t val.            p
# (Intercept)               0.32825861 0.16131732  2.0348628 8.132710e-02
# TreatmentCTRL MiCM GPNMB -0.08880056 0.08948275 -0.9923762 3.540708e-01

Fig_7B_210923_pSTAT2_MiCM_quantification.ctrl.fitg = glm(`Nuclear intensity raw FC from av Un` ~ Treatment + Experiment + Line, data= filter(Fig_7B_210923_pSTAT2_MiCM_quantification.xlsx, Treatment %in% c("VCP MiCM Scr", "VCP MiCM GPNMB")), family = gaussian)
summ(Fig_7B_210923_pSTAT2_MiCM_quantification.ctrl.fitg, digits = 5)$coeftable 
#                               Est.      S.E.     t val.          p
# (Intercept)             -0.1533808 0.2815645 -0.5447448 0.60284286
# TreatmentVCP MiCM GPNMB -0.1266750 0.1561839 -0.8110631 0.44402744

Fig_7B_210923_pSTAT2_MiCM_quantification.violin = violin_plot(data = Fig_7B_210923_pSTAT2_MiCM_quantification.xlsx, color_by = "Treatment", continuous = "Nuclear intensity raw FC from av Un", ylims = c(0.5,1.5), ylabel = "pSTAT2 Nuclear intensity\nfold change from untreated", plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = FALSE, stats_test = "t_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 2.5, tip_length = 0.03) + theme(axis.text.x = element_text(angle = 30, vjust = 0.7, hjust=0.7)) 
Fig_7B_210923_pSTAT2_MiCM_quantification.violin

```

## MN survival after MG132 

Add motor neuron survival after MiCM + MG132 images and quantification as new Fig. 7C-D. May be simpler to plot only values that have been treated with MG132. 

```{r}
# import with magick 
Fig_7C_MiCM_neuronal_survival_images.png = magick::image_read(here(proj_path,"ben-figures/Fig_7C_MiCM_neuronal_survival_images.png"))
Fig_7C_MiCM_neuronal_survival_images.gg = ggdraw() + draw_image(Fig_7C_MiCM_neuronal_survival_images.png)
Fig_7C_MiCM_neuronal_survival_images.gg

# Quantification
read_excel(here(proj_path,"ben-figures/Fig_7D_Mg132MN_data.xlsx"))
read_excel(here(proj_path,"ben-figures/Fig_7D_Mg132MN_data.xlsx")) %>% glimpse
read_excel(here(proj_path,"ben-figures/Fig_7D_Mg132MN_data.xlsx")) %>% print(n=Inf)
read_excel(here(proj_path,"ben-figures/Fig_7D_Mg132MN_data.xlsx")) %>% count(`siRNA Treatment`)
Fig_7D_Mg132MN_data.xlsx = read_excel(here(proj_path,"ben-figures/Fig_7D_Mg132MN_data.xlsx")) %>% mutate(`siRNA Treatment`= gsub("Scrambled", "Scrambled siRNA", `siRNA Treatment`), `siRNA Treatment` = factor(`siRNA Treatment`, levels = c("Untreated","Scrambled siRNA","GPNMB siRNA")), Genotype = paste(Genotype,"MiCM"))
Fig_7D_Mg132MN_data.xlsx %>% glimpse
Fig_7D_Mg132MN_data.xlsx %>% my_summarise(`% motor neuron death`, `siRNA Treatment`)

Fig_7D_Mg132MN_quantification.summarise = Fig_7D_Mg132MN_data.xlsx %>% my_summarise(`% motor neuron death`, `siRNA Treatment`) 
Fig_7D_Mg132MN_quantification.summarise %>% print(n=Inf)

# for stats:
# VCP vs CTRL Scramble
Fig_7D_Mg132MN_quantification.ctrl.fitg = glm(`% motor neuron death` ~ `siRNA Treatment` + Experiment + Line, data= filter(Fig_7D_Mg132MN_data.xlsx, `siRNA Treatment` != "Untreated", `MG132 treatment` != "Untreated", Genotype == "CTRL MiCM"), family = gaussian)
summ(Fig_7D_Mg132MN_quantification.ctrl.fitg, digits = 5)$coeftable 
#                                   Est.     S.E.    t val.            p
# (Intercept)                   94.55820 6.465708 14.624570 1.405627e-07
# `siRNA Treatment`GPNMB siRNA  -8.94249 3.136329 -2.851260 1.904999e-02

Fig_7D_Mg132MN_quantification.vcp.fitg = glm(`% motor neuron death` ~ `siRNA Treatment` + Experiment + Line, data= filter(Fig_7D_Mg132MN_data.xlsx, `siRNA Treatment` != "Untreated", `MG132 treatment` != "Untreated", Genotype == "VCP MiCM"), family = gaussian)
summ(Fig_7D_Mg132MN_quantification.vcp.fitg, digits = 5)$coeftable 
#                                    Est.     S.E.     t val.            p
# (Intercept)                  103.200741 6.547010 15.7630331 7.326259e-08
# `siRNA Treatment`GPNMB siRNA  -1.380155 3.175767 -0.4345895 6.740940e-01

Mg132MN_quantification.contrast_acts <- tibble(facet = c("CTRL MiCM", "VCP MiCM"), .y = "lfc", group1 = c("Scrambled siRNA", "Scrambled siRNA"), group2 = c("GPNMB siRNA", "GPNMB siRNA"), n1=8, n2=8, p.signif = c("*","ns"), y.position = 85, xmin = 1, xmax = 2)

Fig_7D_Mg132MN_quantification.violin = violin_plot(data = filter(Fig_7D_Mg132MN_data.xlsx, `MG132 treatment` != "Untreated", `siRNA Treatment` != "Untreated"), color_by = "siRNA Treatment", continuous = "% motor neuron death", ylabel = "% motor neuron death\nafter MG132 treatment", facet_by = "Genotype", plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 2.5, tip_length = 0.03) + theme(axis.text.x = element_text(angle = 30, vjust = 0.7, hjust=0.7)) +
      stat_pvalue_manual(Mg132MN_quantification.contrast_acts, label = "p.signif", size = 4, hide.ns = TRUE)
Fig_7D_Mg132MN_quantification.violin

# Fig_7D_Mg132MN_quantification.violin = violin_plot(data = Fig_7D_Mg132MN_data.xlsx, color_by = "siRNA Treatment", continuous = "% motor neuron death", ylabel = "% motor neuron survival", facet_by = "Genotype", facet_by2 = "MG132 treatment", plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "t_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 2.5, tip_length = 0.03) + theme(axis.text.x = element_text(angle = 30, vjust = 0.7, hjust=0.7)) 
# Fig_7D_Mg132MN_quantification.violin

Fig_7D_Mg132MN_data.xlsx %>% count(`MG132 treatment`)

```


## Merge

```{r}
pSTAT2_MG132.merged = plot_grid(
  Fig_7A_pSTAT2images.gg, 
  Fig_7B_210923_pSTAT2_MiCM_quantification.violin, 
  Fig_7C_MiCM_neuronal_survival_images.gg,
  Fig_7D_Mg132MN_quantification.violin,
  nrow = 4, rel_heights = c(1,1,0.5,1.2), labels = "AUTO")
# pSTAT2_MG132.merged
ggsave(pSTAT2_MG132.merged, filename = here(proj_path, "expression/deseq2/fig.7.pSTAT2_MG132.merged.png"), height = 10, width = 6)
ggsave(pSTAT2_MG132.merged, filename = here(proj_path, "expression/deseq2/fig.7.pSTAT2_MG132.merged.pdf"), height = 10, width = 6)

```

# Fig S8 GPNMB knocdown in MN survival

```{r}
mn_death_gpnmb_sirna_data = read_excel(here(proj_path, "expression/lysotracker/Rebuttal_data_for Oli.xlsx"), sheet = "Rebuttal_Figure_7") %>% clean_names() %>% mutate(microglia_genotype = gsub("CTRL","CTRL\nMicroglia",microglia_genotype),microglia_genotype = gsub("VCP","VCP\nMicroglia",microglia_genotype), motor_neuron_genotype = gsub("CTRL","CTRL\nMotor neuron",motor_neuron_genotype),motor_neuron_genotype = gsub("VCP","VCP\nMotor neuron",motor_neuron_genotype))
mn_death_gpnmb_sirna_data %>% glimpse

# Basal: MN death in CTRL & VCP MNs treated with scramble & GPNMB siRNA
# for stats, adjust for experiment
percent_motor_neuron_death.untx.fitg = glm(percent_motor_neuron_death ~ motor_neuron_genotype + experiment, data= filter(mn_death_gpnmb_sirna_data, mg132_treatment == "Untreated"), family = gaussian)
summ(percent_motor_neuron_death.untx.fitg, digits = 5)$coeftable 
#                                Est.     S.E.     t val.            p
# (Intercept)              23.7310738 2.984093  7.9525243 4.258113e-12
# motor_neuron_genotypeVCP -1.6696047 2.446467 -0.6824554 4.966474e-01
percent_motor_neuron_death.untx.violin = violin_plot(data = filter(mn_death_gpnmb_sirna_data, mg132_treatment == "Untreated"), color_by = "motor_neuron_genotype", continuous = "percent_motor_neuron_death", ylabel = "MN death %", ylims = c(0,60), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 100, tip_length = 0.05)
percent_motor_neuron_death.untx.violin

# MG132: MN death in CTRL & VCP MNs treated with scramble & GPNMB siRNA
percent_motor_neuron_death.mg132.fitg = glm(percent_motor_neuron_death ~ motor_neuron_genotype + experiment, data= filter(mn_death_gpnmb_sirna_data, mg132_treatment != "Untreated"), family = gaussian)
summ(percent_motor_neuron_death.mg132.fitg, digits = 5)$coeftable 
#                                Est.     S.E.     t val.            p
# (Intercept)              23.7310738 2.984093  7.9525243 4.258113e-12
# motor_neuron_genotypeVCP -1.6696047 2.446467 -0.6824554 4.966474e-01
percent_motor_neuron_death.mg132.violin = violin_plot(data = filter(mn_death_gpnmb_sirna_data, mg132_treatment != "Untreated"), color_by = "motor_neuron_genotype", continuous = "percent_motor_neuron_death", ylabel = "MN death %", ylims = c(0,60), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 100, tip_length = 0.05)
percent_motor_neuron_death.mg132.violin

percent_motor_neuron_death.mg132.fitg = glm(percent_motor_neuron_death ~ si_rna_treatment + experiment, data= filter(mn_death_gpnmb_sirna_data, mg132_treatment == "Untreated", motor_neuron_genotype == "CTRL"), family = gaussian)
summ(percent_motor_neuron_death.mg132.fitg, digits = 5)$coeftable 
percent_motor_neuron_death.mg132.fitg = glm(percent_motor_neuron_death ~ si_rna_treatment + experiment, data= filter(mn_death_gpnmb_sirna_data, mg132_treatment == "Untreated", motor_neuron_genotype == "VCP"), family = gaussian)
summ(percent_motor_neuron_death.mg132.fitg, digits = 5)$coeftable 
percent_motor_neuron_death.mg132.fitg = glm(percent_motor_neuron_death ~ si_rna_treatment + experiment, data= filter(mn_death_gpnmb_sirna_data, mg132_treatment != "Untreated", motor_neuron_genotype == "CTRL"), family = gaussian)
summ(percent_motor_neuron_death.mg132.fitg, digits = 5)$coeftable 
percent_motor_neuron_death.mg132.fitg = glm(percent_motor_neuron_death ~ si_rna_treatment + experiment, data= filter(mn_death_gpnmb_sirna_data, mg132_treatment != "Untreated", motor_neuron_genotype == "VCP"), family = gaussian)
summ(percent_motor_neuron_death.mg132.fitg, digits = 5)$coeftable 

percent_motor_neuron_death.mg132.violin = violin_plot(data = mn_death_gpnmb_sirna_data, facet_by = "mg132_treatment", facet_by2 = "motor_neuron_genotype", color_by = "si_rna_treatment", continuous = "percent_motor_neuron_death", ylabel = "MN death %", ylims = c(0,100), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 100, tip_length = 0.05)
percent_motor_neuron_death.mg132.violin


percent_motor_neuron_death.mg132.violin = violin_plot(data = mn_death_gpnmb_sirna_data, facet_by = "mg132_treatment", facet_by2 = "microglia_genotype", color_by = "si_rna_treatment", continuous = "percent_motor_neuron_death", ylabel = "MN death %", ylims = c(0,100), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 100, tip_length = 0.05)
percent_motor_neuron_death.mg132.violin

 
#untreated only
percent_motor_neuron_death.untx.violin = violin_plot(data = filter(mn_death_gpnmb_sirna_data, mg132_treatment == "Untreated"), facet_by = "microglia_genotype", facet_by2 = "motor_neuron_genotype", color_by = "si_rna_treatment", continuous = "percent_motor_neuron_death", ylabel = "MN death %", ylims = c(0,100), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 100, tip_length = 0.05)
percent_motor_neuron_death.untx.violin

# MG132 only
percent_motor_neuron_death.mg132.violin = violin_plot(data = filter(mn_death_gpnmb_sirna_data, mg132_treatment == "MG132"), facet_by = "microglia_genotype", facet_by2 = "motor_neuron_genotype", color_by = "si_rna_treatment", continuous = "percent_motor_neuron_death", ylabel = "MN death %", ylims = c(0,100), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 100, tip_length = 0.05)
percent_motor_neuron_death.mg132.violin


percent_motor_neuron_death.untx.mg132.violin.merged = plot_grid(percent_motor_neuron_death.untx.violin, percent_motor_neuron_death.mg132.violin, labels = c("Untreated","MG132"), nrow = 2)
percent_motor_neuron_death.untx.mg132.violin.merged
ggsave(percent_motor_neuron_death.untx.mg132.violin.merged, filename = here(proj_path, "expression/deseq2/percent_motor_neuron_death.untx.mg132.violin.merged.png"), height = 8, width = 6)


# relative MN survival
rebuttal_figure_7_data = read_excel(here(proj_path, "ben-figures/Rebuttal_Mg132MN_data_for_Oli_updated.xlsx"), sheet = "Rebuttal figure 7 part 1") %>% clean_names() 
rebuttal_figure_7_data %>% glimpse

percent_motor_neuron_death.untx.violin = violin_plot(data = filter(rebuttal_figure_7_data, mg132_treatment == "Untreated"), facet_by = "microglia_genotype", facet_by2 = "motor_neuron_genotype", color_by = "si_rna_treatment", continuous = "relative_motor_neuron_survival_percent", ylabel = "relative MN death %", ylims = c(50,150), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 100, tip_length = 0.05)
percent_motor_neuron_death.untx.violin

percent_motor_neuron_death.mg132.violin = violin_plot(data = filter(rebuttal_figure_7_data, mg132_treatment == "MG132"), facet_by = "microglia_genotype", facet_by2 = "motor_neuron_genotype", color_by = "si_rna_treatment", continuous = "relative_motor_neuron_survival_percent", ylabel = "relative MN death %", ylims = c(50,150), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 100, tip_length = 0.05)
percent_motor_neuron_death.mg132.violin

percent_motor_neuron_death.untx.mg132.violin.merged = plot_grid(percent_motor_neuron_death.untx.violin, percent_motor_neuron_death.mg132.violin, labels = c("Untreated","MG132"), nrow = 2)
percent_motor_neuron_death.untx.mg132.violin.merged
ggsave(percent_motor_neuron_death.untx.mg132.violin.merged, filename = here(proj_path, "expression/deseq2/percent_motor_neuron_death.untx.mg132.violin.merged.png"), height = 8, width = 6)

```


# Fig S9 Rebuttal MN MG coculture

```{r}
# import the immuno images with magick 
Fig.7A_rebuttal.png = magick::image_read(here(proj_path,"ben-figures/Fig.7A_rebuttal.png"))
Fig.7A_rebuttal.gg = ggdraw() + draw_image(Fig.7A_rebuttal.png)
Fig.7A_rebuttal.gg

rebuttal_figure_7_data = read_excel(here(proj_path, "expression/lysotracker/Rebuttal_data_for Oli.xlsx"), sheet = "Rebuttal_Figure_7_part_2") %>% clean_names()  %>%
  mutate(mg_genotype = gsub("CTRL","CTRL MG",mg_genotype),mg_genotype=gsub("VCP","VCP MG",mg_genotype), mn_genotype=gsub("CTRL","CTRL MN",mn_genotype),mn_genotype=gsub("VCP","VCP MN",mn_genotype),genotype = paste(mg_genotype,"-",mn_genotype),
    si_rna_treatment = gsub("GPNMB", "GPNMB\nsiRNA", si_rna_treatment), si_rna_treatment = gsub("Scr","Scr\nsiRNA", si_rna_treatment), si_rna_treatment = factor(si_rna_treatment, levels = c("Scr\nsiRNA", "GPNMB\nsiRNA")),
    genotype = factor(genotype, levels = c("CTRL MG - CTRL MN", "VCP MG - CTRL MN", "neurons only - CTRL MN", "CTRL MG - VCP MN", "VCP MG - VCP MN", "neurons only - VCP MN"))) %>%
  filter(mn_line != "CTRL1") # remove MN line CTRL1
rebuttal_figure_7_data %>% glimpse


rebuttal_figure_7_data.long = rebuttal_figure_7_data %>% filter(!grepl("neurons only", genotype)) %>%
  mutate(percent_mg = as.numeric(percent_mg), percent_mn = as.numeric(percent_mn), si_rna_treatment = gsub("\n"," ", si_rna_treatment), genotype_si_rna_treatment = paste(genotype,"\n", si_rna_treatment)) %>% 
  group_by(genotype_si_rna_treatment) %>%
  summarize(
    Microglia = mean(percent_mg, na.rm = TRUE),
    se_percent_mg = sd(percent_mg, na.rm = TRUE) / sqrt(n()),
    Motor_neuron = mean(percent_mn, na.rm = TRUE),
    se_percent_mn = sd(percent_mn, na.rm = TRUE) / sqrt(n())
  ) %>%
  pivot_longer(
    cols = c(Microglia, Motor_neuron),
    names_to = "cell_type",
    values_to = "mean_percent"
  ) %>%
  mutate(
    se = ifelse(cell_type == "Microglia", se_percent_mg, se_percent_mn)
  ) %>%
  select(genotype_si_rna_treatment, cell_type, mean_percent, se)


mn.mg.coculture.celltype.barplot = ggbarplot(rebuttal_figure_7_data.long, x="genotype_si_rna_treatment", y="mean_percent", fill="cell_type", position = position_fill(), xlab = "", lab.size = 3, lab.vjust = 0.5, lab.hjust = 1.5)  +
  scale_fill_npg() +
  theme_classic() +  theme(panel.grid = element_blank(), legend.title = element_blank(), legend.position = "top", legend.spacing.x = unit(0.2, "cm"), legend.text=element_text(size=8), legend.key.size = unit(0.4, "cm"), axis.text=element_text(size=8), axis.title=element_text(size=8), strip.text.y = element_text(angle = 0)) +
  guides(fill = guide_legend(reverse = TRUE, nrow = 1)) +
  coord_flip() + ylab(expression(Proportion~of~cells))
mn.mg.coculture.celltype.barplot



rebuttal_figure_7_data %>% my_summarise(percent_mn_caspase_3_ve, experiment) %>% print(n=Inf)

# for stats, adjust for experiment
gpnmb_mean_cytoplasm_intensity.gpnmbkd.fitg = glm(gpnmb_mean_cytoplasm_intensity ~ genotype + experiment, data= rebuttal_figure_7_data, family = gaussian)
summ(gpnmb_mean_cytoplasm_intensity.gpnmbkd.fitg, digits = 6)$coeftable 

percent_mn_caspase_3_ve.coculture.violin = violin_plot(data = rebuttal_figure_7_data, color_by = "si_rna_treatment", continuous = "percent_mn_caspase_3_ve", facet_by= "genotype", ylabel = "% cleaved caspase 3 +ve", ylims = c(0,100), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 100, tip_length = 0.06)
percent_mn_caspase_3_ve.coculture.violin


percent_mn_pyknotic.coculture.violin = violin_plot(data = rebuttal_figure_7_data, color_by = "si_rna_treatment", continuous = "percent_mn_pyknotic", facet_by= "genotype", ylabel = "% pyknotic nuclei", ylims = c(0,100), plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 100, tip_length = 0.06)
percent_mn_pyknotic.coculture.violin

coculture.caspase3.pyknotic = plot_grid(Fig.7A_rebuttal.gg, mn.mg.coculture.celltype.barplot,
  plot_grid(percent_mn_caspase_3_ve.coculture.violin, percent_mn_pyknotic.coculture.violin, ncol = 2, labels = c("C","D")), nrow = 3, labels = c("A","B",""), rel_heights = c(1,0.8,1))
# coculture.caspase3.pyknotic
ggsave(coculture.caspase3.pyknotic, filename = here(proj_path, "expression/deseq2/coculture.caspase3.pyknotic.png"), height = 11.75, width = 9.75)
ggsave(rebuttal_fig6.merged, filename = here(proj_path, "expression/deseq2/coculture.caspase3.pyknotic.pdf"), height = 11.75, width = 9.75)

```




